<div class="mw-parser-output"><div class="thumb tright"><div class="thumbinner" style="width:302px;"><a href="/wiki/File:CRISPR_Sterics.pdf" class="image"><img alt="" src="//upload.wikimedia.org/wikipedia/commons/thumb/0/0c/CRISPR_Sterics.pdf/page1-300px-CRISPR_Sterics.pdf.jpg" decoding="async" width="300" height="288" class="thumbimage" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/0/0c/CRISPR_Sterics.pdf/page1-450px-CRISPR_Sterics.pdf.jpg 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/0/0c/CRISPR_Sterics.pdf/page1-600px-CRISPR_Sterics.pdf.jpg 2x" data-file-width="6711" data-file-height="6449"></a>  <div class="thumbcaption"><div class="magnify"><a href="/wiki/File:CRISPR_Sterics.pdf" class="internal" title="Enlarge"></a></div>Transcriptional repression via steric hindrance</div></div></div>
<p><b>CRISPR interference</b> (<b>CRISPRi</b>) is a genetic perturbation technique that allows for sequence-specific repression of gene expression in <a href="/wiki/Prokaryotic" class="mw-redirect" title="Prokaryotic">prokaryotic</a> and <a href="/wiki/Eukaryotic" class="mw-redirect" title="Eukaryotic">eukaryotic</a> cells. It was first developed in the laboratories of <a href="/wiki/Wendell_Lim" title="Wendell Lim">Wendell Lim</a> and <a href="/wiki/Jonathan_Weissman" title="Jonathan Weissman">Jonathan Weissman</a>
labs at <a href="/wiki/UCSF" class="mw-redirect" title="UCSF">UCSF</a>.<sup id="cite_ref-pmid23452860_1-0" class="reference"><a href="#cite_note-pmid23452860-1">[1]</a></sup> Sequence-specific activation of gene expression refers to <a href="/wiki/DCas9_activation_system" title="DCas9 activation system">CRISPR activation (CRISPRa)</a>. 
</p><p>Based on the bacterial genetic immune system - <a href="/wiki/CRISPR" title="CRISPR">CRISPR</a> (clustered regularly interspaced short palindromic repeats) pathway,<sup id="cite_ref-pmid17379808_2-0" class="reference"><a href="#cite_note-pmid17379808-2">[2]</a></sup> 
the technique provides a complementary approach to <a href="/wiki/RNA_interference" title="RNA interference">RNA interference</a>. The difference between CRISPRi and RNAi, though, is that CRISPRi regulates gene expression primarily on the transcriptional level, while RNAi controls genes on the mRNA level.
</p>
<div id="toc" class="toc" role="navigation" aria-labelledby="mw-toc-heading"><input type="checkbox" role="button" id="toctogglecheckbox" class="toctogglecheckbox" style="display:none"><div class="toctitle" lang="en" dir="ltr"><h2 id="mw-toc-heading">Contents</h2><span class="toctogglespan"><label class="toctogglelabel" for="toctogglecheckbox"></label></span></div>
<ul>
<li class="toclevel-1 tocsection-1"><a href="#Background"><span class="tocnumber">1</span> <span class="toctext">Background</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="#Transcriptional_regulation"><span class="tocnumber">2</span> <span class="toctext">Transcriptional regulation</span></a>
<ul>
<li class="toclevel-2 tocsection-3"><a href="#Repression"><span class="tocnumber">2.1</span> <span class="toctext">Repression</span></a></li>
<li class="toclevel-2 tocsection-4"><a href="#Improvements_on_the_efficiency"><span class="tocnumber">2.2</span> <span class="toctext">Improvements on the efficiency</span></a>
<ul>
<li class="toclevel-3 tocsection-5"><a href="#Other_methods"><span class="tocnumber">2.2.1</span> <span class="toctext">Other methods</span></a></li>
</ul>
</li>
</ul>
</li>
<li class="toclevel-1 tocsection-6"><a href="#Applications"><span class="tocnumber">3</span> <span class="toctext">Applications</span></a>
<ul>
<li class="toclevel-2 tocsection-7"><a href="#Gene_knockdown"><span class="tocnumber">3.1</span> <span class="toctext">Gene knockdown</span></a></li>
<li class="toclevel-2 tocsection-8"><a href="#Allelic_series"><span class="tocnumber">3.2</span> <span class="toctext">Allelic series</span></a></li>
<li class="toclevel-2 tocsection-9"><a href="#Genome_loci_imaging"><span class="tocnumber">3.3</span> <span class="toctext">Genome loci imaging</span></a></li>
<li class="toclevel-2 tocsection-10"><a href="#Stem_cells"><span class="tocnumber">3.4</span> <span class="toctext">Stem cells</span></a></li>
<li class="toclevel-2 tocsection-11"><a href="#Genetic_screening"><span class="tocnumber">3.5</span> <span class="toctext">Genetic screening</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-12"><a href="#Advantages_and_limitations"><span class="tocnumber">4</span> <span class="toctext">Advantages and limitations</span></a>
<ul>
<li class="toclevel-2 tocsection-13"><a href="#Advantages"><span class="tocnumber">4.1</span> <span class="toctext">Advantages</span></a></li>
<li class="toclevel-2 tocsection-14"><a href="#Limitations"><span class="tocnumber">4.2</span> <span class="toctext">Limitations</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-15"><a href="#References"><span class="tocnumber">5</span> <span class="toctext">References</span></a></li>
<li class="toclevel-1 tocsection-16"><a href="#External_links"><span class="tocnumber">6</span> <span class="toctext">External links</span></a></li>
</ul>
</div>

<h2><span class="mw-headline" id="Background">Background</span></h2>
<div role="note" class="hatnote navigation-not-searchable">See also: <a href="/wiki/CRISPR" title="CRISPR">CRISPR</a></div>
<p>Many <a href="/wiki/Bacteria" title="Bacteria">bacteria</a> and most <a href="/wiki/Archaea" title="Archaea">archaea</a> have an adaptive immune system which incorporates CRISPR RNA (crRNA) and CRISPR-associated (cas) genes.
</p><p>The CRISPR interference (CRISPRi) technique was first reported by Lei S. Qi and researchers at the <a href="/wiki/University_of_California_at_San_Francisco" class="mw-redirect" title="University of California at San Francisco">University of California at San Francisco</a> in early 2013.<sup id="cite_ref-pmid23452860_1-1" class="reference"><a href="#cite_note-pmid23452860-1">[1]</a></sup> The technology uses a catalytically dead <a href="/wiki/Cas9" title="Cas9">Cas9</a> (usually denoted as dCas9) protein that lacks endonuclease activity to regulate genes in an RNA-guided manner. Targeting specificity is determined by complementary base-pairing of a single guide RNA (sgRNA) to the genomic loci. sgRNA is a chimeric noncoding RNA that can be subdivided into three regions: a 20 nt base-pairing sequence, a 42 nt dCas9-binding hairpin and a 40 nt terminator (bacteria,<sup id="cite_ref-pmid23360965_3-0" class="reference"><a href="#cite_note-pmid23360965-3">[3]</a></sup>
<sup id="cite_ref-pmid27238023_4-0" class="reference"><a href="#cite_note-pmid27238023-4">[4]</a></sup>
<sup id="cite_ref-pmid27996021_5-0" class="reference"><a href="#cite_note-pmid27996021-5">[5]</a></sup> yeast,<sup id="cite_ref-pmid23460208_6-0" class="reference"><a href="#cite_note-pmid23460208-6">[6]</a></sup> fruit flies,<sup id="cite_ref-pmid2408874_7-0" class="reference"><a href="#cite_note-pmid2408874-7">[7]</a></sup> zebrafish,<sup id="cite_ref-pmid23360964_8-0" class="reference"><a href="#cite_note-pmid23360964-8">[8]</a></sup> mice,<sup id="cite_ref-pmid23643243_9-0" class="reference"><a href="#cite_note-pmid23643243-9">[9]</a></sup>).
</p><p>When designing a synthetic sgRNA, only the 20 nt base-pairing sequence is modified. Secondary variables must also be considered: off-target effects (for which a simple BLAST run of the base-pairing sequence is required), maintenance of the dCas9-binding hairpin structure, and ensuring that no restriction sites are present in the modified sgRNA, as this may pose a problem in downstream cloning steps. Due to the simplicity of sgRNA design, this technology is amenable to genome-wide scaling.<sup id="cite_ref-pmid24136345_10-0" class="reference"><a href="#cite_note-pmid24136345-10">[10]</a></sup>
CRISPRi relies on the generation of catalytically inactive Cas9. This is accomplished by introducing point mutations in the two catalytic residues (D10A and H840A) of the gene encoding Cas9.<sup id="cite_ref-pmid22745249_11-0" class="reference"><a href="#cite_note-pmid22745249-11">[11]</a></sup> In doing so, dCas9 is unable to cleave dsDNA but retains the ability to target DNA. Together, sgRNA and dCas9 constitute a minimal system for gene-specific regulation.<sup id="cite_ref-pmid23452860_1-2" class="reference"><a href="#cite_note-pmid23452860-1">[1]</a></sup>
</p>
<h2><span class="mw-headline" id="Transcriptional_regulation">Transcriptional regulation</span></h2>
<h3><span class="mw-headline" id="Repression">Repression</span></h3>
<p>CRISPRi can sterically repress <a href="/wiki/Transcription_(genetics)" class="mw-redirect" title="Transcription (genetics)">transcription</a> by blocking either transcriptional initiation or elongation. This is accomplished by designing sgRNA complementary to the <a href="/wiki/Promoter_(genetics)" title="Promoter (genetics)">promoter</a> or the <a href="/wiki/Exon" title="Exon">exonic</a> sequences. The level of transcriptional repression for exonic sequences is strand-specific. sgRNA complementary to the non-template strand more strongly represses transcription compared to sgRNA complementary to the template strand. It has been suggested that this is due to the activity of helicase, which unwinds the RNA:DNA heteroduplex ahead of <a href="/wiki/RNA_polymerase_II" title="RNA polymerase II">RNA pol II</a> when the sgRNA is complementary to exons of the template strand. In prokaryotes, this steric inhibition can repress transcription of the target gene by almost 99.9%; in human cells, up to 90% repression was observed.<sup id="cite_ref-pmid23452860_1-3" class="reference"><a href="#cite_note-pmid23452860-1">[1]</a></sup>
</p><p>CRISPRi can also repress transcription via an effector domain. Fusing a repressor domain to dCas9 allows transcription to be further repressed by inducing heterochromatinization. For example, the well-studied <a href="/wiki/Kr%C3%BCppel_associated_box" title="Kr&#252;ppel associated box">Kr&#252;ppel associated box</a> (KRAB) domain can be fused to dCas9 to repress transcription of the target gene up to 99% in human cells.<sup id="cite_ref-pmid23849981_12-0" class="reference"><a href="#cite_note-pmid23849981-12">[12]</a></sup>
</p>
<h3><span class="mw-headline" id="Improvements_on_the_efficiency">Improvements on the efficiency</span></h3>
<p>Whereas genome-editing by the catalytically active Cas9 nuclease can be accompanied by irreversible off-target genomic alterations, CRISPRi is highly specific with minimal off-target reversible effects for two distinct sgRNA sequences.<sup id="cite_ref-pmid23849981_12-1" class="reference"><a href="#cite_note-pmid23849981-12">[12]</a></sup> Nonetheless, several methods have been developed to improve the efficiency of transcriptional modulation. Identification of the <a href="/wiki/Transcription_start_site" class="mw-redirect" title="Transcription start site">transcription start site</a> of a target gene and considering the preferences of sgRNA improves efficiency, as does the presence of accessible <a href="/wiki/Chromatin" title="Chromatin">chromatin</a> at the target site.<sup id="cite_ref-13" class="reference"><a href="#cite_note-13">[13]</a></sup>
</p>
<h4><span class="mw-headline" id="Other_methods">Other methods</span></h4>
<p>Along with other <a href="/wiki/CRISPR_interference#Advantages_and_Limitations" title="CRISPR interference">improvements</a> mentioned, factors such as the distance from the transcription start and the local chromatin state may be critical parameters in determining activation/repression efficiency. Optimization of dCas9 and sgRNA expression, stability, nuclear localization, and interaction will likely allow for further improvement of CRISPRi efficiency in mammalian cells.<sup id="cite_ref-pmid23452860_1-4" class="reference"><a href="#cite_note-pmid23452860-1">[1]</a></sup>
</p>
<h2><span class="mw-headline" id="Applications">Applications</span></h2>
<h3><span class="mw-headline" id="Gene_knockdown">Gene knockdown</span></h3>
<p>A significant portion of the genome (both reporter and endogenous genes) in eukaryotes has been shown to be targetable using lentiviral constructs to express dCas9 and sgRNAs, with comparable efficiency to existing techniques such as RNAi and TALE proteins.<sup id="cite_ref-pmid23849981_12-2" class="reference"><a href="#cite_note-pmid23849981-12">[12]</a></sup> In tandem or as its own system, CRISPRi could be used to achieve the same <a href="/wiki/RNA_Interference#Applications" class="mw-redirect" title="RNA Interference">applications</a> as in RNAi.
</p><p>For bacteria, gene knockdown by CRISPRi has been fully implemented and characterized (off-target analysis, leaky repression) for both Gram-negative <i>E. coli</i> <sup id="cite_ref-pmid23360965_3-1" class="reference"><a href="#cite_note-pmid23360965-3">[3]</a></sup><sup id="cite_ref-pmid27996021_5-1" class="reference"><a href="#cite_note-pmid27996021-5">[5]</a></sup> and Gram-positive <i>B. subtilis</i>.<sup id="cite_ref-pmid27238023_4-1" class="reference"><a href="#cite_note-pmid27238023-4">[4]</a></sup>
</p>
<div class="thumb tright"><div class="thumbinner" style="width:202px;"><a href="/wiki/File:CRISPRflowchart.pdf" class="image"><img alt="" src="//upload.wikimedia.org/wikipedia/commons/thumb/f/f1/CRISPRflowchart.pdf/page1-200px-CRISPRflowchart.pdf.jpg" decoding="async" width="200" height="336" class="thumbimage" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/f/f1/CRISPRflowchart.pdf/page1-300px-CRISPRflowchart.pdf.jpg 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/f/f1/CRISPRflowchart.pdf/page1-400px-CRISPRflowchart.pdf.jpg 2x" data-file-width="2216" data-file-height="3727"></a>  <div class="thumbcaption"><div class="magnify"><a href="/wiki/File:CRISPRflowchart.pdf" class="internal" title="Enlarge"></a></div>CRISPRi construction workflow</div></div></div>
<h3><span class="mw-headline" id="Allelic_series">Allelic series</span></h3>
<p>Differential gene expression can be achieved by modifying the efficiency of sgRNA base-pairing to the target loci.<sup id="cite_ref-pmid24136345_10-1" class="reference"><a href="#cite_note-pmid24136345-10">[10]</a></sup> In theory, modulating this efficiency can be used to create an allelic series for any given gene, in essence creating a collection of hypo- and hypermorphs. These powerful collections can be used to probe any genetic investigation. For <a href="/wiki/Muller%27s_morphs" title="Muller's morphs">hypomorphs</a>, this allows the incremental reduction of gene function as opposed to the binary nature of gene knockouts and the unpredictability of knockdowns. For <a href="/wiki/Muller%27s_morphs" title="Muller's morphs">hypermorphs</a>, this is in contrast to the conventional method of cloning the gene of interest under promoters with variable strength.
</p>
<h3><span class="mw-headline" id="Genome_loci_imaging">Genome loci imaging</span></h3>
<p>Fusing a <a href="/wiki/Green_fluorescent_protein" title="Green fluorescent protein">fluorescent protein</a> to dCas9 allows for imaging of genomic loci in living human cells.<sup id="cite_ref-pmid24360272_14-0" class="reference"><a href="#cite_note-pmid24360272-14">[14]</a></sup> Compared to fluorescence in situ hybridization (FISH), the method uniquely allows for dynamic tracking of chromosome loci. This has been used to study chromatin architecture and nuclear organization dynamics in laboratory cell lines including HeLa cells.
</p>
<h3><span class="mw-headline" id="Stem_cells">Stem cells</span></h3>
<p>Activation of <a href="/wiki/Reprogramming" title="Reprogramming">Yamanaka factors</a> by CRISPRi has been used to <a href="/wiki/Induced_pluripotent_stem_cell" title="Induced pluripotent stem cell">induce pluripotency</a> in human and mouse cells providing an alternative method to iPS technology.<sup id="cite_ref-pmid24346702_15-0" class="reference"><a href="#cite_note-pmid24346702-15">[15]</a></sup><sup id="cite_ref-pmid24500196_16-0" class="reference"><a href="#cite_note-pmid24500196-16">[16]</a></sup> In addition, large-scale activation screens could be used to identify proteins that promote induced pluripotency or, conversely, promote differentiation
to a specific cell lineage.<sup id="cite_ref-pmid16904174_17-0" class="reference"><a href="#cite_note-pmid16904174-17">[17]</a></sup>
</p>
<h3><span class="mw-headline" id="Genetic_screening">Genetic screening</span></h3>
<p>The ability to upregulate gene expression using dCas9-SunTag with a single sgRNA also opens the door to large-scale genetic screens, such as <a href="/wiki/Perturb-seq" title="Perturb-seq">Perturb-seq</a>, to uncover phenotypes that result from increased or decreased gene expression, which will be especially important for understanding the effects of gene regulation in cancer.<sup id="cite_ref-pmid25307933_18-0" class="reference"><a href="#cite_note-pmid25307933-18">[18]</a></sup> Furthermore, CRISPRi systems have been shown to be transferable via horizontal gene transfer mechanisms such as <a href="/wiki/Bacterial_conjugation" title="Bacterial conjugation">bacterial conjugation</a> and specific repression of reporter genes in recipient cells has been demonstrated. CRISPRi could serve as a tool for genetic screening and potentially bacterial population control.<sup id="cite_ref-pmid25409531_19-0" class="reference"><a href="#cite_note-pmid25409531-19">[19]</a></sup>
</p>
<h2><span class="mw-headline" id="Advantages_and_limitations">Advantages and limitations</span></h2>
<h3><span class="mw-headline" id="Advantages">Advantages</span></h3>
<ol><li>CRISPRi can silence a target gene of interest up to 99.9% repression.<sup id="cite_ref-pmid24136345_10-2" class="reference"><a href="#cite_note-pmid24136345-10">[10]</a></sup></li>
<li>Since CRISPRi is based on Watson-Crick base-pairing of sgRNA-DNA and an NGG PAM motif, selection of targetable sites within the genome is straightforward and flexible. Carefully defined protocols have been developed.<sup id="cite_ref-pmid24136345_10-3" class="reference"><a href="#cite_note-pmid24136345-10">[10]</a></sup></li>
<li>Multiple sgRNAs can not only be used to control multiple different genes simultaneously (multiplexing gene targeting), but also to enhance the efficiency of regulating the same gene target.</li>
<li>While the two systems can be complementary, CRISPRi provides advantages over RNAi. As an exogenous system, CRISPRi does not compete with endogenous machinery such as microRNA expression or function. Furthermore, because CRISPRi acts at the DNA level, one can target transcripts such as noncoding RNAs, microRNAs, antisense transcripts, nuclear-localized RNAs, and polymerase III transcripts. Finally, CRISPRi possesses a much larger targetable sequence space; promoters and, in theory, introns can also be targeted.<sup id="cite_ref-pmid23849981_12-3" class="reference"><a href="#cite_note-pmid23849981-12">[12]</a></sup></li>
<li>In <i>E. coli</i>, construction of a gene knockdown strain is extremely fast and requires only one-step oligo <a href="/wiki/Recombineering" title="Recombineering">recombineering</a>.<sup id="cite_ref-pmid27996021_5-2" class="reference"><a href="#cite_note-pmid27996021-5">[5]</a></sup></li></ol>
<h3><span class="mw-headline" id="Limitations">Limitations</span></h3>
<ol><li>The requirement of a <a href="/wiki/Protospacer_adjacent_motif" title="Protospacer adjacent motif">protospacer adjacent motif</a> (PAM) sequence limits the number of potential target sequences. Cas9 and its homologs may use different PAM sequences, and therefore could theoretically be utilized to expand the number of potential target sequences.<sup id="cite_ref-pmid24136345_10-4" class="reference"><a href="#cite_note-pmid24136345-10">[10]</a></sup></li>
<li>Sequence specificity to target loci is only 14 nt long (12 nt of sgRNA and 2nt of the PAM), which can recur around 11 times in a human genome.<sup id="cite_ref-pmid24136345_10-5" class="reference"><a href="#cite_note-pmid24136345-10">[10]</a></sup> Repression is inversely correlated with the distance of the target site from the transcription start site. Genome-wide computational predictions or selection of Cas9 homologs with a longer PAM may reduce nonspecific targeting.</li>
<li>Endogenous chromatin states and modifications may prevent the sequence-specific binding of the dCas9-sgRNA complex.<sup id="cite_ref-pmid24136345_10-6" class="reference"><a href="#cite_note-pmid24136345-10">[10]</a></sup>  The level of transcriptional repression in mammalian cells varies between genes. Much work is needed to understand the role of local DNA conformation and chromatin in relation to binding and regulatory efficiency.</li>
<li>CRISPRi can influence genes that are in close proximity to the target gene. This is especially important when targeting genes that either overlap other genes (sense or antisense overlapping) or are driven by a bidirectional promoter.<sup id="cite_ref-20" class="reference"><a href="#cite_note-20">[20]</a></sup></li></ol>
<h2><span class="mw-headline" id="References">References</span></h2>
<div class="reflist columns references-column-width" style="list-style-type: decimal;">
<ol class="references">
<li id="cite_note-pmid23452860-1"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid23452860_1-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid23452860_1-1"><sup><i><b>b</b></i></sup></a> <a href="#cite_ref-pmid23452860_1-2"><sup><i><b>c</b></i></sup></a> <a href="#cite_ref-pmid23452860_1-3"><sup><i><b>d</b></i></sup></a> <a href="#cite_ref-pmid23452860_1-4"><sup><i><b>e</b></i></sup></a></span> <span class="reference-text"><cite id="CITEREFQiLarsonGilbertDoudna2013" class="citation journal cs1">Qi, L. S.; Larson, M. H.; Gilbert, L. A.; Doudna, J. A.; Weissman, J. S.; Arkin, A. P.; Lim, W. A. (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3664290">"Repurposing CRISPR as an RNA-guided platform for sequence-specific control of gene expression"</a>. <i>Cell</i>. <b>152</b> (5): 1173&#8211;83. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1016%2Fj.cell.2013.02.022">10.1016/j.cell.2013.02.022</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3664290">3664290</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/23452860">23452860</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Cell&amp;rft.atitle=Repurposing+CRISPR+as+an+RNA-guided+platform+for+sequence-specific+control+of+gene+expression&amp;rft.volume=152&amp;rft.issue=5&amp;rft.pages=1173-83&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3664290&amp;rft_id=info%3Apmid%2F23452860&amp;rft_id=info%3Adoi%2F10.1016%2Fj.cell.2013.02.022&amp;rft.aulast=Qi&amp;rft.aufirst=L.+S.&amp;rft.au=Larson%2C+M.+H.&amp;rft.au=Gilbert%2C+L.+A.&amp;rft.au=Doudna%2C+J.+A.&amp;rft.au=Weissman%2C+J.+S.&amp;rft.au=Arkin%2C+A.+P.&amp;rft.au=Lim%2C+W.+A.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3664290&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><style data-mw-deduplicate="TemplateStyles:r982806391">.mw-parser-output cite.citation{font-style:inherit}.mw-parser-output .citation q{quotes:"\"""\"""'""'"}.mw-parser-output .id-lock-free a,.mw-parser-output .citation .cs1-lock-free a{background:linear-gradient(transparent,transparent),url("//upload.wikimedia.org/wikipedia/commons/6/65/Lock-green.svg")right 0.1em center/9px no-repeat}.mw-parser-output .id-lock-limited a,.mw-parser-output .id-lock-registration a,.mw-parser-output .citation .cs1-lock-limited a,.mw-parser-output .citation .cs1-lock-registration a{background:linear-gradient(transparent,transparent),url("//upload.wikimedia.org/wikipedia/commons/d/d6/Lock-gray-alt-2.svg")right 0.1em center/9px no-repeat}.mw-parser-output .id-lock-subscription a,.mw-parser-output .citation .cs1-lock-subscription a{background:linear-gradient(transparent,transparent),url("//upload.wikimedia.org/wikipedia/commons/a/aa/Lock-red-alt-2.svg")right 0.1em center/9px no-repeat}.mw-parser-output .cs1-subscription,.mw-parser-output .cs1-registration{color:#555}.mw-parser-output .cs1-subscription span,.mw-parser-output .cs1-registration span{border-bottom:1px dotted;cursor:help}.mw-parser-output .cs1-ws-icon a{background:linear-gradient(transparent,transparent),url("//upload.wikimedia.org/wikipedia/commons/4/4c/Wikisource-logo.svg")right 0.1em center/12px no-repeat}.mw-parser-output code.cs1-code{color:inherit;background:inherit;border:none;padding:inherit}.mw-parser-output .cs1-hidden-error{display:none;font-size:100%}.mw-parser-output .cs1-visible-error{font-size:100%}.mw-parser-output .cs1-maint{display:none;color:#33aa33;margin-left:0.3em}.mw-parser-output .cs1-subscription,.mw-parser-output .cs1-registration,.mw-parser-output .cs1-format{font-size:95%}.mw-parser-output .cs1-kern-left,.mw-parser-output .cs1-kern-wl-left{padding-left:0.2em}.mw-parser-output .cs1-kern-right,.mw-parser-output .cs1-kern-wl-right{padding-right:0.2em}.mw-parser-output .citation .mw-selflink{font-weight:inherit}</style></span>
</li>
<li id="cite_note-pmid17379808-2"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid17379808_2-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFBarrangouFremauxDeveauRichards2007" class="citation journal cs1">Barrangou, R.; Fremaux, C.; Deveau, H.; Richards, M.; Boyaval, P.; Moineau, S.; Romero, D. A.; Horvath, P. (2007). "CRISPR Provides Acquired Resistance Against Viruses in Prokaryotes". <i>Science</i>. <b>315</b> (5819): 1709&#8211;1712. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1126%2Fscience.1138140">10.1126/science.1138140</a>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/17379808">17379808</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Science&amp;rft.atitle=CRISPR+Provides+Acquired+Resistance+Against+Viruses+in+Prokaryotes&amp;rft.volume=315&amp;rft.issue=5819&amp;rft.pages=1709-1712&amp;rft.date=2007&amp;rft_id=info%3Adoi%2F10.1126%2Fscience.1138140&amp;rft_id=info%3Apmid%2F17379808&amp;rft.aulast=Barrangou&amp;rft.aufirst=R.&amp;rft.au=Fremaux%2C+C.&amp;rft.au=Deveau%2C+H.&amp;rft.au=Richards%2C+M.&amp;rft.au=Boyaval%2C+P.&amp;rft.au=Moineau%2C+S.&amp;rft.au=Romero%2C+D.+A.&amp;rft.au=Horvath%2C+P.&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid23360965-3"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid23360965_3-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid23360965_3-1"><sup><i><b>b</b></i></sup></a></span> <span class="reference-text"><cite id="CITEREFJiangBikardCoxZhang2013" class="citation journal cs1">Jiang, W; Bikard, D; Cox, D; Zhang, F; Marraffini, L. A. (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3748948">"RNA-guided editing of bacterial genomes using CRISPR-Cas systems"</a>. <i>Nature Biotechnology</i>. <b>31</b> (3): 233&#8211;9. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1038%2Fnbt.2508">10.1038/nbt.2508</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3748948">3748948</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/23360965">23360965</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Nature+Biotechnology&amp;rft.atitle=RNA-guided+editing+of+bacterial+genomes+using+CRISPR-Cas+systems&amp;rft.volume=31&amp;rft.issue=3&amp;rft.pages=233-9&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3748948&amp;rft_id=info%3Apmid%2F23360965&amp;rft_id=info%3Adoi%2F10.1038%2Fnbt.2508&amp;rft.aulast=Jiang&amp;rft.aufirst=W&amp;rft.au=Bikard%2C+D&amp;rft.au=Cox%2C+D&amp;rft.au=Zhang%2C+F&amp;rft.au=Marraffini%2C+L.+A.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3748948&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid27238023-4"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid27238023_4-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid27238023_4-1"><sup><i><b>b</b></i></sup></a></span> <span class="reference-text"><cite id="CITEREFPeters2016" class="citation journal cs1">Peters, JM;  et al. (2016). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC4894308">"A Comprehensive, CRISPR-based Functional Analysis of Essential Genes in Bacteria"</a>. <i>Cell</i>. <b>165</b> (6): 1493&#8211;1506. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1016%2Fj.cell.2016.05.003">10.1016/j.cell.2016.05.003</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC4894308">4894308</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/27238023">27238023</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Cell&amp;rft.atitle=A+Comprehensive%2C+CRISPR-based+Functional+Analysis+of+Essential+Genes+in+Bacteria&amp;rft.volume=165&amp;rft.issue=6&amp;rft.pages=1493-1506&amp;rft.date=2016&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4894308&amp;rft_id=info%3Apmid%2F27238023&amp;rft_id=info%3Adoi%2F10.1016%2Fj.cell.2016.05.003&amp;rft.aulast=Peters&amp;rft.aufirst=JM&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4894308&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span> <span class="cs1-visible-error error citation-comment"><span class="cs1-visible-error error citation-comment">Explicit use of et al. in: <code class="cs1-code">|first1=</code> (<a href="/wiki/Help:CS1_errors#explicit_et_al" title="Help:CS1 errors">help</a>)</span></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid27996021-5"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid27996021_5-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid27996021_5-1"><sup><i><b>b</b></i></sup></a> <a href="#cite_ref-pmid27996021_5-2"><sup><i><b>c</b></i></sup></a></span> <span class="reference-text"><cite id="CITEREFLiJunErickstadBrown2016" class="citation journal cs1">Li, X; Jun, Y; Erickstad, M; Brown, S; Parks, A; Court, D; Jun, S (2016). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC5171832">"tCRISPRi: tunable and reversible, one-step control of gene expression"</a>. <i>Scientific Reports</i>. <b>6</b>: 39096. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1038%2Fsrep39076">10.1038/srep39076</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC5171832">5171832</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/27996021">27996021</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Scientific+Reports&amp;rft.atitle=tCRISPRi%3A+tunable+and+reversible%2C+one-step+control+of+gene+expression&amp;rft.volume=6&amp;rft.pages=39096&amp;rft.date=2016&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5171832&amp;rft_id=info%3Apmid%2F27996021&amp;rft_id=info%3Adoi%2F10.1038%2Fsrep39076&amp;rft.aulast=Li&amp;rft.aufirst=X&amp;rft.au=Jun%2C+Y&amp;rft.au=Erickstad%2C+M&amp;rft.au=Brown%2C+S&amp;rft.au=Parks%2C+A&amp;rft.au=Court%2C+D&amp;rft.au=Jun%2C+S&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5171832&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid23460208-6"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid23460208_6-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFDicarloNorvilleMaliRios2013" class="citation journal cs1">Dicarlo, J. E.; Norville, J. E.; Mali, P; Rios, X; Aach, J; Church, G. M. (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3627607">"Genome engineering in Saccharomyces cerevisiae using CRISPR-Cas systems"</a>. <i>Nucleic Acids Research</i>. <b>41</b> (7): 4336&#8211;43. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1093%2Fnar%2Fgkt135">10.1093/nar/gkt135</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3627607">3627607</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/23460208">23460208</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Nucleic+Acids+Research&amp;rft.atitle=Genome+engineering+in+Saccharomyces+cerevisiae+using+CRISPR-Cas+systems&amp;rft.volume=41&amp;rft.issue=7&amp;rft.pages=4336-43&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3627607&amp;rft_id=info%3Apmid%2F23460208&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkt135&amp;rft.aulast=Dicarlo&amp;rft.aufirst=J.+E.&amp;rft.au=Norville%2C+J.+E.&amp;rft.au=Mali%2C+P&amp;rft.au=Rios%2C+X&amp;rft.au=Aach%2C+J&amp;rft.au=Church%2C+G.+M.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3627607&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid2408874-7"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid2408874_7-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFGratzO'Connor-Giles2013" class="citation journal cs1">Gratz, S. J.; O'Connor-Giles, K. M. (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3730909">"Genome engineering of Drosophila with the CRISPR RNA-guided Cas9 nuclease"</a>. <i>Genetics</i>. <b>194</b> (4): 1029&#8211;35. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1534%2Fgenetics.113.152710">10.1534/genetics.113.152710</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3730909">3730909</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/23709638">23709638</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Genetics&amp;rft.atitle=Genome+engineering+of+Drosophila+with+the+CRISPR+RNA-guided+Cas9+nuclease.&amp;rft.volume=194&amp;rft.issue=4&amp;rft.pages=1029-35&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3730909&amp;rft_id=info%3Apmid%2F23709638&amp;rft_id=info%3Adoi%2F10.1534%2Fgenetics.113.152710&amp;rft.aulast=Gratz&amp;rft.aufirst=S.+J.&amp;rft.au=O%27Connor-Giles%2C+K.+M.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3730909&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid23360964-8"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid23360964_8-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFHwangFuReyonMaeder2013" class="citation journal cs1">Hwang, W. Y.; Fu, Y; Reyon, D; Maeder, M. L.; Tsai, S. Q.; Sander, J. D.; Peterson, R. T.; Yeh, J. R.; Joung, J. K. (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3686313">"Efficient genome editing in zebrafish using a CRISPR-Cas system"</a>. <i>Nature Biotechnology</i>. <b>31</b> (3): 227&#8211;9. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1038%2Fnbt.2501">10.1038/nbt.2501</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3686313">3686313</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/23360964">23360964</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Nature+Biotechnology&amp;rft.atitle=Efficient+genome+editing+in+zebrafish+using+a+CRISPR-Cas+system&amp;rft.volume=31&amp;rft.issue=3&amp;rft.pages=227-9&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3686313&amp;rft_id=info%3Apmid%2F23360964&amp;rft_id=info%3Adoi%2F10.1038%2Fnbt.2501&amp;rft.aulast=Hwang&amp;rft.aufirst=W.+Y.&amp;rft.au=Fu%2C+Y&amp;rft.au=Reyon%2C+D&amp;rft.au=Maeder%2C+M.+L.&amp;rft.au=Tsai%2C+S.+Q.&amp;rft.au=Sander%2C+J.+D.&amp;rft.au=Peterson%2C+R.+T.&amp;rft.au=Yeh%2C+J.+R.&amp;rft.au=Joung%2C+J.+K.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3686313&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid23643243-9"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid23643243_9-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFWangYangShivalilaDawlaty2013" class="citation journal cs1">Wang, H.; Yang, H.; Shivalila, C. S.; Dawlaty, M. M.; Cheng, A. W.; Zhang, F.; Jaenisch, R. (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3969854">"One-Step Generation of Mice Carrying Mutations in Multiple Genes by CRISPR/Cas-Mediated Genome Engineering"</a>. <i>Cell</i>. <b>153</b> (4): 910&#8211;918. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1016%2Fj.cell.2013.04.025">10.1016/j.cell.2013.04.025</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3969854">3969854</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/23643243">23643243</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Cell&amp;rft.atitle=One-Step+Generation+of+Mice+Carrying+Mutations+in+Multiple+Genes+by+CRISPR%2FCas-Mediated+Genome+Engineering&amp;rft.volume=153&amp;rft.issue=4&amp;rft.pages=910-918&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3969854&amp;rft_id=info%3Apmid%2F23643243&amp;rft_id=info%3Adoi%2F10.1016%2Fj.cell.2013.04.025&amp;rft.aulast=Wang&amp;rft.aufirst=H.&amp;rft.au=Yang%2C+H.&amp;rft.au=Shivalila%2C+C.+S.&amp;rft.au=Dawlaty%2C+M.+M.&amp;rft.au=Cheng%2C+A.+W.&amp;rft.au=Zhang%2C+F.&amp;rft.au=Jaenisch%2C+R.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3969854&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid24136345-10"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid24136345_10-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid24136345_10-1"><sup><i><b>b</b></i></sup></a> <a href="#cite_ref-pmid24136345_10-2"><sup><i><b>c</b></i></sup></a> <a href="#cite_ref-pmid24136345_10-3"><sup><i><b>d</b></i></sup></a> <a href="#cite_ref-pmid24136345_10-4"><sup><i><b>e</b></i></sup></a> <a href="#cite_ref-pmid24136345_10-5"><sup><i><b>f</b></i></sup></a> <a href="#cite_ref-pmid24136345_10-6"><sup><i><b>g</b></i></sup></a></span> <span class="reference-text"><cite id="CITEREFLarsonGilbertWangLim2013" class="citation journal cs1">Larson, M. H.; Gilbert, L. A.; Wang, X; Lim, W. A.; Weissman, J. S.; Qi, L. S. (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3922765">"CRISPR interference (CRISPRi) for sequence-specific control of gene expression"</a>. <i>Nature Protocols</i>. <b>8</b> (11): 2180&#8211;96. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1038%2Fnprot.2013.132">10.1038/nprot.2013.132</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3922765">3922765</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/24136345">24136345</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Nature+Protocols&amp;rft.atitle=CRISPR+interference+%28CRISPRi%29+for+sequence-specific+control+of+gene+expression&amp;rft.volume=8&amp;rft.issue=11&amp;rft.pages=2180-96&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3922765&amp;rft_id=info%3Apmid%2F24136345&amp;rft_id=info%3Adoi%2F10.1038%2Fnprot.2013.132&amp;rft.aulast=Larson&amp;rft.aufirst=M.+H.&amp;rft.au=Gilbert%2C+L.+A.&amp;rft.au=Wang%2C+X&amp;rft.au=Lim%2C+W.+A.&amp;rft.au=Weissman%2C+J.+S.&amp;rft.au=Qi%2C+L.+S.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3922765&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid22745249-11"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid22745249_11-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFJinekChylinskiFonfaraHauer2012" class="citation journal cs1">Jinek, M.; Chylinski, K.; Fonfara, I.; Hauer, M.; Doudna, J. A.; Charpentier, E. (2012). "A Programmable Dual-RNA-Guided DNA Endonuclease in Adaptive Bacterial Immunity". <i>Science</i>. <b>337</b> (6096): 816&#8211;821. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1126%2Fscience.1225829">10.1126/science.1225829</a>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/22745249">22745249</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Science&amp;rft.atitle=A+Programmable+Dual-RNA-Guided+DNA+Endonuclease+in+Adaptive+Bacterial+Immunity&amp;rft.volume=337&amp;rft.issue=6096&amp;rft.pages=816-821&amp;rft.date=2012&amp;rft_id=info%3Adoi%2F10.1126%2Fscience.1225829&amp;rft_id=info%3Apmid%2F22745249&amp;rft.aulast=Jinek&amp;rft.aufirst=M.&amp;rft.au=Chylinski%2C+K.&amp;rft.au=Fonfara%2C+I.&amp;rft.au=Hauer%2C+M.&amp;rft.au=Doudna%2C+J.+A.&amp;rft.au=Charpentier%2C+E.&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid23849981-12"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid23849981_12-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid23849981_12-1"><sup><i><b>b</b></i></sup></a> <a href="#cite_ref-pmid23849981_12-2"><sup><i><b>c</b></i></sup></a> <a href="#cite_ref-pmid23849981_12-3"><sup><i><b>d</b></i></sup></a></span> <span class="error mw-ext-cite-error" lang="en" dir="ltr">Cite error: The named reference <code>pmid23849981</code> was invoked but never defined (see the <a href="/wiki/Help:Cite_errors/Cite_error_references_no_text" title="Help:Cite errors/Cite error references no text">help page</a>).
</span></li>
<li id="cite_note-13"><span class="mw-cite-backlink"><b><a href="#cite_ref-13">^</a></b></span> <span class="reference-text"><cite id="CITEREFRadzisheuskayaShlyuevaM&#252;ller2016" class="citation journal cs1">Radzisheuskaya, Aliaksandra; Shlyueva, Daria; M&#252;ller, Iris (28 June 2016). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC5062975">"Optimizing sgRNA position markedly improves the efficiency of CRISPR/dCas9-mediated transcriptional repression"</a>. <i>Nucleic Acids Research</i>. <b>44</b> (18): e141. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1093%2Fnar%2Fgkw583">10.1093/nar/gkw583</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC5062975">5062975</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/27353328">27353328</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Nucleic+Acids+Research&amp;rft.atitle=Optimizing+sgRNA+position+markedly+improves+the+efficiency+of+CRISPR%2FdCas9-mediated+transcriptional+repression&amp;rft.volume=44&amp;rft.issue=18&amp;rft.pages=e141&amp;rft.date=2016-06-28&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5062975&amp;rft_id=info%3Apmid%2F27353328&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkw583&amp;rft.aulast=Radzisheuskaya&amp;rft.aufirst=Aliaksandra&amp;rft.au=Shlyueva%2C+Daria&amp;rft.au=M%C3%BCller%2C+Iris&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5062975&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid24360272-14"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid24360272_14-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFChenGilbertCiminiSchnitzbauer2013" class="citation journal cs1">Chen, B; Gilbert, L. A.; Cimini, B. A.; Schnitzbauer, J; Zhang, W; Li, G. W.; Park, J; Blackburn, E. H.; Weissman, J. S.; Qi, L. S.; Huang, B (2013). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3918502">"Dynamic imaging of genomic loci in living human cells by an optimized CRISPR/Cas system"</a>. <i>Cell</i>. <b>155</b> (7): 1479&#8211;91. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1016%2Fj.cell.2013.12.001">10.1016/j.cell.2013.12.001</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3918502">3918502</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/24360272">24360272</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Cell&amp;rft.atitle=Dynamic+imaging+of+genomic+loci+in+living+human+cells+by+an+optimized+CRISPR%2FCas+system&amp;rft.volume=155&amp;rft.issue=7&amp;rft.pages=1479-91&amp;rft.date=2013&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3918502&amp;rft_id=info%3Apmid%2F24360272&amp;rft_id=info%3Adoi%2F10.1016%2Fj.cell.2013.12.001&amp;rft.aulast=Chen&amp;rft.aufirst=B&amp;rft.au=Gilbert%2C+L.+A.&amp;rft.au=Cimini%2C+B.+A.&amp;rft.au=Schnitzbauer%2C+J&amp;rft.au=Zhang%2C+W&amp;rft.au=Li%2C+G.+W.&amp;rft.au=Park%2C+J&amp;rft.au=Blackburn%2C+E.+H.&amp;rft.au=Weissman%2C+J.+S.&amp;rft.au=Qi%2C+L.+S.&amp;rft.au=Huang%2C+B&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3918502&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid24346702-15"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid24346702_15-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFKearnsGengaEnuamehGarber2014" class="citation journal cs1">Kearns, N. A.; Genga, R. M.; Enuameh, M. S.; Garber, M; Wolfe, S. A.; Maehr, R (2014). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3865759">"Cas9 effector-mediated regulation of transcription and differentiation in human pluripotent stem cells"</a>. <i>Development</i>. <b>141</b> (1): 219&#8211;23. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1242%2Fdev.103341">10.1242/dev.103341</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3865759">3865759</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/24346702">24346702</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Development&amp;rft.atitle=Cas9+effector-mediated+regulation+of+transcription+and+differentiation+in+human+pluripotent+stem+cells&amp;rft.volume=141&amp;rft.issue=1&amp;rft.pages=219-23&amp;rft.date=2014&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3865759&amp;rft_id=info%3Apmid%2F24346702&amp;rft_id=info%3Adoi%2F10.1242%2Fdev.103341&amp;rft.aulast=Kearns&amp;rft.aufirst=N.+A.&amp;rft.au=Genga%2C+R.+M.&amp;rft.au=Enuameh%2C+M.+S.&amp;rft.au=Garber%2C+M&amp;rft.au=Wolfe%2C+S.+A.&amp;rft.au=Maehr%2C+R&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3865759&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid24500196-16"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid24500196_16-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFHuLeiWongLiu2014" class="citation journal cs1">Hu, J; Lei, Y; Wong, W. K.; Liu, S; Lee, K. C.; He, X; You, W; Zhou, R; Guo, J. T.; Chen, X; Peng, X; Sun, H; Huang, H; Zhao, H; Feng, B (2014). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3985678">"Direct activation of human and mouse Oct4 genes using engineered TALE and Cas9 transcription factors"</a>. <i>Nucleic Acids Research</i>. <b>42</b> (7): 4375&#8211;90. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1093%2Fnar%2Fgku109">10.1093/nar/gku109</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3985678">3985678</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/24500196">24500196</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Nucleic+Acids+Research&amp;rft.atitle=Direct+activation+of+human+and+mouse+Oct4+genes+using+engineered+TALE+and+Cas9+transcription+factors&amp;rft.volume=42&amp;rft.issue=7&amp;rft.pages=4375-90&amp;rft.date=2014&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3985678&amp;rft_id=info%3Apmid%2F24500196&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgku109&amp;rft.aulast=Hu&amp;rft.aufirst=J&amp;rft.au=Lei%2C+Y&amp;rft.au=Wong%2C+W.+K.&amp;rft.au=Liu%2C+S&amp;rft.au=Lee%2C+K.+C.&amp;rft.au=He%2C+X&amp;rft.au=You%2C+W&amp;rft.au=Zhou%2C+R&amp;rft.au=Guo%2C+J.+T.&amp;rft.au=Chen%2C+X&amp;rft.au=Peng%2C+X&amp;rft.au=Sun%2C+H&amp;rft.au=Huang%2C+H&amp;rft.au=Zhao%2C+H&amp;rft.au=Feng%2C+B&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3985678&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid16904174-17"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid16904174_17-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFTakahashiYamanaka2006" class="citation journal cs1">Takahashi, K.; <a href="/wiki/Shinya_Yamanaka" title="Shinya Yamanaka">Yamanaka, S.</a> (2006). "Induction of Pluripotent Stem Cells from Mouse Embryonic and Adult Fibroblast Cultures by Defined Factors". <i>Cell</i>. <b>126</b> (4): 663&#8211;76. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1016%2Fj.cell.2006.07.024">10.1016/j.cell.2006.07.024</a>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/16904174">16904174</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Cell&amp;rft.atitle=Induction+of+Pluripotent+Stem+Cells+from+Mouse+Embryonic+and+Adult+Fibroblast+Cultures+by+Defined+Factors&amp;rft.volume=126&amp;rft.issue=4&amp;rft.pages=663-76&amp;rft.date=2006&amp;rft_id=info%3Adoi%2F10.1016%2Fj.cell.2006.07.024&amp;rft_id=info%3Apmid%2F16904174&amp;rft.aulast=Takahashi&amp;rft.aufirst=K.&amp;rft.au=Yamanaka%2C+S.&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-pmid25307933-18"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid25307933_18-0">^</a></b></span> <span class="error mw-ext-cite-error" lang="en" dir="ltr">Cite error: The named reference <code>pmid25307933</code> was invoked but never defined (see the <a href="/wiki/Help:Cite_errors/Cite_error_references_no_text" title="Help:Cite errors/Cite error references no text">help page</a>).
</span></li>
<li id="cite_note-pmid25409531-19"><span class="mw-cite-backlink"><b><a href="#cite_ref-pmid25409531_19-0">^</a></b></span> <span class="reference-text"><cite id="CITEREFJiLeeWongDadlani2014" class="citation journal cs1">Ji, W; Lee, D; Wong, E; Dadlani, P; Dinh, D; Huang, V; Kearns, K; Teng, S; Chen, S; Haliburton, J; Heimberg, G; Heineike, B; Ramasubramanian, A; Stevens, T; Helmke, K. J.; Zepeda, V; Qi, L. S.; Lim, W. A. (2014). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC4277763">"Specific gene repression by CRISPRi system transferred through bacterial conjugation"</a>. <i>ACS Synthetic Biology</i>. <b>3</b> (12): 929&#8211;31. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1021%2Fsb500036q">10.1021/sb500036q</a>. <a href="/wiki/PMC_(identifier)" class="mw-redirect" title="PMC (identifier)">PMC</a>&#160;<span class="cs1-lock-free" title="Freely accessible"><a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC4277763">4277763</a></span>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/25409531">25409531</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=ACS+Synthetic+Biology&amp;rft.atitle=Specific+gene+repression+by+CRISPRi+system+transferred+through+bacterial+conjugation&amp;rft.volume=3&amp;rft.issue=12&amp;rft.pages=929-31&amp;rft.date=2014&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4277763&amp;rft_id=info%3Apmid%2F25409531&amp;rft_id=info%3Adoi%2F10.1021%2Fsb500036q&amp;rft.aulast=Ji&amp;rft.aufirst=W&amp;rft.au=Lee%2C+D&amp;rft.au=Wong%2C+E&amp;rft.au=Dadlani%2C+P&amp;rft.au=Dinh%2C+D&amp;rft.au=Huang%2C+V&amp;rft.au=Kearns%2C+K&amp;rft.au=Teng%2C+S&amp;rft.au=Chen%2C+S&amp;rft.au=Haliburton%2C+J&amp;rft.au=Heimberg%2C+G&amp;rft.au=Heineike%2C+B&amp;rft.au=Ramasubramanian%2C+A&amp;rft.au=Stevens%2C+T&amp;rft.au=Helmke%2C+K.+J.&amp;rft.au=Zepeda%2C+V&amp;rft.au=Qi%2C+L.+S.&amp;rft.au=Lim%2C+W.+A.&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4277763&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
<li id="cite_note-20"><span class="mw-cite-backlink"><b><a href="#cite_ref-20">^</a></b></span> <span class="reference-text"><cite id="CITEREFGoyalMyachevaGro&#223;Klingenberg2016" class="citation journal cs1">Goyal, Ashish; Myacheva, Ksenia; Gro&#223;, Matthias; Klingenberg, Marcel; Duran Arqu&#233;, Berta; Diederichs, Sven (2016-09-30). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/content/early/2016/09/30/nar.gkw883">"Challenges of CRISPR/Cas9 applications for long non-coding RNA genes"</a>. <i>Nucleic Acids Research</i>: gkw883. <a href="/wiki/Doi_(identifier)" class="mw-redirect" title="Doi (identifier)">doi</a>:<a rel="nofollow" class="external text" href="https://doi.org/10.1093%2Fnar%2Fgkw883">10.1093/nar/gkw883</a>. <a href="/wiki/ISSN_(identifier)" class="mw-redirect" title="ISSN (identifier)">ISSN</a>&#160;<a rel="nofollow" class="external text" href="//www.worldcat.org/issn/0305-1048">0305-1048</a>. <a href="/wiki/PMID_(identifier)" class="mw-redirect" title="PMID (identifier)">PMID</a>&#160;<a rel="nofollow" class="external text" href="//pubmed.ncbi.nlm.nih.gov/27694625">27694625</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.genre=article&amp;rft.jtitle=Nucleic+Acids+Research&amp;rft.atitle=Challenges+of+CRISPR%2FCas9+applications+for+long+non-coding+RNA+genes&amp;rft.pages=gkw883&amp;rft.date=2016-09-30&amp;rft.issn=0305-1048&amp;rft_id=info%3Apmid%2F27694625&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkw883&amp;rft.aulast=Goyal&amp;rft.aufirst=Ashish&amp;rft.au=Myacheva%2C+Ksenia&amp;rft.au=Gro%C3%9F%2C+Matthias&amp;rft.au=Klingenberg%2C+Marcel&amp;rft.au=Duran+Arqu%C3%A9%2C+Berta&amp;rft.au=Diederichs%2C+Sven&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcontent%2Fearly%2F2016%2F09%2F30%2Fnar.gkw883&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ACRISPR+interference" class="Z3988"></span><link rel="mw-deduplicated-inline-style" href="mw-data:TemplateStyles:r982806391"></span>
</li>
</ol></div>
<h2><span class="mw-headline" id="External_links">External links</span></h2>


</div>
<div id="mw-normal-catlinks" class="mw-normal-catlinks"><a href="/wiki/Help:Category" title="Help:Category">Categories</a>: <ul><li><a href="/wiki/Category:Repetitive_DNA_sequences" title="Category:Repetitive DNA sequences">Repetitive DNA sequences</a></li><li><a href="/wiki/Category:Non-coding_RNA" title="Category:Non-coding RNA">Non-coding RNA</a></li></ul></div>