{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Copy of arno_experiments.ipynb","provenance":[{"file_id":"1IK-QF90VpGAZmfH0XqDrD8doJfuC6B92","timestamp":1610636648085}],"collapsed_sections":[],"authorship_tag":"ABX9TyNv/4Yxv01qke7oe210YgPE"},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"n4eNz7RIrzow"},"source":["## Set globals"]},{"cell_type":"code","metadata":{"id":"PJwyfS7eiPWw","colab":{"base_uri":"https://localhost:8080/","height":227},"executionInfo":{"status":"ok","timestamp":1610625274125,"user_tz":-60,"elapsed":33138,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"0cd1888e-1207-442c-fcc5-281fa4e7786e"},"source":["# set paths\n","path_extractions = '/content/drive/My Drive/WIKING/extractions'\n","path_git = '/content/drive/My Drive/WIKING/WIKING_git'\n","path_my_tools = '/content/drive/My Drive/my_tools'\n","\n","# infiles\n","file_CRISPR_de = path_extractions + '/CRISPR_de' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/\n","file_CRISPR_en = path_extractions + '/CRISPR_en' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/\n","file_CRISPR_gene_editing_en = path_extractions + '/CRISPR_gene_editing_en' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/ \n","file_events = path_git + '/data/CRISPR_events - events.csv'\n","file_accounts = path_git + '/data/CRISPR_events - accounts.csv'\n","\n","# outfiles\n","file_contexts_CRISPR_en = path_extractions + '/contexts_CRISPR_en.xlsx'\n","file_contexts_CRISPR_gene_editing_en = path_extractions + '/contexts_CRISPR_gene_editing_en.xlsx'\n","\n","# mount Google Drive\n","from google.colab import drive\n","drive.mount('/content/drive')\n","\n","# add git folder to path variable\n","import sys\n","sys.path.append(path_git + '/code')\n","\n","# import Wolfgang's classes:\n","from article.article import Article\n","\n","# import other modules\n","!pip install fuzzywuzzy # https://github.com/seatgeek/fuzzywuzzy\n","from fuzzywuzzy import fuzz\n","from fuzzywuzzy import process\n","import networkx as nx\n","!pip install python-louvain\n","from community import community_louvain\n","import re\n","import os\n","import json\n","import pandas as pd\n","import numpy as np\n","import matplotlib.pyplot as plt\n","import itertools\n","import pprint\n","pp = pprint.PrettyPrinter(indent=4)\n","from lxml import etree\n","\n","# set working directory\n","os.chdir(path_git)\n","os.getcwd()"],"execution_count":1,"outputs":[{"output_type":"stream","text":["Mounted at /content/drive\n","Collecting fuzzywuzzy\n","  Downloading https://files.pythonhosted.org/packages/43/ff/74f23998ad2f93b945c0309f825be92e04e0348e062026998b5eefef4c33/fuzzywuzzy-0.18.0-py2.py3-none-any.whl\n","Installing collected packages: fuzzywuzzy\n","Successfully installed fuzzywuzzy-0.18.0\n"],"name":"stdout"},{"output_type":"stream","text":["/usr/local/lib/python3.6/dist-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning\n","  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')\n"],"name":"stderr"},{"output_type":"stream","text":["Requirement already satisfied: python-louvain in /usr/local/lib/python3.6/dist-packages (0.15)\n","Requirement already satisfied: networkx in /usr/local/lib/python3.6/dist-packages (from python-louvain) (2.5)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages (from python-louvain) (1.19.5)\n","Requirement already satisfied: decorator>=4.3.0 in /usr/local/lib/python3.6/dist-packages (from networkx->python-louvain) (4.4.2)\n"],"name":"stdout"},{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'/content/drive/My Drive/WIKING/WIKING_git'"]},"metadata":{"tags":[]},"execution_count":1}]},{"cell_type":"markdown","metadata":{"id":"BG3rjcwdsAvH"},"source":["## Functions"]},{"cell_type":"code","metadata":{"id":"VA1WhsvGiqjB","executionInfo":{"status":"ok","timestamp":1610625274842,"user_tz":-60,"elapsed":712,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}}},"source":["def find_all(text, keyphrase): # https://stackoverflow.com/questions/4664850/how-to-find-all-occurrences-of-a-substring\n","  ''' Returns indices for all matches of keyphrase '''\n","  start = 0\n","  while True:\n","    start = text.find(keyphrase, start)\n","    if start == -1: \n","      return\n","    yield start\n","    start += len(keyphrase) # use start += 1 to find overlapping matches"],"execution_count":2,"outputs":[]},{"cell_type":"code","metadata":{"id":"eLjQwf0rnLmi","executionInfo":{"status":"ok","timestamp":1610625275350,"user_tz":-60,"elapsed":1219,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}}},"source":["def keyphrase_in_context(keyphrase, article, width=50, score_cutoff=75, scorer=fuzz.ratio):\n","  ''' \n","  Returns df for contexts to the left and to the right of keyphrase\n","  '''\n","  gen = article.yield_revisions()\n","  contexts = []\n","  for revision in gen:\n","    text = revision.get_text()\n","    text = text.lower()\n","    if keyphrase in text:\n","      for indx in find_all(text, keyphrase):\n","        # context left\n","        left = ''\n","        for char in text[indx - width : indx][::-1]:\n","          if char == '\\n': # make '\\n' the boundary for context\n","            break\n","          left += char\n","        left = left[::-1].strip()\n","        # context right\n","        right = ''\n","        for char in text[indx + len(keyphrase): indx + len(keyphrase) + width]:\n","          if char == '\\n': # make '\\n' the boundary for context\n","            break\n","          right += char\n","        right = right.strip()\n","        contexts.append((revision.index, revision.timestamp.string, 0, left, keyphrase, right, 0, revision.revid, revision.url, revision.user, revision.userid, revision.comment,))\n","  df = pd.DataFrame(contexts, columns=['revindx','timestamp','left_id','left','keyphrase','right','right_id','revid','url','user','userid','comment',])\n","  \n","  # determine communities of similarity for left and right contexts\n","  for side in ['left', 'right']:\n","    G = nx.Graph()\n","    unique_contexts = list(df[side].unique())\n","    length = len(unique_contexts)\n","    for query in unique_contexts:\n","      indx = unique_contexts.index(query) + 1\n","      if indx <= length:\n","        scores = process.extractBests(query, unique_contexts[unique_contexts.index(query) + 1:], score_cutoff=score_cutoff, limit=length, scorer=scorer) # https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/process.py\n","        edges = [(query, i[0]) for i in scores]\n","        G.add_edges_from(edges)\n","    # add community_ids for lonesome contexts as well\n","    unique_contexts = set(unique_contexts)\n","    community_dic = community_louvain.best_partition(G)\n","    if community_dic:\n","      community_contexts = set(community_dic.keys())\n","      lonesome_contexts = unique_contexts - community_contexts\n","      lonesome_dic = {k:v for v,k in enumerate(lonesome_contexts,start=max(community_dic.values())+1)}\n","      community_dic.update(lonesome_dic) # hiermit nun 'lid' und 'rid' Spalten befüllen...\n","    else:\n","      community_dic = {k:v for v,k in enumerate(unique_contexts)}\n","    # add community_ids to df\n","    column = df.apply(lambda row: community_dic[row[side]], axis=1)\n","    df['{}_id'.format(side)] = column\n","  return df\n","  # # add community_ids for lonesome contexts as well\n","  # unique_contexts = set(unique_contexts['left'] + unique_contexts['right'])\n","  # community_dic = community_louvain.best_partition(G)\n","  # community_contexts = set(community_dic.keys())\n","  # lonesome_contexts = unique_contexts - community_contexts\n","  # lonesome_dic = {k:v for v,k in enumerate(lonesome_contexts,start=max(community_dic.values())+1)}\n","  # community_dic.update(lonesome_dic) # hiermit nun 'lid' und 'rid' Spalten befüllen...\n","  # # add community_ids for all left and right contexts in df\n","  # for side in ['left', 'right']:\n","  #   column = df.apply(lambda row: community_dic[row[side]], axis=1)\n","  #   df['{}_id'.format(side)] = column\n","  # return df"],"execution_count":3,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"aodKMqgBxmb8"},"source":["## Load data"]},{"cell_type":"code","metadata":{"id":"gD4IDC16epDS","executionInfo":{"status":"ok","timestamp":1610625275351,"user_tz":-60,"elapsed":1219,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}}},"source":["CRISPR_en = Article(file_CRISPR_en)\n","CRISPR_gene_editing_en = Article(file_CRISPR_gene_editing_en)\n","# events = pd.read_csv(file_events)\n","# accounts = pd.read_csv(file_accounts)"],"execution_count":4,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"kr9rzPh5zNIA"},"source":["## Experiments"]},{"cell_type":"code","metadata":{"id":"vpnxPlxrzPbt","executionInfo":{"status":"ok","timestamp":1610629088859,"user_tz":-60,"elapsed":927,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}}},"source":["article = CRISPR_en\n","gen = article.yield_revisions()\n","for revision in gen:\n","  # print(revision.revid, type(revision.revid))\n","  if revision.revid == 603962693:\n","    text1 = revision.get_text().lower()\n","    text2 = revision.get_wikitext()\n","    text3 = revision.etree_from_html()\n","    # pp.pprint(text)\n","    break"],"execution_count":12,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"_l9TB3a-Uu8o"},"source":["## Citation context"]},{"cell_type":"markdown","metadata":{"id":"F661L0u7UxeY"},"source":["### Find all intext citations"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"c-sdy2RPOINS","executionInfo":{"status":"ok","timestamp":1610636550005,"user_tz":-60,"elapsed":1050,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"36651491-44d6-4306-e145-2de0d5ccf9bb"},"source":["# intext citations in Wikipedia html are sup-tags of class='refreence'\n","for i in text3.xpath(\".//sup[@class='reference']\"):\n","  print(etree.tostring(i))\n","  # supid = i.get('id')\n","  # print('\\t', supid)\n","  for a in i.xpath(\".//a\"): # Use \".\" to search relatively. OTherwise it searches the whole tree\n","    # print(etree.tostring(a))\n","    refid = a.get('href')\n","    print('\\t',refid)\n","    refnum = a.xpath(\".//text()\")\n","    if len(refnum) == 1:\n","      print('\\t',refnum[0])\n","    else:\n","      print('\\tMore than one refnum:',refnum[0])\n"],"execution_count":79,"outputs":[{"output_type":"stream","text":["b'<sup id=\"cite_ref-pmid20056882_1-0\" class=\"reference\"><a href=\"#cite_note-pmid20056882-1\">[1]</a></sup>'\n","\t #cite_note-pmid20056882-1\n","\t [1]\n","b'<sup id=\"cite_ref-pmid20125085_2-0\" class=\"reference\"><a href=\"#cite_note-pmid20125085-2\">[2]</a></sup>\\n'\n","\t #cite_note-pmid20125085-2\n","\t [2]\n","b'<sup id=\"cite_ref-3\" class=\"reference\"><a href=\"#cite_note-3\">[3]</a></sup>'\n","\t #cite_note-3\n","\t [3]\n","b'<sup id=\"cite_ref-pmid17521438_4-0\" class=\"reference\"><a href=\"#cite_note-pmid17521438-4\">[4]</a></sup>\\n'\n","\t #cite_note-pmid17521438-4\n","\t [4]\n","b'<sup id=\"cite_ref-pmid17379808_5-0\" class=\"reference\"><a href=\"#cite_note-pmid17379808-5\">[5]</a></sup>'\n","\t #cite_note-pmid17379808-5\n","\t [5]\n","b'<sup id=\"cite_ref-pmid19095942_6-0\" class=\"reference\"><a href=\"#cite_note-pmid19095942-6\">[6]</a></sup> and provides a form of '\n","\t #cite_note-pmid19095942-6\n","\t [6]\n","b'<sup id=\"cite_ref-pmid20125085_2-1\" class=\"reference\"><a href=\"#cite_note-pmid20125085-2\">[2]</a></sup>\\n'\n","\t #cite_note-pmid20125085-2\n","\t [2]\n","b'<sup id=\"cite_ref-7\" class=\"reference\"><a href=\"#cite_note-7\">[7]</a></sup> By inserting a '\n","\t #cite_note-7\n","\t [7]\n","b'<sup id=\"cite_ref-8\" class=\"reference\"><a href=\"#cite_note-8\">[8]</a></sup>\\n'\n","\t #cite_note-8\n","\t [8]\n","b'<sup id=\"cite_ref-pmid3316184_9-0\" class=\"reference\"><a href=\"#cite_note-pmid3316184-9\">[9]</a></sup> In 2000, similar clustered repeats were identified in additional bacteria and archaea and were termed Short Regularly Spaced Repeats (SRSR).'\n","\t #cite_note-pmid3316184-9\n","\t [9]\n","b'<sup id=\"cite_ref-pmid10760181_10-0\" class=\"reference\"><a href=\"#cite_note-pmid10760181-10\">[10]</a></sup> SRSR were renamed CRISPR in 2002.'\n","\t #cite_note-pmid10760181-10\n","\t [10]\n","b'<sup id=\"cite_ref-pmid11952905_11-0\" class=\"reference\"><a href=\"#cite_note-pmid11952905-11\">[11]</a></sup> A set of genes, some encoding putative '\n","\t #cite_note-pmid11952905-11\n","\t [11]\n","b'<sup id=\"cite_ref-pmid11952905_11-1\" class=\"reference\"><a href=\"#cite_note-pmid11952905-11\">[11]</a></sup>\\n'\n","\t #cite_note-pmid11952905-11\n","\t [11]\n","b'<sup id=\"cite_ref-pmid20056882_1-1\" class=\"reference\"><a href=\"#cite_note-pmid20056882-1\">[1]</a></sup>'\n","\t #cite_note-pmid20056882-1\n","\t [1]\n","b'<sup id=\"cite_ref-pmid20125085_2-2\" class=\"reference\"><a href=\"#cite_note-pmid20125085-2\">[2]</a></sup>'\n","\t #cite_note-pmid20125085-2\n","\t [2]\n","b'<sup id=\"cite_ref-pmid20056882_1-2\" class=\"reference\"><a href=\"#cite_note-pmid20056882-1\">[1]</a></sup> Koonin and colleagues proposed that spacers serve as a template for RNA molecules, analogously to eukaryotic cells that use a system called '\n","\t #cite_note-pmid20056882-1\n","\t [1]\n","b'<sup id=\"cite_ref-craze_12-0\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-craze_12-1\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-craze_12-2\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-craze_12-3\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-13\" class=\"reference\"><a href=\"#cite_note-13\">[13]</a></sup> It has since been used in a wide range of organisms including bakers yeast (S. cerevisiae),'\n","\t #cite_note-13\n","\t [13]\n","b'<sup id=\"cite_ref-14\" class=\"reference\"><a href=\"#cite_note-14\">[14]</a></sup> zebra fish,'\n","\t #cite_note-14\n","\t [14]\n","b'<sup id=\"cite_ref-15\" class=\"reference\"><a href=\"#cite_note-15\">[15]</a></sup> nematodes (C. elegans),'\n","\t #cite_note-15\n","\t [15]\n","b'<sup id=\"cite_ref-16\" class=\"reference\"><a href=\"#cite_note-16\">[16]</a></sup> plants,'\n","\t #cite_note-16\n","\t [16]\n","b'<sup id=\"cite_ref-17\" class=\"reference\"><a href=\"#cite_note-17\">[17]</a></sup> mice,'\n","\t #cite_note-17\n","\t [17]\n","b'<sup id=\"cite_ref-pmid23643243_18-0\" class=\"reference\"><a href=\"#cite_note-pmid23643243-18\">[18]</a></sup> and several other organisms.\\n'\n","\t #cite_note-pmid23643243-18\n","\t [18]\n","b'<sup id=\"cite_ref-19\" class=\"reference\"><a href=\"#cite_note-19\">[19]</a></sup>\\n'\n","\t #cite_note-19\n","\t [19]\n","b'<sup id=\"cite_ref-craze_12-4\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-20\" class=\"reference\"><a href=\"#cite_note-20\">[20]</a></sup>\\n'\n","\t #cite_note-20\n","\t [20]\n","b'<sup id=\"cite_ref-pmid16292354_21-0\" class=\"reference\"><a href=\"#cite_note-pmid16292354-21\">[21]</a></sup> They usually show some '\n","\t #cite_note-pmid16292354-21\n","\t [21]\n","b'<sup id=\"cite_ref-pmid17442114_22-0\" class=\"reference\"><a href=\"#cite_note-pmid17442114-22\">[22]</a></sup> Repeats are separated by spacers of similar length.'\n","\t #cite_note-pmid17442114-22\n","\t [22]\n","b'<sup id=\"cite_ref-pmid16292354_21-1\" class=\"reference\"><a href=\"#cite_note-pmid16292354-21\">[21]</a></sup> Some CRISPR spacer sequences exactly match sequences from '\n","\t #cite_note-pmid16292354-21\n","\t [21]\n","b'<sup id=\"cite_ref-pmid15791728_23-0\" class=\"reference\"><a href=\"#cite_note-pmid15791728-23\">[23]</a></sup>'\n","\t #cite_note-pmid15791728-23\n","\t [23]\n","b'<sup id=\"cite_ref-pmid16079334_24-0\" class=\"reference\"><a href=\"#cite_note-pmid16079334-24\">[24]</a></sup>'\n","\t #cite_note-pmid16079334-24\n","\t [24]\n","b'<sup id=\"cite_ref-25\" class=\"reference\"><a href=\"#cite_note-25\">[25]</a></sup> although some spacers match the prokaryote\\'s genome (self-targeting spacers).'\n","\t #cite_note-25\n","\t [25]\n","b'<sup id=\"cite_ref-pmid20598393_26-0\" class=\"reference\"><a href=\"#cite_note-pmid20598393-26\">[26]</a></sup> New spacers can be added rapidly in response to phage infection.'\n","\t #cite_note-pmid20598393-26\n","\t [26]\n","b'<sup id=\"cite_ref-27\" class=\"reference\"><a href=\"#cite_note-27\">[27]</a></sup>\\n'\n","\t #cite_note-27\n","\t [27]\n","b'<sup id=\"cite_ref-pmid16292354_21-2\" class=\"reference\"><a href=\"#cite_note-pmid16292354-21\">[21]</a></sup> Of these protein families, Cas1 appears to be ubiquitous among different CRISPR/Cas systems. Particular combinations of '\n","\t #cite_note-pmid16292354-21\n","\t [21]\n","b'<sup id=\"cite_ref-pmid16292354_21-3\" class=\"reference\"><a href=\"#cite_note-pmid16292354-21\">[21]</a></sup> More than one CRISPR subtype may occur in a single genome. The sporadic distribution of the CRISPR/Cas subtypes suggests that the system is subject to '\n","\t #cite_note-pmid16292354-21\n","\t [21]\n","b'<sup id=\"cite_ref-pmid20056882_1-3\" class=\"reference\"><a href=\"#cite_note-pmid20056882-1\">[1]</a></sup>'\n","\t #cite_note-pmid20056882-1\n","\t [1]\n","b'<sup id=\"cite_ref-pmid16545108_28-0\" class=\"reference\"><a href=\"#cite_note-pmid16545108-28\">[28]</a></sup> Evidence suggests functional diversity among CRISPR subtypes. The Cse (Cas subtype '\n","\t #cite_note-pmid16545108-28\n","\t [28]\n","b'<sup id=\"cite_ref-pmid18703739_29-0\" class=\"reference\"><a href=\"#cite_note-pmid18703739-29\">[29]</a></sup> In other prokaryotes, Cas6 processes the CRISPR transcripts. Interestingly, CRISPR-based phage inactivation in '\n","\t #cite_note-pmid18703739-29\n","\t [29]\n","b'<sup id=\"cite_ref-pmid17442114_22-1\" class=\"reference\"><a href=\"#cite_note-pmid17442114-22\">[22]</a></sup>\\n'\n","\t #cite_note-pmid17442114-22\n","\t [22]\n","b'<sup id=\"cite_ref-pmid19906303_30-0\" class=\"reference\"><a href=\"#cite_note-pmid19906303-30\">[30]</a></sup> Others investigated the '\n","\t #cite_note-pmid19906303-30\n","\t [30]\n","b'<sup id=\"cite_ref-31\" class=\"reference\"><a href=\"#cite_note-31\">[31]</a></sup>\\n'\n","\t #cite_note-31\n","\t [31]\n","b'<sup id=\"cite_ref-32\" class=\"reference\"><a href=\"#cite_note-32\">[32]</a></sup>\\n'\n","\t #cite_note-32\n","\t [32]\n","b'<sup id=\"cite_ref-halemajumder2012_33-0\" class=\"reference\"><a href=\"#cite_note-halemajumder2012-33\">[33]</a></sup> provided a first step toward realization of proposals for CRISPR-derived biotechnology:'\n","\t #cite_note-halemajumder2012-33\n","\t [33]\n","b'<sup id=\"cite_ref-pmid18157154_34-0\" class=\"reference\"><a href=\"#cite_note-pmid18157154-34\">[34]</a></sup>\\n'\n","\t #cite_note-pmid18157154-34\n","\t [34]\n","b'<sup id=\"cite_ref-35\" class=\"reference\"><a href=\"#cite_note-35\">[35]</a></sup> and '\n","\t #cite_note-35\n","\t [35]\n","b'<sup id=\"cite_ref-pmid23643243_18-1\" class=\"reference\"><a href=\"#cite_note-pmid23643243-18\">[18]</a></sup>'\n","\t #cite_note-pmid23643243-18\n","\t [18]\n","b'<sup id=\"cite_ref-36\" class=\"reference\"><a href=\"#cite_note-36\">[36]</a></sup>'\n","\t #cite_note-36\n","\t [36]\n","b'<sup id=\"cite_ref-37\" class=\"reference\"><a href=\"#cite_note-37\">[37]</a></sup>'\n","\t #cite_note-37\n","\t [37]\n","b'<sup id=\"cite_ref-pmid20634809_38-0\" class=\"reference\"><a href=\"#cite_note-pmid20634809-38\">[38]</a></sup> The mechanisms that induce PICI excision and how PICI inhibits phage replication are not well understood. One study showed that lytic ICP1 phage that specifically targets'\n","\t #cite_note-pmid20634809-38\n","\t [38]\n","b'<sup id=\"cite_ref-pmid23446421_39-0\" class=\"reference\"><a href=\"#cite_note-pmid23446421-39\">[39]</a></sup>\\n'\n","\t #cite_note-pmid23446421-39\n","\t [39]\n","b'<sup id=\"cite_ref-craze_12-5\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-craze_12-6\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-craze_12-7\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-craze_12-8\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-craze_12-9\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-tr1311_40-0\" class=\"reference\"><a href=\"#cite_note-tr1311-40\">[40]</a></sup>\\n'\n","\t #cite_note-tr1311-40\n","\t [40]\n","b'<sup id=\"cite_ref-41\" class=\"reference\"><a href=\"#cite_note-41\">[41]</a></sup>\\n'\n","\t #cite_note-41\n","\t [41]\n","b'<sup id=\"cite_ref-tr1311_40-1\" class=\"reference\"><a href=\"#cite_note-tr1311-40\">[40]</a></sup>\\n'\n","\t #cite_note-tr1311-40\n","\t [40]\n","b'<sup id=\"cite_ref-craze_12-10\" class=\"reference\"><a href=\"#cite_note-craze-12\">[12]</a></sup>\\n'\n","\t #cite_note-craze-12\n","\t [12]\n","b'<sup id=\"cite_ref-MIT_42-0\" class=\"reference\"><a href=\"#cite_note-MIT-42\">[42]</a></sup> and can target more genes than these earlier techniques.'\n","\t #cite_note-MIT-42\n","\t [42]\n","b'<sup id=\"cite_ref-MIT_42-1\" class=\"reference\"><a href=\"#cite_note-MIT-42\">[42]</a></sup>\\n'\n","\t #cite_note-MIT-42\n","\t [42]\n","b'<sup id=\"cite_ref-43\" class=\"reference\"><a href=\"#cite_note-43\">[43]</a></sup> The patent rights have been assigned by the inventors to the Regents of the University of California and to the University of Vienna.\\n'\n","\t #cite_note-43\n","\t [43]\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":105},"id":"qteACwSlNMpW","executionInfo":{"status":"ok","timestamp":1610633002851,"user_tz":-60,"elapsed":815,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"badabbf4-5613-4fc6-d809-9c03fc9c48fb"},"source":["\"\".join(text3.xpath(\".//div[@class='mw-parser-output']\")[0].xpath(\"//text()\")).strip()"],"execution_count":31,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'Diagram of the possible mechanism for CRISPR.[1]\\nCRISPRs (clustered regularly interspaced short palindromic repeats) are DNA loci containing short repetitions of base sequences. Each repetition is followed by short segments of \"spacer DNA\" from previous exposures to a virus.[2]\\nCRISPRs are found in approximately 40% of sequenced eubacteria genomes and 90% of sequenced archaea.[3][4]\\nCRISPRs are often associated with cas genes that code for proteins related to CRISPRs. The CRISPR-Cas system is a prokaryotic immune system that confers resistance to foreign genetic elements such as plasmids and phages[5][6] and provides a form of acquired immunity. CRISPR spacers recognize and silence these exogenous genetic elements like RNAi in eukaryotic organisms.[2]\\nSince 2012, the CRISPR-Cas system has been used for gene editing (silencing, enhancing or changing specific genes) that even works in eukaryotes like mice and primates.[7] By inserting a plasmid containing cas genes and specifically designed CRISPRs, an organism\\'s genome can be cut at any desired location.\\n\\nContents\\n\\n1 History\\n2 Locus structure\\n\\n2.1 Repeats and spacers\\n2.2 Cas genes and CRISPR subtypes\\n\\n\\n3 Mechanism\\n4 Evolutionary significance\\n5 Applications\\n\\n5.1 Mouse models\\n\\n\\n6 CRISPR/Cas system in phage\\n7 Multi-gene targeting\\n8 Reversible knockdown\\n9 Gene activation\\n10 Infrastructure\\n11 Therapeutics\\n12 Safety\\n13 Alternatives\\n14 Patent\\n15 References\\n16 Further reading\\n17 External links\\n\\n\\n\\nHistory\\nBacteria may incorporate foreign DNA in other circumstances and even to scavenge damaged DNA from their environment.[8]\\nRepeats were first described in 1987 for the bacterium Escherichia coli.[9] In 2000, similar clustered repeats were identified in additional bacteria and archaea and were termed Short Regularly Spaced Repeats (SRSR).[10] SRSR were renamed CRISPR in 2002.[11] A set of genes, some encoding putative nuclease or helicase proteins, were found to be associated with CRISPR repeats (the cas, or CRISPR-associated genes).[11]\\n\\n  Simplified diagram of a CRISPR locus. The three major components of a CRISPR locus are shown: cas genes, a leader sequence, and a repeat-spacer array. Repeats are shown as grey boxes and spacers are colored bars. While most CRISPR loci contain each of the three components, the arrangement is not always as shown.[1][2]\\nIn 2005, three independent researchers showed that CRISPR spacers showed homology to several phage DNA and extrachromosomal DNA such as plasmids. This was an indication that the CRISPR/cas system could have a role  in adaptive immunity in bacteria.[1] Koonin and colleagues proposed that spacers serve as a template for RNA molecules, analogously to eukaryotic cells that use a system called RNA interference.[12]\\nIn 2007 Barrangou, Horvath (food industry scientists at Danisco) and others showed that they could alter the resistance of Streptococcus thermophilus to phage attack with spacer DNA.[12]\\nDoudna and Charpentier had independently been exploring CRISPR-associated proteins to learn how bacteria deploy spacers in their immune defenses. They jointly studied a simpler CRISPR system that relies on a protein called Cas9. They found that bacteria respond to an invading phage by transcribing spacers and palindromic DNA into a long RNA molecule that the cell then uses tracrRNA and Cas9 to cut it into pieces called crRNAs.[12]\\nCas9 is a nuclease, an enzyme specialized for cutting DNA, with two active cutting sites, one for each strand of the double helix. The team demonstrated that they could disable one or both sites while preserving Cas9\\'s ability to home located its target DNA. Jinek combined tracrRNA and spacer RNA into a \"single-guide RNA\" molecule that, mixed with Cas9, could find and cut the correct DNA targets. Jinek et al proposed that such synthetic guide RNAs might be able to be used for gene editing.[12]\\nCRISPR was first shown to work as a genome engineering/editing tool in human cell culture by Jennifer Doudna in 2012 (PMID 22745249) and George M. Church\\'s group at Harvard University.[13] It has since been used in a wide range of organisms including bakers yeast (S. cerevisiae),[14] zebra fish,[15] nematodes (C. elegans),[16] plants,[17] mice,[18] and several other organisms.\\nAdditionally CRISPR has been modified to make programmable transcription factors that allow scientists to target and activate or silence specific genes.[19]\\nLibraries of tens of thousands of guide RNAs are now available.[12]\\nThe first evidence that CRISPR can reverse disease symptoms in living animals was demonstrated in March 2014, when MIT researchers cured mice of a rare liver disorder.[20]\\n\\nLocus structure\\nRepeats and spacers\\nCRISPR repeats range in size from 24 to 48 base pairs.[21] They usually show some dyad symmetry, implying the formation of a secondary structure such as a hairpin, but are not truly palindromic.[22] Repeats are separated by spacers of similar length.[21] Some CRISPR spacer sequences exactly match sequences from plasmids and phages,[23][24][25] although some spacers match the prokaryote\\'s genome (self-targeting spacers).[26] New spacers can be added rapidly in response to phage infection.[27]\\n\\nCas genes and CRISPR subtypes\\nCRISPR-associated (cas) genes are often associated with CRISPR repeat-spacer arrays. As of 2013, more than forty different Cas protein families had been described.[21] Of these protein families, Cas1 appears to be ubiquitous among different CRISPR/Cas systems. Particular combinations of cas genes and repeat structures have been used to define 8 CRISPR subtypes (Ecoli, Ypest, Nmeni, Dvulg, Tneap, Hmari, Apern, and Mtube), some of which are associated with an additional gene module encoding repeat-associated mysterious proteins (RAMPs).[21] More than one CRISPR subtype may occur in a single genome. The sporadic distribution of the CRISPR/Cas subtypes suggests that the system is subject to horizontal gene transfer during microbial evolution.\\n\\nCRISPR associated proteincrystal structure of a crispr-associated protein from thermus thermophilusIdentifiersSymbolCRISPR_assocPfamPF08798Pfam clanCL0362InterProIPR010179CDDcd09727Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nCRISPR associated protein Cas2crystal structure of a hypothetical protein tt1823 from thermus thermophilusIdentifiersSymbolCRISPR_Cas2PfamPF09827InterProIPR019199CDDcd09638Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nCRISPR-associated protein Cse1IdentifiersSymbolCRISPR_Cse1PfamPF09481InterProIPR013381CDDcd09729Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nCRISPR-associated protein Cse2IdentifiersSymbolCRISPR_Cse2PfamPF09485InterProIPR013382CDDcd09670Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nMechanism\\nExogenous DNA is apparently processed by proteins encoded by Cas genes into small elements (~30 base pairs in length), which are then somehow inserted into the CRISPR locus near the leader sequence. RNAs from the CRISPR loci are constitutively expressed and are processed by Cas proteins to small RNAs composed of individual, exogenously-derived sequence elements with a flanking repeat sequence. The RNAs guide other Cas proteins to silence exogenous genetic elements at the RNA or DNA level.[1][28] Evidence suggests functional diversity among CRISPR subtypes. The Cse (Cas subtype Ecoli) proteins (called CasA-E in E. coli) form a functional complex, Cascade, that processes CRISPR RNA transcripts into spacer-repeat units that Cascade retains.[29] In other prokaryotes, Cas6 processes the CRISPR transcripts. Interestingly, CRISPR-based phage inactivation in E. coli requires Cascade and Cas3, but not Cas1 and Cas2. The Cmr (Cas RAMP module) proteins found in Pyrococcus furiosus and other prokaryotes form a functional complex with small CRISPR RNAs that recognizes and cleaves complementary target RNAs. RNA-guided CRISPR enzymes are classified as type V restriction enzymes.\\n\\nEvolutionary significance\\nA bioinformatic study showed that CRISPRs are evolutionarily conserved and cluster into related types. Many show signs of a conserved secondary structure.[22]\\nThrough the CRISPR/Cas mechanism, bacteria can acquire immunity to certain phages and thus halt further transmission of targeted phages. For this reason, some researchers have proposed that CRISPR/Cas is a Lamarckian inheritance mechanism.[30] Others investigated the coevolution of host and viral genomes by analysis of CRISPR sequences.[31]\\nCas9 proteins are highly enriched in pathogenic and commensal bacteria. CRISPR/Cas-mediated gene regulation may contribute to the regulation of endogenous bacterial genes, particularly during bacterial interaction with eukaryotic hosts. For example, Cas protein Cas9 of Francisella novicida uses a unique, small, CRISPR/Cas-associated RNA (scaRNA) to repress an endogenous transcript encoding a bacterial lipoprotein that is critical for F. novicida to dampen host response and promote virulence.[32]\\n\\nApplications\\nThe proof-of-principle demonstration of selective engineered redirection of the CRISPR-Cas system in 2012[33] provided a first step toward realization of proposals for CRISPR-derived biotechnology:[34]\\n\\nArtificial immunization against phage by introduction of engineered CRISPR loci in industrially important bacteria, including those used in food production and large-scale fermentation\\nGenome engineering at cellular or organismic level by reprogramming a CRISPR-Cas system to achieve RNA-guided genome engineering. Proof of concept studies demonstrated examples both in vitro[35] and in vivo[18][36][37]\\nDiscrimination of bacterial strains by comparison of spacer sequences\\nMouse models\\nCRISPR simplifies creation of mouse models and reduces the time required to a matter of weeks from months or longer. Knockdown of endogenous genes has been achieved by transfection with a plasmid that contains a CRISPR area with a spacer, which inhibits a target gene. Injecting mouse zygotes with Cas9 and two guide RNAs was able to disable two genes with 80% efficiency. So-called homology-directed repair involves using Cas9 to \"nick\" DNA, to introduce new gene parts to the zygote.[citation needed]\\n\\nCRISPR/Cas system in phage\\nAnother way that bacteria can defend against phage infection is by having chromosomal islands. A subtype of chromosomal islands called phage-inducible chromosomal island (PICI) is excised from bacterial chromosome upon phage infection and can inhibit phage replication.[38] The mechanisms that induce PICI excision and how PICI inhibits phage replication are not well understood. One study showed that lytic ICP1 phage that specifically targets Vibrio cholerae serogroup O1 has acquired a CRISPR/Cas system that targets a V. cholera PICI-like element. The system has 2 CRISPR loci and 9 Cas genes. It seems to be homologous to the 1-F system found in Yersinia pestis. Moreover, like the bacterial CRISPR/Cas system, ICP1 CRISPR/Cas system can also acquire new sequences, which allows the phage to co-evolve with its host.[39]\\n\\nMulti-gene targeting\\nCRISPRs have been used to cut as many as five genes at once.[12]\\n\\nReversible knockdown\\nFurther information: CRISPR interference\\n\"CRISPRi\" like RNAi, turns off genes in a reversible fashion by targeting a site but not cutting. In bacteria, the presence of Cas9 alone is enough to block transcription, but for mammalian applications, a section of protein is added. Its guide RNA targets regulatory DNA, called promoters that immediately precede the gene target.[12]\\n\\nGene activation\\nFurther information: CRISPR interference\\nCas9 has been used to carry synthetic transcription factors (protein fragments that turn on genes.) This enabled the activation of specific human genes. The technique achieved a strong effect by targeting multiple CRISPR constructs to slightly different spots on the gene\\'s promoter.[12]\\nThe genes included some tied to human diseases, including those involved in muscle differentiation, cancer, inflammation and producing fetal hemoglobin.[12]\\n\\nInfrastructure\\nFree software is available to design guide RNA to target any desired gene. The Addgene repository offers academics the DNA to make their own CRISPR system for $65. Since the beginning of the year, In 2013 alone Addgene distributed more than 10,000 CRISPR constructs. The facility has received CRISPR-enabling DNA sequences from 11 independent research teams.[12]\\n\\nTherapeutics\\nEditas Medicine, a $43 million startup, aims to develop treatments that employ CRISPR/Cas to make edits to single base pairs and larger stretches of DNA. Inherited diseases such as cystic fibrosis, sickle-cell anemia and Huntington\\'s disease are caused by single base pair mutations; CRISPR/Cas technology has the potential to correct these errors. The \"corrected\" gene remains in its normal location on its chromosome, which preserves the way the cell normally activate inhibits its expression.[40]\\n\\nAfter harvesting blood cell precursors called hematopoietic stem cells from a patient\\'s bone marrow, CRISPR gene surgery would correct the defective gene. Then the gene-\\xadcorrected stem cells would be returned to the patient\\'s marrow, which would then produce healthy red blood cells. Replacing 70% of the sickle cells would produce a cure.[41]\\n\\nBefore it can be used clinically, the company must be able to guarantee that only the targeted region will be affected and determine how to deliver the therapy to a patient’s cells.[40]\\n\\nSafety\\nBefore work proceeds much further on human genomes, improved targeting is required. Guide RNAs have been observed targeting sequences that differ by multiple base pairs from the intended sequence.[12]\\n\\nAlternatives\\nIn the early 2000s, researchers developed zinc finger nucleases, synthetic proteins whose DNA-binding domains enable them to cut DNA at specific spots. Later, synthetic nucleases called TALENs provided an easier way to target specific DNA and were predicted to surpass zinc fingers. They both depend on custom-making new proteins for each DNA target, a more cumbersome procedure than guide RNAs. CRISPRs are more efficient[42] and can target more genes than these earlier techniques.[42]\\n\\nPatent\\nA provisional US patent application on the use of the CRISPR system for editing genes and regulating gene expression was filed on May 12, 2012 by the inventors. Subsequent applications were combined and on March 6, 2014 the resulting patent application was published by the USPTO.[43] The patent rights have been assigned by the inventors to the Regents of the University of California and to the University of Vienna.\\n\\nReferences\\n\\n\\n^ a b c d Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20056882, please use {{cite journal}} with |pmid=20056882 instead.\\n\\n^ a b c Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20125085, please use {{cite journal}} with |pmid=20125085 instead.\\n\\n^ 71/79 Archaea, 463/1008 Bacteria CRISPRdb, Date: 19.6.2010\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17521438, please use {{cite journal}} with |pmid=17521438 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17379808, please use {{cite journal}} with |pmid=17379808 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19095942, please use {{cite journal}} with |pmid=19095942 instead.\\n\\n^ Wiedenheft, B (2012 Feb 15). \"RNA-guided genetic silencing systems in bacteria and archaea\". Nature. 482 (7385): 331–8. PMID\\xa022337052. Unknown parameter |coauthors= ignored (|author= suggested) (help); Check date values in: |date= (help).mw-parser-output cite.citation{font-style:inherit}.mw-parser-output .citation q{quotes:\"\\\\\"\"\"\\\\\"\"\"\\'\"\"\\'\"}.mw-parser-output .id-lock-free a,.mw-parser-output .citation .cs1-lock-free a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/6/65/Lock-green.svg\")right 0.1em center/9px no-repeat}.mw-parser-output .id-lock-limited a,.mw-parser-output .id-lock-registration a,.mw-parser-output .citation .cs1-lock-limited a,.mw-parser-output .citation .cs1-lock-registration a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/d/d6/Lock-gray-alt-2.svg\")right 0.1em center/9px no-repeat}.mw-parser-output .id-lock-subscription a,.mw-parser-output .citation .cs1-lock-subscription a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/a/aa/Lock-red-alt-2.svg\")right 0.1em center/9px no-repeat}.mw-parser-output .cs1-subscription,.mw-parser-output .cs1-registration{color:#555}.mw-parser-output .cs1-subscription span,.mw-parser-output .cs1-registration span{border-bottom:1px dotted;cursor:help}.mw-parser-output .cs1-ws-icon a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/4/4c/Wikisource-logo.svg\")right 0.1em center/12px no-repeat}.mw-parser-output code.cs1-code{color:inherit;background:inherit;border:none;padding:inherit}.mw-parser-output .cs1-hidden-error{display:none;font-size:100%}.mw-parser-output .cs1-visible-error{font-size:100%}.mw-parser-output .cs1-maint{display:none;color:#33aa33;margin-left:0.3em}.mw-parser-output .cs1-subscription,.mw-parser-output .cs1-registration,.mw-parser-output .cs1-format{font-size:95%}.mw-parser-output .cs1-kern-left,.mw-parser-output .cs1-kern-wl-left{padding-left:0.2em}.mw-parser-output .cs1-kern-right,.mw-parser-output .cs1-kern-wl-right{padding-right:0.2em}.mw-parser-output .citation .mw-selflink{font-weight:inherit}\\n\\n^ Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1073/pnas.1315278110, please use {{cite journal}} with |doi=10.1073/pnas.1315278110 instead.\"Bacteria incorporate pieces of old DNA in their own genome, scientists discover\". KurzweilAI. doi:10.1073/pnas.1315278110. Retrieved 2013-11-26.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 3316184, please use {{cite journal}} with |pmid=3316184 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 10760181, please use {{cite journal}} with |pmid=10760181 instead.\\n\\n^ a b Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 11952905, please use {{cite journal}} with |pmid=11952905 instead.\\n\\n^ a b c d e f g h i j k Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1126/science.341.6148.833, please use {{cite journal}} with |doi=10.1126/science.341.6148.833 instead.\\n\\n^ \"CRISPR gene therapy: Scientists call for more public debate around breakthrough technique - Science - News\". The Independent. 2013-11-07. Retrieved 2013-11-25.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23460208, please use {{cite journal}} with |pmid=23460208 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23360964, please use {{cite journal}} with |pmid=23360964 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23817069, please use {{cite journal}} with |pmid=23817069 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23999092, please use {{cite journal}} with |pmid=23999092 instead.\\n\\n^ a b Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23643243, please use {{cite journal}} with |pmid=23643243 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 24136345, please use {{cite journal}} with |pmid=24136345 instead.\\n\\n^ \"Researchers reverse a liver disorder in mice by correcting a mutated gene\". PhysOrg. 30 March 2014. Retrieved 31 March 2014.\\n\\n^ a b c d Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 16292354, please use {{cite journal}} with |pmid=16292354 instead.\\n\\n^ a b Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17442114, please use {{cite journal}} with |pmid=17442114 instead.\\n\\n^ Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1007/s00239-004-0046-3, please use {{cite journal}} with |doi=10.1007/s00239-004-0046-3 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 16079334 , please use {{cite journal}} with |pmid=16079334  instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 15758212, please use {{cite journal}} with |pmid=15758212 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20598393, please use {{cite journal}} with |pmid=20598393 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17894817, please use {{cite journal}} with |pmid=17894817 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 16545108, please use {{cite journal}} with |pmid=16545108 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18703739, please use {{cite journal}} with |pmid=18703739 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19906303, please use {{cite journal}} with |pmid=19906303 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19132092, please use {{cite journal}} with |pmid=19132092 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23584588, please use {{cite journal}} with |pmid=23584588 instead.\\n\\n^ Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1016/j.molcel.2011.10.023, please use {{cite journal}} with |doi=10.1016/j.molcel.2011.10.023 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18157154, please use {{cite journal}} with |pmid=18157154 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 22745249, please use {{cite journal}} with |pmid=22745249 instead.Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 22949671, please use {{cite journal}} with |pmid=22949671 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23287718, please use {{cite journal}} with |pmid=23287718 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23287722, please use {{cite journal}} with |pmid=23287722 instead.Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1073/pnas.1313587110, please use {{cite journal}} with |doi=10.1073/pnas.1313587110 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20634809, please use {{cite journal}} with |pmid=20634809 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23446421, please use {{cite journal}} with |pmid=23446421 instead.\\n\\n^ a b Young, Susan. \"Biotech Startup Editas Medicine Wants to Cure Grievous Genetic Diseases with New Genome Editing Technology | MIT Technology Review\". Technologyreview.com. Retrieved 2013-11-30.\\n\\n^ Young, Susan (2014-02-11). \"CRISPR and Other Genome Editing Tools Boost Medical Research and Gene Therapy\\'s Reach | MIT Technology Review\". Technologyreview.com. Retrieved 2014-04-13.\\n\\n^ a b Young, Susan (11 February 2014) Genome Surgery MIT Technology Review, Retrieved 17 February 2014\\n\\n^ \"Methods And Compositions For Rna-directed Target Dna Modification And For Rna-directed Modulation Of Transcription\". Freshpatents.com. Retrieved 2014-04-13.\\n\\n\\nFurther reading\\n.mw-parser-output .refbegin{font-size:90%;margin-bottom:0.5em}.mw-parser-output .refbegin-hanging-indents>ul{list-style-type:none;margin-left:0}.mw-parser-output .refbegin-hanging-indents>ul>li,.mw-parser-output .refbegin-hanging-indents>dl>dd{margin-left:0;padding-left:3.2em;text-indent:-3.2em;list-style:none}.mw-parser-output .refbegin-100{font-size:100%}.mw-parser-output .refbegin-columns{margin-top:0.3em}.mw-parser-output .refbegin-columns dl,.mw-parser-output .refbegin-columns ol,.mw-parser-output .refbegin-columns ul{margin-top:0}.mw-parser-output .refbegin-columns li,.mw-parser-output .refbegin-columns dd{page-break-inside:avoid;break-inside:avoid-column}\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 24584096, please use {{cite journal}} with |pmid=24584096 instead.\\nAttention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1016/j.tig.2014.01.003, please use {{cite journal}} with |doi=10.1016/j.tig.2014.01.003 instead.\\nAttention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1038/nrmicro3241, please use {{cite journal}} with |doi=10.1038/nrmicro3241 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18065539, please use {{cite journal}} with |pmid=18065539 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18065545, please use {{cite journal}} with |pmid=18065545 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18497291, please use {{cite journal}} with |pmid=18497291 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18971321, please use {{cite journal}} with |pmid=18971321 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19141480, please use {{cite journal}} with |pmid=19141480 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19143596, please use {{cite journal}} with |pmid=19143596 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19239620, please use {{cite journal}} with |pmid=19239620 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19246744, please use {{cite journal}} with |pmid=19246744 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19383692, please use {{cite journal}} with |pmid=19383692 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19945378, please use {{cite journal}} with |pmid=19945378 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19945373, please use {{cite journal}} with |pmid=19945373 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20072129, please use {{cite journal}} with |pmid=20072129 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20129051, please use {{cite journal}} with |pmid=20129051 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20132443, please use {{cite journal}} with |pmid=20132443 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20133361, please use {{cite journal}} with |pmid=20133361 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20528693, please use {{cite journal}} with |pmid=20528693 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20556198, please use {{cite journal}} with |pmid=20556198 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20559554, please use {{cite journal}} with |pmid=20559554 instead.\\n\\nExternal links\\nE-CRISP.org A comprehensive software for CRISPR gRNA design\\nCRISPR Design Tool\\nCRISPR Design Tool with scoring algorithms\\nTool for finding CRISPRs\\nTool for finding CRISPR targets in other nucleic acids\\nRfam page for the CRISPR entries\\n.mw-parser-output .navbar{display:inline;font-size:88%;font-weight:normal}.mw-parser-output .navbar-collapse{float:left;text-align:left}.mw-parser-output .navbar-boxtext{word-spacing:0}.mw-parser-output .navbar ul{display:inline-block;white-space:nowrap;line-height:inherit}.mw-parser-output .navbar-brackets::before{margin-right:-0.125em;content:\"[ \"}.mw-parser-output .navbar-brackets::after{margin-left:-0.125em;content:\" ]\"}.mw-parser-output .navbar li{word-spacing:-0.125em}.mw-parser-output .navbar-mini abbr{font-variant:small-caps;border-bottom:none;text-decoration:none;cursor:inherit}.mw-parser-output .navbar-ct-full{font-size:114%;margin:0 7em}.mw-parser-output .navbar-ct-mini{font-size:114%;margin:0 4em}.mw-parser-output .infobox .navbar{font-size:100%}.mw-parser-output .navbox .navbar{display:block;font-size:100%}.mw-parser-output .navbox-title .navbar{float:left;text-align:left;margin-right:0.5em}vteGallery of secondary structure images@media all and (max-width:720px){.mw-parser-output .mod-gallery{width:100%!important}}.mw-parser-output .mod-gallery{display:table}.mw-parser-output .mod-gallery-default{background:transparent;margin-top:.3em}.mw-parser-output .mod-gallery-center{margin-left:auto;margin-right:auto}.mw-parser-output .mod-gallery-left{float:left}.mw-parser-output .mod-gallery-right{float:right}.mw-parser-output .mod-gallery-none{float:none}.mw-parser-output .mod-gallery-collapsible{width:100%}.mw-parser-output .mod-gallery .title,.mw-parser-output .mod-gallery .main,.mw-parser-output .mod-gallery .footer{display:table-row}.mw-parser-output .mod-gallery .title>div{display:table-cell;text-align:center;font-weight:bold}.mw-parser-output .mod-gallery .main>div{display:table-cell}.mw-parser-output .mod-gallery .gallery{line-height:1.35em}.mw-parser-output .mod-gallery .footer>div{display:table-cell;text-align:right;font-size:80%;line-height:1em}.mw-parser-output .mod-gallery .title>div *,.mw-parser-output .mod-gallery .footer>div *{overflow:visible}.mw-parser-output .mod-gallery .gallerybox img{background:none!important}.mw-parser-output .mod-gallery .bordered-images .thumb img{border:solid #eaecf0 1px}.mw-parser-output .mod-gallery .whitebg .thumb{background:#fff!important}\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR2:  Secondary structure taken from the Rfam database. Family RF01315.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR5:  Secondary structure taken from the Rfam database. Family RF011318.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR6:  Secondary structure taken from the Rfam database. Family RF01319.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR8:  Secondary structure taken from the Rfam database. Family RF01321.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR9:  Secondary structure taken from the Rfam database. Family RF01322.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR19:  Secondary structure taken from the Rfam database. Family RF01332.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR41:  Secondary structure taken from the Rfam database. Family RF01350.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR52:  Secondary structure taken from the Rfam database. Family RF01365.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR57:  Secondary structure taken from the Rfam database. Family RF01370.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR65:  Secondary structure taken from the Rfam database. Family RF01378.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\n\\n\\n\\n\\n\\nCategories: Repetitive DNA sequencesNon-coding RNA'"]},"metadata":{"tags":[]},"execution_count":31}]},{"cell_type":"markdown","metadata":{"id":"kGgAMOM6zulH"},"source":["## Extract contexts"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":414},"id":"ghiJjtnuvUIB","executionInfo":{"status":"error","timestamp":1610559404039,"user_tz":-60,"elapsed":1479,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"7fd0caa7-a102-4eaa-97f5-19e7cd919bc7"},"source":["for article, article_name in [\n","                # (CRISPR_en, 'CRISPR_en'),\n","                # (CRISPR_gene_editing_en, 'CRISPR_gene_editing_en'),\n","                ]:\n","  print(article_name)\n","  for keyphrase in [\n","                                    \n","                # discoveries\n","                'first',\n","                'discover',\n","                \n","                # naming things\n","                'name', # CRISPR acronym\n","                'acronym', # CRISPR acronym\n","                \n","                # patents\n","                'patent',\n","                'property',\n","                'right',\n","\n","                # clinical trials\n","                'trial',\n","                'clinical',\n","                'patient',\n","                'approve', # e.g. FDA approved...\n","\n","                # awards\n","                'award',\n","                'prize',\n","                'nobel',\n","\n","                # people:\n","                'doudna', \n","                'charpentier', \n","                'zhang', \n","                'šikšnys', # with all accents\n","                  'šiksnys', # only first accent\n","                  'sikšnys', # only second accent\n","                  'siksnys', # no accents\n","                'gašiūnas', # with all accents\n","                  'gasiūnas', # only first accent\n","                  'gašiunas', # only second accent\n","                  'gasiunas', # no accents\n","                'horvath', # auch im Zusammenhang mit Šikšnys, aber auch so interessant\n","                'barrangou', # auch im Zusammenhang mit Šikšnys, aber auch so interessant\n","                'church',\n","                'brouns',\n","                'mojica',\n","                'ishino',\n","                'jansen',\n","                'marraffini',\n","                'sontheimer',\n","                'koonin',\n","                'moineau',\n","                'bolotin',\n","                'sorokin',\n","                'makarova',\n","                'pourcel',\n","                'overbeek',\n","                'liang',\n","                ' xu', # added space!\n","                'jiankui',\n","                'nishimasu',\n","                'shibata',\n","                'lander', # review\n","                \n","                # institutions:\n","                'osaka',\n","                'alicante',\n","                'danisco',\n","                'broad',\n","                'harvard',\n","                'university of california', \n","                'berkeley',\n","                \n","                ]:\n","    print('\\t' + keyphrase)\n","    df = keyphrase_in_context(keyphrase, article, width=100, score_cutoff=75)\n","    df.sort_values(by=['left_id','timestamp'])\n","    df.to_excel(path_extractions + '/contexts_{}_{}.xlsx'.format(article_name, keyphrase))"],"execution_count":null,"outputs":[{"output_type":"stream","text":["CRISPR_gene_editing_en\n","\tarno\n"],"name":"stdout"},{"output_type":"error","ename":"TypeError","evalue":"ignored","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m<ipython-input-13-4d864295b87a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     78\u001b[0m                 ]:\n\u001b[1;32m     79\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\\t'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mkeyphrase\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 80\u001b[0;31m     \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkeyphrase_in_context\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkeyphrase\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marticle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwidth\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m100\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscore_cutoff\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m75\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     81\u001b[0m     \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mby\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'left_id'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'timestamp'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     82\u001b[0m     \u001b[0;31m# df.to_excel(path_extractions + '/contexts_{}_{}.xlsx'.format(article_name, keyphrase))\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m<ipython-input-12-6d40ff7cce69>\u001b[0m in \u001b[0;36mkeyphrase_in_context\u001b[0;34m(keyphrase, article, width, score_cutoff, scorer)\u001b[0m\n\u001b[1;32m      5\u001b[0m   \u001b[0mgen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marticle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0myield_revisions\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m   \u001b[0mcontexts\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m   \u001b[0;32mfor\u001b[0m \u001b[0mrevision\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgen\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m     \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrevision\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/content/drive/My Drive/WIKING/WIKING_git/code/article/article.py\u001b[0m in \u001b[0;36myield_revisions\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     44\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m                 \u001b[0mrevision\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mloads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 46\u001b[0;31m                 \u001b[0;32myield\u001b[0m \u001b[0mRevision\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mrevision\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     47\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     48\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mtrack_field_values_in_article\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfields\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbibliography\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mTypeError\u001b[0m: __init__() missing 1 required positional argument: 'wikitext'"]}]},{"cell_type":"markdown","metadata":{"id":"4k2Me_0RyhFO"},"source":["### CRISPR_en: \"first\"\n","\n"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":788},"id":"CPm0Mh-syhFP","executionInfo":{"status":"ok","timestamp":1606682187152,"user_tz":-60,"elapsed":97739,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"4885e2f4-70b3-4a97-da4a-cf6c8843f258"},"source":["df = keyphrase_in_context('first', CRISPR_en, width=50, score_cutoff=75)\n","df.to_excel(file_contexts_CRISPR_en)\n","df.sort_values(by=['left_id','timestamp'])"],"execution_count":null,"outputs":[{"output_type":"stream","text":["WARNING:root:Applied processor reduces input query to empty string, all comparisons will have score 0. [Query: '']\n"],"name":"stderr"},{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>revindx</th>\n","      <th>timestamp</th>\n","      <th>left_id</th>\n","      <th>left</th>\n","      <th>keyphrase</th>\n","      <th>right</th>\n","      <th>right_id</th>\n","      <th>revid</th>\n","      <th>url</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>25</th>\n","      <td>133</td>\n","      <td>2012-01-07 04:36:56</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>470025144</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>27</th>\n","      <td>134</td>\n","      <td>2012-08-01 19:27:33</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>505309163</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>29</th>\n","      <td>135</td>\n","      <td>2012-08-22 21:44:35</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>508685743</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>31</th>\n","      <td>136</td>\n","      <td>2012-12-06 19:12:03</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>526751001</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>33</th>\n","      <td>137</td>\n","      <td>2012-12-11 08:13:19</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>527490124</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>26284</th>\n","      <td>2039</td>\n","      <td>2020-11-02 21:24:41</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986762186</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26297</th>\n","      <td>2040</td>\n","      <td>2020-11-03 09:53:58</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986842914</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26310</th>\n","      <td>2041</td>\n","      <td>2020-11-03 20:16:34</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986927023</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26323</th>\n","      <td>2042</td>\n","      <td>2020-11-03 22:56:29</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986949106</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26336</th>\n","      <td>2043</td>\n","      <td>2020-11-04 21:00:44</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>987088136</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>26340 rows × 9 columns</p>\n","</div>"],"text/plain":["       revindx  ...                                                url\n","25         133  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","27         134  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","29         135  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","31         136  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","33         137  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","...        ...  ...                                                ...\n","26284     2039  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26297     2040  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26310     2041  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26323     2042  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26336     2043  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","\n","[26340 rows x 9 columns]"]},"metadata":{"tags":[]},"execution_count":18}]},{"cell_type":"markdown","metadata":{"id":"fc1dPCidx823"},"source":["### CRISPR_gene_editing_en: \"first\"\n","\n"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":771},"id":"dB8a9egkrWyg","executionInfo":{"status":"ok","timestamp":1606681779613,"user_tz":-60,"elapsed":6031,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"508e83c7-d9e5-4ccc-8689-f495287716c9"},"source":["df = keyphrase_in_context('first', CRISPR_gene_editing_en, width=50, score_cutoff=75)\n","df.to_excel(file_contexts_CRISPR_gene_editing_en)\n","df.sort_values(by=['left_id','timestamp'])"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>revindx</th>\n","      <th>timestamp</th>\n","      <th>left_id</th>\n","      <th>left</th>\n","      <th>keyphrase</th>\n","      <th>right</th>\n","      <th>right_id</th>\n","      <th>revid</th>\n","      <th>url</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>3</th>\n","      <td>0</td>\n","      <td>2019-02-17 06:32:48</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727671</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>10</th>\n","      <td>1</td>\n","      <td>2019-02-17 06:33:54</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727741</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>17</th>\n","      <td>2</td>\n","      <td>2019-02-17 06:35:41</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727860</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>24</th>\n","      <td>3</td>\n","      <td>2019-02-17 06:37:14</td>\n","      <td>0</td>\n","      <td>340.2.[132] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727959</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>31</th>\n","      <td>4</td>\n","      <td>2019-02-17 06:39:37</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883728113</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>1447</th>\n","      <td>184</td>\n","      <td>2020-10-23 09:36:41</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>984993980</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1458</th>\n","      <td>185</td>\n","      <td>2020-11-02 04:29:36</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986649176</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1469</th>\n","      <td>186</td>\n","      <td>2020-11-02 10:34:00</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986682078</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1480</th>\n","      <td>187</td>\n","      <td>2020-11-02 10:35:18</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986682164</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1491</th>\n","      <td>188</td>\n","      <td>2020-11-02 10:35:49</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986682206</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>1497 rows × 9 columns</p>\n","</div>"],"text/plain":["      revindx  ...                                                url\n","3           0  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","10          1  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","17          2  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","24          3  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","31          4  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","...       ...  ...                                                ...\n","1447      184  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1458      185  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1469      186  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1480      187  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1491      188  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","\n","[1497 rows x 9 columns]"]},"metadata":{"tags":[]},"execution_count":14}]},{"cell_type":"markdown","metadata":{"id":"PgVJr7mPzwj_"},"source":["# Old stuff..."]},{"cell_type":"markdown","metadata":{"id":"6QRPtrvO9Mvp"},"source":["### Occurrence spans"]},{"cell_type":"code","metadata":{"id":"PrlDj91nsCLY"},"source":["def occurance_spans(keyphrase, article):\n","  ''' \n","  returns a list of time spans during which a keyphrase 'keyphrase' continously appears in an article 'a'\n","  '''\n","  gen = article.yield_revisions()\n","  last_rev_was_match = False\n","  spans = []\n","  for revision in gen:\n","    if keyphrase in revision.get_text():\n","    # if keyphrase in i.get_text().split('\\nLocus structure\\n')[0]: # Trick 17 to exclude everything after history section and practially only search in histroy section\n","      last_rev_that_matched = revision\n","      if not last_rev_was_match:\n","        span_beg = revision\n","        last_rev_was_match = True\n","    else:\n","      if last_rev_was_match:\n","        spans.append((span_beg, last_rev_that_matched))\n","        last_rev_was_match = False\n","  if last_rev_was_match:\n","    spans.append((span_beg, last_rev_that_matched))\n","  print(\n","      '\"{}\" was found in the following revisions of {}:\\n{}'.format(keyphrase, 'Test', '\\n'.join(['Span {}: {} - {}'.format(indx + 1, revision[0].timestamp, revision[1].timestamp) for indx,revision in enumerate(spans)])) \n","      if spans else '\"{}\" was not found in any revision of \"{}\"'.format(keyphrase, 'Test')\n","      )\n","  return spans"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"Y5HIHBrA9Sak"},"source":["### Unique contexts"]},{"cell_type":"code","metadata":{"id":"_eUn_KzFheAn"},"source":["def unique_contexts(keyphrase, article, width=50):\n","  ''' \n","  returns a pandas df with three columns:\n","  1) unique contexts of width in which keyphrase appears in article\n","  2) timestamp for first occurrence of each context\n","  3) url for first occurrence of each context\n","  '''\n","  kill = re.compile(r'\\s*\\[.*?\\]')\n","  gen = article.yield_revisions()\n","  unique_contexts = []\n","  timestamps = []\n","  urls = []\n","  for revision in gen:\n","    text = revision.get_text()\n","    # if s in text.split('\\nLocus structure\\n')[0]: # Trick 17 to exclude everything after history section and practially only search in histroy section\n","    if keyphrase in text:\n","      text = kill.sub('', text) # kill intext refs like '[23]'\n","      text = text.lower()\n","      for indx in find_all(text, keyphrase):\n","        # make '/n' the boundary for context\n","        before = text[indx - width : indx]\n","        after = text[indx + len(keyphrase): indx + len(keyphrase) + width]\n","        context = keyphrase[::-1]\n","        for char in before[::-1]:\n","          if char == '\\n':\n","            break\n","          context += char\n","        context = context[::-1]\n","        for char in after:\n","          if char == '\\n':\n","            break\n","          context += char\n","        # add unique contexts and add their meta data\n","        context = context.strip()\n","        if not context in unique_contexts:\n","          unique_contexts.append(context)\n","          timestamps.append(revision.timestamp.string)\n","          urls.append(revision.url)    \n","  print('Found {} unique contexts for \"{}\" in article \"{}\".'.format(len(unique_contexts), keyphrase, article.name))\n","  # return list(zip(unique_contexts, timestamps, urls))\n","  return pd.DataFrame({'Context':unique_contexts, \n","        'Timestamp':timestamps,\n","        'Url':urls}) "],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"ajWDEbMWHHSw"},"source":["### Left-right contexts"]},{"cell_type":"code","metadata":{"id":"7elYtyzQr-PP"},"source":["def lr_context_graph(keyphrase, article, width=50):\n","  ''' \n","  Returns networkx DiGraph for links between contexts to the left and to the right of keyphrase\n","  '''\n","  gen = article.yield_revisions()\n","  DG = nx.DiGraph(keyphrase=keyphrase)\n","  for revision in gen:\n","    text = revision.get_text()\n","    text = text.lower()\n","    # text = re.sub(r'\\s*\\[.*?\\]','', text) # kill intext refs like '[23]'\n","    metadata = (revision.timestamp.string, revision.url)\n","    if keyphrase in text:\n","      for indx in find_all(text, keyphrase):\n","        # context left\n","        left = ''\n","        for char in text[indx - width : indx][::-1]:\n","          if char == '\\n': # make '/n' the boundary for context\n","            break\n","          left += char\n","        left = left[::-1].strip()\n","        # add/update node for context left\n","        if not left in DG:\n","          DG.add_node(left, as_left=[metadata], as_right=[])\n","        else:\n","          DG.nodes[left]['as_left'].append(metadata)\n","        # context right\n","        right = ''\n","        for char in text[indx + len(keyphrase): indx + len(keyphrase) + width]:\n","          if char == '\\n': # make '/n' the boundary for context\n","            break\n","          right += char\n","        right = right.strip()\n","        # add/update node for context right\n","        if not right in DG:\n","          DG.add_node(right, as_right=[metadata], as_left=[])\n","        else:\n","          DG.nodes[right]['as_right'].append(metadata)\n","        # add/update edge from context left to context right\n","        if not [left, right] in DG.edges:\n","          DG.add_edge(left, right, occurrences=[metadata])\n","        else:\n","            DG.edges[left,right]['occurrences'].append(metadata)\n","  return DG"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"o3QeZcjHpN83"},"source":["### Node communities"]},{"cell_type":"code","metadata":{"id":"4cJ9mGfolzDZ"},"source":["def node_community_dic(DG, score_cutoff=90, scorer=fuzz.ratio):\n","  keyphrase = DG.graph['keyphrase']\n","  G = nx.Graph()\n","  for side in ['left', 'right']:\n","    choices = [node[0] for node in DG.nodes.data() if node[1]['as_{}'.format(side)]]\n","    length = len(choices)\n","    for query in choices:\n","      indx = choices.index(query) + 1\n","      if indx <= length:\n","        scores = process.extractBests(query, choices[choices.index(query) + 1:], score_cutoff=score_cutoff, limit=length, scorer=scorer) # https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/process.py\n","        edges = [(query, i[0]) for i in scores]\n","        G.add_edges_from(edges)\n","  return community_louvain.best_partition(G)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"xTaYKRvg6hN-"},"source":["### Master node for each community"]},{"cell_type":"code","metadata":{"id":"ATQugc1nqHFy"},"source":["def master_nodes_dic(DG, ncd):\n","  '''\n","  Returns dictionary with community_ids as keys and nodes + latest occurrence as value\n","  '''\n","  result = {}\n","  reverse_mnd = {v:[k for k in mnd if mnd[k] == v] for v in mnd.values()}\n","  for community_id, nodes in reverse_mnd.items():\n","    candidates = []\n","    for node in nodes:\n","      latest = max([DG.nodes[node]['as_{}'.format(side)][-1][0] for side in ['left', 'right'] if DG.nodes[node]['as_{}'.format(side)]])\n","      candidates.append((node, latest))\n","    winner = sorted(candidates, key= lambda x: (x[1], x[0]), reverse=True)[0]\n","    result[community_id] = winner\n","  return result"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"KpfcElE48O3W"},"source":["### Reduced LR contexts graph (using ``lr_context_graph``, ``node_community_dic``, and ``master_nodes_dic``)"]},{"cell_type":"code","metadata":{"id":"wgGSr0Fl8ii7"},"source":["def reduce_lr_context_graph(DG, ncd, mnd):\n","  pass"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"bYzX8wKFrxkB"},"source":["## Load data"]},{"cell_type":"code","metadata":{"id":"5pFtlIlNiYtw"},"source":["CRISPR_en = Article(file_CRISPR_en)\n","CRISPR_gene_editing_en = Article(file_CRISPR_gene_editing_en)\n","\n","# events = pd.read_csv(file_events)\n","# accounts = pd.read_csv(file_accounts)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"iN1eI6MICMu_"},"source":["## Experiments"]},{"cell_type":"markdown","metadata":{"id":"CPg3sesxo00n"},"source":["### Extract LR contexts and merge nodes "]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ZAA1h3_wuO3D","executionInfo":{"status":"ok","timestamp":1606500810339,"user_tz":-60,"elapsed":615,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"6b665a2a-d41b-4e75-9d90-9da12af436f8"},"source":["DG = lr_context_graph('first', CRISPR_gene_editing_en, width=50)\n","ncd = node_community_dic(DG)\n","# sorted(ncd.items(), key=lambda item:(item[1], item[0]))\n","mnd = master_nodes_dic(DG, ncd)\n","mnd"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["{0: (\"340.2.[145] yang's white button mushroom was the\",\n","  '2020-11-02 10:35:49'),\n"," 1: ('t position 8 (help); no-break space character in |',\n","  '2020-01-31 21:33:44'),\n"," 2: ('demonstrated in 2012 in:[25][26]', '2019-02-18 01:48:30'),\n"," 3: ('organism genetically modified with the crispr-cas',\n","  '2020-11-02 10:35:49'),\n"," 4: ('= at position 5 (help); no-break space character i',\n","  '2020-01-31 21:33:44')}"]},"metadata":{"tags":[]},"execution_count":255}]},{"cell_type":"markdown","metadata":{"id":"IJppJpuY6ZAF"},"source":["### Older stuff"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"D_vRSWezNP31","executionInfo":{"status":"ok","timestamp":1606423116262,"user_tz":-60,"elapsed":76291,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"fe9f6fea-2e38-4840-ab2c-d17b6f5b9009"},"source":["contexts_first = unique_contexts('first', CRISPR_en, width=50)\n","contexts_first.to_excel(file_contexts_CRISPR_en)\n","\n","contexts_first = unique_contexts('first', CRISPR_gene_editing_en, width=50)\n","contexts_first.to_excel(file_contexts_CRISPR_gene_editing_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Found 143 unique contexts for \"first\" in article \"CRISPR_en\".\n","Found 23 unique contexts for \"first\" in article \"CRISPR_gene_editing_en\".\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"yJguDbiEIxKv","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1606410107435,"user_tz":-60,"elapsed":85293,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"a30c97b9-40ce-41d7-9431-988943b84a8a"},"source":["timeline = occurance_spans('CRISPR Timeline', CRISPR_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"CRISPR Timeline\" was found in the following revisions of Test:\n","Span 1: {'datetime': datetime.datetime(2016, 9, 28, 9, 26, 29),\n"," 'day': 28,\n"," 'hour': 9,\n"," 'minute': 26,\n"," 'month': 9,\n"," 'second': 29,\n"," 'string': '2016-09-28 09:26:29',\n"," 'year': 2016} - {'datetime': datetime.datetime(2016, 12, 28, 8, 5, 49),\n"," 'day': 28,\n"," 'hour': 8,\n"," 'minute': 5,\n"," 'month': 12,\n"," 'second': 49,\n"," 'string': '2016-12-28 08:05:49',\n"," 'year': 2016}\n","Span 2: {'datetime': datetime.datetime(2019, 3, 18, 14, 33, 42),\n"," 'day': 18,\n"," 'hour': 14,\n"," 'minute': 33,\n"," 'month': 3,\n"," 'second': 42,\n"," 'string': '2019-03-18 14:33:42',\n"," 'year': 2019} - {'datetime': datetime.datetime(2019, 3, 18, 14, 34, 29),\n"," 'day': 18,\n"," 'hour': 14,\n"," 'minute': 34,\n"," 'month': 3,\n"," 'second': 29,\n"," 'string': '2019-03-18 14:34:29',\n"," 'year': 2019}\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"MtcXB1nBIkkq","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1604323635576,"user_tz":-60,"elapsed":74224,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"5367bc1b-4e68-442c-e636-2702ceb5a7c9"},"source":["# history_trick17 = occurance_spans('\\nLocus structure\\n', CRISPR_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"\n","Locus structure\n","\" was not found in any revision of \"Test\"\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"R2nt3LFoDezQ","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1604322549498,"user_tz":-60,"elapsed":78372,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"439e1385-bad6-41a0-e927-2043fe561133"},"source":["history = occurance_spans('\\nLocus structure\\n', CRISPR_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"\n","Locus structure\n","\" was found in the following revisions of Test:\n","Span 1: {'datetime': datetime.datetime(2013, 11, 24, 21, 3, 53),\n"," 'day': 24,\n"," 'hour': 21,\n"," 'minute': 3,\n"," 'month': 11,\n"," 'second': 53,\n"," 'string': '2013-11-24 21:03:53',\n"," 'year': 2013} - {'datetime': datetime.datetime(2016, 9, 22, 22, 1, 43),\n"," 'day': 22,\n"," 'hour': 22,\n"," 'minute': 1,\n"," 'month': 9,\n"," 'second': 43,\n"," 'string': '2016-09-22 22:01:43',\n"," 'year': 2016}\n","Span 2: {'datetime': datetime.datetime(2016, 9, 23, 0, 51, 24),\n"," 'day': 23,\n"," 'hour': 0,\n"," 'minute': 51,\n"," 'month': 9,\n"," 'second': 24,\n"," 'string': '2016-09-23 00:51:24',\n"," 'year': 2016} - {'datetime': datetime.datetime(2017, 9, 18, 8, 58, 26),\n"," 'day': 18,\n"," 'hour': 8,\n"," 'minute': 58,\n"," 'month': 9,\n"," 'second': 26,\n"," 'string': '2017-09-18 08:58:26',\n"," 'year': 2017}\n","Span 3: {'datetime': datetime.datetime(2017, 9, 18, 9, 0, 40),\n"," 'day': 18,\n"," 'hour': 9,\n"," 'minute': 0,\n"," 'month': 9,\n"," 'second': 40,\n"," 'string': '2017-09-18 09:00:40',\n"," 'year': 2017} - {'datetime': datetime.datetime(2018, 9, 24, 16, 48, 11),\n"," 'day': 24,\n"," 'hour': 16,\n"," 'minute': 48,\n"," 'month': 9,\n"," 'second': 11,\n"," 'string': '2018-09-24 16:48:11',\n"," 'year': 2018}\n","Span 4: {'datetime': datetime.datetime(2018, 9, 29, 6, 55, 25),\n"," 'day': 29,\n"," 'hour': 6,\n"," 'minute': 55,\n"," 'month': 9,\n"," 'second': 25,\n"," 'string': '2018-09-29 06:55:25',\n"," 'year': 2018} - {'datetime': datetime.datetime(2018, 9, 29, 6, 55, 25),\n"," 'day': 29,\n"," 'hour': 6,\n"," 'minute': 55,\n"," 'month': 9,\n"," 'second': 25,\n"," 'string': '2018-09-29 06:55:25',\n"," 'year': 2018}\n","Span 5: {'datetime': datetime.datetime(2018, 9, 30, 12, 48, 13),\n"," 'day': 30,\n"," 'hour': 12,\n"," 'minute': 48,\n"," 'month': 9,\n"," 'second': 13,\n"," 'string': '2018-09-30 12:48:13',\n"," 'year': 2018} - {'datetime': datetime.datetime(2019, 2, 28, 15, 35, 54),\n"," 'day': 28,\n"," 'hour': 15,\n"," 'minute': 35,\n"," 'month': 2,\n"," 'second': 54,\n"," 'string': '2019-02-28 15:35:54',\n"," 'year': 2019}\n","Span 6: {'datetime': datetime.datetime(2019, 2, 28, 15, 38, 47),\n"," 'day': 28,\n"," 'hour': 15,\n"," 'minute': 38,\n"," 'month': 2,\n"," 'second': 47,\n"," 'string': '2019-02-28 15:38:47',\n"," 'year': 2019} - {'datetime': datetime.datetime(2019, 5, 20, 16, 37, 18),\n"," 'day': 20,\n"," 'hour': 16,\n"," 'minute': 37,\n"," 'month': 5,\n"," 'second': 18,\n"," 'string': '2019-05-20 16:37:18',\n"," 'year': 2019}\n","Span 7: {'datetime': datetime.datetime(2019, 5, 21, 17, 37, 36),\n"," 'day': 21,\n"," 'hour': 17,\n"," 'minute': 37,\n"," 'month': 5,\n"," 'second': 36,\n"," 'string': '2019-05-21 17:37:36',\n"," 'year': 2019} - {'datetime': datetime.datetime(2020, 10, 7, 14, 20, 24),\n"," 'day': 7,\n"," 'hour': 14,\n"," 'minute': 20,\n"," 'month': 10,\n"," 'second': 24,\n"," 'string': '2020-10-07 14:20:24',\n"," 'year': 2020}\n","Span 8: {'datetime': datetime.datetime(2020, 10, 8, 1, 22, 24),\n"," 'day': 8,\n"," 'hour': 1,\n"," 'minute': 22,\n"," 'month': 10,\n"," 'second': 24,\n"," 'string': '2020-10-08 01:22:24',\n"," 'year': 2020} - {'datetime': datetime.datetime(2020, 10, 10, 21, 4, 18),\n"," 'day': 10,\n"," 'hour': 21,\n"," 'minute': 4,\n"," 'month': 10,\n"," 'second': 18,\n"," 'string': '2020-10-10 21:04:18',\n"," 'year': 2020}\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"XPiKH4pzvbRs","colab":{"base_uri":"https://localhost:8080/","height":52},"executionInfo":{"status":"ok","timestamp":1603732761615,"user_tz":-60,"elapsed":937,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"5bed8eb7-3188-4821-db88-9ab8092531c6"},"source":["mojica = occurance_spans('Mojica', CRISPR_en)\n","# for span in doudna:\n","#   print(span[0].timestamp, span[1].timestamp)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"Mojica\" was found in the following revisions of Test:\n","Span 1: 2010-01-11T02:11:54Z - 2020-09-02T08:32:27Z\n"],"name":"stdout"}]}]}