{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"arno_experiments.ipynb","provenance":[{"file_id":"1IK-QF90VpGAZmfH0XqDrD8doJfuC6B92","timestamp":1610636648085}],"collapsed_sections":["BG3rjcwdsAvH"],"authorship_tag":"ABX9TyNz4w90FaLnfQlnCyFcOUkh"},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"n4eNz7RIrzow"},"source":["## Set globals"]},{"cell_type":"code","metadata":{"id":"PJwyfS7eiPWw","colab":{"base_uri":"https://localhost:8080/","height":175},"executionInfo":{"status":"ok","timestamp":1610655807084,"user_tz":-60,"elapsed":6667,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"a378edc5-ef34-4db8-dc93-e118648e2bd8"},"source":["# set paths\n","path_extractions = '/content/drive/My Drive/WIKING/extractions'\n","path_git = '/content/drive/My Drive/WIKING/WIKING_git'\n","path_my_tools = '/content/drive/My Drive/my_tools'\n","\n","# infiles\n","file_CRISPR_de = path_extractions + '/CRISPR_de' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/\n","file_CRISPR_en = path_extractions + '/CRISPR_en' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/\n","file_CRISPR_gene_editing_en = path_extractions + '/CRISPR_gene_editing_en' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/ \n","file_events = path_git + '/data/CRISPR_events - events.csv'\n","file_accounts = path_git + '/data/CRISPR_events - accounts.csv'\n","\n","# outfiles\n","file_contexts_CRISPR_en = path_extractions + '/contexts_CRISPR_en.xlsx'\n","file_contexts_CRISPR_gene_editing_en = path_extractions + '/contexts_CRISPR_gene_editing_en.xlsx'\n","\n","# mount Google Drive\n","from google.colab import drive\n","drive.mount('/content/drive')\n","\n","# add git folder to path variable\n","import sys\n","sys.path.append(path_git + '/code')\n","\n","# import Wolfgang's classes:\n","from article.article import Article\n","\n","# import other modules\n","!pip install fuzzywuzzy # https://github.com/seatgeek/fuzzywuzzy\n","from fuzzywuzzy import fuzz\n","from fuzzywuzzy import process\n","import networkx as nx\n","!pip install python-louvain\n","from community import community_louvain\n","import re\n","import os\n","import json\n","import pandas as pd\n","import numpy as np\n","import matplotlib.pyplot as plt\n","import itertools\n","import pprint\n","pp = pprint.PrettyPrinter(indent=4)\n","from lxml import etree\n","\n","# set working directory\n","os.chdir(path_git)\n","os.getcwd()"],"execution_count":1,"outputs":[{"output_type":"stream","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n","Requirement already satisfied: fuzzywuzzy in /usr/local/lib/python3.6/dist-packages (0.18.0)\n"],"name":"stdout"},{"output_type":"stream","text":["/usr/local/lib/python3.6/dist-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning\n","  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')\n"],"name":"stderr"},{"output_type":"stream","text":["Requirement already satisfied: python-louvain in /usr/local/lib/python3.6/dist-packages (0.15)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages (from python-louvain) (1.19.5)\n","Requirement already satisfied: networkx in /usr/local/lib/python3.6/dist-packages (from python-louvain) (2.5)\n","Requirement already satisfied: decorator>=4.3.0 in /usr/local/lib/python3.6/dist-packages (from networkx->python-louvain) (4.4.2)\n"],"name":"stdout"},{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'/content/drive/My Drive/WIKING/WIKING_git'"]},"metadata":{"tags":[]},"execution_count":1}]},{"cell_type":"markdown","metadata":{"id":"BG3rjcwdsAvH"},"source":["## Functions"]},{"cell_type":"code","metadata":{"id":"VA1WhsvGiqjB"},"source":["def find_all(text, keyphrase): # https://stackoverflow.com/questions/4664850/how-to-find-all-occurrences-of-a-substring\n","  ''' Returns indices for all matches of keyphrase '''\n","  start = 0\n","  while True:\n","    start = text.find(keyphrase, start)\n","    if start == -1: \n","      return\n","    yield start\n","    start += len(keyphrase) # use start += 1 to find overlapping matches"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"eLjQwf0rnLmi"},"source":["def keyphrase_in_context(keyphrase, article, width=50, score_cutoff=75, scorer=fuzz.ratio):\n","  ''' \n","  Returns df for contexts to the left and to the right of keyphrase\n","  '''\n","  gen = article.yield_revisions()\n","  contexts = []\n","  for revision in gen:\n","    text = revision.get_text()\n","    text = text.lower()\n","    if keyphrase in text:\n","      for indx in find_all(text, keyphrase):\n","        # context left\n","        left = ''\n","        for char in text[indx - width : indx][::-1]:\n","          if char == '\\n': # make '\\n' the boundary for context\n","            break\n","          left += char\n","        left = left[::-1].strip()\n","        # context right\n","        right = ''\n","        for char in text[indx + len(keyphrase): indx + len(keyphrase) + width]:\n","          if char == '\\n': # make '\\n' the boundary for context\n","            break\n","          right += char\n","        right = right.strip()\n","        contexts.append((revision.index, revision.timestamp.string, 0, left, keyphrase, right, 0, revision.revid, revision.url, revision.user, revision.userid, revision.comment,))\n","  df = pd.DataFrame(contexts, columns=['revindx','timestamp','left_id','left','keyphrase','right','right_id','revid','url','user','userid','comment',])\n","  \n","  # determine communities of similarity for left and right contexts\n","  for side in ['left', 'right']:\n","    G = nx.Graph()\n","    unique_contexts = list(df[side].unique())\n","    length = len(unique_contexts)\n","    for query in unique_contexts:\n","      indx = unique_contexts.index(query) + 1\n","      if indx <= length:\n","        scores = process.extractBests(query, unique_contexts[unique_contexts.index(query) + 1:], score_cutoff=score_cutoff, limit=length, scorer=scorer) # https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/process.py\n","        edges = [(query, i[0]) for i in scores]\n","        G.add_edges_from(edges)\n","    # add community_ids for lonesome contexts as well\n","    unique_contexts = set(unique_contexts)\n","    community_dic = community_louvain.best_partition(G)\n","    if community_dic:\n","      community_contexts = set(community_dic.keys())\n","      lonesome_contexts = unique_contexts - community_contexts\n","      lonesome_dic = {k:v for v,k in enumerate(lonesome_contexts,start=max(community_dic.values())+1)}\n","      community_dic.update(lonesome_dic) # hiermit nun 'lid' und 'rid' Spalten befüllen...\n","    else:\n","      community_dic = {k:v for v,k in enumerate(unique_contexts)}\n","    # add community_ids to df\n","    column = df.apply(lambda row: community_dic[row[side]], axis=1)\n","    df['{}_id'.format(side)] = column\n","  return df\n","  # # add community_ids for lonesome contexts as well\n","  # unique_contexts = set(unique_contexts['left'] + unique_contexts['right'])\n","  # community_dic = community_louvain.best_partition(G)\n","  # community_contexts = set(community_dic.keys())\n","  # lonesome_contexts = unique_contexts - community_contexts\n","  # lonesome_dic = {k:v for v,k in enumerate(lonesome_contexts,start=max(community_dic.values())+1)}\n","  # community_dic.update(lonesome_dic) # hiermit nun 'lid' und 'rid' Spalten befüllen...\n","  # # add community_ids for all left and right contexts in df\n","  # for side in ['left', 'right']:\n","  #   column = df.apply(lambda row: community_dic[row[side]], axis=1)\n","  #   df['{}_id'.format(side)] = column\n","  # return df"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"aodKMqgBxmb8"},"source":["## Load data"]},{"cell_type":"code","metadata":{"id":"gD4IDC16epDS","executionInfo":{"status":"ok","timestamp":1610655812345,"user_tz":-60,"elapsed":815,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}}},"source":["CRISPR_en = Article(file_CRISPR_en)\n","CRISPR_gene_editing_en = Article(file_CRISPR_gene_editing_en)\n","# events = pd.read_csv(file_events)\n","# accounts = pd.read_csv(file_accounts)"],"execution_count":2,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"kr9rzPh5zNIA"},"source":["## Experiments"]},{"cell_type":"markdown","metadata":{"id":"_l9TB3a-Uu8o"},"source":["## Citation context"]},{"cell_type":"code","metadata":{"id":"UX675SB6iQbt","executionInfo":{"status":"ok","timestamp":1610655824955,"user_tz":-60,"elapsed":11739,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}}},"source":["# # get revisions\n","article = CRISPR_en\n","revision1 = article.get_revision(revid=603962693)\n","revision2 = article.get_revision(revid=998336424)\n","\n","target_reference_id = 20056882 # PMID von Horvath P, Barrangou R (January 2010). \"CRISPR/Cas, the immune system of bacteria and archaea\". Science. 327 (5962): 167–170. "],"execution_count":3,"outputs":[]},{"cell_type":"code","metadata":{"id":"lj6EHU3_oXHq"},"source":["revisions = [revision1, revision2]\n","for revision in revisions:\n","  references = revision.get_references()\n","  if target_reference_id in []"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"baN9q9M5pZHj","executionInfo":{"status":"ok","timestamp":1610657247084,"user_tz":-60,"elapsed":680,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"43dceaf7-962e-4447-a748-4893bbba16f2"},"source":["d = {k:'' for k in ['DOI','PMC','PMID']}\n","ids = {'PMC': 'PMC6628903', 'DOI': '10.1126%2Fscience.aar6245', }\n","d.update(ids)\n","d"],"execution_count":9,"outputs":[{"output_type":"execute_result","data":{"text/plain":["{'DOI': '10.1126%2Fscience.aar6245', 'PMC': 'PMC6628903', 'PMID': ''}"]},"metadata":{"tags":[]},"execution_count":9}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"s6HmznMJY1Pw","executionInfo":{"status":"ok","timestamp":1610655923747,"user_tz":-60,"elapsed":1278,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"a885beb3-3404-422a-a2bd-7f8ae384d511"},"source":["# pick one revision\n","revision = revision2\n","print(revision.url, \"\\n\")\n","\n","#get references\n","references = revision.get_references()\n","for reference in references:\n","  print(reference.get_text())\n","  print(reference.get_dois())\n","  print(reference.get_pmcs())\n","  print(reference.get_pmids())\n","  print(reference.get_identifiers())\n","\n","  # # dict comprehension (same as reference.get_identifiers())\n","  # nofollows = reference.source.xpath(\".//a[@rel='nofollow']\")\n","  # external_ids = {'DOI' if 'doi.org' in tag.get('href') \n","  #           else 'PMC' if 'pmc' in tag.get('href') \n","  #           else 'PMID':\n","  #                        tag.get('href').split('/')[-1] if 'doi.org' in tag.get('href')\n","  #                   else tag.get('href').split('/')[-1].split('PMC')[-1] if 'pmc' in tag.get('href')\n","  #                   else tag.get('href').split('/')[-1].split('?')[0] # = 'pubmed'\n","  #           for tag in nofollows \n","  #           if any(i in tag.get('href') for i in ['doi.org','pmc','pubmed'])}\n","  # print('\\t',external_ids)\n","\n"],"execution_count":5,"outputs":[{"output_type":"stream","text":["https://en.wikipedia.org/w/index.php?title=CRISPR&oldid=998336424 \n","\n","^  Subtype I-G was previously known as subtype I-U.[73] \n","\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","^  Subtype V-K was previously known as subtype V-U5.[83] \n","\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Mulepati S, Héroux A, Bailey S (2014). \"Crystal structure of a CRISPR RNA–guided surveillance complex bound to a ssDNA target\". Science. 345 (6203): 1479–1484. Bibcode:2014Sci...345.1479M. doi:10.1126/science.1256996. PMC 4427192. PMID 25123481.\n","['10.1126/science.1256996']\n","['4427192']\n","['25123481']\n","{'PMC': 'PMC4427192', 'DOI': '10.1126%2Fscience.1256996', 'PMID': '25123481'}\n","\t {'PMC': '4427192', 'DOI': '10.1126%2Fscience.1256996', 'PMID': '25123481'}\n","Barrangou R (2015). \"The roles of CRISPR-Cas systems in adaptive immunity and beyond\". Current Opinion in Immunology. 32: 36–41. doi:10.1016/j.coi.2014.12.008. PMID 25574773.\n","['10.1016/j.coi.2014.12.008']\n","[]\n","['25574773']\n","{'DOI': '10.1016%2Fj.coi.2014.12.008', 'PMID': '25574773'}\n","\t {'DOI': '10.1016%2Fj.coi.2014.12.008', 'PMID': '25574773'}\n","Horvath P, Barrangou R (January 2010). \"CRISPR/Cas, the immune system of bacteria and archaea\". Science. 327 (5962): 167–170. Bibcode:2010Sci...327..167H. doi:10.1126/science.1179555. PMID 20056882. S2CID 17960960.\n","['10.1126/science.1179555']\n","[]\n","['20056882']\n","{'DOI': '10.1126%2Fscience.1179555', 'PMID': '20056882'}\n","\t {'DOI': '10.1126%2Fscience.1179555', 'PMID': '20056882'}\n","Redman M, King A, Watson C, King D (August 2016). \"What is CRISPR/Cas9?\". Archives of Disease in Childhood. Education and Practice Edition. 101 (4): 213–215. doi:10.1136/archdischild-2016-310459. PMC 4975809. PMID 27059283.\n","['10.1136/archdischild-2016-310459']\n","['4975809']\n","['27059283']\n","{'PMC': 'PMC4975809', 'DOI': '10.1136%2Farchdischild-2016-310459', 'PMID': '27059283'}\n","\t {'PMC': '4975809', 'DOI': '10.1136%2Farchdischild-2016-310459', 'PMID': '27059283'}\n","Barrangou R, Fremaux C, Deveau H, Richards M, Boyaval P, Moineau S,  et al. (March 2007). \"CRISPR provides acquired resistance against viruses in prokaryotes\". Science. 315 (5819): 1709–1712. Bibcode:2007Sci...315.1709B. doi:10.1126/science.1138140. hdl:20.500.11794/38902. PMID 17379808. S2CID 3888761.\n","['10.1126/science.1138140']\n","[]\n","['17379808']\n","{'DOI': '10.1126%2Fscience.1138140', 'PMID': '17379808'}\n","\t {'DOI': '10.1126%2Fscience.1138140', 'PMID': '17379808'}\n","Marraffini LA, Sontheimer EJ (December 2008). \"CRISPR interference limits horizontal gene transfer in staphylococci by targeting DNA\". Science. 322 (5909): 1843–1845. Bibcode:2008Sci...322.1843M. doi:10.1126/science.1165771. PMC 2695655. PMID 19095942.\n","['10.1126/science.1165771']\n","['2695655']\n","['19095942']\n","{'PMC': 'PMC2695655', 'DOI': '10.1126%2Fscience.1165771', 'PMID': '19095942'}\n","\t {'PMC': '2695655', 'DOI': '10.1126%2Fscience.1165771', 'PMID': '19095942'}\n","Mohanraju P, Makarova KS, Zetsche B, Zhang F, Koonin EV, van der Oost J (2016). \"Diverse evolutionary roots and mechanistic variations of the CRISPR-Cas systems\" (PDF). Science. 353 (6299): aad5147. doi:10.1126/science.aad5147. hdl:1721.1/113195. PMID 27493190. S2CID 11086282.\n","['10.1126/science.aad5147']\n","[]\n","['27493190']\n","{'DOI': '10.1126%2Fscience.aad5147', 'PMID': '27493190'}\n","\t {'DOI': '10.1126%2Fscience.aad5147', 'PMID': '27493190'}\n","Hille F, Richter H, Wong SP, Bratovič M, Ressel S, Charpentier E (March 2018). \"The Biology of CRISPR-Cas: Backward and Forward\". Cell. 172 (6): 1239–1259. doi:10.1016/j.cell.2017.11.032. hdl:21.11116/0000-0003-FC0D-4. PMID 29522745. S2CID 3777503.\n","['10.1016/j.cell.2017.11.032']\n","[]\n","['29522745']\n","{'DOI': '10.1016%2Fj.cell.2017.11.032', 'PMID': '29522745'}\n","\t {'DOI': '10.1016%2Fj.cell.2017.11.032', 'PMID': '29522745'}\n","Zhang F, Wen Y, Guo X (2014). \"CRISPR/Cas9 for genome editing: progress, implications and challenges\". Human Molecular Genetics. 23 (R1): R40–6. doi:10.1093/hmg/ddu125. PMID 24651067.\n","['10.1093/hmg/ddu125']\n","[]\n","['24651067']\n","{'DOI': '10.1093%2Fhmg%2Fddu125', 'PMID': '24651067'}\n","\t {'DOI': '10.1093%2Fhmg%2Fddu125', 'PMID': '24651067'}\n","^ CRISPR-CAS9, TALENS and ZFNS - the battle in gene editing https://www.ptglab.com/news/blog/crispr-cas9-talens-and-zfns-the-battle-in-gene-editing/\n","\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Hsu PD, Lander ES, Zhang F (June 2014). \"Development and applications of CRISPR-Cas9 for genome engineering\". Cell. 157 (6): 1262–1278. doi:10.1016/j.cell.2014.05.010. PMC 4343198. PMID 24906146.\n","['10.1016/j.cell.2014.05.010']\n","['4343198']\n","['24906146']\n","{'PMC': 'PMC4343198', 'DOI': '10.1016%2Fj.cell.2014.05.010', 'PMID': '24906146'}\n","\t {'PMC': '4343198', 'DOI': '10.1016%2Fj.cell.2014.05.010', 'PMID': '24906146'}\n","\"Press release: The Nobel Prize in Chemistry 2020\". Nobel Foundation. Retrieved 7 October 2020.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Wu KJ, Peltier E (7 October 2020). \"Nobel Prize in Chemistry Awarded to 2 Scientists for Work on Genome Editing – Emmanuelle Charpentier and Jennifer A. Doudna developed the Crispr tool, which can alter the DNA of animals, plants and microorganisms with high precision\". The New York Times. Retrieved 7 October 2020.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Ishino Y, Shinagawa H, Makino K, Amemura M, Nakata A (December 1987). \"Nucleotide sequence of the iap gene, responsible for alkaline phosphatase isozyme conversion in Escherichia coli, and identification of the gene product\". Journal of Bacteriology. 169 (12): 5429–5433. doi:10.1128/jb.169.12.5429-5433.1987. PMC 213968. PMID 3316184.\n","['10.1128/jb.169.12.5429-5433.1987']\n","['213968']\n","['3316184']\n","{'PMC': 'PMC213968', 'DOI': '10.1128%2Fjb.169.12.5429-5433.1987', 'PMID': '3316184'}\n","\t {'PMC': '213968', 'DOI': '10.1128%2Fjb.169.12.5429-5433.1987', 'PMID': '3316184'}\n","van Soolingen D, de Haas PE, Hermans PW, Groenen PM, van Embden JD (August 1993). \"Comparison of various repetitive DNA elements as genetic markers for strain differentiation and epidemiology of Mycobacterium tuberculosis\". Journal of Clinical Microbiology. 31 (8): 1987–1995. doi:10.1128/JCM.31.8.1987-1995.1993. PMC 265684. PMID 7690367.\n","['10.1128/JCM.31.8.1987-1995.1993']\n","['265684']\n","['7690367']\n","{'PMC': 'PMC265684', 'DOI': '10.1128%2FJCM.31.8.1987-1995.1993', 'PMID': '7690367'}\n","\t {'PMC': '265684', 'DOI': '10.1128%2FJCM.31.8.1987-1995.1993', 'PMID': '7690367'}\n","Groenen PM, Bunschoten AE, van Soolingen D, van Embden JD (December 1993). \"Nature of DNA polymorphism in the direct repeat cluster of Mycobacterium tuberculosis; application for strain differentiation by a novel typing method\". Molecular Microbiology. 10 (5): 1057–1065. doi:10.1111/j.1365-2958.1993.tb00976.x. PMID 7934856. S2CID 25304723.\n","['10.1111/j.1365-2958.1993.tb00976.x']\n","[]\n","['7934856']\n","{'DOI': '10.1111%2Fj.1365-2958.1993.tb00976.x', 'PMID': '7934856'}\n","\t {'DOI': '10.1111%2Fj.1365-2958.1993.tb00976.x', 'PMID': '7934856'}\n","Mojica FJ, Montoliu L (2016). \"On the Origin of CRISPR-Cas Technology: From Prokaryotes to Mammals\". Trends in Microbiology. 24 (10): 811–820. doi:10.1016/j.tim.2016.06.005. PMID 27401123.\n","['10.1016/j.tim.2016.06.005']\n","[]\n","['27401123']\n","{'DOI': '10.1016%2Fj.tim.2016.06.005', 'PMID': '27401123'}\n","\t {'DOI': '10.1016%2Fj.tim.2016.06.005', 'PMID': '27401123'}\n","Mojica FJ, Rodriguez-Valera F (2016). \"The discovery of CRISPR in archaea and bacteria\" (PDF). The FEBS Journal. 283 (17): 3162–3169. doi:10.1111/febs.13766. hdl:10045/57676. PMID 27234458. S2CID 42827598.\n","['10.1111/febs.13766']\n","[]\n","['27234458']\n","{'DOI': '10.1111%2Ffebs.13766', 'PMID': '27234458'}\n","\t {'DOI': '10.1111%2Ffebs.13766', 'PMID': '27234458'}\n","Mojica FJ, Díez-Villaseñor C, Soria E, Juez G (April 2000). \"Biological significance of a family of regularly spaced repeats in the genomes of Archaea, Bacteria and mitochondria\". Molecular Microbiology. 36 (1): 244–246. doi:10.1046/j.1365-2958.2000.01838.x. PMID 10760181.\n","['10.1046/j.1365-2958.2000.01838.x']\n","[]\n","['10760181']\n","{'DOI': '10.1046%2Fj.1365-2958.2000.01838.x', 'PMID': '10760181'}\n","\t {'DOI': '10.1046%2Fj.1365-2958.2000.01838.x', 'PMID': '10760181'}\n","Barrangou R, van der Oost J (2013). CRISPR-Cas Systems : RNA-mediated Adaptive Immunity in Bacteria and Archaea. Heidelberg: Springer. p. 6. ISBN 978-3-642-34656-9.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Tang TH, Bachellerie JP, Rozhdestvensky T, Bortolin ML, Huber H, Drungowski M;  et al. (2002). \"Identification of 86 candidates for small non-messenger RNAs from the archaeon Archaeoglobus fulgidus\". Proc Natl Acad Sci U S A. 99 (11): 7536–41. Bibcode:2002PNAS...99.7536T. doi:10.1073/pnas.112047299. PMC 124276. PMID 12032318.\n","['10.1073/pnas.112047299']\n","['124276']\n","['12032318']\n","{'PMID': '12032318', 'DOI': '10.1073%2Fpnas.112047299', 'PMC': 'PMC124276'}\n","\t {'PMID': '12032318', 'DOI': '10.1073%2Fpnas.112047299', 'PMC': '124276'}\n","Charpentier E, Richter H, van der Oost J, White MF (May 2015). \"Biogenesis pathways of RNA guides in archaeal and bacterial CRISPR-Cas adaptive immunity\". FEMS Microbiology Reviews. 39 (3): 428–441. doi:10.1093/femsre/fuv023. PMC 5965381. PMID 25994611.\n","['10.1093/femsre/fuv023']\n","['5965381']\n","['25994611']\n","{'PMC': 'PMC5965381', 'DOI': '10.1093%2Ffemsre%2Ffuv023', 'PMID': '25994611'}\n","\t {'PMC': '5965381', 'DOI': '10.1093%2Ffemsre%2Ffuv023', 'PMID': '25994611'}\n","Jansen R, Embden JD, Gaastra W, Schouls LM (March 2002). \"Identification of genes that are associated with DNA repeats in prokaryotes\". Molecular Microbiology. 43 (6): 1565–1575. doi:10.1046/j.1365-2958.2002.02839.x. PMID 11952905. S2CID 23196085.\n","['10.1046/j.1365-2958.2002.02839.x']\n","[]\n","['11952905']\n","{'DOI': '10.1046%2Fj.1365-2958.2002.02839.x', 'PMID': '11952905'}\n","\t {'DOI': '10.1046%2Fj.1365-2958.2002.02839.x', 'PMID': '11952905'}\n","Horvath P, Barrangou R (January 2010). \"CRISPR/Cas, the immune system of bacteria and archaea\". Science. 327 (5962): 167–170. Bibcode:2010Sci...327..167H. doi:10.1126/Science.1179555. PMID 20056882. S2CID 17960960.\n","['10.1126/Science.1179555']\n","[]\n","['20056882']\n","{'DOI': '10.1126%2FScience.1179555', 'PMID': '20056882'}\n","\t {'DOI': '10.1126%2FScience.1179555', 'PMID': '20056882'}\n","Marraffini LA, Sontheimer EJ (March 2010). \"CRISPR interference: RNA-directed adaptive immunity in bacteria and archaea\". Nature Reviews Genetics. 11 (3): 181–190. doi:10.1038/nrg2749. PMC 2928866. PMID 20125085.\n","['10.1038/nrg2749']\n","['2928866']\n","['20125085']\n","{'PMC': 'PMC2928866', 'DOI': '10.1038%2Fnrg2749', 'PMID': '20125085'}\n","\t {'PMC': '2928866', 'DOI': '10.1038%2Fnrg2749', 'PMID': '20125085'}\n","Grissa I, Vergnaud G, Pourcel C (May 2007). \"The CRISPRdb database and tools to display CRISPRs and to generate dictionaries of spacers and repeats\". BMC Bioinformatics. 8: 172. doi:10.1186/1471-2105-8-172. PMC 1892036. PMID 17521438.\n","['10.1186/1471-2105-8-172']\n","['1892036']\n","['17521438']\n","{'PMC': 'PMC1892036', 'DOI': '10.1186%2F1471-2105-8-172', 'PMID': '17521438'}\n","\t {'PMC': '1892036', 'DOI': '10.1186%2F1471-2105-8-172', 'PMID': '17521438'}\n","Pourcel C, Salvignol G, Vergnaud G (March 2005). \"CRISPR elements in Yersinia pestis acquire new repeats by preferential uptake of bacteriophage DNA, and provide additional tools for evolutionary studies\". Microbiology. 151 (Pt 3): 653–663. doi:10.1099/mic.0.27437-0. PMID 15758212.\n","['10.1099/mic.0.27437-0']\n","[]\n","['15758212']\n","{'DOI': '10.1099%2Fmic.0.27437-0', 'PMID': '15758212'}\n","\t {'DOI': '10.1099%2Fmic.0.27437-0', 'PMID': '15758212'}\n","Mojica FJ, Díez-Villaseñor C, García-Martínez J, Soria E (February 2005). \"Intervening sequences of regularly spaced prokaryotic repeats derive from foreign genetic elements\". Journal of Molecular Evolution. 60 (2): 174–182. Bibcode:2005JMolE..60..174M. doi:10.1007/s00239-004-0046-3. PMID 15791728. S2CID 27481111.\n","['10.1007/s00239-004-0046-3']\n","[]\n","['15791728']\n","{'DOI': '10.1007%2Fs00239-004-0046-3', 'PMID': '15791728'}\n","\t {'DOI': '10.1007%2Fs00239-004-0046-3', 'PMID': '15791728'}\n","Bolotin A, Quinquis B, Sorokin A, Ehrlich SD (August 2005). \"Clustered regularly interspaced short palindrome repeats (CRISPRs) have spacers of extrachromosomal origin\". Microbiology. 151 (Pt 8): 2551–2561. doi:10.1099/mic.0.28048-0. PMID 16079334.\n","['10.1099/mic.0.28048-0']\n","[]\n","['16079334']\n","{'DOI': '10.1099%2Fmic.0.28048-0', 'PMID': '16079334'}\n","\t {'DOI': '10.1099%2Fmic.0.28048-0', 'PMID': '16079334'}\n","Morange M (June 2015). \"What history tells us XXXVII. CRISPR-Cas: The discovery of an immune system in prokaryotes\". Journal of Biosciences. 40 (2): 221–223. doi:10.1007/s12038-015-9532-6. PMID 25963251.\n","['10.1007/s12038-015-9532-6']\n","[]\n","['25963251']\n","{'DOI': '10.1007%2Fs12038-015-9532-6', 'PMID': '25963251'}\n","\t {'DOI': '10.1007%2Fs12038-015-9532-6', 'PMID': '25963251'}\n","Lander ES (January 2016). \"The Heroes of CRISPR\". Cell. 164 (1–2): 18–28. doi:10.1016/j.cell.2015.12.041. PMID 26771483.\n","['10.1016/j.cell.2015.12.041']\n","[]\n","['26771483']\n","{'DOI': '10.1016%2Fj.cell.2015.12.041', 'PMID': '26771483'}\n","\t {'DOI': '10.1016%2Fj.cell.2015.12.041', 'PMID': '26771483'}\n","Makarova KS, Grishin NV, Shabalina SA, Wolf YI, Koonin EV (March 2006). \"A putative RNA-interference-based immune system in prokaryotes: computational analysis of the predicted enzymatic machinery, functional analogies with eukaryotic RNAi, and hypothetical mechanisms of action\". Biology Direct. 1: 7. doi:10.1186/1745-6150-1-7. PMC 1462988. PMID 16545108.\n","['10.1186/1745-6150-1-7']\n","['1462988']\n","['16545108']\n","{'PMC': 'PMC1462988', 'DOI': '10.1186%2F1745-6150-1-7', 'PMID': '16545108'}\n","\t {'PMC': '1462988', 'DOI': '10.1186%2F1745-6150-1-7', 'PMID': '16545108'}\n","Marraffini LA (October 2015). \"CRISPR-Cas immunity in prokaryotes\". Nature. 526 (7571): 55–61. Bibcode:2015Natur.526...55M. doi:10.1038/nature15386. PMID 26432244. S2CID 3718361.\n","['10.1038/nature15386']\n","[]\n","['26432244']\n","{'DOI': '10.1038%2Fnature15386', 'PMID': '26432244'}\n","\t {'DOI': '10.1038%2Fnature15386', 'PMID': '26432244'}\n","Pennisi E (August 2013). \"The CRISPR craze\". News Focus. Science. 341 (6148): 833–836. Bibcode:2013Sci...341..833P. doi:10.1126/science.341.6148.833. PMID 23970676.\n","['10.1126/science.341.6148.833']\n","[]\n","['23970676']\n","{'DOI': '10.1126%2Fscience.341.6148.833', 'PMID': '23970676'}\n","\t {'DOI': '10.1126%2Fscience.341.6148.833', 'PMID': '23970676'}\n","Brouns SJ, Jore MM, Lundgren M, Westra ER, Slijkhuis RJ, Snijders AP, Dickman MJ, Makarova KS, Koonin EV, van der Oost J (August 2008). \"Small CRISPR RNAs guide antiviral defense in prokaryotes\". Science. 321 (5891): 960–964. Bibcode:2008Sci...321..960B. doi:10.1126/science.1159689. PMC 5898235. PMID 18703739.\n","['10.1126/science.1159689']\n","['5898235']\n","['18703739']\n","{'PMC': 'PMC5898235', 'DOI': '10.1126%2Fscience.1159689', 'PMID': '18703739'}\n","\t {'PMC': '5898235', 'DOI': '10.1126%2Fscience.1159689', 'PMID': '18703739'}\n","Garneau JE, Dupuis MÈ, Villion M, Romero DA, Barrangou R, Boyaval P,  et al. (November 2010). \"The CRISPR/Cas bacterial immune system cleaves bacteriophage and plasmid DNA\". Nature. 468 (7320): 67–71. Bibcode:2010Natur.468...67G. CiteSeerX 10.1.1.451.9645. doi:10.1038/nature09523. PMID 21048762. S2CID 205222849.\n","['10.1038/nature09523']\n","[]\n","['21048762']\n","{'DOI': '10.1038%2Fnature09523', 'PMID': '21048762'}\n","\t {'DOI': '10.1038%2Fnature09523', 'PMID': '21048762'}\n","Deltcheva E, Chylinski K, Sharma CM, Gonzales K, Chao Y, Pirzada ZA, Eckert MR, Vogel J, Charpentier E (March 2011). \"CRISPR RNA maturation by trans-encoded small RNA and host factor RNase III\". Nature. 471 (7340): 602–607. Bibcode:2011Natur.471..602D. doi:10.1038/nature09886. PMC 3070239. PMID 21455174.\n","['10.1038/nature09886']\n","['3070239']\n","['21455174']\n","{'PMC': 'PMC3070239', 'DOI': '10.1038%2Fnature09886', 'PMID': '21455174'}\n","\t {'PMC': '3070239', 'DOI': '10.1038%2Fnature09886', 'PMID': '21455174'}\n","Barrangou R (November 2015). \"Diversity of CRISPR-Cas immune systems and molecular machines\". Genome Biology. 16: 247. doi:10.1186/s13059-015-0816-9. PMC 4638107. PMID 26549499.\n","['10.1186/s13059-015-0816-9']\n","['4638107']\n","['26549499']\n","{'PMC': 'PMC4638107', 'DOI': '10.1186%2Fs13059-015-0816-9', 'PMID': '26549499'}\n","\t {'PMC': '4638107', 'DOI': '10.1186%2Fs13059-015-0816-9', 'PMID': '26549499'}\n","Jinek M, Chylinski K, Fonfara I, Hauer M, Doudna JA, Charpentier E (August 2012). \"A programmable dual-RNA-guided DNA endonuclease in adaptive bacterial immunity\". Science. 337 (6096): 816–821. Bibcode:2012Sci...337..816J. doi:10.1126/science.1225829. PMC 6286148. PMID 22745249.\n","['10.1126/science.1225829']\n","['6286148']\n","['22745249']\n","{'PMC': 'PMC6286148', 'DOI': '10.1126%2Fscience.1225829', 'PMID': '22745249'}\n","\t {'PMC': '6286148', 'DOI': '10.1126%2Fscience.1225829', 'PMID': '22745249'}\n","Cong L, Ran FA, Cox D, Lin S, Barretto R, Habib N, Hsu PD, Wu X, Jiang W, Marraffini LA, Zhang F (February 2013). \"Multiplex genome engineering using CRISPR/Cas systems\". Science. 339 (6121): 819–823. Bibcode:2013Sci...339..819C. doi:10.1126/science.1231143. PMC 3795411. PMID 23287718.\n","['10.1126/science.1231143']\n","['3795411']\n","['23287718']\n","{'PMC': 'PMC3795411', 'DOI': '10.1126%2Fscience.1231143', 'PMID': '23287718'}\n","\t {'PMC': '3795411', 'DOI': '10.1126%2Fscience.1231143', 'PMID': '23287718'}\n","Mali P, Yang L, Esvelt KM, Aach J, Guell M, DiCarlo JE, Norville JE, Church GM (February 2013). \"RNA-guided human genome engineering via Cas9\". Science. 339 (6121): 823–826. Bibcode:2013Sci...339..823M. doi:10.1126/science.1232033. PMC 3712628. PMID 23287722.\n","['10.1126/science.1232033']\n","['3712628']\n","['23287722']\n","{'PMC': 'PMC3712628', 'DOI': '10.1126%2Fscience.1232033', 'PMID': '23287722'}\n","\t {'PMC': '3712628', 'DOI': '10.1126%2Fscience.1232033', 'PMID': '23287722'}\n","DiCarlo JE, Norville JE, Mali P, Rios X, Aach J, Church GM (April 2013). \"Genome engineering in Saccharomyces cerevisiae using CRISPR-Cas systems\". Nucleic Acids Research. 41 (7): 4336–4343. doi:10.1093/nar/gkt135. PMC 3627607. PMID 23460208.\n","['10.1093/nar/gkt135']\n","['3627607']\n","['23460208']\n","{'PMC': 'PMC3627607', 'DOI': '10.1093%2Fnar%2Fgkt135', 'PMID': '23460208'}\n","\t {'PMC': '3627607', 'DOI': '10.1093%2Fnar%2Fgkt135', 'PMID': '23460208'}\n","Zhang GC, Kong II, Kim H, Liu JJ, Cate JH, Jin YS (December 2014). \"Construction of a quadruple auxotrophic mutant of an industrial polyploid saccharomyces cerevisiae strain by using RNA-guided Cas9 nuclease\". Applied and Environmental Microbiology. 80 (24): 7694–7701. doi:10.1128/AEM.02310-14. PMC 4249234. PMID 25281382.\n","['10.1128/AEM.02310-14']\n","['4249234']\n","['25281382']\n","{'PMC': 'PMC4249234', 'DOI': '10.1128%2FAEM.02310-14', 'PMID': '25281382'}\n","\t {'PMC': '4249234', 'DOI': '10.1128%2FAEM.02310-14', 'PMID': '25281382'}\n","Liu JJ, Kong II, Zhang GC, Jayakody LN, Kim H, Xia PF, Kwak S, Sung BH, Sohn JH, Walukiewicz HE, Rao CV, Jin YS (April 2016). \"Metabolic Engineering of Probiotic Saccharomyces boulardii\". Applied and Environmental Microbiology. 82 (8): 2280–2287. doi:10.1128/AEM.00057-16. PMC 4959471. PMID 26850302.\n","['10.1128/AEM.00057-16']\n","['4959471']\n","['26850302']\n","{'PMC': 'PMC4959471', 'DOI': '10.1128%2FAEM.00057-16', 'PMID': '26850302'}\n","\t {'PMC': '4959471', 'DOI': '10.1128%2FAEM.00057-16', 'PMID': '26850302'}\n","Vyas VK, Barrasa MI, Fink GR (2015). \"Candida albicans CRISPR system permits genetic engineering of essential genes and gene families\". Science Advances. 1 (3): e1500248. Bibcode:2015SciA....1E0248V. doi:10.1126/sciadv.1500248. PMC 4428347. PMID 25977940.\n","['10.1126/sciadv.1500248']\n","['4428347']\n","['25977940']\n","{'PMC': 'PMC4428347', 'DOI': '10.1126%2Fsciadv.1500248', 'PMID': '25977940'}\n","\t {'PMC': '4428347', 'DOI': '10.1126%2Fsciadv.1500248', 'PMID': '25977940'}\n","Ng H, Dean N (2017). \"Candida albicans by Increased Single Guide RNA Expression\". mSphere. 2 (2): e00385–16. doi:10.1128/mSphere.00385-16. PMC 5397569. PMID 28435892.\n","['10.1128/mSphere.00385-16']\n","['5397569']\n","['28435892']\n","{'PMC': 'PMC5397569', 'DOI': '10.1128%2FmSphere.00385-16', 'PMID': '28435892'}\n","\t {'PMC': '5397569', 'DOI': '10.1128%2FmSphere.00385-16', 'PMID': '28435892'}\n","Hwang WY, Fu Y, Reyon D, Maeder ML, Tsai SQ, Sander JD, Peterson RT, Yeh JR, Joung JK (March 2013). \"Efficient genome editing in zebrafish using a CRISPR-Cas system\". Nature Biotechnology. 31 (3): 227–229. doi:10.1038/nbt.2501. PMC 3686313. PMID 23360964.\n","['10.1038/nbt.2501']\n","['3686313']\n","['23360964']\n","{'PMC': 'PMC3686313', 'DOI': '10.1038%2Fnbt.2501', 'PMID': '23360964'}\n","\t {'PMC': '3686313', 'DOI': '10.1038%2Fnbt.2501', 'PMID': '23360964'}\n","Gratz SJ, Cummings AM, Nguyen JN, Hamm DC, Donohue LK, Harrison MM, Wildonger J, O'Connor-Giles KM (August 2013). \"Genome engineering of Drosophila with the CRISPR RNA-guided Cas9 nuclease\". Genetics. 194 (4): 1029–1035. doi:10.1534/genetics.113.152710. PMC 3730909. PMID 23709638.\n","['10.1534/genetics.113.152710']\n","['3730909']\n","['23709638']\n","{'PMC': 'PMC3730909', 'DOI': '10.1534%2Fgenetics.113.152710', 'PMID': '23709638'}\n","\t {'PMC': '3730909', 'DOI': '10.1534%2Fgenetics.113.152710', 'PMID': '23709638'}\n","Bassett AR, Tibbit C, Ponting CP, Liu JL (July 2013). \"Highly efficient targeted mutagenesis of Drosophila with the CRISPR/Cas9 system\". Cell Reports. 4 (1): 220–228. doi:10.1016/j.celrep.2013.06.020. PMC 3714591. PMID 23827738.\n","['10.1016/j.celrep.2013.06.020']\n","['3714591']\n","['23827738']\n","{'PMC': 'PMC3714591', 'DOI': '10.1016%2Fj.celrep.2013.06.020', 'PMID': '23827738'}\n","\t {'PMC': '3714591', 'DOI': '10.1016%2Fj.celrep.2013.06.020', 'PMID': '23827738'}\n","Yan H, Opachaloemphan C, Mancini G, Yang H, Gallitto M, Mlejnek J, Leibholz A, Haight K, Ghaninia M, Huo L, Perry M, Slone J, Zhou X, Traficante M, Penick CA, Dolezal K, Gokhale K, Stevens K, Fetter-Pruneda I, Bonasio R, Zwiebel LJ, Berger SL, Liebig J, Reinberg D, Desplan C (August 2017). \"An Engineered orco Mutation Produces Aberrant Social Behavior and Defective Neural Development in Ants\". Cell. 170 (4): 736–747.e9. doi:10.1016/j.cell.2017.06.051. PMC 5587193. PMID 28802043.\n","['10.1016/j.cell.2017.06.051']\n","['5587193']\n","['28802043']\n","{'PMID': '28802043', 'DOI': '10.1016%2Fj.cell.2017.06.051', 'PMC': 'PMC5587193'}\n","\t {'PMID': '28802043', 'DOI': '10.1016%2Fj.cell.2017.06.051', 'PMC': '5587193'}\n","Trible W, Olivos-Cisneros L, McKenzie SK, Saragosti J, Chang NC, Matthews BJ, Oxley PR, Kronauer DJ (August 2017). \"orco Mutagenesis Causes Loss of Antennal Lobe Glomeruli and Impaired Social Behavior in Ants\". Cell. 170 (4): 727–735.e10. doi:10.1016/j.cell.2017.07.001. PMC 5556950. PMID 28802042.\n","['10.1016/j.cell.2017.07.001']\n","['5556950']\n","['28802042']\n","{'PMID': '28802042', 'DOI': '10.1016%2Fj.cell.2017.07.001', 'PMC': 'PMC5556950'}\n","\t {'PMID': '28802042', 'DOI': '10.1016%2Fj.cell.2017.07.001', 'PMC': '5556950'}\n","Kistler KE, Vosshall LB, Matthews BJ (April 2015). \"Genome engineering with CRISPR-Cas9 in the mosquito Aedes aegypti\". Cell Reports. 11 (1): 51–60. doi:10.1016/j.celrep.2015.03.009. PMC 4394034. PMID 25818303.\n","['10.1016/j.celrep.2015.03.009']\n","['4394034']\n","['25818303']\n","{'PMID': '25818303', 'DOI': '10.1016%2Fj.celrep.2015.03.009', 'PMC': 'PMC4394034'}\n","\t {'PMID': '25818303', 'DOI': '10.1016%2Fj.celrep.2015.03.009', 'PMC': '4394034'}\n","Friedland AE, Tzur YB, Esvelt KM, Colaiácovo MP, Church GM, Calarco JA (August 2013). \"Heritable genome editing in C. elegans via a CRISPR-Cas9 system\". Nature Methods. 10 (8): 741–743. doi:10.1038/nmeth.2532. PMC 3822328. PMID 23817069.\n","['10.1038/nmeth.2532']\n","['3822328']\n","['23817069']\n","{'PMC': 'PMC3822328', 'DOI': '10.1038%2Fnmeth.2532', 'PMID': '23817069'}\n","\t {'PMC': '3822328', 'DOI': '10.1038%2Fnmeth.2532', 'PMID': '23817069'}\n","Jiang W, Zhou H, Bi H, Fromm M, Yang B, Weeks DP (November 2013). \"Demonstration of CRISPR/Cas9/sgRNA-mediated targeted gene modification in Arabidopsis, tobacco, sorghum and rice\". Nucleic Acids Research. 41 (20): e188. doi:10.1093/nar/gkt780. PMC 3814374. PMID 23999092.\n","['10.1093/nar/gkt780']\n","['3814374']\n","['23999092']\n","{'PMC': 'PMC3814374', 'DOI': '10.1093%2Fnar%2Fgkt780', 'PMID': '23999092'}\n","\t {'PMC': '3814374', 'DOI': '10.1093%2Fnar%2Fgkt780', 'PMID': '23999092'}\n","Wang H, Yang H, Shivalila CS, Dawlaty MM, Cheng AW, Zhang F, Jaenisch R (May 2013). \"One-step generation of mice carrying mutations in multiple genes by CRISPR/Cas-mediated genome engineering\". Cell. 153 (4): 910–918. doi:10.1016/j.cell.2013.04.025. PMC 3969854. PMID 23643243.\n","['10.1016/j.cell.2013.04.025']\n","['3969854']\n","['23643243']\n","{'PMC': 'PMC3969854', 'DOI': '10.1016%2Fj.cell.2013.04.025', 'PMID': '23643243'}\n","\t {'PMC': '3969854', 'DOI': '10.1016%2Fj.cell.2013.04.025', 'PMID': '23643243'}\n","Soni D, Wang DM, Regmi SC, Mittal M, Vogel SM, Schlüter D, Tiruppathi C (May 2018). \"Deubiquitinase function of A20 maintains and repairs endothelial barrier after lung vascular injury\". Cell Death Discovery. 4 (60): 60. doi:10.1038/s41420-018-0056-3. PMC 5955943. PMID 29796309.\n","['10.1038/s41420-018-0056-3']\n","['5955943']\n","['29796309']\n","{'PMC': 'PMC5955943', 'DOI': '10.1038%2Fs41420-018-0056-3', 'PMID': '29796309'}\n","\t {'PMC': '5955943', 'DOI': '10.1038%2Fs41420-018-0056-3', 'PMID': '29796309'}\n","Guo X, Li XJ (July 2015). \"Targeted genome editing in primate embryos\". Cell Research. 25 (7): 767–768. doi:10.1038/cr.2015.64. PMC 4493275. PMID 26032266.\n","['10.1038/cr.2015.64']\n","['4493275']\n","['26032266']\n","{'PMC': 'PMC4493275', 'DOI': '10.1038%2Fcr.2015.64', 'PMID': '26032266'}\n","\t {'PMC': '4493275', 'DOI': '10.1038%2Fcr.2015.64', 'PMID': '26032266'}\n","Baltimore D, Berg P, Botchan M, Carroll D, Charo RA, Church G, Corn JE, Daley GQ, Doudna JA, Fenner M, Greely HT, Jinek M, Martin GS, Penhoet E, Puck J, Sternberg SH, Weissman JS, Yamamoto KR (April 2015). \"Biotechnology. A prudent path forward for genomic engineering and germline gene modification\". Science. 348 (6230): 36–38. Bibcode:2015Sci...348...36B. doi:10.1126/science.aab1028. PMC 4394183. PMID 25791083.\n","['10.1126/science.aab1028']\n","['4394183']\n","['25791083']\n","{'PMC': 'PMC4394183', 'DOI': '10.1126%2Fscience.aab1028', 'PMID': '25791083'}\n","\t {'PMC': '4394183', 'DOI': '10.1126%2Fscience.aab1028', 'PMID': '25791083'}\n","Larson MH, Gilbert LA, Wang X, Lim WA, Weissman JS, Qi LS (November 2013). \"CRISPR interference (CRISPRi) for sequence-specific control of gene expression\". Nature Protocols. 8 (11): 2180–2196. doi:10.1038/nprot.2013.132. PMC 3922765. PMID 24136345.\n","['10.1038/nprot.2013.132']\n","['3922765']\n","['24136345']\n","{'PMC': 'PMC3922765', 'DOI': '10.1038%2Fnprot.2013.132', 'PMID': '24136345'}\n","\t {'PMC': '3922765', 'DOI': '10.1038%2Fnprot.2013.132', 'PMID': '24136345'}\n","Liang P, Xu Y, Zhang X, Ding C, Huang R, Zhang Z,  et al. (May 2015). \"CRISPR/Cas9-mediated gene editing in human tripronuclear zygotes\". Protein & Cell. 6 (5): 363–372. doi:10.1007/s13238-015-0153-5. PMC 4417674. PMID 25894090.\n","['10.1007/s13238-015-0153-5']\n","['4417674']\n","['25894090']\n","{'PMC': 'PMC4417674', 'DOI': '10.1007%2Fs13238-015-0153-5', 'PMID': '25894090'}\n","\t {'PMC': '4417674', 'DOI': '10.1007%2Fs13238-015-0153-5', 'PMID': '25894090'}\n","Yan MY, Yan HQ, Ren GX, Zhao JP, Guo XP, Sun YC (September 2017). \"CRISPR-Cas12a-Assisted Recombineering in Bacteria\". Applied and Environmental Microbiology. 83 (17). doi:10.1128/AEM.00947-17. PMC 5561284. PMID 28646112.\n","['10.1128/AEM.00947-17']\n","['5561284']\n","['28646112']\n","{'PMC': 'PMC5561284', 'DOI': '10.1128%2FAEM.00947-17', 'PMID': '28646112'}\n","\t {'PMC': '5561284', 'DOI': '10.1128%2FAEM.00947-17', 'PMID': '28646112'}\n","Zetsche B, Gootenberg JS, Abudayyeh OO, Slaymaker IM, Makarova KS, Essletzbichler P, Volz SE, Joung J, van der Oost J, Regev A, Koonin EV, Zhang F (October 2015). \"Cpf1 is a single RNA-guided endonuclease of a class 2 CRISPR-Cas system\". Cell. 163 (3): 759–771. doi:10.1016/j.cell.2015.09.038. PMC 4638220. PMID 26422227.\n","['10.1016/j.cell.2015.09.038']\n","['4638220']\n","['26422227']\n","{'PMC': 'PMC4638220', 'DOI': '10.1016%2Fj.cell.2015.09.038', 'PMID': '26422227'}\n","\t {'PMC': '4638220', 'DOI': '10.1016%2Fj.cell.2015.09.038', 'PMID': '26422227'}\n","Fonfara I, Richter H, Bratovič M, Le Rhun A, Charpentier E (April 2016). \"The CRISPR-associated DNA-cleaving enzyme Cpf1 also processes precursor CRISPR RNA\". Nature. 532 (7600): 517–521. Bibcode:2016Natur.532..517F. doi:10.1038/nature17945. PMID 27096362. S2CID 2271552.\n","['10.1038/nature17945']\n","[]\n","['27096362']\n","{'DOI': '10.1038%2Fnature17945', 'PMID': '27096362'}\n","\t {'DOI': '10.1038%2Fnature17945', 'PMID': '27096362'}\n","Kim H, Kim ST, Ryu J, Kang BC, Kim JS, and Kim SG (February 2017). \"CRISPR/Cpf1-mediated DNA-free plant genome editing\". Nature Communications. 8 (14406): 14406. Bibcode:2017NatCo...814406K. doi:10.1038/ncomms14406. PMC 5316869. PMID 28205546.\n","['10.1038/ncomms14406']\n","['5316869']\n","['28205546']\n","{'PMC': 'PMC5316869', 'DOI': '10.1038%2Fncomms14406', 'PMID': '28205546'}\n","\t {'PMC': '5316869', 'DOI': '10.1038%2Fncomms14406', 'PMID': '28205546'}\n","\"Cpf1 Nuclease\". abmgood.com. Retrieved 2017-12-14.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Abudayyeh OO, Gootenberg JS, Konermann S, Joung J, Slaymaker IM, Cox DB,  et al. (August 2016). \"C2c2 is a single-component programmable RNA-guided RNA-targeting CRISPR effector\". Science. 353 (6299): aaf5573. doi:10.1126/science.aaf5573. PMC 5127784. PMID 27256883.\n","['10.1126/science.aaf5573']\n","['5127784']\n","['27256883']\n","{'PMC': 'PMC5127784', 'DOI': '10.1126%2Fscience.aaf5573', 'PMID': '27256883'}\n","\t {'PMC': '5127784', 'DOI': '10.1126%2Fscience.aaf5573', 'PMID': '27256883'}\n","Gootenberg JS, Abudayyeh OO, Lee JW, Essletzbichler P, Dy AJ, Joung J,  et al. (April 2017). \"Nucleic acid detection with CRISPR-Cas13a/C2c2\". Science. 356 (6336): 438–442. Bibcode:2017Sci...356..438G. doi:10.1126/science.aam9321. PMC 5526198. PMID 28408723.\n","['10.1126/science.aam9321']\n","['5526198']\n","['28408723']\n","{'PMC': 'PMC5526198', 'DOI': '10.1126%2Fscience.aam9321', 'PMID': '28408723'}\n","\t {'PMC': '5526198', 'DOI': '10.1126%2Fscience.aam9321', 'PMID': '28408723'}\n","Gootenberg JS, Abudayyeh OO, Kellner MJ, Joung J, Collins JJ, Zhang F (April 2018). \"Multiplexed and portable nucleic acid detection platform with Cas13, Cas12a, and Csm6\". Science. 360 (6387): 439–444. Bibcode:2018Sci...360..439G. doi:10.1126/science.aaq0179. PMC 5961727. PMID 29449508.\n","['10.1126/science.aaq0179']\n","['5961727']\n","['29449508']\n","{'PMC': 'PMC5961727', 'DOI': '10.1126%2Fscience.aaq0179', 'PMID': '29449508'}\n","\t {'PMC': '5961727', 'DOI': '10.1126%2Fscience.aaq0179', 'PMID': '29449508'}\n","Iwasaki RS, Batey RT (2020). \"SPRINT: a Cas13a-based platform for detection of small molecules\". Nucleic Acids Research. 48 (17): e101. doi:10.1093/nar/gkaa673. PMC 7515716. PMID 32797156.\n","['10.1093/nar/gkaa673']\n","['7515716']\n","['32797156']\n","{'PMC': 'PMC7515716', 'DOI': '10.1093%2Fnar%2Fgkaa673', 'PMID': '32797156'}\n","\t {'PMC': '7515716', 'DOI': '10.1093%2Fnar%2Fgkaa673', 'PMID': '32797156'}\n","Hille F, Charpentier E (November 2016). \"CRISPR-Cas: biology, mechanisms and relevance\". Philosophical Transactions of the Royal Society of London. Series B, Biological Sciences. 371 (1707): 20150496. doi:10.1098/rstb.2015.0496. PMC 5052741. PMID 27672148.\n","['10.1098/rstb.2015.0496']\n","['5052741']\n","['27672148']\n","{'PMC': 'PMC5052741', 'DOI': '10.1098%2Frstb.2015.0496', 'PMID': '27672148'}\n","\t {'PMC': '5052741', 'DOI': '10.1098%2Frstb.2015.0496', 'PMID': '27672148'}\n","Barrangou R, Marraffini LA (April 2014). \"CRISPR-Cas systems: Prokaryotes upgrade to adaptive immunity\". Molecular Cell. 54 (2): 234–244. doi:10.1016/j.molcel.2014.03.011. PMC 4025954. PMID 24766887.\n","['10.1016/j.molcel.2014.03.011']\n","['4025954']\n","['24766887']\n","{'PMC': 'PMC4025954', 'DOI': '10.1016%2Fj.molcel.2014.03.011', 'PMID': '24766887'}\n","\t {'PMC': '4025954', 'DOI': '10.1016%2Fj.molcel.2014.03.011', 'PMID': '24766887'}\n","Tyson GW, Banfield JF (January 2008). \"Rapidly evolving CRISPRs implicated in acquired resistance of microorganisms to viruses\". Environmental Microbiology. 10 (1): 200–207. doi:10.1111/j.1462-2920.2007.01444.x. PMID 17894817.\n","['10.1111/j.1462-2920.2007.01444.x']\n","[]\n","['17894817']\n","{'DOI': '10.1111%2Fj.1462-2920.2007.01444.x', 'PMID': '17894817'}\n","\t {'DOI': '10.1111%2Fj.1462-2920.2007.01444.x', 'PMID': '17894817'}\n","Makarova KS, Wolf YI, Alkhnbashi OS, Costa F, Shah SA, Saunders SJ,  et al. (November 2015). \"An updated evolutionary classification of CRISPR-Cas systems\". Nature Reviews. Microbiology. 13 (11): 722–736. doi:10.1038/nrmicro3569. PMC 5426118. PMID 26411297.\n","['10.1038/nrmicro3569']\n","['5426118']\n","['26411297']\n","{'PMC': 'PMC5426118', 'DOI': '10.1038%2Fnrmicro3569', 'PMID': '26411297'}\n","\t {'PMC': '5426118', 'DOI': '10.1038%2Fnrmicro3569', 'PMID': '26411297'}\n","Wright AV, Nuñez JK, Doudna JA (January 2016). \"Biology and Applications of CRISPR Systems: Harnessing Nature's Toolbox for Genome Engineering\". Cell. 164 (1–2): 29–44. doi:10.1016/j.cell.2015.12.035. PMID 26771484.\n","['10.1016/j.cell.2015.12.035']\n","[]\n","['26771484']\n","{'DOI': '10.1016%2Fj.cell.2015.12.035', 'PMID': '26771484'}\n","\t {'DOI': '10.1016%2Fj.cell.2015.12.035', 'PMID': '26771484'}\n","Westra ER, Dowling AJ, Broniewski JM, van Houte S (November 2016). \"Evolution and Ecology of CRISPR\". Annual Review of Ecology, Evolution, and Systematics. 47 (1): 307–331. doi:10.1146/annurev-ecolsys-121415-032428.\n","['10.1146/annurev-ecolsys-121415-032428']\n","[]\n","[]\n","{'DOI': '10.1146%2Fannurev-ecolsys-121415-032428'}\n","\t {'DOI': '10.1146%2Fannurev-ecolsys-121415-032428'}\n","Wiedenheft B, Sternberg SH, Doudna JA (February 2012). \"RNA-guided genetic silencing systems in bacteria and archaea\". Nature. 482 (7385): 331–338. Bibcode:2012Natur.482..331W. doi:10.1038/nature10886. PMID 22337052. S2CID 205227944.\n","['10.1038/nature10886']\n","[]\n","['22337052']\n","{'DOI': '10.1038%2Fnature10886', 'PMID': '22337052'}\n","\t {'DOI': '10.1038%2Fnature10886', 'PMID': '22337052'}\n","Deng L, Garrett RA, Shah SA, Peng X, She Q (March 2013). \"A novel interference mechanism by a type IIIB CRISPR-Cmr module in Sulfolobus\". Molecular Microbiology. 87 (5): 1088–1099. doi:10.1111/mmi.12152. PMID 23320564.\n","['10.1111/mmi.12152']\n","[]\n","['23320564']\n","{'DOI': '10.1111%2Fmmi.12152', 'PMID': '23320564'}\n","\t {'DOI': '10.1111%2Fmmi.12152', 'PMID': '23320564'}\n","Sinkunas T, Gasiunas G, Fremaux C, Barrangou R, Horvath P, Siksnys V (April 2011). \"Cas3 is a single-stranded DNA nuclease and ATP-dependent helicase in the CRISPR/Cas immune system\". The EMBO Journal. 30 (7): 1335–1342. doi:10.1038/emboj.2011.41. PMC 3094125. PMID 21343909.\n","['10.1038/emboj.2011.41']\n","['3094125']\n","['21343909']\n","{'PMC': 'PMC3094125', 'DOI': '10.1038%2Femboj.2011.41', 'PMID': '21343909'}\n","\t {'PMC': '3094125', 'DOI': '10.1038%2Femboj.2011.41', 'PMID': '21343909'}\n","Huo Y, Nam KH, Ding F, Lee H, Wu L, Xiao Y, Farchione MD, Zhou S, Rajashankar K, Kurinov I, Zhang R, Ke A (September 2014). \"Structures of CRISPR Cas3 offer mechanistic insights into Cascade-activated DNA unwinding and degradation\". Nature Structural & Molecular Biology. 21 (9): 771–777. doi:10.1038/nsmb.2875. PMC 4156918. PMID 25132177.\n","['10.1038/nsmb.2875']\n","['4156918']\n","['25132177']\n","{'PMC': 'PMC4156918', 'DOI': '10.1038%2Fnsmb.2875', 'PMID': '25132177'}\n","\t {'PMC': '4156918', 'DOI': '10.1038%2Fnsmb.2875', 'PMID': '25132177'}\n","Brendel J, Stoll B, Lange SJ, Sharma K, Lenz C, Stachler AE,  et al. (March 2014). \"A complex of Cas proteins 5, 6, and 7 is required for the biogenesis and stability of clustered regularly interspaced short palindromic repeats (crispr)-derived rnas (crrnas) in Haloferax volcanii\". The Journal of Biological Chemistry. 289 (10): 7164–77. doi:10.1074/jbc.M113.508184. PMC 3945376. PMID 24459147.\n","['10.1074/jbc.M113.508184']\n","['3945376']\n","['24459147']\n","{'PMC': 'PMC3945376', 'DOI': '10.1074%2Fjbc.M113.508184', 'PMID': '24459147'}\n","\t {'PMC': '3945376', 'DOI': '10.1074%2Fjbc.M113.508184', 'PMID': '24459147'}\n","Chylinski K, Makarova KS, Charpentier E, Koonin EV (June 2014). \"Classification and evolution of type II CRISPR-Cas systems\". Nucleic Acids Research. 42 (10): 6091–6105. doi:10.1093/nar/gku241. PMC 4041416. PMID 24728998.\n","['10.1093/nar/gku241']\n","['4041416']\n","['24728998']\n","{'PMC': 'PMC4041416', 'DOI': '10.1093%2Fnar%2Fgku241', 'PMID': '24728998'}\n","\t {'PMC': '4041416', 'DOI': '10.1093%2Fnar%2Fgku241', 'PMID': '24728998'}\n","Makarova KS, Aravind L, Wolf YI, Koonin EV (July 2011). \"Unification of Cas protein families and a simple scenario for the origin and evolution of CRISPR-Cas systems\". Biology Direct. 6: 38. doi:10.1186/1745-6150-6-38. PMC 3150331. PMID 21756346.\n","['10.1186/1745-6150-6-38']\n","['3150331']\n","['21756346']\n","{'PMC': 'PMC3150331', 'DOI': '10.1186%2F1745-6150-6-38', 'PMID': '21756346'}\n","\t {'PMC': '3150331', 'DOI': '10.1186%2F1745-6150-6-38', 'PMID': '21756346'}\n","Makarova KS, Wolf YI, Iranzo J, Shmakov SA, Alkhnbashi OS, Brouns SJJ, Charpentier E, Cheng D, Haft DH, Horvath P, Moineau S, Mojica FJM, Scott D, Shah SA, Siksnys V, Terns MP, Venclovas Č, White MF, Yakunin AF, Yan W, Zhang F, Garrett RA, Backofen R, van der Oost J, Barrangou R, Koonin EV. (2019). \"Evolutionary classification of CRISPR–Cas systems: A burst of class 2 and derived variants\". Nature Reviews Microbiology. 18 (2): 67–83. doi:10.1038/s41579-019-0299-x. hdl:10045/102627. PMID 31857715. S2CID 209420490.\n","['10.1038/s41579-019-0299-x']\n","[]\n","['31857715']\n","{'DOI': '10.1038%2Fs41579-019-0299-x', 'PMID': '31857715'}\n","\t {'DOI': '10.1038%2Fs41579-019-0299-x', 'PMID': '31857715'}\n","Mogila I, Kazlauskiene M, Valinskyte S, Tamulaitiene G, Tamulaitis G, Siksnys V (March 2019). \"Genetic Dissection of the Type III-A CRISPR-Cas System Csm Complex Reveals Roles of Individual Subunits\". Cell Reports. 26 (10): 2753–2765.e4. doi:10.1016/j.celrep.2019.02.029. PMID 30840895.\n","['10.1016/j.celrep.2019.02.029']\n","[]\n","['30840895']\n","{'DOI': '10.1016%2Fj.celrep.2019.02.029', 'PMID': '30840895'}\n","\t {'DOI': '10.1016%2Fj.celrep.2019.02.029', 'PMID': '30840895'}\n","Gasiunas G, Barrangou R, Horvath P, Siksnys V (September 2012). \"Cas9-crRNA ribonucleoprotein complex mediates specific DNA cleavage for adaptive immunity in bacteria\". Proceedings of the National Academy of Sciences of the United States of America. 109 (39): E2579–2586. Bibcode:2012PNAS..109E2579G. doi:10.1073/pnas.1208507109. PMC 3465414. PMID 22949671.\n","['10.1073/pnas.1208507109']\n","['3465414']\n","['22949671']\n","{'PMC': 'PMC3465414', 'DOI': '10.1073%2Fpnas.1208507109', 'PMID': '22949671'}\n","\t {'PMC': '3465414', 'DOI': '10.1073%2Fpnas.1208507109', 'PMID': '22949671'}\n","Heler R, Samai P, Modell JW, Weiner C, Goldberg GW, Bikard D, Marraffini LA (March 2015). \"Cas9 specifies functional viral targets during CRISPR-Cas adaptation\". Nature. 519 (7542): 199–202. Bibcode:2015Natur.519..199H. doi:10.1038/nature14245. PMC 4385744. PMID 25707807.\n","['10.1038/nature14245']\n","['4385744']\n","['25707807']\n","{'PMC': 'PMC4385744', 'DOI': '10.1038%2Fnature14245', 'PMID': '25707807'}\n","\t {'PMC': '4385744', 'DOI': '10.1038%2Fnature14245', 'PMID': '25707807'}\n","Nam KH, Kurinov I, Ke A (September 2011). \"Crystal structure of clustered regularly interspaced short palindromic repeats (CRISPR)-associated Csn2 protein revealed Ca2+-dependent double-stranded DNA binding activity\". The Journal of Biological Chemistry. 286 (35): 30759–30768. doi:10.1074/jbc.M111.256263. PMC 3162437. PMID 21697083.\n","['10.1074/jbc.M111.256263']\n","['3162437']\n","['21697083']\n","{'PMC': 'PMC3162437', 'DOI': '10.1074%2Fjbc.M111.256263', 'PMID': '21697083'}\n","\t {'PMC': '3162437', 'DOI': '10.1074%2Fjbc.M111.256263', 'PMID': '21697083'}\n","Lee H, Dhingra Y, Sashital DG (April 2019). \"The Cas4-Cas1-Cas2 complex mediates precise prespacer processing during CRISPR adaptation\". eLife. 8. doi:10.7554/eLife.44248. PMC 6519985. PMID 31021314.\n","['10.7554/eLife.44248']\n","['6519985']\n","['31021314']\n","{'PMC': 'PMC6519985', 'DOI': '10.7554%2FeLife.44248', 'PMID': '31021314'}\n","\t {'PMC': '6519985', 'DOI': '10.7554%2FeLife.44248', 'PMID': '31021314'}\n","Chylinski K, Le Rhun A, Charpentier E (May 2013). \"The tracrRNA and Cas9 families of type II CRISPR-Cas immunity systems\". RNA Biology. 10 (5): 726–737. doi:10.4161/rna.24321. PMC 3737331. PMID 23563642.\n","['10.4161/rna.24321']\n","['3737331']\n","['23563642']\n","{'PMC': 'PMC3737331', 'DOI': '10.4161%2Frna.24321', 'PMID': '23563642'}\n","\t {'PMC': '3737331', 'DOI': '10.4161%2Frna.24321', 'PMID': '23563642'}\n","Makarova KS, Zhang F, Koonin EV (January 2017). \"SnapShot: Class 2 CRISPR-Cas Systems\". Cell. 168 (1–2): 328–328.e1. doi:10.1016/j.cell.2016.12.038. PMID 28086097.\n","['10.1016/j.cell.2016.12.038']\n","[]\n","['28086097']\n","{'DOI': '10.1016%2Fj.cell.2016.12.038', 'PMID': '28086097'}\n","\t {'DOI': '10.1016%2Fj.cell.2016.12.038', 'PMID': '28086097'}\n","Cox DB, Gootenberg JS, Abudayyeh OO, Franklin B, Kellner MJ, Joung J, Zhang F (November 2017). \"RNA editing with CRISPR-Cas13\". Science. 358 (6366): 1019–1027. Bibcode:2017Sci...358.1019C. doi:10.1126/science.aaq0180. PMC 5793859. PMID 29070703.\n","['10.1126/science.aaq0180']\n","['5793859']\n","['29070703']\n","{'PMC': 'PMC5793859', 'DOI': '10.1126%2Fscience.aaq0180', 'PMID': '29070703'}\n","\t {'PMC': '5793859', 'DOI': '10.1126%2Fscience.aaq0180', 'PMID': '29070703'}\n","^ Azangou-Khyavy, M. et al. (2020) ‘CRISPR/Cas: From Tumor Gene Editing to T Cell-Based Immunotherapy of Cancer’, Frontiers in Immunology, 11. doi: 10.3389/fimmu.2020.02062.\n","\n","['10.3389/fimmu.2020.02062']\n","[]\n","[]\n","{}\n","\t {}\n","Aliyari R, Ding SW (January 2009). \"RNA-based viral immunity initiated by the Dicer family of host immune receptors\". Immunological Reviews. 227 (1): 176–188. doi:10.1111/j.1600-065X.2008.00722.x. PMC 2676720. PMID 19120484.\n","['10.1111/j.1600-065X.2008.00722.x']\n","['2676720']\n","['19120484']\n","{'PMC': 'PMC2676720', 'DOI': '10.1111%2Fj.1600-065X.2008.00722.x', 'PMID': '19120484'}\n","\t {'PMC': '2676720', 'DOI': '10.1111%2Fj.1600-065X.2008.00722.x', 'PMID': '19120484'}\n","Dugar G, Herbig A, Förstner KU, Heidrich N, Reinhardt R, Nieselt K, Sharma CM (May 2013). \"High-resolution transcriptome maps reveal strain-specific regulatory features of multiple Campylobacter jejuni isolates\". PLOS Genetics. 9 (5): e1003495. doi:10.1371/journal.pgen.1003495. PMC 3656092. PMID 23696746.\n","['10.1371/journal.pgen.1003495']\n","['3656092']\n","['23696746']\n","{'PMC': 'PMC3656092', 'DOI': '10.1371%2Fjournal.pgen.1003495', 'PMID': '23696746'}\n","\t {'PMC': '3656092', 'DOI': '10.1371%2Fjournal.pgen.1003495', 'PMID': '23696746'}\n","Hatoum-Aslan A, Maniv I, Marraffini LA (December 2011). \"Mature clustered, regularly interspaced, short palindromic repeats RNA (crRNA) length is measured by a ruler mechanism anchored at the precursor processing site\". Proceedings of the National Academy of Sciences of the United States of America. 108 (52): 21218–21222. Bibcode:2011PNAS..10821218H. doi:10.1073/pnas.1112832108. PMC 3248500. PMID 22160698.\n","['10.1073/pnas.1112832108']\n","['3248500']\n","['22160698']\n","{'PMC': 'PMC3248500', 'DOI': '10.1073%2Fpnas.1112832108', 'PMID': '22160698'}\n","\t {'PMC': '3248500', 'DOI': '10.1073%2Fpnas.1112832108', 'PMID': '22160698'}\n","Yosef I, Goren MG, Qimron U (July 2012). \"Proteins and DNA elements essential for the CRISPR adaptation process in Escherichia coli\". Nucleic Acids Research. 40 (12): 5569–5576. doi:10.1093/nar/gks216. PMC 3384332. PMID 22402487.\n","['10.1093/nar/gks216']\n","['3384332']\n","['22402487']\n","{'PMC': 'PMC3384332', 'DOI': '10.1093%2Fnar%2Fgks216', 'PMID': '22402487'}\n","\t {'PMC': '3384332', 'DOI': '10.1093%2Fnar%2Fgks216', 'PMID': '22402487'}\n","Swarts DC, Mosterd C, van Passel MW, Brouns SJ (2012). \"CRISPR interference directs strand specific spacer acquisition\". PLOS ONE. 7 (4): e35888. Bibcode:2012PLoSO...735888S. doi:10.1371/journal.pone.0035888. PMC 3338789. PMID 22558257.\n","['10.1371/journal.pone.0035888']\n","['3338789']\n","['22558257']\n","{'PMC': 'PMC3338789', 'DOI': '10.1371%2Fjournal.pone.0035888', 'PMID': '22558257'}\n","\t {'PMC': '3338789', 'DOI': '10.1371%2Fjournal.pone.0035888', 'PMID': '22558257'}\n","Babu M, Beloglazova N, Flick R, Graham C, Skarina T, Nocek B,  et al. (January 2011). \"A dual function of the CRISPR-Cas system in bacterial antivirus immunity and DNA repair\". Molecular Microbiology. 79 (2): 484–502. doi:10.1111/j.1365-2958.2010.07465.x. PMC 3071548. PMID 21219465.\n","['10.1111/j.1365-2958.2010.07465.x']\n","['3071548']\n","['21219465']\n","{'PMC': 'PMC3071548', 'DOI': '10.1111%2Fj.1365-2958.2010.07465.x', 'PMID': '21219465'}\n","\t {'PMC': '3071548', 'DOI': '10.1111%2Fj.1365-2958.2010.07465.x', 'PMID': '21219465'}\n","Han D, Lehmann K, Krauss G (June 2009). \"SSO1450—a CAS1 protein from Sulfolobus solfataricus P2 with high affinity for RNA and DNA\". FEBS Letters. 583 (12): 1928–1932. doi:10.1016/j.febslet.2009.04.047. PMID 19427858. S2CID 22279972.\n","['10.1016/j.febslet.2009.04.047']\n","[]\n","['19427858']\n","{'DOI': '10.1016%2Fj.febslet.2009.04.047', 'PMID': '19427858'}\n","\t {'DOI': '10.1016%2Fj.febslet.2009.04.047', 'PMID': '19427858'}\n","Wiedenheft B, Zhou K, Jinek M, Coyle SM, Ma W, Doudna JA (June 2009). \"Structural basis for DNase activity of a conserved protein implicated in CRISPR-mediated genome defense\". Structure. 17 (6): 904–912. doi:10.1016/j.str.2009.03.019. PMID 19523907.\n","['10.1016/j.str.2009.03.019']\n","[]\n","['19523907']\n","{'DOI': '10.1016%2Fj.str.2009.03.019', 'PMID': '19523907'}\n","\t {'DOI': '10.1016%2Fj.str.2009.03.019', 'PMID': '19523907'}\n","Beloglazova N, Brown G, Zimmerman MD, Proudfoot M, Makarova KS, Kudritska M,  et al. (July 2008). \"A novel family of sequence-specific endoribonucleases associated with the clustered regularly interspaced short palindromic repeats\". The Journal of Biological Chemistry. 283 (29): 20361–20371. doi:10.1074/jbc.M803225200. PMC 2459268. PMID 18482976.\n","['10.1074/jbc.M803225200']\n","['2459268']\n","['18482976']\n","{'PMC': 'PMC2459268', 'DOI': '10.1074%2Fjbc.M803225200', 'PMID': '18482976'}\n","\t {'PMC': '2459268', 'DOI': '10.1074%2Fjbc.M803225200', 'PMID': '18482976'}\n","Samai P, Smith P, Shuman S (December 2010). \"Structure of a CRISPR-associated protein Cas2 from Desulfovibrio vulgaris\". Acta Crystallographica Section F. 66 (Pt 12): 1552–1556. doi:10.1107/S1744309110039801. PMC 2998353. PMID 21139194.\n","['10.1107/S1744309110039801']\n","['2998353']\n","['21139194']\n","{'PMC': 'PMC2998353', 'DOI': '10.1107%2FS1744309110039801', 'PMID': '21139194'}\n","\t {'PMC': '2998353', 'DOI': '10.1107%2FS1744309110039801', 'PMID': '21139194'}\n","Nam KH, Ding F, Haitjema C, Huang Q, DeLisa MP, Ke A (October 2012). \"Double-stranded endonuclease activity in Bacillus halodurans clustered regularly interspaced short palindromic repeats (CRISPR)-associated Cas2 protein\". The Journal of Biological Chemistry. 287 (43): 35943–35952. doi:10.1074/jbc.M112.382598. PMC 3476262. PMID 22942283.\n","['10.1074/jbc.M112.382598']\n","['3476262']\n","['22942283']\n","{'PMC': 'PMC3476262', 'DOI': '10.1074%2Fjbc.M112.382598', 'PMID': '22942283'}\n","\t {'PMC': '3476262', 'DOI': '10.1074%2Fjbc.M112.382598', 'PMID': '22942283'}\n","Nuñez JK, Kranzusch PJ, Noeske J, Wright AV, Davies CW, Doudna JA (June 2014). \"Cas1-Cas2 complex formation mediates spacer acquisition during CRISPR-Cas adaptive immunity\". Nature Structural & Molecular Biology. 21 (6): 528–534. doi:10.1038/nsmb.2820. PMC 4075942. PMID 24793649.\n","['10.1038/nsmb.2820']\n","['4075942']\n","['24793649']\n","{'PMC': 'PMC4075942', 'DOI': '10.1038%2Fnsmb.2820', 'PMID': '24793649'}\n","\t {'PMC': '4075942', 'DOI': '10.1038%2Fnsmb.2820', 'PMID': '24793649'}\n","Nuñez JK, Lee AS, Engelman A, Doudna JA (March 2015). \"Integrase-mediated spacer acquisition during CRISPR-Cas adaptive immunity\". Nature. 519 (7542): 193–198. Bibcode:2015Natur.519..193N. doi:10.1038/nature14237. PMC 4359072. PMID 25707795.\n","['10.1038/nature14237']\n","['4359072']\n","['25707795']\n","{'PMC': 'PMC4359072', 'DOI': '10.1038%2Fnature14237', 'PMID': '25707795'}\n","\t {'PMC': '4359072', 'DOI': '10.1038%2Fnature14237', 'PMID': '25707795'}\n","Wang J, Li J, Zhao H, Sheng G, Wang M, Yin M, Wang Y (November 2015). \"Structural and Mechanistic Basis of PAM-Dependent Spacer Acquisition in CRISPR-Cas Systems\". Cell. 163 (4): 840–853. doi:10.1016/j.cell.2015.10.008. PMID 26478180.\n","['10.1016/j.cell.2015.10.008']\n","[]\n","['26478180']\n","{'DOI': '10.1016%2Fj.cell.2015.10.008', 'PMID': '26478180'}\n","\t {'DOI': '10.1016%2Fj.cell.2015.10.008', 'PMID': '26478180'}\n","Nuñez JK, Harrington LB, Kranzusch PJ, Engelman AN, Doudna JA (November 2015). \"Foreign DNA capture during CRISPR-Cas adaptive immunity\". Nature. 527 (7579): 535–538. Bibcode:2015Natur.527..535N. doi:10.1038/nature15760. PMC 4662619. PMID 26503043.\n","['10.1038/nature15760']\n","['4662619']\n","['26503043']\n","{'PMC': 'PMC4662619', 'DOI': '10.1038%2Fnature15760', 'PMID': '26503043'}\n","\t {'PMC': '4662619', 'DOI': '10.1038%2Fnature15760', 'PMID': '26503043'}\n","Sorek R, Lawrence CM, Wiedenheft B (2013). \"CRISPR-mediated adaptive immune systems in bacteria and archaea\". Annual Review of Biochemistry. 82 (1): 237–266. doi:10.1146/annurev-biochem-072911-172315. PMID 23495939.\n","['10.1146/annurev-biochem-072911-172315']\n","[]\n","['23495939']\n","{'DOI': '10.1146%2Fannurev-biochem-072911-172315', 'PMID': '23495939'}\n","\t {'DOI': '10.1146%2Fannurev-biochem-072911-172315', 'PMID': '23495939'}\n","Nuñez JK, Bai L, Harrington LB, Hinder TL, Doudna JA (June 2016). \"CRISPR Immunological Memory Requires a Host Factor for Specificity\". Molecular Cell. 62 (6): 824–833. doi:10.1016/j.molcel.2016.04.027. PMID 27211867.\n","['10.1016/j.molcel.2016.04.027']\n","[]\n","['27211867']\n","{'DOI': '10.1016%2Fj.molcel.2016.04.027', 'PMID': '27211867'}\n","\t {'DOI': '10.1016%2Fj.molcel.2016.04.027', 'PMID': '27211867'}\n","Fagerlund RD, Wilkinson ME, Klykov O, Barendregt A, Pearce FG, Kieper SN, Maxwell HW, Capolupo A, Heck AJ, Krause KL, Bostina M, Scheltema RA, Staals RH, Fineran PC (June 2017). \"Spacer capture and integration by a type I-F Cas1-Cas2–3 CRISPR adaptation complex\". Proceedings of the National Academy of Sciences of the United States of America. 114 (26): E5122–E5128. doi:10.1073/pnas.1618421114. PMC 5495228. PMID 28611213.\n","['10.1073/pnas.1618421114']\n","['5495228']\n","['28611213']\n","{'PMC': 'PMC5495228', 'DOI': '10.1073%2Fpnas.1618421114', 'PMID': '28611213'}\n","\t {'PMC': '5495228', 'DOI': '10.1073%2Fpnas.1618421114', 'PMID': '28611213'}\n","Rollie C, Graham S, Rouillon C, White MF (February 2018). \"Prespacer processing and specific integration in a Type I-A CRISPR system\". Nucleic Acids Research. 46 (3): 1007–1020. doi:10.1093/nar/gkx1232. PMC 5815122. PMID 29228332.\n","['10.1093/nar/gkx1232']\n","['5815122']\n","['29228332']\n","{'PMC': 'PMC5815122', 'DOI': '10.1093%2Fnar%2Fgkx1232', 'PMID': '29228332'}\n","\t {'PMC': '5815122', 'DOI': '10.1093%2Fnar%2Fgkx1232', 'PMID': '29228332'}\n","Horvath P, Romero DA, Coûté-Monvoisin AC, Richards M, Deveau H, Moineau S,  et al. (February 2008). \"Diversity, activity, and evolution of CRISPR loci in Streptococcus thermophilus\". Journal of Bacteriology. 190 (4): 1401–1412. doi:10.1128/JB.01415-07. PMC 2238196. PMID 18065539.\n","['10.1128/JB.01415-07']\n","['2238196']\n","['18065539']\n","{'PMC': 'PMC2238196', 'DOI': '10.1128%2FJB.01415-07', 'PMID': '18065539'}\n","\t {'PMC': '2238196', 'DOI': '10.1128%2FJB.01415-07', 'PMID': '18065539'}\n","Deveau H, Barrangou R, Garneau JE, Labonté J, Fremaux C, Boyaval P, Romero DA, Horvath P, Moineau S (February 2008). \"Phage response to CRISPR-encoded resistance in Streptococcus thermophilus\". Journal of Bacteriology. 190 (4): 1390–1400. doi:10.1128/JB.01412-07. PMC 2238228. PMID 18065545.\n","['10.1128/JB.01412-07']\n","['2238228']\n","['18065545']\n","{'PMC': 'PMC2238228', 'DOI': '10.1128%2FJB.01412-07', 'PMID': '18065545'}\n","\t {'PMC': '2238228', 'DOI': '10.1128%2FJB.01412-07', 'PMID': '18065545'}\n","Mojica FJ, Díez-Villaseñor C, García-Martínez J, Almendros C (March 2009). \"Short motif sequences determine the targets of the prokaryotic CRISPR defence system\". Microbiology. 155 (Pt 3): 733–740. doi:10.1099/mic.0.023960-0. PMID 19246744.\n","['10.1099/mic.0.023960-0']\n","[]\n","['19246744']\n","{'DOI': '10.1099%2Fmic.0.023960-0', 'PMID': '19246744'}\n","\t {'DOI': '10.1099%2Fmic.0.023960-0', 'PMID': '19246744'}\n","Lillestøl RK, Shah SA, Brügger K, Redder P, Phan H, Christiansen J, Garrett RA (April 2009). \"CRISPR families of the crenarchaeal genus Sulfolobus: bidirectional transcription and dynamic properties\". Molecular Microbiology. 72 (1): 259–272. doi:10.1111/j.1365-2958.2009.06641.x. PMID 19239620. S2CID 36258923.\n","['10.1111/j.1365-2958.2009.06641.x']\n","[]\n","['19239620']\n","{'DOI': '10.1111%2Fj.1365-2958.2009.06641.x', 'PMID': '19239620'}\n","\t {'DOI': '10.1111%2Fj.1365-2958.2009.06641.x', 'PMID': '19239620'}\n","Shah SA, Hansen NR, Garrett RA (February 2009). \"Distribution of CRISPR spacer matches in viruses and plasmids of crenarchaeal acidothermophiles and implications for their inhibitory mechanism\". Biochemical Society Transactions. 37 (Pt 1): 23–28. doi:10.1042/BST0370023. PMID 19143596. S2CID 19093261.\n","['10.1042/BST0370023']\n","[]\n","['19143596']\n","{'DOI': '10.1042%2FBST0370023', 'PMID': '19143596'}\n","\t {'DOI': '10.1042%2FBST0370023', 'PMID': '19143596'}\n","Díez-Villaseñor C, Guzmán NM, Almendros C, García-Martínez J, Mojica FJ (May 2013). \"CRISPR-spacer integration reporter plasmids reveal distinct genuine acquisition specificities among CRISPR-Cas I-E variants of Escherichia coli\". RNA Biology. 10 (5): 792–802. doi:10.4161/rna.24023. PMC 3737337. PMID 23445770.\n","['10.4161/rna.24023']\n","['3737337']\n","['23445770']\n","{'PMC': 'PMC3737337', 'DOI': '10.4161%2Frna.24023', 'PMID': '23445770'}\n","\t {'PMC': '3737337', 'DOI': '10.4161%2Frna.24023', 'PMID': '23445770'}\n","Erdmann S, Garrett RA (September 2012). \"Selective and hyperactive uptake of foreign DNA by adaptive immune systems of an archaeon via two distinct mechanisms\". Molecular Microbiology. 85 (6): 1044–1056. doi:10.1111/j.1365-2958.2012.08171.x. PMC 3468723. PMID 22834906.\n","['10.1111/j.1365-2958.2012.08171.x']\n","['3468723']\n","['22834906']\n","{'PMC': 'PMC3468723', 'DOI': '10.1111%2Fj.1365-2958.2012.08171.x', 'PMID': '22834906'}\n","\t {'PMC': '3468723', 'DOI': '10.1111%2Fj.1365-2958.2012.08171.x', 'PMID': '22834906'}\n","Shah SA, Erdmann S, Mojica FJ, Garrett RA (May 2013). \"Protospacer recognition motifs: mixed identities and functional diversity\". RNA Biology. 10 (5): 891–899. doi:10.4161/rna.23764. PMC 3737346. PMID 23403393.\n","['10.4161/rna.23764']\n","['3737346']\n","['23403393']\n","{'PMC': 'PMC3737346', 'DOI': '10.4161%2Frna.23764', 'PMID': '23403393'}\n","\t {'PMC': '3737346', 'DOI': '10.4161%2Frna.23764', 'PMID': '23403393'}\n","Andersson AF, Banfield JF (May 2008). \"Virus population dynamics and acquired virus resistance in natural microbial communities\". Science. 320 (5879): 1047–1050. Bibcode:2008Sci...320.1047A. doi:10.1126/science.1157358. PMID 18497291. S2CID 26209623.\n","['10.1126/science.1157358']\n","[]\n","['18497291']\n","{'DOI': '10.1126%2Fscience.1157358', 'PMID': '18497291'}\n","\t {'DOI': '10.1126%2Fscience.1157358', 'PMID': '18497291'}\n","Pride DT, Sun CL, Salzman J, Rao N, Loomer P, Armitage GC,  et al. (January 2011). \"Analysis of streptococcal CRISPRs from human saliva reveals substantial sequence diversity within and between subjects over time\". Genome Research. 21 (1): 126–136. doi:10.1101/gr.111732.110. PMC 3012920. PMID 21149389.\n","['10.1101/gr.111732.110']\n","['3012920']\n","['21149389']\n","{'PMC': 'PMC3012920', 'DOI': '10.1101%2Fgr.111732.110', 'PMID': '21149389'}\n","\t {'PMC': '3012920', 'DOI': '10.1101%2Fgr.111732.110', 'PMID': '21149389'}\n","Goren MG, Yosef I, Auster O, Qimron U (October 2012). \"Experimental definition of a clustered regularly interspaced short palindromic duplicon in Escherichia coli\". Journal of Molecular Biology. 423 (1): 14–16. doi:10.1016/j.jmb.2012.06.037. PMID 22771574.\n","['10.1016/j.jmb.2012.06.037']\n","[]\n","['22771574']\n","{'DOI': '10.1016%2Fj.jmb.2012.06.037', 'PMID': '22771574'}\n","\t {'DOI': '10.1016%2Fj.jmb.2012.06.037', 'PMID': '22771574'}\n","Datsenko KA, Pougach K, Tikhonov A, Wanner BL, Severinov K, Semenova E (July 2012). \"Molecular memory of prior infections activates the CRISPR/Cas adaptive bacterial immunity system\". Nature Communications. 3: 945. Bibcode:2012NatCo...3..945D. doi:10.1038/ncomms1937. PMID 22781758.\n","['10.1038/ncomms1937']\n","[]\n","['22781758']\n","{'DOI': '10.1038%2Fncomms1937', 'PMID': '22781758'}\n","\t {'DOI': '10.1038%2Fncomms1937', 'PMID': '22781758'}\n","Gesner EM, Schellenberg MJ, Garside EL, George MM, Macmillan AM (June 2011). \"Recognition and maturation of effector RNAs in a CRISPR interference pathway\". Nature Structural & Molecular Biology. 18 (6): 688–692. doi:10.1038/nsmb.2042. PMID 21572444. S2CID 677704.\n","['10.1038/nsmb.2042']\n","[]\n","['21572444']\n","{'DOI': '10.1038%2Fnsmb.2042', 'PMID': '21572444'}\n","\t {'DOI': '10.1038%2Fnsmb.2042', 'PMID': '21572444'}\n","Sashital DG, Jinek M, Doudna JA (June 2011). \"An RNA-induced conformational change required for CRISPR RNA cleavage by the endoribonuclease Cse3\". Nature Structural & Molecular Biology. 18 (6): 680–687. doi:10.1038/nsmb.2043. PMID 21572442. S2CID 5538195.\n","['10.1038/nsmb.2043']\n","[]\n","['21572442']\n","{'DOI': '10.1038%2Fnsmb.2043', 'PMID': '21572442'}\n","\t {'DOI': '10.1038%2Fnsmb.2043', 'PMID': '21572442'}\n","Haurwitz RE, Jinek M, Wiedenheft B, Zhou K, Doudna JA (September 2010). \"Sequence- and structure-specific RNA processing by a CRISPR endonuclease\". Science. 329 (5997): 1355–1358. Bibcode:2010Sci...329.1355H. doi:10.1126/science.1192272. PMC 3133607. PMID 20829488.\n","['10.1126/science.1192272']\n","['3133607']\n","['20829488']\n","{'PMC': 'PMC3133607', 'DOI': '10.1126%2Fscience.1192272', 'PMID': '20829488'}\n","\t {'PMC': '3133607', 'DOI': '10.1126%2Fscience.1192272', 'PMID': '20829488'}\n","Kunin V, Sorek R, Hugenholtz P (2007). \"Evolutionary conservation of sequence and secondary structures in CRISPR repeats\". Genome Biology. 8 (4): R61. doi:10.1186/gb-2007-8-4-r61. PMC 1896005. PMID 17442114.\n","['10.1186/gb-2007-8-4-r61']\n","['1896005']\n","['17442114']\n","{'PMC': 'PMC1896005', 'DOI': '10.1186%2Fgb-2007-8-4-r61', 'PMID': '17442114'}\n","\t {'PMC': '1896005', 'DOI': '10.1186%2Fgb-2007-8-4-r61', 'PMID': '17442114'}\n","Carte J, Wang R, Li H, Terns RM, Terns MP (December 2008). \"Cas6 is an endoribonuclease that generates guide RNAs for invader defense in prokaryotes\". Genes & Development. 22 (24): 3489–3496. doi:10.1101/gad.1742908. PMC 2607076. PMID 19141480.\n","['10.1101/gad.1742908']\n","['2607076']\n","['19141480']\n","{'PMC': 'PMC2607076', 'DOI': '10.1101%2Fgad.1742908', 'PMID': '19141480'}\n","\t {'PMC': '2607076', 'DOI': '10.1101%2Fgad.1742908', 'PMID': '19141480'}\n","Wang R, Preamplume G, Terns MP, Terns RM, Li H (February 2011). \"Interaction of the Cas6 riboendonuclease with CRISPR RNAs: recognition and cleavage\". Structure. 19 (2): 257–264. doi:10.1016/j.str.2010.11.014. PMC 3154685. PMID 21300293.\n","['10.1016/j.str.2010.11.014']\n","['3154685']\n","['21300293']\n","{'PMC': 'PMC3154685', 'DOI': '10.1016%2Fj.str.2010.11.014', 'PMID': '21300293'}\n","\t {'PMC': '3154685', 'DOI': '10.1016%2Fj.str.2010.11.014', 'PMID': '21300293'}\n","Niewoehner O, Jinek M, Doudna JA (January 2014). \"Evolution of CRISPR RNA recognition and processing by Cas6 endonucleases\". Nucleic Acids Research. 42 (2): 1341–1353. doi:10.1093/nar/gkt922. PMC 3902920. PMID 24150936.\n","['10.1093/nar/gkt922']\n","['3902920']\n","['24150936']\n","{'PMC': 'PMC3902920', 'DOI': '10.1093%2Fnar%2Fgkt922', 'PMID': '24150936'}\n","\t {'PMC': '3902920', 'DOI': '10.1093%2Fnar%2Fgkt922', 'PMID': '24150936'}\n","Semenova E, Jore MM, Datsenko KA, Semenova A, Westra ER, Wanner B,  et al. (June 2011). \"Interference by clustered regularly interspaced short palindromic repeat (CRISPR) RNA is governed by a seed sequence\". Proceedings of the National Academy of Sciences of the United States of America. 108 (25): 10098–10103. Bibcode:2011PNAS..10810098S. doi:10.1073/pnas.1104144108. PMC 3121866. PMID 21646539.\n","['10.1073/pnas.1104144108']\n","['3121866']\n","['21646539']\n","{'PMC': 'PMC3121866', 'DOI': '10.1073%2Fpnas.1104144108', 'PMID': '21646539'}\n","\t {'PMC': '3121866', 'DOI': '10.1073%2Fpnas.1104144108', 'PMID': '21646539'}\n","Gudbergsdottir S, Deng L, Chen Z, Jensen JV, Jensen LR, She Q, Garrett RA (January 2011). \"Dynamic properties of the Sulfolobus CRISPR/Cas and CRISPR/Cmr systems when challenged with vector-borne viral and plasmid genes and protospacers\". Molecular Microbiology. 79 (1): 35–49. doi:10.1111/j.1365-2958.2010.07452.x. PMC 3025118. PMID 21166892.\n","['10.1111/j.1365-2958.2010.07452.x']\n","['3025118']\n","['21166892']\n","{'PMC': 'PMC3025118', 'DOI': '10.1111%2Fj.1365-2958.2010.07452.x', 'PMID': '21166892'}\n","\t {'PMC': '3025118', 'DOI': '10.1111%2Fj.1365-2958.2010.07452.x', 'PMID': '21166892'}\n","Manica A, Zebec Z, Teichmann D, Schleper C (April 2011). \"In vivo activity of CRISPR-mediated virus defence in a hyperthermophilic archaeon\". Molecular Microbiology. 80 (2): 481–491. doi:10.1111/j.1365-2958.2011.07586.x. PMID 21385233. S2CID 41442419.\n","['10.1111/j.1365-2958.2011.07586.x']\n","[]\n","['21385233']\n","{'DOI': '10.1111%2Fj.1365-2958.2011.07586.x', 'PMID': '21385233'}\n","\t {'DOI': '10.1111%2Fj.1365-2958.2011.07586.x', 'PMID': '21385233'}\n","Jore MM, Lundgren M, van Duijn E, Bultema JB, Westra ER, Waghmare SP,  et al. (May 2011). \"Structural basis for CRISPR RNA-guided DNA recognition by Cascade\" (PDF). Nature Structural & Molecular Biology. 18 (5): 529–536. doi:10.1038/nsmb.2019. PMID 21460843. S2CID 10987554.\n","['10.1038/nsmb.2019']\n","[]\n","['21460843']\n","{'DOI': '10.1038%2Fnsmb.2019', 'PMID': '21460843'}\n","\t {'DOI': '10.1038%2Fnsmb.2019', 'PMID': '21460843'}\n","Wiedenheft B, Lander GC, Zhou K, Jore MM, Brouns SJ, van der Oost J, Doudna JA, Nogales E (September 2011). \"Structures of the RNA-guided surveillance complex from a bacterial immune system\". Nature. 477 (7365): 486–489. Bibcode:2011Natur.477..486W. doi:10.1038/nature10402. PMC 4165517. PMID 21938068.\n","['10.1038/nature10402']\n","['4165517']\n","['21938068']\n","{'PMC': 'PMC4165517', 'DOI': '10.1038%2Fnature10402', 'PMID': '21938068'}\n","\t {'PMC': '4165517', 'DOI': '10.1038%2Fnature10402', 'PMID': '21938068'}\n","Zhang J, Rouillon C, Kerou M, Reeks J, Brugger K, Graham S, Reimann J, Cannone G, Liu H, Albers SV, Naismith JH, Spagnolo L, White MF (February 2012). \"Structure and mechanism of the CMR complex for CRISPR-mediated antiviral immunity\". Molecular Cell. 45 (3): 303–313. doi:10.1016/j.molcel.2011.12.013. PMC 3381847. PMID 22227115.\n","['10.1016/j.molcel.2011.12.013']\n","['3381847']\n","['22227115']\n","{'PMC': 'PMC3381847', 'DOI': '10.1016%2Fj.molcel.2011.12.013', 'PMID': '22227115'}\n","\t {'PMC': '3381847', 'DOI': '10.1016%2Fj.molcel.2011.12.013', 'PMID': '22227115'}\n","Hale CR, Zhao P, Olson S, Duff MO, Graveley BR, Wells L, Terns RM, Terns MP (November 2009). \"RNA-guided RNA cleavage by a CRISPR RNA-Cas protein complex\". Cell. 139 (5): 945–956. doi:10.1016/j.cell.2009.07.040. PMC 2951265. PMID 19945378.\n","['10.1016/j.cell.2009.07.040']\n","['2951265']\n","['19945378']\n","{'PMC': 'PMC2951265', 'DOI': '10.1016%2Fj.cell.2009.07.040', 'PMID': '19945378'}\n","\t {'PMC': '2951265', 'DOI': '10.1016%2Fj.cell.2009.07.040', 'PMID': '19945378'}\n","Estrella MA, Kuo FT, Bailey S (2016). \"RNA-activated DNA cleavage by the Type III-B CRISPR–Cas effector complex\". Genes and Development. 30 (4): 460–470. doi:10.1101/gad.273722.115. PMC 4762430. PMID 26848046.\n","['10.1101/gad.273722.115']\n","['4762430']\n","['26848046']\n","{'PMC': 'PMC4762430', 'DOI': '10.1101%2Fgad.273722.115', 'PMID': '26848046'}\n","\t {'PMC': '4762430', 'DOI': '10.1101%2Fgad.273722.115', 'PMID': '26848046'}\n","Samai P, Pyenson N, Jiang W, Goldberg GW, Hatoum-Aslan A, Marraffini LA (2015). \"Co-transcriptional DNA and RNA Cleavage during Type III CRISPR-Cas Immunity\". Cell. 161 (5): 1164–1174. doi:10.1016/j.cell.2015.04.027. PMC 4594840. PMID 25959775.\n","['10.1016/j.cell.2015.04.027']\n","['4594840']\n","['25959775']\n","{'PMC': 'PMC4594840', 'DOI': '10.1016%2Fj.cell.2015.04.027', 'PMID': '25959775'}\n","\t {'PMC': '4594840', 'DOI': '10.1016%2Fj.cell.2015.04.027', 'PMID': '25959775'}\n","Marraffini LA, Sontheimer EJ (January 2010). \"Self versus non-self discrimination during CRISPR RNA-directed immunity\". Nature. 463 (7280): 568–571. Bibcode:2010Natur.463..568M. doi:10.1038/nature08703. PMC 2813891. PMID 20072129.\n","['10.1038/nature08703']\n","['2813891']\n","['20072129']\n","{'PMC': 'PMC2813891', 'DOI': '10.1038%2Fnature08703', 'PMID': '20072129'}\n","\t {'PMC': '2813891', 'DOI': '10.1038%2Fnature08703', 'PMID': '20072129'}\n","Krupovic M, Béguin P, Koonin EV (August 2017). \"Casposons: mobile genetic elements that gave rise to the CRISPR-Cas adaptation machinery\". Current Opinion in Microbiology. 38: 36–43. doi:10.1016/j.mib.2017.04.004. PMC 5665730. PMID 28472712.\n","['10.1016/j.mib.2017.04.004']\n","['5665730']\n","['28472712']\n","{'PMC': 'PMC5665730', 'DOI': '10.1016%2Fj.mib.2017.04.004', 'PMID': '28472712'}\n","\t {'PMC': '5665730', 'DOI': '10.1016%2Fj.mib.2017.04.004', 'PMID': '28472712'}\n","Koonin EV, Makarova KS (May 2013). \"CRISPR-Cas: evolution of an RNA-based adaptive immunity system in prokaryotes\". RNA Biology. 10 (5): 679–686. doi:10.4161/rna.24022. PMC 3737325. PMID 23439366.\n","['10.4161/rna.24022']\n","['3737325']\n","['23439366']\n","{'PMC': 'PMC3737325', 'DOI': '10.4161%2Frna.24022', 'PMID': '23439366'}\n","\t {'PMC': '3737325', 'DOI': '10.4161%2Frna.24022', 'PMID': '23439366'}\n","Koonin EV, Makarova KS, Zhang F (June 2017). \"Diversity, classification and evolution of CRISPR-Cas systems\". Current Opinion in Microbiology. 37: 67–78. doi:10.1016/j.mib.2017.05.008. PMC 5776717. PMID 28605718.\n","['10.1016/j.mib.2017.05.008']\n","['5776717']\n","['28605718']\n","{'PMC': 'PMC5776717', 'DOI': '10.1016%2Fj.mib.2017.05.008', 'PMID': '28605718'}\n","\t {'PMC': '5776717', 'DOI': '10.1016%2Fj.mib.2017.05.008', 'PMID': '28605718'}\n","Shmakov S, Smargon A, Scott D, Cox D, Pyzocha N, Yan W, Abudayyeh OO, Gootenberg JS, Makarova KS, Wolf YI, Severinov K, Zhang F, Koonin EV (March 2017). \"Diversity and evolution of class 2 CRISPR-Cas systems\". Nature Reviews. Microbiology. 15 (3): 169–182. doi:10.1038/nrmicro.2016.184. PMC 5851899. PMID 28111461.\n","['10.1038/nrmicro.2016.184']\n","['5851899']\n","['28111461']\n","{'PMC': 'PMC5851899', 'DOI': '10.1038%2Fnrmicro.2016.184', 'PMID': '28111461'}\n","\t {'PMC': '5851899', 'DOI': '10.1038%2Fnrmicro.2016.184', 'PMID': '28111461'}\n","Kupczok A, Bollback JP (February 2013). \"Probabilistic models for CRISPR spacer content evolution\". BMC Evolutionary Biology. 13 (1): 54. doi:10.1186/1471-2148-13-54. PMC 3704272. PMID 23442002.\n","['10.1186/1471-2148-13-54']\n","['3704272']\n","['23442002']\n","{'PMC': 'PMC3704272', 'DOI': '10.1186%2F1471-2148-13-54', 'PMID': '23442002'}\n","\t {'PMC': '3704272', 'DOI': '10.1186%2F1471-2148-13-54', 'PMID': '23442002'}\n","Sternberg SH, Richter H, Charpentier E, Qimron U (March 2016). \"Adaptation in CRISPR-Cas Systems\". Molecular Cell. 61 (6): 797–808. doi:10.1016/j.molcel.2016.01.030. hdl:21.11116/0000-0003-E74E-2. PMID 26949040.\n","['10.1016/j.molcel.2016.01.030']\n","[]\n","['26949040']\n","{'DOI': '10.1016%2Fj.molcel.2016.01.030', 'PMID': '26949040'}\n","\t {'DOI': '10.1016%2Fj.molcel.2016.01.030', 'PMID': '26949040'}\n","Koonin EV, Wolf YI (November 2009). \"Is evolution Darwinian or/and Lamarckian?\". Biology Direct. 4: 42. doi:10.1186/1745-6150-4-42. PMC 2781790. PMID 19906303.\n","['10.1186/1745-6150-4-42']\n","['2781790']\n","['19906303']\n","{'PMC': 'PMC2781790', 'DOI': '10.1186%2F1745-6150-4-42', 'PMID': '19906303'}\n","\t {'PMC': '2781790', 'DOI': '10.1186%2F1745-6150-4-42', 'PMID': '19906303'}\n","Weiss A (October 2015). \"Lamarckian Illusions\". Trends in Ecology & Evolution. 30 (10): 566–568. doi:10.1016/j.tree.2015.08.003. PMID 26411613.\n","['10.1016/j.tree.2015.08.003']\n","[]\n","['26411613']\n","{'DOI': '10.1016%2Fj.tree.2015.08.003', 'PMID': '26411613'}\n","\t {'DOI': '10.1016%2Fj.tree.2015.08.003', 'PMID': '26411613'}\n","Koonin EV, Wolf YI (February 2016). \"Just how Lamarckian is CRISPR-Cas immunity: the continuum of evolvability mechanisms\". Biology Direct. 11 (1): 9. doi:10.1186/s13062-016-0111-z. PMC 4765028. PMID 26912144.\n","['10.1186/s13062-016-0111-z']\n","['4765028']\n","['26912144']\n","{'PMC': 'PMC4765028', 'DOI': '10.1186%2Fs13062-016-0111-z', 'PMID': '26912144'}\n","\t {'PMC': '4765028', 'DOI': '10.1186%2Fs13062-016-0111-z', 'PMID': '26912144'}\n","Heidelberg JF, Nelson WC, Schoenfeld T, Bhaya D (2009).  Ahmed N (ed.). \"Germ warfare in a microbial mat community: CRISPRs provide insights into the co-evolution of host and viral genomes\". PLOS ONE. 4 (1): e4169. Bibcode:2009PLoSO...4.4169H. doi:10.1371/journal.pone.0004169. PMC 2612747. PMID 19132092.\n","['10.1371/journal.pone.0004169']\n","['2612747']\n","['19132092']\n","{'PMC': 'PMC2612747', 'DOI': '10.1371%2Fjournal.pone.0004169', 'PMID': '19132092'}\n","\t {'PMC': '2612747', 'DOI': '10.1371%2Fjournal.pone.0004169', 'PMID': '19132092'}\n","Sampson TR, Saroj SD, Llewellyn AC, Tzeng YL, Weiss DS (May 2013). \"A CRISPR/Cas system mediates bacterial innate immune evasion and virulence\". Nature. 497 (7448): 254–257. Bibcode:2013Natur.497..254S. doi:10.1038/nature12048. PMC 3651764. PMID 23584588.\n","['10.1038/nature12048']\n","['3651764']\n","['23584588']\n","{'PMC': 'PMC3651764', 'DOI': '10.1038%2Fnature12048', 'PMID': '23584588'}\n","\t {'PMC': '3651764', 'DOI': '10.1038%2Fnature12048', 'PMID': '23584588'}\n","Touchon M, Rocha EP (June 2010).  Randau L (ed.). \"The small, slow and specialized CRISPR and anti-CRISPR of Escherichia and Salmonella\". PLOS ONE. 5 (6): e11126. Bibcode:2010PLoSO...511126T. doi:10.1371/journal.pone.0011126. PMC 2886076. PMID 20559554.\n","['10.1371/journal.pone.0011126']\n","['2886076']\n","['20559554']\n","{'PMC': 'PMC2886076', 'DOI': '10.1371%2Fjournal.pone.0011126', 'PMID': '20559554'}\n","\t {'PMC': '2886076', 'DOI': '10.1371%2Fjournal.pone.0011126', 'PMID': '20559554'}\n","Rho M, Wu YW, Tang H, Doak TG, Ye Y (2012). \"Diverse CRISPRs evolving in human microbiomes\". PLOS Genetics. 8 (6): e1002441. doi:10.1371/journal.pgen.1002441. PMC 3374615. PMID 22719260.\n","['10.1371/journal.pgen.1002441']\n","['3374615']\n","['22719260']\n","{'PMC': 'PMC3374615', 'DOI': '10.1371%2Fjournal.pgen.1002441', 'PMID': '22719260'}\n","\t {'PMC': '3374615', 'DOI': '10.1371%2Fjournal.pgen.1002441', 'PMID': '22719260'}\n","Sun CL, Barrangou R, Thomas BC, Horvath P, Fremaux C, Banfield JF (February 2013). \"Phage mutations in response to CRISPR diversification in a bacterial population\". Environmental Microbiology. 15 (2): 463–470. doi:10.1111/j.1462-2920.2012.02879.x. PMID 23057534.\n","['10.1111/j.1462-2920.2012.02879.x']\n","[]\n","['23057534']\n","{'DOI': '10.1111%2Fj.1462-2920.2012.02879.x', 'PMID': '23057534'}\n","\t {'DOI': '10.1111%2Fj.1462-2920.2012.02879.x', 'PMID': '23057534'}\n","Kuno S, Sako Y, Yoshida T (May 2014). \"Diversification of CRISPR within coexisting genotypes in a natural population of the bloom-forming cyanobacterium Microcystis aeruginosa\". Microbiology. 160 (Pt 5): 903–916. doi:10.1099/mic.0.073494-0. PMID 24586036.\n","['10.1099/mic.0.073494-0']\n","[]\n","['24586036']\n","{'DOI': '10.1099%2Fmic.0.073494-0', 'PMID': '24586036'}\n","\t {'DOI': '10.1099%2Fmic.0.073494-0', 'PMID': '24586036'}\n","Sorek R, Kunin V, Hugenholtz P (March 2008). \"CRISPR—a widespread system that provides acquired resistance against phages in bacteria and archaea\". Nature Reviews. Microbiology. 6 (3): 181–186. doi:10.1038/nrmicro1793. PMID 18157154. S2CID 3538077. Table 1: Web resources for CRISPR analysis\n","['10.1038/nrmicro1793']\n","[]\n","['18157154']\n","{'DOI': '10.1038%2Fnrmicro1793', 'PMID': '18157154'}\n","\t {'DOI': '10.1038%2Fnrmicro1793', 'PMID': '18157154'}\n","Pride DT, Salzman J, Relman DA (September 2012). \"Comparisons of clustered regularly interspaced short palindromic repeats and viromes in human saliva reveal bacterial adaptations to salivary viruses\". Environmental Microbiology. 14 (9): 2564–2576. doi:10.1111/j.1462-2920.2012.02775.x. PMC 3424356. PMID 22583485.\n","['10.1111/j.1462-2920.2012.02775.x']\n","['3424356']\n","['22583485']\n","{'PMC': 'PMC3424356', 'DOI': '10.1111%2Fj.1462-2920.2012.02775.x', 'PMID': '22583485'}\n","\t {'PMC': '3424356', 'DOI': '10.1111%2Fj.1462-2920.2012.02775.x', 'PMID': '22583485'}\n","Held NL, Herrera A, Whitaker RJ (November 2013). \"Reassortment of CRISPR repeat-spacer loci in Sulfolobus islandicus\". Environmental Microbiology. 15 (11): 3065–3076. doi:10.1111/1462-2920.12146. PMID 23701169.\n","['10.1111/1462-2920.12146']\n","[]\n","['23701169']\n","{'DOI': '10.1111%2F1462-2920.12146', 'PMID': '23701169'}\n","\t {'DOI': '10.1111%2F1462-2920.12146', 'PMID': '23701169'}\n","Held NL, Herrera A, Cadillo-Quiroz H, Whitaker RJ (September 2010). \"CRISPR associated diversity within a population of Sulfolobus islandicus\". PLOS ONE. 5 (9): e12988. Bibcode:2010PLoSO...512988H. doi:10.1371/journal.pone.0012988. PMC 2946923. PMID 20927396.\n","['10.1371/journal.pone.0012988']\n","['2946923']\n","['20927396']\n","{'PMC': 'PMC2946923', 'DOI': '10.1371%2Fjournal.pone.0012988', 'PMID': '20927396'}\n","\t {'PMC': '2946923', 'DOI': '10.1371%2Fjournal.pone.0012988', 'PMID': '20927396'}\n","Medvedeva S, Liu Y, Koonin EV, Severinov K, Prangishvili D, Krupovic M (November 2019). \"Virus-borne mini-CRISPR arrays are involved in interviral conflicts\". Nature Communications. 10 (1): 5204. Bibcode:2019NatCo..10.5204M. doi:10.1038/s41467-019-13205-2. PMC 6858448. PMID 31729390.\n","['10.1038/s41467-019-13205-2']\n","['6858448']\n","['31729390']\n","{'PMC': 'PMC6858448', 'DOI': '10.1038%2Fs41467-019-13205-2', 'PMID': '31729390'}\n","\t {'PMC': '6858448', 'DOI': '10.1038%2Fs41467-019-13205-2', 'PMID': '31729390'}\n","Skennerton CT, Imelfort M, Tyson GW (May 2013). \"Crass: identification and reconstruction of CRISPR from unassembled metagenomic data\". Nucleic Acids Research. 41 (10): e105. doi:10.1093/nar/gkt183. PMC 3664793. PMID 23511966.\n","['10.1093/nar/gkt183']\n","['3664793']\n","['23511966']\n","{'PMC': 'PMC3664793', 'DOI': '10.1093%2Fnar%2Fgkt183', 'PMID': '23511966'}\n","\t {'PMC': '3664793', 'DOI': '10.1093%2Fnar%2Fgkt183', 'PMID': '23511966'}\n","Stern A, Mick E, Tirosh I, Sagy O, Sorek R (October 2012). \"CRISPR targeting reveals a reservoir of common phages associated with the human gut microbiome\". Genome Research. 22 (10): 1985–1994. doi:10.1101/gr.138297.112. PMC 3460193. PMID 22732228.\n","['10.1101/gr.138297.112']\n","['3460193']\n","['22732228']\n","{'PMC': 'PMC3460193', 'DOI': '10.1101%2Fgr.138297.112', 'PMID': '22732228'}\n","\t {'PMC': '3460193', 'DOI': '10.1101%2Fgr.138297.112', 'PMID': '22732228'}\n","Novick RP, Christie GE, Penadés JR (August 2010). \"The phage-related chromosomal islands of Gram-positive bacteria\". Nature Reviews Microbiology. 8 (8): 541–551. doi:10.1038/nrmicro2393. PMC 3522866. PMID 20634809.\n","['10.1038/nrmicro2393']\n","['3522866']\n","['20634809']\n","{'PMC': 'PMC3522866', 'DOI': '10.1038%2Fnrmicro2393', 'PMID': '20634809'}\n","\t {'PMC': '3522866', 'DOI': '10.1038%2Fnrmicro2393', 'PMID': '20634809'}\n","Ram G, Chen J, Kumar K, Ross HF, Ubeda C, Damle PK, Lane KD, Penadés JR, Christie GE, Novick RP (October 2012). \"Staphylococcal pathogenicity island interference with helper phage reproduction is a paradigm of molecular parasitism\". Proceedings of the National Academy of Sciences of the United States of America. 109 (40): 16300–16305. Bibcode:2012PNAS..10916300R. doi:10.1073/pnas.1204615109. PMC 3479557. PMID 22991467.\n","['10.1073/pnas.1204615109']\n","['3479557']\n","['22991467']\n","{'PMC': 'PMC3479557', 'DOI': '10.1073%2Fpnas.1204615109', 'PMID': '22991467'}\n","\t {'PMC': '3479557', 'DOI': '10.1073%2Fpnas.1204615109', 'PMID': '22991467'}\n","Ram G, Chen J, Ross HF, Novick RP (October 2014). \"Precisely modulated pathogenicity island interference with late phage gene transcription\". Proceedings of the National Academy of Sciences of the United States of America. 111 (40): 14536–14541. Bibcode:2014PNAS..11114536R. doi:10.1073/pnas.1406749111. PMC 4209980. PMID 25246539.\n","['10.1073/pnas.1406749111']\n","['4209980']\n","['25246539']\n","{'PMC': 'PMC4209980', 'DOI': '10.1073%2Fpnas.1406749111', 'PMID': '25246539'}\n","\t {'PMC': '4209980', 'DOI': '10.1073%2Fpnas.1406749111', 'PMID': '25246539'}\n","Seed KD, Lazinski DW, Calderwood SB, Camilli A (February 2013). \"A bacteriophage encodes its own CRISPR/Cas adaptive response to evade host innate immunity\". Nature. 494 (7438): 489–491. Bibcode:2013Natur.494..489S. doi:10.1038/nature11927. PMC 3587790. PMID 23446421.\n","['10.1038/nature11927']\n","['3587790']\n","['23446421']\n","{'PMC': 'PMC3587790', 'DOI': '10.1038%2Fnature11927', 'PMID': '23446421'}\n","\t {'PMC': '3587790', 'DOI': '10.1038%2Fnature11927', 'PMID': '23446421'}\n","\"What is CRISPR and How does it work?\". Livescience.Tech. Retrieved 2019-12-14.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Doudna JA, Charpentier E (November 2014). \"Genome editing. The new frontier of genome engineering with CRISPR-Cas9\". Science. 346 (6213): 1258096. doi:10.1126/science.1258096. PMID 25430774. S2CID 6299381.\n","['10.1126/science.1258096']\n","[]\n","['25430774']\n","{'DOI': '10.1126%2Fscience.1258096', 'PMID': '25430774'}\n","\t {'DOI': '10.1126%2Fscience.1258096', 'PMID': '25430774'}\n","Ledford H (June 2015). \"CRISPR, the disruptor\". Nature. 522 (7554): 20–24. Bibcode:2015Natur.522...20L. doi:10.1038/522020a. PMID 26040877.\n","['10.1038/522020a']\n","[]\n","['26040877']\n","{'DOI': '10.1038%2F522020a', 'PMID': '26040877'}\n","\t {'DOI': '10.1038%2F522020a', 'PMID': '26040877'}\n","Alphey L (2016). \"Can CRISPR-Cas9 gene drives curb malaria?\". Nature Biotechnology. 34 (2): 149–150. doi:10.1038/nbt.3473. PMID 26849518. S2CID 10014014.\n","['10.1038/nbt.3473']\n","[]\n","['26849518']\n","{'DOI': '10.1038%2Fnbt.3473', 'PMID': '26849518'}\n","\t {'DOI': '10.1038%2Fnbt.3473', 'PMID': '26849518'}\n","Bernabé-Orts JM, Casas-Rodrigo I, Minguet EG, Landolfi V, Garcia-Carpintero V, Gianoglio S,  et al. (April 2019). \"Assessment of Cas12a-mediated gene editing efficiency in plants\". Plant Biotechnology Journal. 17 (10): 1971–1984. doi:10.1111/pbi.13113. PMC 6737022. PMID 30950179.\n","['10.1111/pbi.13113']\n","['6737022']\n","['30950179']\n","{'PMC': 'PMC6737022', 'DOI': '10.1111%2Fpbi.13113', 'PMID': '30950179'}\n","\t {'PMC': '6737022', 'DOI': '10.1111%2Fpbi.13113', 'PMID': '30950179'}\n","\"In A 1st, Doctors In U.S. Use CRISPR Tool To Treat Patient With Genetic Disorder\". NPR.org. Retrieved 2019-07-31.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Abuse, National Institute on Drug (2020-02-14). \"Antiretroviral Therapy Combined With CRISPR Gene Editing Can Eliminate HIV Infection in Mice\". National Institute on Drug Abuse. Retrieved 2020-11-15.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","^ In A 1st, Scientists Use Revolutionary Gene-Editing Tool To Edit Inside A Patient\n","\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","The-Crispr (2019-07-15). \"Listen Radiolab CRISPR podcast\". The Crispr. Retrieved 2019-07-15.\n","[]\n","[]\n","[]\n","{}\n","\t {}\n","Ledford H (2017). \"CRISPR studies muddy results of older gene research\". Nature. doi:10.1038/nature.2017.21763. S2CID 90757972.\n","['10.1038/nature.2017.21763']\n","[]\n","[]\n","{'DOI': '10.1038%2Fnature.2017.21763'}\n","\t {'DOI': '10.1038%2Fnature.2017.21763'}\n","Gu W, Crawford ED, O'Donovan BD, Wilson MR, Chow ED, Retallack H, DeRisi JL (March 2016). \"Depletion of Abundant Sequences by Hybridization (DASH): using Cas9 to remove unwanted high-abundance species in sequencing libraries and molecular counting applications\". Genome Biology. 17 (1): 41. doi:10.1186/s13059-016-0904-5. PMC 4778327. PMID 26944702.\n","['10.1186/s13059-016-0904-5']\n","['4778327']\n","['26944702']\n","{'PMC': 'PMC4778327', 'DOI': '10.1186%2Fs13059-016-0904-5', 'PMID': '26944702'}\n","\t {'PMC': '4778327', 'DOI': '10.1186%2Fs13059-016-0904-5', 'PMID': '26944702'}\n","Gootenberg JS, Abudayyeh OO, Lee JW, Essletzbichler P, Dy AJ, Joung J,  et al. (April 2017). \"Nucleic acid detection with CRISPR-Cas13a/C2c2\". Science. 356 (6336): 438–442. Bibcode:2017Sci...356..438G. doi:10.1126/science.aam9321. PMC 5526198. PMID 28408723.\n","['10.1126/science.aam9321']\n","['5526198']\n","['28408723']\n","{'PMC': 'PMC5526198', 'DOI': '10.1126%2Fscience.aam9321', 'PMID': '28408723'}\n","\t {'PMC': '5526198', 'DOI': '10.1126%2Fscience.aam9321', 'PMID': '28408723'}\n","Chen JS, Ma E, Harrington LB, Da Costa M, Tian X, Palefsky JM, Doudna JA (April 2018). \"CRISPR-Cas12a target binding unleashes indiscriminate single-stranded DNase activity\". Science. 360 (6387): 436–439. Bibcode:2018Sci...360..436C. doi:10.1126/science.aar6245. PMC 6628903. PMID 29449511.\n","['10.1126/science.aar6245']\n","['6628903']\n","['29449511']\n","{'PMC': 'PMC6628903', 'DOI': '10.1126%2Fscience.aar6245', 'PMID': '29449511'}\n","\t {'PMC': '6628903', 'DOI': '10.1126%2Fscience.aar6245', 'PMID': '29449511'}\n","Iwasaki RS, Batey RT (September 2020). \"SPRINT: a Cas13a-based platform for detection of small molecules\". Nucleic Acids Research. 48 (17): e101. doi:10.1093/nar/gkaa673. PMID 32797156.\n","['10.1093/nar/gkaa673']\n","[]\n","['32797156']\n","{'DOI': '10.1093%2Fnar%2Fgkaa673', 'PMID': '32797156'}\n","\t {'DOI': '10.1093%2Fnar%2Fgkaa673', 'PMID': '32797156'}\n","Dhamad AE, Abdal Rhida MA (2020). \"COVID-19: molecular and serological detection methods\". PeerJ. 8: e10180. doi:10.7717/peerj.10180. PMC 7547594. PMID 33083156.\n","['10.7717/peerj.10180']\n","['7547594']\n","['33083156']\n","{'PMC': 'PMC7547594', 'DOI': '10.7717%2Fpeerj.10180', 'PMID': '33083156'}\n","\t {'PMC': '7547594', 'DOI': '10.7717%2Fpeerj.10180', 'PMID': '33083156'}\n","Konwarh R (September 2020). \"Can CRISPR/Cas Technology Be a Felicitous Stratagem Against the COVID-19 Fiasco? Prospects and Hitches\". Frontiers in Molecular Biosciences. 7: 557377. doi:10.3389/fmolb.2020.557377. PMC 7511716. PMID 33134311.\n","['10.3389/fmolb.2020.557377']\n","['7511716']\n","['33134311']\n","{'PMC': 'PMC7511716', 'DOI': '10.3389%2Ffmolb.2020.557377', 'PMID': '33134311'}\n","\t {'PMC': '7511716', 'DOI': '10.3389%2Ffmolb.2020.557377', 'PMID': '33134311'}\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":36},"id":"Wz7IU9yGKmdI","executionInfo":{"status":"ok","timestamp":1610649155670,"user_tz":-60,"elapsed":567,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"cd71d060-cf06-40e5-856b-aa0b77234572"},"source":["''.split('-')[-1]"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["''"]},"metadata":{"tags":[]},"execution_count":17}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"6-sHsP8THGMz","executionInfo":{"status":"ok","timestamp":1610648355430,"user_tz":-60,"elapsed":604,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"3bca7f85-bbf4-4cd5-89e3-29c075756081"},"source":["# refs = revision.get_references()\n","# refs[0].get_text()\n","etree.tostring(refs[0].source)"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["b'<li id=\"cite_note-pmid20056882-1\"><span class=\"mw-cite-backlink\">^ <a href=\"#cite_ref-pmid20056882_1-0\"><sup><i><b>a</b></i></sup></a> <a href=\"#cite_ref-pmid20056882_1-1\"><sup><i><b>b</b></i></sup></a> <a href=\"#cite_ref-pmid20056882_1-2\"><sup><i><b>c</b></i></sup></a> <a href=\"#cite_ref-pmid20056882_1-3\"><sup><i><b>d</b></i></sup></a></span> <span class=\"reference-text\"><span class=\"error\">Attention: This template ({{<a href=\"/wiki/Template:Cite_pmid\" title=\"Template:Cite pmid\">cite pmid</a>}}) is deprecated. To cite the publication identified by <a class=\"external mw-magiclink-pmid\" rel=\"nofollow\" href=\"//www.ncbi.nlm.nih.gov/pubmed/20056882?dopt=Abstract\">PMID 20056882</a>, please use {{<a href=\"/wiki/Template:Cite_journal\" title=\"Template:Cite journal\">cite journal</a>}} with <code class=\"nowrap\" style=\"\">|pmid=20056882</code> instead.</span></span>\\n</li>\\n'"]},"metadata":{"tags":[]},"execution_count":12}]},{"cell_type":"markdown","metadata":{"id":"F661L0u7UxeY"},"source":["### Find all intext citations"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":234},"id":"c-sdy2RPOINS","executionInfo":{"status":"error","timestamp":1610648064493,"user_tz":-60,"elapsed":1149,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"151c2710-7ead-4af0-d4b4-9437aea5f8e3"},"source":["# intext citations in Wikipedia html are sup-tags of class='refreence'\n","for i in text3.xpath(\".//sup[@class='reference']\"):\n","  print(etree.tostring(i))\n","  # supid = i.get('id')\n","  # print('\\t', supid)\n","  for a in i.xpath(\".//a\"): # Use \".\" to search relatively. OTherwise it searches the whole tree\n","    # print(etree.tostring(a))\n","    refid = a.get('href')\n","    print('\\t',refid)\n","    refnum = a.xpath(\".//text()\")\n","    if len(refnum) == 1:\n","      print('\\t',refnum[0])\n","    else:\n","      print('\\tMore than one refnum:',refnum[0])\n"],"execution_count":null,"outputs":[{"output_type":"error","ename":"NameError","evalue":"ignored","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m<ipython-input-1-50abab0e058d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# intext citations in Wikipedia html are sup-tags of class='refreence'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtext3\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxpath\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\".//sup[@class='reference']\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m   \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0metree\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtostring\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m   \u001b[0;31m# supid = i.get('id')\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m   \u001b[0;31m# print('\\t', supid)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mNameError\u001b[0m: name 'text3' is not defined"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":105},"id":"qteACwSlNMpW","executionInfo":{"status":"ok","timestamp":1610633002851,"user_tz":-60,"elapsed":815,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"badabbf4-5613-4fc6-d809-9c03fc9c48fb"},"source":["\"\".join(text3.xpath(\".//div[@class='mw-parser-output']\")[0].xpath(\"//text()\")).strip()"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'Diagram of the possible mechanism for CRISPR.[1]\\nCRISPRs (clustered regularly interspaced short palindromic repeats) are DNA loci containing short repetitions of base sequences. Each repetition is followed by short segments of \"spacer DNA\" from previous exposures to a virus.[2]\\nCRISPRs are found in approximately 40% of sequenced eubacteria genomes and 90% of sequenced archaea.[3][4]\\nCRISPRs are often associated with cas genes that code for proteins related to CRISPRs. The CRISPR-Cas system is a prokaryotic immune system that confers resistance to foreign genetic elements such as plasmids and phages[5][6] and provides a form of acquired immunity. CRISPR spacers recognize and silence these exogenous genetic elements like RNAi in eukaryotic organisms.[2]\\nSince 2012, the CRISPR-Cas system has been used for gene editing (silencing, enhancing or changing specific genes) that even works in eukaryotes like mice and primates.[7] By inserting a plasmid containing cas genes and specifically designed CRISPRs, an organism\\'s genome can be cut at any desired location.\\n\\nContents\\n\\n1 History\\n2 Locus structure\\n\\n2.1 Repeats and spacers\\n2.2 Cas genes and CRISPR subtypes\\n\\n\\n3 Mechanism\\n4 Evolutionary significance\\n5 Applications\\n\\n5.1 Mouse models\\n\\n\\n6 CRISPR/Cas system in phage\\n7 Multi-gene targeting\\n8 Reversible knockdown\\n9 Gene activation\\n10 Infrastructure\\n11 Therapeutics\\n12 Safety\\n13 Alternatives\\n14 Patent\\n15 References\\n16 Further reading\\n17 External links\\n\\n\\n\\nHistory\\nBacteria may incorporate foreign DNA in other circumstances and even to scavenge damaged DNA from their environment.[8]\\nRepeats were first described in 1987 for the bacterium Escherichia coli.[9] In 2000, similar clustered repeats were identified in additional bacteria and archaea and were termed Short Regularly Spaced Repeats (SRSR).[10] SRSR were renamed CRISPR in 2002.[11] A set of genes, some encoding putative nuclease or helicase proteins, were found to be associated with CRISPR repeats (the cas, or CRISPR-associated genes).[11]\\n\\n  Simplified diagram of a CRISPR locus. The three major components of a CRISPR locus are shown: cas genes, a leader sequence, and a repeat-spacer array. Repeats are shown as grey boxes and spacers are colored bars. While most CRISPR loci contain each of the three components, the arrangement is not always as shown.[1][2]\\nIn 2005, three independent researchers showed that CRISPR spacers showed homology to several phage DNA and extrachromosomal DNA such as plasmids. This was an indication that the CRISPR/cas system could have a role  in adaptive immunity in bacteria.[1] Koonin and colleagues proposed that spacers serve as a template for RNA molecules, analogously to eukaryotic cells that use a system called RNA interference.[12]\\nIn 2007 Barrangou, Horvath (food industry scientists at Danisco) and others showed that they could alter the resistance of Streptococcus thermophilus to phage attack with spacer DNA.[12]\\nDoudna and Charpentier had independently been exploring CRISPR-associated proteins to learn how bacteria deploy spacers in their immune defenses. They jointly studied a simpler CRISPR system that relies on a protein called Cas9. They found that bacteria respond to an invading phage by transcribing spacers and palindromic DNA into a long RNA molecule that the cell then uses tracrRNA and Cas9 to cut it into pieces called crRNAs.[12]\\nCas9 is a nuclease, an enzyme specialized for cutting DNA, with two active cutting sites, one for each strand of the double helix. The team demonstrated that they could disable one or both sites while preserving Cas9\\'s ability to home located its target DNA. Jinek combined tracrRNA and spacer RNA into a \"single-guide RNA\" molecule that, mixed with Cas9, could find and cut the correct DNA targets. Jinek et al proposed that such synthetic guide RNAs might be able to be used for gene editing.[12]\\nCRISPR was first shown to work as a genome engineering/editing tool in human cell culture by Jennifer Doudna in 2012 (PMID 22745249) and George M. Church\\'s group at Harvard University.[13] It has since been used in a wide range of organisms including bakers yeast (S. cerevisiae),[14] zebra fish,[15] nematodes (C. elegans),[16] plants,[17] mice,[18] and several other organisms.\\nAdditionally CRISPR has been modified to make programmable transcription factors that allow scientists to target and activate or silence specific genes.[19]\\nLibraries of tens of thousands of guide RNAs are now available.[12]\\nThe first evidence that CRISPR can reverse disease symptoms in living animals was demonstrated in March 2014, when MIT researchers cured mice of a rare liver disorder.[20]\\n\\nLocus structure\\nRepeats and spacers\\nCRISPR repeats range in size from 24 to 48 base pairs.[21] They usually show some dyad symmetry, implying the formation of a secondary structure such as a hairpin, but are not truly palindromic.[22] Repeats are separated by spacers of similar length.[21] Some CRISPR spacer sequences exactly match sequences from plasmids and phages,[23][24][25] although some spacers match the prokaryote\\'s genome (self-targeting spacers).[26] New spacers can be added rapidly in response to phage infection.[27]\\n\\nCas genes and CRISPR subtypes\\nCRISPR-associated (cas) genes are often associated with CRISPR repeat-spacer arrays. As of 2013, more than forty different Cas protein families had been described.[21] Of these protein families, Cas1 appears to be ubiquitous among different CRISPR/Cas systems. Particular combinations of cas genes and repeat structures have been used to define 8 CRISPR subtypes (Ecoli, Ypest, Nmeni, Dvulg, Tneap, Hmari, Apern, and Mtube), some of which are associated with an additional gene module encoding repeat-associated mysterious proteins (RAMPs).[21] More than one CRISPR subtype may occur in a single genome. The sporadic distribution of the CRISPR/Cas subtypes suggests that the system is subject to horizontal gene transfer during microbial evolution.\\n\\nCRISPR associated proteincrystal structure of a crispr-associated protein from thermus thermophilusIdentifiersSymbolCRISPR_assocPfamPF08798Pfam clanCL0362InterProIPR010179CDDcd09727Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nCRISPR associated protein Cas2crystal structure of a hypothetical protein tt1823 from thermus thermophilusIdentifiersSymbolCRISPR_Cas2PfamPF09827InterProIPR019199CDDcd09638Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nCRISPR-associated protein Cse1IdentifiersSymbolCRISPR_Cse1PfamPF09481InterProIPR013381CDDcd09729Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nCRISPR-associated protein Cse2IdentifiersSymbolCRISPR_Cse2PfamPF09485InterProIPR013382CDDcd09670Available protein structures:Pfam\\n \\xa0structures / ECOD\\n \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\nMechanism\\nExogenous DNA is apparently processed by proteins encoded by Cas genes into small elements (~30 base pairs in length), which are then somehow inserted into the CRISPR locus near the leader sequence. RNAs from the CRISPR loci are constitutively expressed and are processed by Cas proteins to small RNAs composed of individual, exogenously-derived sequence elements with a flanking repeat sequence. The RNAs guide other Cas proteins to silence exogenous genetic elements at the RNA or DNA level.[1][28] Evidence suggests functional diversity among CRISPR subtypes. The Cse (Cas subtype Ecoli) proteins (called CasA-E in E. coli) form a functional complex, Cascade, that processes CRISPR RNA transcripts into spacer-repeat units that Cascade retains.[29] In other prokaryotes, Cas6 processes the CRISPR transcripts. Interestingly, CRISPR-based phage inactivation in E. coli requires Cascade and Cas3, but not Cas1 and Cas2. The Cmr (Cas RAMP module) proteins found in Pyrococcus furiosus and other prokaryotes form a functional complex with small CRISPR RNAs that recognizes and cleaves complementary target RNAs. RNA-guided CRISPR enzymes are classified as type V restriction enzymes.\\n\\nEvolutionary significance\\nA bioinformatic study showed that CRISPRs are evolutionarily conserved and cluster into related types. Many show signs of a conserved secondary structure.[22]\\nThrough the CRISPR/Cas mechanism, bacteria can acquire immunity to certain phages and thus halt further transmission of targeted phages. For this reason, some researchers have proposed that CRISPR/Cas is a Lamarckian inheritance mechanism.[30] Others investigated the coevolution of host and viral genomes by analysis of CRISPR sequences.[31]\\nCas9 proteins are highly enriched in pathogenic and commensal bacteria. CRISPR/Cas-mediated gene regulation may contribute to the regulation of endogenous bacterial genes, particularly during bacterial interaction with eukaryotic hosts. For example, Cas protein Cas9 of Francisella novicida uses a unique, small, CRISPR/Cas-associated RNA (scaRNA) to repress an endogenous transcript encoding a bacterial lipoprotein that is critical for F. novicida to dampen host response and promote virulence.[32]\\n\\nApplications\\nThe proof-of-principle demonstration of selective engineered redirection of the CRISPR-Cas system in 2012[33] provided a first step toward realization of proposals for CRISPR-derived biotechnology:[34]\\n\\nArtificial immunization against phage by introduction of engineered CRISPR loci in industrially important bacteria, including those used in food production and large-scale fermentation\\nGenome engineering at cellular or organismic level by reprogramming a CRISPR-Cas system to achieve RNA-guided genome engineering. Proof of concept studies demonstrated examples both in vitro[35] and in vivo[18][36][37]\\nDiscrimination of bacterial strains by comparison of spacer sequences\\nMouse models\\nCRISPR simplifies creation of mouse models and reduces the time required to a matter of weeks from months or longer. Knockdown of endogenous genes has been achieved by transfection with a plasmid that contains a CRISPR area with a spacer, which inhibits a target gene. Injecting mouse zygotes with Cas9 and two guide RNAs was able to disable two genes with 80% efficiency. So-called homology-directed repair involves using Cas9 to \"nick\" DNA, to introduce new gene parts to the zygote.[citation needed]\\n\\nCRISPR/Cas system in phage\\nAnother way that bacteria can defend against phage infection is by having chromosomal islands. A subtype of chromosomal islands called phage-inducible chromosomal island (PICI) is excised from bacterial chromosome upon phage infection and can inhibit phage replication.[38] The mechanisms that induce PICI excision and how PICI inhibits phage replication are not well understood. One study showed that lytic ICP1 phage that specifically targets Vibrio cholerae serogroup O1 has acquired a CRISPR/Cas system that targets a V. cholera PICI-like element. The system has 2 CRISPR loci and 9 Cas genes. It seems to be homologous to the 1-F system found in Yersinia pestis. Moreover, like the bacterial CRISPR/Cas system, ICP1 CRISPR/Cas system can also acquire new sequences, which allows the phage to co-evolve with its host.[39]\\n\\nMulti-gene targeting\\nCRISPRs have been used to cut as many as five genes at once.[12]\\n\\nReversible knockdown\\nFurther information: CRISPR interference\\n\"CRISPRi\" like RNAi, turns off genes in a reversible fashion by targeting a site but not cutting. In bacteria, the presence of Cas9 alone is enough to block transcription, but for mammalian applications, a section of protein is added. Its guide RNA targets regulatory DNA, called promoters that immediately precede the gene target.[12]\\n\\nGene activation\\nFurther information: CRISPR interference\\nCas9 has been used to carry synthetic transcription factors (protein fragments that turn on genes.) This enabled the activation of specific human genes. The technique achieved a strong effect by targeting multiple CRISPR constructs to slightly different spots on the gene\\'s promoter.[12]\\nThe genes included some tied to human diseases, including those involved in muscle differentiation, cancer, inflammation and producing fetal hemoglobin.[12]\\n\\nInfrastructure\\nFree software is available to design guide RNA to target any desired gene. The Addgene repository offers academics the DNA to make their own CRISPR system for $65. Since the beginning of the year, In 2013 alone Addgene distributed more than 10,000 CRISPR constructs. The facility has received CRISPR-enabling DNA sequences from 11 independent research teams.[12]\\n\\nTherapeutics\\nEditas Medicine, a $43 million startup, aims to develop treatments that employ CRISPR/Cas to make edits to single base pairs and larger stretches of DNA. Inherited diseases such as cystic fibrosis, sickle-cell anemia and Huntington\\'s disease are caused by single base pair mutations; CRISPR/Cas technology has the potential to correct these errors. The \"corrected\" gene remains in its normal location on its chromosome, which preserves the way the cell normally activate inhibits its expression.[40]\\n\\nAfter harvesting blood cell precursors called hematopoietic stem cells from a patient\\'s bone marrow, CRISPR gene surgery would correct the defective gene. Then the gene-\\xadcorrected stem cells would be returned to the patient\\'s marrow, which would then produce healthy red blood cells. Replacing 70% of the sickle cells would produce a cure.[41]\\n\\nBefore it can be used clinically, the company must be able to guarantee that only the targeted region will be affected and determine how to deliver the therapy to a patient’s cells.[40]\\n\\nSafety\\nBefore work proceeds much further on human genomes, improved targeting is required. Guide RNAs have been observed targeting sequences that differ by multiple base pairs from the intended sequence.[12]\\n\\nAlternatives\\nIn the early 2000s, researchers developed zinc finger nucleases, synthetic proteins whose DNA-binding domains enable them to cut DNA at specific spots. Later, synthetic nucleases called TALENs provided an easier way to target specific DNA and were predicted to surpass zinc fingers. They both depend on custom-making new proteins for each DNA target, a more cumbersome procedure than guide RNAs. CRISPRs are more efficient[42] and can target more genes than these earlier techniques.[42]\\n\\nPatent\\nA provisional US patent application on the use of the CRISPR system for editing genes and regulating gene expression was filed on May 12, 2012 by the inventors. Subsequent applications were combined and on March 6, 2014 the resulting patent application was published by the USPTO.[43] The patent rights have been assigned by the inventors to the Regents of the University of California and to the University of Vienna.\\n\\nReferences\\n\\n\\n^ a b c d Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20056882, please use {{cite journal}} with |pmid=20056882 instead.\\n\\n^ a b c Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20125085, please use {{cite journal}} with |pmid=20125085 instead.\\n\\n^ 71/79 Archaea, 463/1008 Bacteria CRISPRdb, Date: 19.6.2010\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17521438, please use {{cite journal}} with |pmid=17521438 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17379808, please use {{cite journal}} with |pmid=17379808 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19095942, please use {{cite journal}} with |pmid=19095942 instead.\\n\\n^ Wiedenheft, B (2012 Feb 15). \"RNA-guided genetic silencing systems in bacteria and archaea\". Nature. 482 (7385): 331–8. PMID\\xa022337052. Unknown parameter |coauthors= ignored (|author= suggested) (help); Check date values in: |date= (help).mw-parser-output cite.citation{font-style:inherit}.mw-parser-output .citation q{quotes:\"\\\\\"\"\"\\\\\"\"\"\\'\"\"\\'\"}.mw-parser-output .id-lock-free a,.mw-parser-output .citation .cs1-lock-free a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/6/65/Lock-green.svg\")right 0.1em center/9px no-repeat}.mw-parser-output .id-lock-limited a,.mw-parser-output .id-lock-registration a,.mw-parser-output .citation .cs1-lock-limited a,.mw-parser-output .citation .cs1-lock-registration a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/d/d6/Lock-gray-alt-2.svg\")right 0.1em center/9px no-repeat}.mw-parser-output .id-lock-subscription a,.mw-parser-output .citation .cs1-lock-subscription a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/a/aa/Lock-red-alt-2.svg\")right 0.1em center/9px no-repeat}.mw-parser-output .cs1-subscription,.mw-parser-output .cs1-registration{color:#555}.mw-parser-output .cs1-subscription span,.mw-parser-output .cs1-registration span{border-bottom:1px dotted;cursor:help}.mw-parser-output .cs1-ws-icon a{background:linear-gradient(transparent,transparent),url(\"//upload.wikimedia.org/wikipedia/commons/4/4c/Wikisource-logo.svg\")right 0.1em center/12px no-repeat}.mw-parser-output code.cs1-code{color:inherit;background:inherit;border:none;padding:inherit}.mw-parser-output .cs1-hidden-error{display:none;font-size:100%}.mw-parser-output .cs1-visible-error{font-size:100%}.mw-parser-output .cs1-maint{display:none;color:#33aa33;margin-left:0.3em}.mw-parser-output .cs1-subscription,.mw-parser-output .cs1-registration,.mw-parser-output .cs1-format{font-size:95%}.mw-parser-output .cs1-kern-left,.mw-parser-output .cs1-kern-wl-left{padding-left:0.2em}.mw-parser-output .cs1-kern-right,.mw-parser-output .cs1-kern-wl-right{padding-right:0.2em}.mw-parser-output .citation .mw-selflink{font-weight:inherit}\\n\\n^ Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1073/pnas.1315278110, please use {{cite journal}} with |doi=10.1073/pnas.1315278110 instead.\"Bacteria incorporate pieces of old DNA in their own genome, scientists discover\". KurzweilAI. doi:10.1073/pnas.1315278110. Retrieved 2013-11-26.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 3316184, please use {{cite journal}} with |pmid=3316184 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 10760181, please use {{cite journal}} with |pmid=10760181 instead.\\n\\n^ a b Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 11952905, please use {{cite journal}} with |pmid=11952905 instead.\\n\\n^ a b c d e f g h i j k Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1126/science.341.6148.833, please use {{cite journal}} with |doi=10.1126/science.341.6148.833 instead.\\n\\n^ \"CRISPR gene therapy: Scientists call for more public debate around breakthrough technique - Science - News\". The Independent. 2013-11-07. Retrieved 2013-11-25.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23460208, please use {{cite journal}} with |pmid=23460208 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23360964, please use {{cite journal}} with |pmid=23360964 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23817069, please use {{cite journal}} with |pmid=23817069 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23999092, please use {{cite journal}} with |pmid=23999092 instead.\\n\\n^ a b Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23643243, please use {{cite journal}} with |pmid=23643243 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 24136345, please use {{cite journal}} with |pmid=24136345 instead.\\n\\n^ \"Researchers reverse a liver disorder in mice by correcting a mutated gene\". PhysOrg. 30 March 2014. Retrieved 31 March 2014.\\n\\n^ a b c d Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 16292354, please use {{cite journal}} with |pmid=16292354 instead.\\n\\n^ a b Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17442114, please use {{cite journal}} with |pmid=17442114 instead.\\n\\n^ Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1007/s00239-004-0046-3, please use {{cite journal}} with |doi=10.1007/s00239-004-0046-3 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 16079334 , please use {{cite journal}} with |pmid=16079334  instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 15758212, please use {{cite journal}} with |pmid=15758212 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20598393, please use {{cite journal}} with |pmid=20598393 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 17894817, please use {{cite journal}} with |pmid=17894817 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 16545108, please use {{cite journal}} with |pmid=16545108 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18703739, please use {{cite journal}} with |pmid=18703739 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19906303, please use {{cite journal}} with |pmid=19906303 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19132092, please use {{cite journal}} with |pmid=19132092 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23584588, please use {{cite journal}} with |pmid=23584588 instead.\\n\\n^ Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1016/j.molcel.2011.10.023, please use {{cite journal}} with |doi=10.1016/j.molcel.2011.10.023 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18157154, please use {{cite journal}} with |pmid=18157154 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 22745249, please use {{cite journal}} with |pmid=22745249 instead.Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 22949671, please use {{cite journal}} with |pmid=22949671 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23287718, please use {{cite journal}} with |pmid=23287718 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23287722, please use {{cite journal}} with |pmid=23287722 instead.Attention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1073/pnas.1313587110, please use {{cite journal}} with |doi=10.1073/pnas.1313587110 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20634809, please use {{cite journal}} with |pmid=20634809 instead.\\n\\n^ Attention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 23446421, please use {{cite journal}} with |pmid=23446421 instead.\\n\\n^ a b Young, Susan. \"Biotech Startup Editas Medicine Wants to Cure Grievous Genetic Diseases with New Genome Editing Technology | MIT Technology Review\". Technologyreview.com. Retrieved 2013-11-30.\\n\\n^ Young, Susan (2014-02-11). \"CRISPR and Other Genome Editing Tools Boost Medical Research and Gene Therapy\\'s Reach | MIT Technology Review\". Technologyreview.com. Retrieved 2014-04-13.\\n\\n^ a b Young, Susan (11 February 2014) Genome Surgery MIT Technology Review, Retrieved 17 February 2014\\n\\n^ \"Methods And Compositions For Rna-directed Target Dna Modification And For Rna-directed Modulation Of Transcription\". Freshpatents.com. Retrieved 2014-04-13.\\n\\n\\nFurther reading\\n.mw-parser-output .refbegin{font-size:90%;margin-bottom:0.5em}.mw-parser-output .refbegin-hanging-indents>ul{list-style-type:none;margin-left:0}.mw-parser-output .refbegin-hanging-indents>ul>li,.mw-parser-output .refbegin-hanging-indents>dl>dd{margin-left:0;padding-left:3.2em;text-indent:-3.2em;list-style:none}.mw-parser-output .refbegin-100{font-size:100%}.mw-parser-output .refbegin-columns{margin-top:0.3em}.mw-parser-output .refbegin-columns dl,.mw-parser-output .refbegin-columns ol,.mw-parser-output .refbegin-columns ul{margin-top:0}.mw-parser-output .refbegin-columns li,.mw-parser-output .refbegin-columns dd{page-break-inside:avoid;break-inside:avoid-column}\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 24584096, please use {{cite journal}} with |pmid=24584096 instead.\\nAttention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1016/j.tig.2014.01.003, please use {{cite journal}} with |doi=10.1016/j.tig.2014.01.003 instead.\\nAttention: This template ({{cite doi}}) is deprecated. To cite the publication identified by doi:10.1038/nrmicro3241, please use {{cite journal}} with |doi=10.1038/nrmicro3241 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18065539, please use {{cite journal}} with |pmid=18065539 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18065545, please use {{cite journal}} with |pmid=18065545 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18497291, please use {{cite journal}} with |pmid=18497291 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 18971321, please use {{cite journal}} with |pmid=18971321 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19141480, please use {{cite journal}} with |pmid=19141480 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19143596, please use {{cite journal}} with |pmid=19143596 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19239620, please use {{cite journal}} with |pmid=19239620 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19246744, please use {{cite journal}} with |pmid=19246744 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19383692, please use {{cite journal}} with |pmid=19383692 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19945378, please use {{cite journal}} with |pmid=19945378 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 19945373, please use {{cite journal}} with |pmid=19945373 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20072129, please use {{cite journal}} with |pmid=20072129 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20129051, please use {{cite journal}} with |pmid=20129051 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20132443, please use {{cite journal}} with |pmid=20132443 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20133361, please use {{cite journal}} with |pmid=20133361 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20528693, please use {{cite journal}} with |pmid=20528693 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20556198, please use {{cite journal}} with |pmid=20556198 instead.\\nAttention: This template ({{cite pmid}}) is deprecated. To cite the publication identified by PMID 20559554, please use {{cite journal}} with |pmid=20559554 instead.\\n\\nExternal links\\nE-CRISP.org A comprehensive software for CRISPR gRNA design\\nCRISPR Design Tool\\nCRISPR Design Tool with scoring algorithms\\nTool for finding CRISPRs\\nTool for finding CRISPR targets in other nucleic acids\\nRfam page for the CRISPR entries\\n.mw-parser-output .navbar{display:inline;font-size:88%;font-weight:normal}.mw-parser-output .navbar-collapse{float:left;text-align:left}.mw-parser-output .navbar-boxtext{word-spacing:0}.mw-parser-output .navbar ul{display:inline-block;white-space:nowrap;line-height:inherit}.mw-parser-output .navbar-brackets::before{margin-right:-0.125em;content:\"[ \"}.mw-parser-output .navbar-brackets::after{margin-left:-0.125em;content:\" ]\"}.mw-parser-output .navbar li{word-spacing:-0.125em}.mw-parser-output .navbar-mini abbr{font-variant:small-caps;border-bottom:none;text-decoration:none;cursor:inherit}.mw-parser-output .navbar-ct-full{font-size:114%;margin:0 7em}.mw-parser-output .navbar-ct-mini{font-size:114%;margin:0 4em}.mw-parser-output .infobox .navbar{font-size:100%}.mw-parser-output .navbox .navbar{display:block;font-size:100%}.mw-parser-output .navbox-title .navbar{float:left;text-align:left;margin-right:0.5em}vteGallery of secondary structure images@media all and (max-width:720px){.mw-parser-output .mod-gallery{width:100%!important}}.mw-parser-output .mod-gallery{display:table}.mw-parser-output .mod-gallery-default{background:transparent;margin-top:.3em}.mw-parser-output .mod-gallery-center{margin-left:auto;margin-right:auto}.mw-parser-output .mod-gallery-left{float:left}.mw-parser-output .mod-gallery-right{float:right}.mw-parser-output .mod-gallery-none{float:none}.mw-parser-output .mod-gallery-collapsible{width:100%}.mw-parser-output .mod-gallery .title,.mw-parser-output .mod-gallery .main,.mw-parser-output .mod-gallery .footer{display:table-row}.mw-parser-output .mod-gallery .title>div{display:table-cell;text-align:center;font-weight:bold}.mw-parser-output .mod-gallery .main>div{display:table-cell}.mw-parser-output .mod-gallery .gallery{line-height:1.35em}.mw-parser-output .mod-gallery .footer>div{display:table-cell;text-align:right;font-size:80%;line-height:1em}.mw-parser-output .mod-gallery .title>div *,.mw-parser-output .mod-gallery .footer>div *{overflow:visible}.mw-parser-output .mod-gallery .gallerybox img{background:none!important}.mw-parser-output .mod-gallery .bordered-images .thumb img{border:solid #eaecf0 1px}.mw-parser-output .mod-gallery .whitebg .thumb{background:#fff!important}\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR2:  Secondary structure taken from the Rfam database. Family RF01315.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR5:  Secondary structure taken from the Rfam database. Family RF011318.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR6:  Secondary structure taken from the Rfam database. Family RF01319.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR8:  Secondary structure taken from the Rfam database. Family RF01321.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR9:  Secondary structure taken from the Rfam database. Family RF01322.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR19:  Secondary structure taken from the Rfam database. Family RF01332.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR41:  Secondary structure taken from the Rfam database. Family RF01350.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR52:  Secondary structure taken from the Rfam database. Family RF01365.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR57:  Secondary structure taken from the Rfam database. Family RF01370.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t\\nCRISPR-DR65:  Secondary structure taken from the Rfam database. Family RF01378.\\n\\n\\t\\t\\t\\n\\t\\t\\n\\n\\n\\n\\n\\n\\nCategories: Repetitive DNA sequencesNon-coding RNA'"]},"metadata":{"tags":[]},"execution_count":31}]},{"cell_type":"markdown","metadata":{"id":"kGgAMOM6zulH"},"source":["## Extract contexts"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":414},"id":"ghiJjtnuvUIB","executionInfo":{"status":"error","timestamp":1610559404039,"user_tz":-60,"elapsed":1479,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"7fd0caa7-a102-4eaa-97f5-19e7cd919bc7"},"source":["for article, article_name in [\n","                # (CRISPR_en, 'CRISPR_en'),\n","                # (CRISPR_gene_editing_en, 'CRISPR_gene_editing_en'),\n","                ]:\n","  print(article_name)\n","  for keyphrase in [\n","                                    \n","                # discoveries\n","                'first',\n","                'discover',\n","                \n","                # naming things\n","                'name', # CRISPR acronym\n","                'acronym', # CRISPR acronym\n","                \n","                # patents\n","                'patent',\n","                'property',\n","                'right',\n","\n","                # clinical trials\n","                'trial',\n","                'clinical',\n","                'patient',\n","                'approve', # e.g. FDA approved...\n","\n","                # awards\n","                'award',\n","                'prize',\n","                'nobel',\n","\n","                # people:\n","                'doudna', \n","                'charpentier', \n","                'zhang', \n","                'šikšnys', # with all accents\n","                  'šiksnys', # only first accent\n","                  'sikšnys', # only second accent\n","                  'siksnys', # no accents\n","                'gašiūnas', # with all accents\n","                  'gasiūnas', # only first accent\n","                  'gašiunas', # only second accent\n","                  'gasiunas', # no accents\n","                'horvath', # auch im Zusammenhang mit Šikšnys, aber auch so interessant\n","                'barrangou', # auch im Zusammenhang mit Šikšnys, aber auch so interessant\n","                'church',\n","                'brouns',\n","                'mojica',\n","                'ishino',\n","                'jansen',\n","                'marraffini',\n","                'sontheimer',\n","                'koonin',\n","                'moineau',\n","                'bolotin',\n","                'sorokin',\n","                'makarova',\n","                'pourcel',\n","                'overbeek',\n","                'liang',\n","                ' xu', # added space!\n","                'jiankui',\n","                'nishimasu',\n","                'shibata',\n","                'lander', # review\n","                \n","                # institutions:\n","                'osaka',\n","                'alicante',\n","                'danisco',\n","                'broad',\n","                'harvard',\n","                'university of california', \n","                'berkeley',\n","                \n","                ]:\n","    print('\\t' + keyphrase)\n","    df = keyphrase_in_context(keyphrase, article, width=100, score_cutoff=75)\n","    df.sort_values(by=['left_id','timestamp'])\n","    df.to_excel(path_extractions + '/contexts_{}_{}.xlsx'.format(article_name, keyphrase))"],"execution_count":null,"outputs":[{"output_type":"stream","text":["CRISPR_gene_editing_en\n","\tarno\n"],"name":"stdout"},{"output_type":"error","ename":"TypeError","evalue":"ignored","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m<ipython-input-13-4d864295b87a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     78\u001b[0m                 ]:\n\u001b[1;32m     79\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\\t'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mkeyphrase\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 80\u001b[0;31m     \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkeyphrase_in_context\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkeyphrase\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marticle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwidth\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m100\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscore_cutoff\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m75\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     81\u001b[0m     \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mby\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'left_id'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'timestamp'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     82\u001b[0m     \u001b[0;31m# df.to_excel(path_extractions + '/contexts_{}_{}.xlsx'.format(article_name, keyphrase))\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m<ipython-input-12-6d40ff7cce69>\u001b[0m in \u001b[0;36mkeyphrase_in_context\u001b[0;34m(keyphrase, article, width, score_cutoff, scorer)\u001b[0m\n\u001b[1;32m      5\u001b[0m   \u001b[0mgen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marticle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0myield_revisions\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m   \u001b[0mcontexts\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m   \u001b[0;32mfor\u001b[0m \u001b[0mrevision\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgen\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m     \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrevision\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/content/drive/My Drive/WIKING/WIKING_git/code/article/article.py\u001b[0m in \u001b[0;36myield_revisions\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     44\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m                 \u001b[0mrevision\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mloads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 46\u001b[0;31m                 \u001b[0;32myield\u001b[0m \u001b[0mRevision\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mrevision\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     47\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     48\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mtrack_field_values_in_article\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfields\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbibliography\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mTypeError\u001b[0m: __init__() missing 1 required positional argument: 'wikitext'"]}]},{"cell_type":"markdown","metadata":{"id":"4k2Me_0RyhFO"},"source":["### CRISPR_en: \"first\"\n","\n"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":788},"id":"CPm0Mh-syhFP","executionInfo":{"status":"ok","timestamp":1606682187152,"user_tz":-60,"elapsed":97739,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"4885e2f4-70b3-4a97-da4a-cf6c8843f258"},"source":["df = keyphrase_in_context('first', CRISPR_en, width=50, score_cutoff=75)\n","df.to_excel(file_contexts_CRISPR_en)\n","df.sort_values(by=['left_id','timestamp'])"],"execution_count":null,"outputs":[{"output_type":"stream","text":["WARNING:root:Applied processor reduces input query to empty string, all comparisons will have score 0. [Query: '']\n"],"name":"stderr"},{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>revindx</th>\n","      <th>timestamp</th>\n","      <th>left_id</th>\n","      <th>left</th>\n","      <th>keyphrase</th>\n","      <th>right</th>\n","      <th>right_id</th>\n","      <th>revid</th>\n","      <th>url</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>25</th>\n","      <td>133</td>\n","      <td>2012-01-07 04:36:56</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>470025144</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>27</th>\n","      <td>134</td>\n","      <td>2012-08-01 19:27:33</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>505309163</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>29</th>\n","      <td>135</td>\n","      <td>2012-08-22 21:44:35</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>508685743</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>31</th>\n","      <td>136</td>\n","      <td>2012-12-06 19:12:03</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>526751001</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>33</th>\n","      <td>137</td>\n","      <td>2012-12-11 08:13:19</td>\n","      <td>0</td>\n","      <td>n of the crispr-cas system in 2012[21] provided a</td>\n","      <td>first</td>\n","      <td>step toward realization of some of the several pr</td>\n","      <td>33</td>\n","      <td>527490124</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>26284</th>\n","      <td>2039</td>\n","      <td>2020-11-02 21:24:41</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986762186</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26297</th>\n","      <td>2040</td>\n","      <td>2020-11-03 09:53:58</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986842914</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26310</th>\n","      <td>2041</td>\n","      <td>2020-11-03 20:16:34</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986927023</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26323</th>\n","      <td>2042</td>\n","      <td>2020-11-03 22:56:29</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>986949106</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>26336</th>\n","      <td>2043</td>\n","      <td>2020-11-04 21:00:44</td>\n","      <td>75</td>\n","      <td>th the newly acquired spacer inserted between the</td>\n","      <td>first</td>\n","      <td>and second direct repeats.[95][116]</td>\n","      <td>5</td>\n","      <td>987088136</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>26340 rows × 9 columns</p>\n","</div>"],"text/plain":["       revindx  ...                                                url\n","25         133  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","27         134  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","29         135  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","31         136  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","33         137  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","...        ...  ...                                                ...\n","26284     2039  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26297     2040  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26310     2041  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26323     2042  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","26336     2043  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","\n","[26340 rows x 9 columns]"]},"metadata":{"tags":[]},"execution_count":18}]},{"cell_type":"markdown","metadata":{"id":"fc1dPCidx823"},"source":["### CRISPR_gene_editing_en: \"first\"\n","\n"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":771},"id":"dB8a9egkrWyg","executionInfo":{"status":"ok","timestamp":1606681779613,"user_tz":-60,"elapsed":6031,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"508e83c7-d9e5-4ccc-8689-f495287716c9"},"source":["df = keyphrase_in_context('first', CRISPR_gene_editing_en, width=50, score_cutoff=75)\n","df.to_excel(file_contexts_CRISPR_gene_editing_en)\n","df.sort_values(by=['left_id','timestamp'])"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>revindx</th>\n","      <th>timestamp</th>\n","      <th>left_id</th>\n","      <th>left</th>\n","      <th>keyphrase</th>\n","      <th>right</th>\n","      <th>right_id</th>\n","      <th>revid</th>\n","      <th>url</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>3</th>\n","      <td>0</td>\n","      <td>2019-02-17 06:32:48</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727671</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>10</th>\n","      <td>1</td>\n","      <td>2019-02-17 06:33:54</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727741</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>17</th>\n","      <td>2</td>\n","      <td>2019-02-17 06:35:41</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727860</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>24</th>\n","      <td>3</td>\n","      <td>2019-02-17 06:37:14</td>\n","      <td>0</td>\n","      <td>340.2.[132] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883727959</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>31</th>\n","      <td>4</td>\n","      <td>2019-02-17 06:39:37</td>\n","      <td>0</td>\n","      <td>340.2.[131] yang's white button mushroom was the</td>\n","      <td>first</td>\n","      <td>organism genetically modified with the crispr/cas</td>\n","      <td>3</td>\n","      <td>883728113</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>1447</th>\n","      <td>184</td>\n","      <td>2020-10-23 09:36:41</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>984993980</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1458</th>\n","      <td>185</td>\n","      <td>2020-11-02 04:29:36</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986649176</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1469</th>\n","      <td>186</td>\n","      <td>2020-11-02 10:34:00</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986682078</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1480</th>\n","      <td>187</td>\n","      <td>2020-11-02 10:35:18</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986682164</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","    <tr>\n","      <th>1491</th>\n","      <td>188</td>\n","      <td>2020-11-02 10:35:49</td>\n","      <td>13</td>\n","      <td>ay:table-row}.mw-parser-output .portal&gt;ul&gt;li&gt;s...</td>\n","      <td>first</td>\n","      <td>-child{display:table-cell;padding:0.2em;vertic...</td>\n","      <td>15</td>\n","      <td>986682206</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>1497 rows × 9 columns</p>\n","</div>"],"text/plain":["      revindx  ...                                                url\n","3           0  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","10          1  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","17          2  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","24          3  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","31          4  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","...       ...  ...                                                ...\n","1447      184  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1458      185  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1469      186  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1480      187  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","1491      188  ...  https://en.wikipedia.org/w/index.php?title=CRI...\n","\n","[1497 rows x 9 columns]"]},"metadata":{"tags":[]},"execution_count":14}]},{"cell_type":"markdown","metadata":{"id":"PgVJr7mPzwj_"},"source":["# Old stuff..."]},{"cell_type":"markdown","metadata":{"id":"6QRPtrvO9Mvp"},"source":["### Occurrence spans"]},{"cell_type":"code","metadata":{"id":"PrlDj91nsCLY"},"source":["def occurance_spans(keyphrase, article):\n","  ''' \n","  returns a list of time spans during which a keyphrase 'keyphrase' continously appears in an article 'a'\n","  '''\n","  gen = article.yield_revisions()\n","  last_rev_was_match = False\n","  spans = []\n","  for revision in gen:\n","    if keyphrase in revision.get_text():\n","    # if keyphrase in i.get_text().split('\\nLocus structure\\n')[0]: # Trick 17 to exclude everything after history section and practially only search in histroy section\n","      last_rev_that_matched = revision\n","      if not last_rev_was_match:\n","        span_beg = revision\n","        last_rev_was_match = True\n","    else:\n","      if last_rev_was_match:\n","        spans.append((span_beg, last_rev_that_matched))\n","        last_rev_was_match = False\n","  if last_rev_was_match:\n","    spans.append((span_beg, last_rev_that_matched))\n","  print(\n","      '\"{}\" was found in the following revisions of {}:\\n{}'.format(keyphrase, 'Test', '\\n'.join(['Span {}: {} - {}'.format(indx + 1, revision[0].timestamp, revision[1].timestamp) for indx,revision in enumerate(spans)])) \n","      if spans else '\"{}\" was not found in any revision of \"{}\"'.format(keyphrase, 'Test')\n","      )\n","  return spans"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"Y5HIHBrA9Sak"},"source":["### Unique contexts"]},{"cell_type":"code","metadata":{"id":"_eUn_KzFheAn"},"source":["def unique_contexts(keyphrase, article, width=50):\n","  ''' \n","  returns a pandas df with three columns:\n","  1) unique contexts of width in which keyphrase appears in article\n","  2) timestamp for first occurrence of each context\n","  3) url for first occurrence of each context\n","  '''\n","  kill = re.compile(r'\\s*\\[.*?\\]')\n","  gen = article.yield_revisions()\n","  unique_contexts = []\n","  timestamps = []\n","  urls = []\n","  for revision in gen:\n","    text = revision.get_text()\n","    # if s in text.split('\\nLocus structure\\n')[0]: # Trick 17 to exclude everything after history section and practially only search in histroy section\n","    if keyphrase in text:\n","      text = kill.sub('', text) # kill intext refs like '[23]'\n","      text = text.lower()\n","      for indx in find_all(text, keyphrase):\n","        # make '/n' the boundary for context\n","        before = text[indx - width : indx]\n","        after = text[indx + len(keyphrase): indx + len(keyphrase) + width]\n","        context = keyphrase[::-1]\n","        for char in before[::-1]:\n","          if char == '\\n':\n","            break\n","          context += char\n","        context = context[::-1]\n","        for char in after:\n","          if char == '\\n':\n","            break\n","          context += char\n","        # add unique contexts and add their meta data\n","        context = context.strip()\n","        if not context in unique_contexts:\n","          unique_contexts.append(context)\n","          timestamps.append(revision.timestamp.string)\n","          urls.append(revision.url)    \n","  print('Found {} unique contexts for \"{}\" in article \"{}\".'.format(len(unique_contexts), keyphrase, article.name))\n","  # return list(zip(unique_contexts, timestamps, urls))\n","  return pd.DataFrame({'Context':unique_contexts, \n","        'Timestamp':timestamps,\n","        'Url':urls}) "],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"ajWDEbMWHHSw"},"source":["### Left-right contexts"]},{"cell_type":"code","metadata":{"id":"7elYtyzQr-PP"},"source":["def lr_context_graph(keyphrase, article, width=50):\n","  ''' \n","  Returns networkx DiGraph for links between contexts to the left and to the right of keyphrase\n","  '''\n","  gen = article.yield_revisions()\n","  DG = nx.DiGraph(keyphrase=keyphrase)\n","  for revision in gen:\n","    text = revision.get_text()\n","    text = text.lower()\n","    # text = re.sub(r'\\s*\\[.*?\\]','', text) # kill intext refs like '[23]'\n","    metadata = (revision.timestamp.string, revision.url)\n","    if keyphrase in text:\n","      for indx in find_all(text, keyphrase):\n","        # context left\n","        left = ''\n","        for char in text[indx - width : indx][::-1]:\n","          if char == '\\n': # make '/n' the boundary for context\n","            break\n","          left += char\n","        left = left[::-1].strip()\n","        # add/update node for context left\n","        if not left in DG:\n","          DG.add_node(left, as_left=[metadata], as_right=[])\n","        else:\n","          DG.nodes[left]['as_left'].append(metadata)\n","        # context right\n","        right = ''\n","        for char in text[indx + len(keyphrase): indx + len(keyphrase) + width]:\n","          if char == '\\n': # make '/n' the boundary for context\n","            break\n","          right += char\n","        right = right.strip()\n","        # add/update node for context right\n","        if not right in DG:\n","          DG.add_node(right, as_right=[metadata], as_left=[])\n","        else:\n","          DG.nodes[right]['as_right'].append(metadata)\n","        # add/update edge from context left to context right\n","        if not [left, right] in DG.edges:\n","          DG.add_edge(left, right, occurrences=[metadata])\n","        else:\n","            DG.edges[left,right]['occurrences'].append(metadata)\n","  return DG"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"o3QeZcjHpN83"},"source":["### Node communities"]},{"cell_type":"code","metadata":{"id":"4cJ9mGfolzDZ"},"source":["def node_community_dic(DG, score_cutoff=90, scorer=fuzz.ratio):\n","  keyphrase = DG.graph['keyphrase']\n","  G = nx.Graph()\n","  for side in ['left', 'right']:\n","    choices = [node[0] for node in DG.nodes.data() if node[1]['as_{}'.format(side)]]\n","    length = len(choices)\n","    for query in choices:\n","      indx = choices.index(query) + 1\n","      if indx <= length:\n","        scores = process.extractBests(query, choices[choices.index(query) + 1:], score_cutoff=score_cutoff, limit=length, scorer=scorer) # https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/process.py\n","        edges = [(query, i[0]) for i in scores]\n","        G.add_edges_from(edges)\n","  return community_louvain.best_partition(G)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"xTaYKRvg6hN-"},"source":["### Master node for each community"]},{"cell_type":"code","metadata":{"id":"ATQugc1nqHFy"},"source":["def master_nodes_dic(DG, ncd):\n","  '''\n","  Returns dictionary with community_ids as keys and nodes + latest occurrence as value\n","  '''\n","  result = {}\n","  reverse_mnd = {v:[k for k in mnd if mnd[k] == v] for v in mnd.values()}\n","  for community_id, nodes in reverse_mnd.items():\n","    candidates = []\n","    for node in nodes:\n","      latest = max([DG.nodes[node]['as_{}'.format(side)][-1][0] for side in ['left', 'right'] if DG.nodes[node]['as_{}'.format(side)]])\n","      candidates.append((node, latest))\n","    winner = sorted(candidates, key= lambda x: (x[1], x[0]), reverse=True)[0]\n","    result[community_id] = winner\n","  return result"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"KpfcElE48O3W"},"source":["### Reduced LR contexts graph (using ``lr_context_graph``, ``node_community_dic``, and ``master_nodes_dic``)"]},{"cell_type":"code","metadata":{"id":"wgGSr0Fl8ii7"},"source":["def reduce_lr_context_graph(DG, ncd, mnd):\n","  pass"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"bYzX8wKFrxkB"},"source":["## Load data"]},{"cell_type":"code","metadata":{"id":"5pFtlIlNiYtw"},"source":["CRISPR_en = Article(file_CRISPR_en)\n","CRISPR_gene_editing_en = Article(file_CRISPR_gene_editing_en)\n","\n","# events = pd.read_csv(file_events)\n","# accounts = pd.read_csv(file_accounts)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"iN1eI6MICMu_"},"source":["## Experiments"]},{"cell_type":"markdown","metadata":{"id":"CPg3sesxo00n"},"source":["### Extract LR contexts and merge nodes "]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ZAA1h3_wuO3D","executionInfo":{"status":"ok","timestamp":1606500810339,"user_tz":-60,"elapsed":615,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"6b665a2a-d41b-4e75-9d90-9da12af436f8"},"source":["DG = lr_context_graph('first', CRISPR_gene_editing_en, width=50)\n","ncd = node_community_dic(DG)\n","# sorted(ncd.items(), key=lambda item:(item[1], item[0]))\n","mnd = master_nodes_dic(DG, ncd)\n","mnd"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["{0: (\"340.2.[145] yang's white button mushroom was the\",\n","  '2020-11-02 10:35:49'),\n"," 1: ('t position 8 (help); no-break space character in |',\n","  '2020-01-31 21:33:44'),\n"," 2: ('demonstrated in 2012 in:[25][26]', '2019-02-18 01:48:30'),\n"," 3: ('organism genetically modified with the crispr-cas',\n","  '2020-11-02 10:35:49'),\n"," 4: ('= at position 5 (help); no-break space character i',\n","  '2020-01-31 21:33:44')}"]},"metadata":{"tags":[]},"execution_count":255}]},{"cell_type":"markdown","metadata":{"id":"IJppJpuY6ZAF"},"source":["### Older stuff"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"D_vRSWezNP31","executionInfo":{"status":"ok","timestamp":1606423116262,"user_tz":-60,"elapsed":76291,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"fe9f6fea-2e38-4840-ab2c-d17b6f5b9009"},"source":["contexts_first = unique_contexts('first', CRISPR_en, width=50)\n","contexts_first.to_excel(file_contexts_CRISPR_en)\n","\n","contexts_first = unique_contexts('first', CRISPR_gene_editing_en, width=50)\n","contexts_first.to_excel(file_contexts_CRISPR_gene_editing_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Found 143 unique contexts for \"first\" in article \"CRISPR_en\".\n","Found 23 unique contexts for \"first\" in article \"CRISPR_gene_editing_en\".\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"yJguDbiEIxKv","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1606410107435,"user_tz":-60,"elapsed":85293,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"a30c97b9-40ce-41d7-9431-988943b84a8a"},"source":["timeline = occurance_spans('CRISPR Timeline', CRISPR_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"CRISPR Timeline\" was found in the following revisions of Test:\n","Span 1: {'datetime': datetime.datetime(2016, 9, 28, 9, 26, 29),\n"," 'day': 28,\n"," 'hour': 9,\n"," 'minute': 26,\n"," 'month': 9,\n"," 'second': 29,\n"," 'string': '2016-09-28 09:26:29',\n"," 'year': 2016} - {'datetime': datetime.datetime(2016, 12, 28, 8, 5, 49),\n"," 'day': 28,\n"," 'hour': 8,\n"," 'minute': 5,\n"," 'month': 12,\n"," 'second': 49,\n"," 'string': '2016-12-28 08:05:49',\n"," 'year': 2016}\n","Span 2: {'datetime': datetime.datetime(2019, 3, 18, 14, 33, 42),\n"," 'day': 18,\n"," 'hour': 14,\n"," 'minute': 33,\n"," 'month': 3,\n"," 'second': 42,\n"," 'string': '2019-03-18 14:33:42',\n"," 'year': 2019} - {'datetime': datetime.datetime(2019, 3, 18, 14, 34, 29),\n"," 'day': 18,\n"," 'hour': 14,\n"," 'minute': 34,\n"," 'month': 3,\n"," 'second': 29,\n"," 'string': '2019-03-18 14:34:29',\n"," 'year': 2019}\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"MtcXB1nBIkkq","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1604323635576,"user_tz":-60,"elapsed":74224,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"5367bc1b-4e68-442c-e636-2702ceb5a7c9"},"source":["# history_trick17 = occurance_spans('\\nLocus structure\\n', CRISPR_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"\n","Locus structure\n","\" was not found in any revision of \"Test\"\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"R2nt3LFoDezQ","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1604322549498,"user_tz":-60,"elapsed":78372,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"439e1385-bad6-41a0-e927-2043fe561133"},"source":["history = occurance_spans('\\nLocus structure\\n', CRISPR_en)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"\n","Locus structure\n","\" was found in the following revisions of Test:\n","Span 1: {'datetime': datetime.datetime(2013, 11, 24, 21, 3, 53),\n"," 'day': 24,\n"," 'hour': 21,\n"," 'minute': 3,\n"," 'month': 11,\n"," 'second': 53,\n"," 'string': '2013-11-24 21:03:53',\n"," 'year': 2013} - {'datetime': datetime.datetime(2016, 9, 22, 22, 1, 43),\n"," 'day': 22,\n"," 'hour': 22,\n"," 'minute': 1,\n"," 'month': 9,\n"," 'second': 43,\n"," 'string': '2016-09-22 22:01:43',\n"," 'year': 2016}\n","Span 2: {'datetime': datetime.datetime(2016, 9, 23, 0, 51, 24),\n"," 'day': 23,\n"," 'hour': 0,\n"," 'minute': 51,\n"," 'month': 9,\n"," 'second': 24,\n"," 'string': '2016-09-23 00:51:24',\n"," 'year': 2016} - {'datetime': datetime.datetime(2017, 9, 18, 8, 58, 26),\n"," 'day': 18,\n"," 'hour': 8,\n"," 'minute': 58,\n"," 'month': 9,\n"," 'second': 26,\n"," 'string': '2017-09-18 08:58:26',\n"," 'year': 2017}\n","Span 3: {'datetime': datetime.datetime(2017, 9, 18, 9, 0, 40),\n"," 'day': 18,\n"," 'hour': 9,\n"," 'minute': 0,\n"," 'month': 9,\n"," 'second': 40,\n"," 'string': '2017-09-18 09:00:40',\n"," 'year': 2017} - {'datetime': datetime.datetime(2018, 9, 24, 16, 48, 11),\n"," 'day': 24,\n"," 'hour': 16,\n"," 'minute': 48,\n"," 'month': 9,\n"," 'second': 11,\n"," 'string': '2018-09-24 16:48:11',\n"," 'year': 2018}\n","Span 4: {'datetime': datetime.datetime(2018, 9, 29, 6, 55, 25),\n"," 'day': 29,\n"," 'hour': 6,\n"," 'minute': 55,\n"," 'month': 9,\n"," 'second': 25,\n"," 'string': '2018-09-29 06:55:25',\n"," 'year': 2018} - {'datetime': datetime.datetime(2018, 9, 29, 6, 55, 25),\n"," 'day': 29,\n"," 'hour': 6,\n"," 'minute': 55,\n"," 'month': 9,\n"," 'second': 25,\n"," 'string': '2018-09-29 06:55:25',\n"," 'year': 2018}\n","Span 5: {'datetime': datetime.datetime(2018, 9, 30, 12, 48, 13),\n"," 'day': 30,\n"," 'hour': 12,\n"," 'minute': 48,\n"," 'month': 9,\n"," 'second': 13,\n"," 'string': '2018-09-30 12:48:13',\n"," 'year': 2018} - {'datetime': datetime.datetime(2019, 2, 28, 15, 35, 54),\n"," 'day': 28,\n"," 'hour': 15,\n"," 'minute': 35,\n"," 'month': 2,\n"," 'second': 54,\n"," 'string': '2019-02-28 15:35:54',\n"," 'year': 2019}\n","Span 6: {'datetime': datetime.datetime(2019, 2, 28, 15, 38, 47),\n"," 'day': 28,\n"," 'hour': 15,\n"," 'minute': 38,\n"," 'month': 2,\n"," 'second': 47,\n"," 'string': '2019-02-28 15:38:47',\n"," 'year': 2019} - {'datetime': datetime.datetime(2019, 5, 20, 16, 37, 18),\n"," 'day': 20,\n"," 'hour': 16,\n"," 'minute': 37,\n"," 'month': 5,\n"," 'second': 18,\n"," 'string': '2019-05-20 16:37:18',\n"," 'year': 2019}\n","Span 7: {'datetime': datetime.datetime(2019, 5, 21, 17, 37, 36),\n"," 'day': 21,\n"," 'hour': 17,\n"," 'minute': 37,\n"," 'month': 5,\n"," 'second': 36,\n"," 'string': '2019-05-21 17:37:36',\n"," 'year': 2019} - {'datetime': datetime.datetime(2020, 10, 7, 14, 20, 24),\n"," 'day': 7,\n"," 'hour': 14,\n"," 'minute': 20,\n"," 'month': 10,\n"," 'second': 24,\n"," 'string': '2020-10-07 14:20:24',\n"," 'year': 2020}\n","Span 8: {'datetime': datetime.datetime(2020, 10, 8, 1, 22, 24),\n"," 'day': 8,\n"," 'hour': 1,\n"," 'minute': 22,\n"," 'month': 10,\n"," 'second': 24,\n"," 'string': '2020-10-08 01:22:24',\n"," 'year': 2020} - {'datetime': datetime.datetime(2020, 10, 10, 21, 4, 18),\n"," 'day': 10,\n"," 'hour': 21,\n"," 'minute': 4,\n"," 'month': 10,\n"," 'second': 18,\n"," 'string': '2020-10-10 21:04:18',\n"," 'year': 2020}\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"XPiKH4pzvbRs","colab":{"base_uri":"https://localhost:8080/","height":52},"executionInfo":{"status":"ok","timestamp":1603732761615,"user_tz":-60,"elapsed":937,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"5bed8eb7-3188-4821-db88-9ab8092531c6"},"source":["mojica = occurance_spans('Mojica', CRISPR_en)\n","# for span in doudna:\n","#   print(span[0].timestamp, span[1].timestamp)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"Mojica\" was found in the following revisions of Test:\n","Span 1: 2010-01-11T02:11:54Z - 2020-09-02T08:32:27Z\n"],"name":"stdout"}]}]}