{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"arno_experiments.ipynb","provenance":[{"file_id":"1IK-QF90VpGAZmfH0XqDrD8doJfuC6B92","timestamp":1610636648085}],"collapsed_sections":["BG3rjcwdsAvH","oqOIgx_RBue9","n-kefUUG2Lj3","IpZaf0Otpqth","rRJN-eU96egO","RYiX-CAS8F31","4rG5Bsw18IoO"],"authorship_tag":"ABX9TyMH3O026ItuZys+9WX4s4bQ"},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"n4eNz7RIrzow"},"source":["# Set globals"]},{"cell_type":"code","metadata":{"id":"PJwyfS7eiPWw","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1614777826311,"user_tz":-60,"elapsed":8039,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"978597f5-a344-4b5a-e26e-f899310bcdc6"},"source":["# set paths\n","path_extractions = '/content/drive/My Drive/WIKING/extractions'\n","path_git = '/content/drive/My Drive/WIKING/WIKING_git'\n","path_my_tools = '/content/drive/My Drive/my_tools'\n","\n","# infiles\n","file_CRISPR_de = path_extractions + '/wikipedia_entries/CRISPR_de' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/\n","file_CRISPR_en = path_extractions + '/wikipedia_entries/CRISPR_en' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/\n","file_CRISPR_gene_editing_en = path_extractions + '/wikipedia_entries/CRISPR_gene_editing_en' # get latest file version here: https://files.webis.de/wikipedia-tracing-innovations/ \n","file_events = path_git + '/data/CRISPR_events - events.csv'\n","file_accounts = path_git + '/data/CRISPR_events - accounts.csv'\n","\n","# mount Google Drive\n","from google.colab import drive\n","drive.mount('/content/drive')\n","\n","# add additional folders to path variable\n","import sys\n","sys.path.append(path_git + '/code')\n","\n","# import Wolfgang's classes:\n","!pip install Levenshtein # required by Wolfgang's code\n","from article.article import Article\n","# from scraper.scraper import Scraper\n","\n","# import other modules\n","!pip install fuzzywuzzy # https://github.com/seatgeek/fuzzywuzzy\n","from fuzzywuzzy import fuzz\n","from fuzzywuzzy import process\n","import networkx as nx\n","!pip install python-louvain\n","from community import community_louvain\n","import re\n","import os\n","import json\n","import requests\n","from bs4 import BeautifulSoup\n","import pandas as pd\n","import numpy as np\n","import matplotlib.pyplot as plt\n","import itertools\n","import pprint\n","pp = pprint.PrettyPrinter(indent=4)\n","from lxml import etree\n","import lxml # because I want to be able to call lxml.html to avoid a name conflict with solo html \n","\n","# # import pywikibot\n","# os.chdir(path_extractions)\n","# !pip install pywikibot\n","# pywikibot_config = r\"\"\"# -*- coding: utf-8  -*-\n","\n","# mylang = 'en'\n","# family = 'wikipedia'\n","# usernames['wikipedia']['en'] = 'test'\"\"\"\n","# with open('user-config.py', 'w', encoding=\"utf-8\") as f:\n","#     f.write(pywikibot_config)\n","# import pywikibot # https://doc.wikimedia.org/pywikibot/master/api_ref/pywikibot.page.html\n","\n","# set working directory\n","os.chdir(path_git)\n","os.getcwd()\n","\n","# Load data\n","CRISPR_en = Article(file_CRISPR_en)\n","CRISPR_gene_editing_en = Article(file_CRISPR_gene_editing_en)\n","# events = pd.read_csv(file_events)\n","# accounts = pd.read_csv(file_accounts)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n","Requirement already satisfied: Levenshtein in /usr/local/lib/python3.7/dist-packages (0.12.0)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from Levenshtein) (54.0.0)\n","Requirement already satisfied: fuzzywuzzy in /usr/local/lib/python3.7/dist-packages (0.18.0)\n","Requirement already satisfied: python-louvain in /usr/local/lib/python3.7/dist-packages (0.15)\n","Requirement already satisfied: networkx in /usr/local/lib/python3.7/dist-packages (from python-louvain) (2.5)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from python-louvain) (1.19.5)\n","Requirement already satisfied: decorator>=4.3.0 in /usr/local/lib/python3.7/dist-packages (from networkx->python-louvain) (4.4.2)\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"BG3rjcwdsAvH"},"source":["# Functions"]},{"cell_type":"code","metadata":{"id":"9MsA6ct8QIwV"},"source":["def make_soup(url, headers={'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1)'}):\n","  # return BeautifulSoup(requests.get(url).text, 'html.parser') \n","  return BeautifulSoup(requests.get(url, headers=headers).text, 'html.parser')"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"ZCghcj7J1tZb"},"source":["def cluster_lr_contexts(lr_contexts, score_cutoff=75, scorer=fuzz.ratio, sort_by=['left_id','timestamp']):\n","  # determine communities of similarity for left and right contexts\n","  df = pd.DataFrame(lr_contexts, columns=['revindx','timestamp','left_id','left','keyphrase','right','right_id','revid','url','user','userid','comment',])\n","  for side in ['left', 'right']:\n","    G = nx.Graph()\n","    unique_contexts = list(df[side].unique())\n","    length = len(unique_contexts)\n","    for query in unique_contexts:\n","      indx = unique_contexts.index(query) + 1\n","      if indx <= length:\n","        scores = process.extractBests(query, unique_contexts[unique_contexts.index(query) + 1:], score_cutoff=score_cutoff, limit=length, scorer=scorer) # https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/process.py\n","        edges = [(query, i[0]) for i in scores]\n","        G.add_edges_from(edges)\n","    # add community_ids for lonesome contexts as well\n","    unique_contexts = set(unique_contexts)\n","    community_dic = community_louvain.best_partition(G)\n","    if community_dic:\n","      community_contexts = set(community_dic.keys())\n","      lonesome_contexts = unique_contexts - community_contexts\n","      lonesome_dic = {k:v for v,k in enumerate(lonesome_contexts,start=max(community_dic.values())+1)}\n","      community_dic.update(lonesome_dic) # hiermit nun 'lid' und 'rid' Spalten befüllen...\n","    else:\n","      community_dic = {k:v for v,k in enumerate(unique_contexts)}\n","    # add community_ids to df\n","    column = df.apply(lambda row: community_dic[row[side]], axis=1)\n","    df['{}_id'.format(side)] = column\n","  df.sort_values(by=sort_by)\n","  return df"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"wDV3TrWOpci9"},"source":["# https://stackoverflow.com/questions/6822725/rolling-or-sliding-window-iterator\n","def window(seq, n=2):\n","    \"Returns a sliding window (of width n) over data from the iterable\"\n","    \"   s -> (s0,s1,...s[n-1]), (s1,s2,...,sn), ...                   \"\n","    it = iter(seq)\n","    result = tuple(itertools.islice(it, n))\n","    if len(result) == n:\n","        yield result\n","    for elem in it:\n","        result = result[1:] + (elem,)\n","        yield result"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"-VYLnLUxsZRG"},"source":["# Check names\n"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Bj_srtMb362l","executionInfo":{"status":"ok","timestamp":1614787741537,"user_tz":-60,"elapsed":526,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"728dad8d-1cc2-4204-8bca-8890f6e1ce60"},"source":["with open(path_extractions + '/author_names_CRISPR.txt', 'r') as reader:\n","  names = sorted(set(name[1].title() # normalize style (what about accents?!)\n","              for name in [line.strip().split('\\t') for line in reader.readlines()][1:] \n","              if len(name[1].title()) > 2 # exclude short names\n","              ))\n","len(names)"],"execution_count":41,"outputs":[{"output_type":"execute_result","data":{"text/plain":["1195"]},"metadata":{"tags":[]},"execution_count":41}]},{"cell_type":"code","metadata":{"id":"IGNCrwxSscF2","colab":{"base_uri":"https://localhost:8080/","height":606},"executionInfo":{"status":"error","timestamp":1614798041933,"user_tz":-60,"elapsed":15261,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"49372218-5ae0-44a6-fc76-c5ca3dd87155"},"source":["with open(path_extractions + '/author_names_CRISPR.txt', 'r') as reader:\n","  names = sorted(set(name[1].title() # normalize style (what about accents?!)\n","              for name in [line.strip().split('\\t') for line in reader.readlines()][1:] \n","              if len(name[1].title()) > 2 # exclude short names\n","              ))\n","\n","# names = ['Doudna','Charpentier']\n","\n","include_sections = ['History', 'Applications', 'Discovery and properties', 'The significance for evolution and possible applications']\n","exclude_sections = ['Contents', 'References', 'Reference', 'Further reading', 'External links', 'See also', 'Notes', 'Web sites', ]\n","\n","articles = [    \n","  # CRISPR_gene_editing_en,\n","  CRISPR_en,\n","]\n","\n","for article in articles:\n","  print(article.name)\n","  names_found = sorted(set(\n","      name\n","      for article in articles\n","      for revision in article.yield_revisions()\n","      for name in names\n","      for section in revision.get_arno_sections()\n","      if section.level == 2\n","      if not section.heading in exclude_sections\n","      if name in section.get_all_text()\n","  ))\n","  print('\\tNames found:',len(names_found))\n","\n","  with open(path_extractions + f'/author_names_in_{article.name}.txt', 'w') as writer:\n","    for name in names_found:\n","      writer.write(name + '\\n')\n","\n","\n","\n","# def occurance_spans(keyphrase, article):\n","#   ''' \n","#   returns a list of time spans during which a keyphrase 'keyphrase' continously appears in an article 'a'\n","#   '''\n","#   gen = article.yield_revisions()\n","#   last_rev_was_match = False\n","#   spans = []\n","#   for revision in gen:\n","#     if keyphrase in revision.get_text():\n","#     # if keyphrase in i.get_text().split('\\nLocus structure\\n')[0]: # Trick 17 to exclude everything after history section and practially only search in histroy section\n","#       last_rev_that_matched = revision\n","#       if not last_rev_was_match:\n","#         span_beg = revision\n","#         last_rev_was_match = True\n","#     else:\n","#       if last_rev_was_match:\n","#         spans.append((span_beg, last_rev_that_matched))\n","#         last_rev_was_match = False\n","#   if last_rev_was_match:\n","#     spans.append((span_beg, last_rev_that_matched))\n","#   print(\n","#       '\"{}\" was found in the following revisions of {}:\\n{}'.format(keyphrase, 'Test', '\\n'.join(['Span {}: {} - {}'.format(indx + 1, revision[0].timestamp, revision[1].timestamp) for indx,revision in enumerate(spans)])) \n","#       if spans else '\"{}\" was not found in any revision of \"{}\"'.format(keyphrase, 'Test')\n","#       )\n","#   return spans"],"execution_count":56,"outputs":[{"output_type":"stream","text":["CRISPR_en\n"],"name":"stdout"},{"output_type":"error","ename":"KeyboardInterrupt","evalue":"ignored","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)","\u001b[0;32m<ipython-input-56-d90860ba59a9>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     19\u001b[0m   names_found = sorted(set(\n\u001b[1;32m     20\u001b[0m       \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 21\u001b[0;31m       \u001b[0;32mfor\u001b[0m \u001b[0marticle\u001b[0m \u001b[0;32min\u001b[0m \u001b[0marticles\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     22\u001b[0m       \u001b[0;32mfor\u001b[0m \u001b[0mrevision\u001b[0m \u001b[0;32min\u001b[0m \u001b[0marticle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0myield_revisions\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     23\u001b[0m       \u001b[0;32mfor\u001b[0m \u001b[0mname\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mnames\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m<ipython-input-56-d90860ba59a9>\u001b[0m in \u001b[0;36m<genexpr>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m     22\u001b[0m       \u001b[0;32mfor\u001b[0m \u001b[0mrevision\u001b[0m \u001b[0;32min\u001b[0m \u001b[0marticle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0myield_revisions\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     23\u001b[0m       \u001b[0;32mfor\u001b[0m \u001b[0mname\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mnames\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 24\u001b[0;31m       \u001b[0;32mfor\u001b[0m \u001b[0msection\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrevision\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_arno_sections\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     25\u001b[0m       \u001b[0;32mif\u001b[0m \u001b[0msection\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlevel\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m       \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0msection\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mheading\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mexclude_sections\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/content/drive/My Drive/WIKING/WIKING_git/code/article/revision/revision.py\u001b[0m in \u001b[0;36mget_arno_sections\u001b[0;34m(self, headline_range)\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0msections\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    190\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 191\u001b[0;31m     \u001b[0;32mdef\u001b[0m \u001b[0mget_specific_sections\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mselection\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    192\u001b[0m         \"\"\"\n\u001b[1;32m    193\u001b[0m         \u001b[0mGet\u001b[0m \u001b[0mspecific\u001b[0m \u001b[0msections\u001b[0m \u001b[0mbased\u001b[0m \u001b[0mon\u001b[0m \u001b[0ma\u001b[0m \u001b[0mselection\u001b[0m \u001b[0mof\u001b[0m \u001b[0mheadings\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/content/drive/My Drive/WIKING/WIKING_git/code/article/revision/revision.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0msections\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    190\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 191\u001b[0;31m     \u001b[0;32mdef\u001b[0m \u001b[0mget_specific_sections\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mselection\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    192\u001b[0m         \"\"\"\n\u001b[1;32m    193\u001b[0m         \u001b[0mGet\u001b[0m \u001b[0mspecific\u001b[0m \u001b[0msections\u001b[0m \u001b[0mbased\u001b[0m \u001b[0mon\u001b[0m \u001b[0ma\u001b[0m \u001b[0mselection\u001b[0m \u001b[0mof\u001b[0m \u001b[0mheadings\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/content/drive/My Drive/WIKING/WIKING_git/code/article/revision/arno_section.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, revision, heading_html, text_html)\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/lxml/html/__init__.py\u001b[0m in \u001b[0;36mfromstring\u001b[0;34m(html, base_url, parser, **kw)\u001b[0m\n\u001b[1;32m    877\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    878\u001b[0m     \u001b[0;31m# otherwise, lets parse it out...\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 879\u001b[0;31m     \u001b[0mbodies\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfindall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'body'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    880\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mbodies\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    881\u001b[0m         \u001b[0mbodies\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfindall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'{%s}body'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mXHTML_NAMESPACE\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree._Element.findall\u001b[0;34m()\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/lxml/_elementpath.cpython-37m-x86_64-linux-gnu.so\u001b[0m in \u001b[0;36mlxml._elementpath.findall\u001b[0;34m()\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/lxml/_elementpath.cpython-37m-x86_64-linux-gnu.so\u001b[0m in \u001b[0;36mselect\u001b[0;34m()\u001b[0m\n","\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree._Element.iterchildren\u001b[0;34m()\u001b[0m\n","\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree.ElementChildIterator.__cinit__\u001b[0;34m()\u001b[0m\n","\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree._elementFactory\u001b[0;34m()\u001b[0m\n","\u001b[0;32msrc/lxml/classlookup.pxi\u001b[0m in \u001b[0;36mlxml.etree._parser_class_lookup\u001b[0;34m()\u001b[0m\n","\u001b[0;32msrc/lxml/classlookup.pxi\u001b[0m in \u001b[0;36mlxml.etree._custom_class_lookup\u001b[0;34m()\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.7/dist-packages/lxml/html/__init__.py\u001b[0m in \u001b[0;36mlookup\u001b[0;34m(self, node_type, document, namespace, name)\u001b[0m\n\u001b[1;32m    733\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_element_classes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclasses\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    734\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 735\u001b[0;31m     \u001b[0;32mdef\u001b[0m \u001b[0mlookup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnode_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdocument\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    736\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mnode_type\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'element'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    737\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_element_classes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mHtmlElement\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mKeyboardInterrupt\u001b[0m: "]}]},{"cell_type":"markdown","metadata":{"id":"oqOIgx_RBue9"},"source":["# Extract sections"]},{"cell_type":"markdown","metadata":{"id":"QTl82RWF2BYR"},"source":["## Using the functions I already pushed to the git"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":36},"id":"kHhiUdPRknZT","executionInfo":{"status":"ok","timestamp":1613770164722,"user_tz":-60,"elapsed":3328,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"83276eea-6f4f-43ce-90fb-5304ea181e45"},"source":["# revision = CRISPR_en.get_revision(revid=701817377)\n","revision = CRISPR_gene_editing_en.get_revision(revid=986682164)\n","revision.url"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'https://en.wikipedia.org/w/index.php?title=CRISPR gene editing&oldid=986682164'"]},"metadata":{"tags":[]},"execution_count":2}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":105},"id":"cRaiZlum8V2i","executionInfo":{"status":"ok","timestamp":1613770175400,"user_tz":-60,"elapsed":760,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"a4bdd442-d53b-4ebd-c328-974573e6d08f"},"source":["section = revision.get_specific_sections(['History'])[0]\n","section.get_all_text()"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'Predecessors\\n\\nIn the early 2000s, researchers developed zinc finger nucleases (ZFNs), synthetic proteins whose DNA-binding domains enable them to create double-stranded breaks in DNA at specific points. In 2010, synthetic nucleases called transcription activator-like effector nucleases (TALENs) provided an easier way to target a double-stranded break to a specific location on the DNA strand. Both zinc finger nucleases and TALENs require the design and creation of a custom protein for each targeted DNA sequence, which is a much more difficult and time-consuming process than that of designing guide RNAs. CRISPRs are much easier to design because the process requires synthesizing only a short RNA sequence, a procedure that is already widely used for many other molecular biology techniques (e.g. creating oligonucleotide primers).[12]\\nWhereas methods such as RNA interference (RNAi) do not fully suppress gene function, CRISPR, ZFNs, and TALENs provide full irreversible gene knockout.[13] CRISPR can also target several DNA sites simultaneously simply by introducing different gRNAs. In addition, the costs of employing CRISPR are relatively low.[13][14][15]\\n\\n\\nDiscovery\\n\\nIn 2012 Jennifer Doudna and Emmanuelle Charpentier published their finding that CRISPR-Cas9 could be programmed with RNA to edit genomic DNA, now considered one of the most significant discoveries in the history of biology.[16]\\n\\n\\nPatents and commercialization\\n\\nAs of November\\xa02013[update], SAGE Labs (part of Horizon Discovery group) had exclusive rights from one of those companies to produce and sell genetically engineered rats and non-exclusive rights for mouse and rabbit models.[17] By 2015[update], Thermo Fisher Scientific had licensed intellectual property from ToolGen to develop CRISPR reagent kits.[18]\\nAs of December\\xa02014[update], patent rights to CRISPR were contested. Several companies formed to develop related drugs and research tools.[19] As companies ramped up financing, doubts as to whether CRISPR could be quickly monetized were raised.[20] In February 2017 the US Patent Office ruled on a patent interference case brought by University of California with respect to patents issued to the Broad Institute, and found that the Broad patents, with claims covering the application of CRISPR-Cas9 in eukaryotic cells, were distinct from the inventions claimed by University of California.[21][22][23]\\nShortly after, University of California filed an appeal of this ruling.[24][25]\\n\\n\\nRecent events\\n\\nIn March 2017, the European Patent Office (EPO) announced its intention to allow broad claims for editing all kinds of cells to Max-Planck Institute in Berlin, University of California, and University of Vienna,[26][27] and in August 2017, the EPO announced its intention to allow CRISPR claims in a patent application that MilliporeSigma had filed.[26] As of August\\xa02017[update] the patent situation in Europe was complex, with MilliporeSigma, ToolGen, Vilnius University, and Harvard contending for claims, along with University of California and Broad.[28]\\nIn July 2018, the ECJ ruled that gene editing for plants was a sub-category of GMO foods and therefore that the CRISPR technique would henceforth be regulated in the European Union by their rules and regulations for GMOs.[29]\\nIn February 2020, a US trial safely showed CRISPR gene editing on three cancer patients.[30]\\nIn October 2020, researchers Emmanuelle Charpentier and Jennifer Doudna were awarded the Nobel Prize in Chemistry for their work in this field. [31]\\n'"]},"metadata":{"tags":[]},"execution_count":4}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"8rh9iBoYUAi6","executionInfo":{"status":"ok","timestamp":1613770184097,"user_tz":-60,"elapsed":764,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"db316a37-4cd2-4e0b-bdab-7f73d2315d88"},"source":["revision.get_section_headings()"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["['Contents',\n"," 'Synopsis',\n"," 'History',\n"," 'History/Predecessors',\n"," 'History/Discovery',\n"," 'History/Patents and commercialization',\n"," 'History/Recent events',\n"," 'Genome engineering',\n"," 'Genome engineering/Major components',\n"," 'Genome engineering/Structure',\n"," 'Genome engineering/Delivery',\n"," 'Genome engineering/Controlled genome editing',\n"," 'CRISPR screening',\n"," 'Applications',\n"," 'Applications/Disease models',\n"," 'Applications/Biomedicine',\n"," 'Applications/Biomedicine/CRISPR in the treatment of infection',\n"," 'Applications/Biomedicine/CRISPR and cancer',\n"," 'Applications/Knockdown/activation',\n"," 'Applications/RNA editing',\n"," 'Applications/Gene drive',\n"," 'Applications/In vitro genetic depletion',\n"," 'Applications/Prime editing',\n"," 'Society and culture',\n"," 'Society and culture/Human germline modification',\n"," 'Society and culture/Policy barriers to genetic engineering',\n"," 'Society and culture/Recognition',\n"," 'See also',\n"," 'References']"]},"metadata":{"tags":[]},"execution_count":5}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":105},"id":"kX7w99x4nquQ","executionInfo":{"status":"ok","timestamp":1613692375458,"user_tz":-60,"elapsed":521,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"5cef1f0d-bb4d-4e12-c70c-389a4b4cda7c"},"source":["[section for section in revision.get_arno_sections() if any(section.heading == heading for heading in ['History'])][0].get_all_text()"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'Repeated sequences\\n\\nThe discovery of clustered DNA repeats occurred independently in three parts of the world. The first description of what would later be called CRISPR is from Osaka University researcher Yoshizumi Ishino and his colleagues in 1987. They accidentally cloned part of a CRISPR sequence together with the \"iap\" gene (isozyme conversion of alkaline phosphatase)[14] that was their target. The organization of the repeats was unusual. Repeated sequences are typically arranged consecutively, without interspersed different sequences.[14][11] They did not know the function of the interrupted clustered repeats.\\nIn 1993, researchers of Mycobacterium tuberculosis in the Netherlands published two articles about a cluster of interrupted direct repeats (DR) in that bacterium. They recognized the diversity of the sequences that intervened the direct repeats among different strains of M. tuberculosis[15] and used this property to design a typing method that was named spoligotyping, which is still in use today.[16][17]\\nFrancisco Mojica at the University of Alicante in Spain studied repeats observed in the archaeal organisms of Haloferax and Haloarcula species, and their function. Mojica\\'s supervisor surmised at the time that the clustered repeats had a role in correctly segregating replicated DNA into daughter cells during cell division because plasmids and chromosomes with identical repeat arrays could not coexist in Haloferax volcanii. Transcription of the interrupted repeats was also noted for the first time; this was the first full characterization of CRISPR.[17][18] By 2000, Mojica performed a survey of scientific literature and one of his students performed a search in published genomes with a program devised by himself. They identified interrupted repeats in 20 species of microbes as belonging to the same family.[19] In 2001, Mojica and Ruud Jansen, who were searching for additional interrupted repeats, proposed the acronym CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) to alleviate the confusion stemming from the numerous acronyms used to describe the sequences in the scientific literature.[18][20] In 2002, Tang, et al. showed evidence that CRISPR repeat regions from the genome of Archaeoglobus fulgidus were transcribed into long RNA molecules that were subsequently processed into unit-length small RNAs, plus some longer forms of 2, 3, or more spacer-repeat units.[21][22]\\n\\n\\nCRISPR-associated systems\\n\\nA major addition to the understanding of CRISPR came with Jansen\\'s observation that the prokaryote repeat cluster was accompanied by a set of homologous genes that make up CRISPR-associated systems or cas genes. Four cas genes (cas 1–4) were initially recognized. The Cas proteins showed helicase and nuclease motifs, suggesting a role in the dynamic structure of the CRISPR loci.[23] In this publication the acronym CRISPR was used as the universal name of this pattern. However, the CRISPR function remained enigmatic.\\n\\n  Simplified diagram of a CRISPR locus. The three major components of a CRISPR locus are shown: cas genes, a leader sequence, and a repeat-spacer array. Repeats are shown as gray boxes and spacers are colored bars. The arrangement of the three components is not always as shown.[24][25] In addition, several CRISPRs with similar sequences can be present in a single genome, only one of which is associated with cas genes.[26]\\nIn 2005, three independent research groups showed that some CRISPR spacers are derived from phage DNA and extrachromosomal DNA such as plasmids.[27][28][29] In effect, the spacers are fragments of DNA gathered from viruses that previously tried to attack the cell. The source of the spacers was a sign that the CRISPR/cas system could have a role in adaptive immunity in bacteria.[24][30] All three studies proposing this idea were initially rejected by high-profile journals, but eventually appeared in other journals.[31]\\nThe first publication[28] proposing a role of CRISPR-Cas in microbial immunity, by Mojica and collaborators at the University of Alicante, predicted a role for the RNA transcript of spacers on target recognition in a mechanism that could be analogous to the RNA interference system used by eukaryotic cells. Koonin and colleagues extended this RNA interference hypothesis by proposing mechanisms of action for the different CRISPR-Cas subtypes according to the predicted function of their proteins.[32]\\nExperimental work by several groups revealed the basic mechanisms of CRISPR-Cas immunity. In 2007, the first experimental evidence that CRISPR was an adaptive immune system was published.[11][5] A CRISPR region in Streptococcus thermophilus acquired spacers from the DNA of an infecting bacteriophage. The researchers manipulated the resistance of S. thermophilus to different types of phage by adding and deleting spacers whose sequence matched those found in the tested phages.[33][34] In 2008, Brouns and Van der Oost identified a complex of Cas proteins (called Cascade) that in E. coli cut the CRISPR RNA precursor within the repeats into mature spacer-containing RNA molecules called CRISPR RNA (crRNA), which remained bound to the protein complex.[35] Moreover, it was found that Cascade, crRNA and a helicase/nuclease (Cas3) were required to provide a bacterial host with immunity against infection by a DNA virus. By designing an anti-virus CRISPR, they demonstrated that two orientations of the crRNA (sense/antisense) provided immunity, indicating that the crRNA guides were targeting dsDNA. That year Marraffini and Sontheimer confirmed that a CRISPR sequence of S. epidermidis targeted DNA and not RNA to prevent conjugation. This finding was at odds with the proposed RNA-interference-like mechanism of CRISPR-Cas immunity, although a CRISPR-Cas system that targets foreign RNA was later found in Pyrococcus furiosus.[11][33] A 2010 study showed that CRISPR-Cas cuts both strands of phage and plasmid DNA in S. thermophilus.[36]\\n\\n\\nCas9\\n\\nMain article: Cas9\\nResearchers studied a simpler CRISPR system from Streptococcus pyogenes that relies on the protein Cas9. The Cas9 endonuclease is a four-component system that includes two small crRNA molecules and trans-activating CRISPR RNA (tracrRNA).[37][38] Jennifer Doudna and Emmanuelle Charpentier re-engineered the Cas9 endonuclease into a more manageable two-component system by fusing the two RNA molecules into a \"single-guide RNA\" that, when combined with Cas9, could find and cut the DNA target specified by the guide RNA. This contribution was so significant that it was recognized by the Nobel Prize in Chemistry in 2020. By manipulating the nucleotide sequence of the guide RNA, the artificial Cas9 system could be programmed to target any DNA sequence for cleavage.[39] Another group of collaborators comprising Virginijus Šikšnys together with Gasiūnas, Barrangou and Horvath showed that Cas9 from the S. thermophilus CRISPR system can also be reprogrammed to target a site of their choosing by changing the sequence of its crRNA. These advances fueled efforts to edit genomes with the modified CRISPR-Cas9 system.[17]\\nGroups led by Feng Zhang and George Church simultaneously published descriptions of genome editing in human cell cultures using CRISPR-Cas9 for the first time.[11][40][41] It has since been used in a wide range of organisms, including baker\\'s yeast (Saccharomyces cerevisiae),[42][43][44] the opportunistic pathogen Candida albicans,[45][46] zebrafish (Danio rerio),[47] fruit flies (Drosophila melanogaster),[48][49] ants (Harpegnathos saltator[50] and Ooceraea biroi[51]), mosquitoes (Aedes aegypti[52]), nematodes (Caenorhabditis elegans),[53] plants,[54] mice,[55][56] monkeys[57] and human embryos.[58]\\nCRISPR has been modified to make programmable transcription factors that allow scientists to target and activate or silence specific genes.[59]\\nThe CRISPR-Cas9 system has shown to make effective gene edits in Human tripronuclear zygotes first described in a 2015 paper by Chinese scientists P. Liang and Y. Xu. The system made a successful cleavage of mutant Beta-Hemoglobin (HBB) in 28 out of 54 embryos. 4 out of the 28 embryos were successfully recombined using a donor template given by the scientists. The scientists showed that during DNA recombination of the cleaved strand, the homologous endogenous sequence HBD competes with the exogenous donor template. DNA repair in human embryos is much more complicated and particular than in derived stem cells.[60]\\n\\n\\nCas12a (formerly Cpf1)\\n\\nIn 2015, the nuclease Cas12a (formerly known as Cpf1[61]) was characterized in the CRISPR/Cpf1 system of the bacterium Francisella novicida.[62][63] Its original name, from a TIGRFAMs protein family definition built in 2012, reflects the prevalence of its CRISPR-Cas subtype in the Prevotella and Francisella lineages. Cas12a showed several key differences from Cas9 including: causing a \\'staggered\\' cut in double stranded DNA as opposed to the \\'blunt\\' cut produced by Cas9, relying on a \\'T rich\\' PAM (providing alternative targeting sites to Cas9) and requiring only a CRISPR RNA (crRNA) for successful targeting. By contrast Cas9 requires both crRNA and a transactivating crRNA (tracrRNA).\\nThese differences may give Cas12a some advantages over Cas9. For example, Cas12a\\'s small crRNAs are ideal for multiplexed genome editing, as more of them can be packaged in one vector than can Cas9\\'s sgRNAs. As well, the sticky 5′ overhangs left by Cas12a can be used for DNA assembly that is much more target-specific than traditional Restriction Enzyme cloning.[64] Finally, Cas12a cleaves DNA 18–23 base pairs downstream from the PAM site. This means there is no disruption to the recognition sequence after repair, and so Cas12a enables multiple rounds of DNA cleavage. By contrast, since Cas9 cuts only 3 base pairs upstream of the PAM site, the NHEJ pathway results in indel mutations that destroy the recognition sequence, thereby preventing further rounds of cutting. In theory, repeated rounds of DNA cleavage should cause an increased opportunity for the desired genomic editing to occur.[65]\\n\\n\\nCas13 (formerly C2c2)\\n\\nIn 2016, the nuclease Cas13a (formerly known as C2c2) from the bacterium Leptotrichia shahii was characterized. Cas13 is an RNA-guided RNA endonuclease, which means that it does not cleave DNA, but only single-stranded RNA. Cas13 is guided by its crRNA to a ssRNA target and binds and cleaves the target. A distinctive feature of Cas13, as compared to Cas9, is that after cutting its target, Cas13 remains bound to the target and then cleaves other ssRNA molecules non-discriminately. [66] This property is called \"collateral cleavage\" and has been exploited for the development of various diagnostic technologies. [67][68][69]\\n'"]},"metadata":{"tags":[]},"execution_count":3}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":36},"id":"IPe8KBEszJ2B","executionInfo":{"status":"ok","timestamp":1613612926612,"user_tz":-60,"elapsed":729,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"559bfac8-dccb-48d6-cc62-53905740907d"},"source":["revision.get_arno_sections()[1].heading"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'History'"]},"metadata":{"tags":[]},"execution_count":5}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"DmsUbe3WyvDI","executionInfo":{"status":"ok","timestamp":1613612939612,"user_tz":-60,"elapsed":512,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"f5e13ad8-b1cf-48bf-bf61-18077f2e8acb"},"source":["for ref in revision.get_arno_sections()[1].get_all_references():\n","  print(ref.get_text())"],"execution_count":null,"outputs":[{"output_type":"stream","text":["\"Cpf1 Nuclease\". abmgood.com. Retrieved 2017-12-14.\n","Abudayyeh OO, Gootenberg JS, Konermann S, Joung J, Slaymaker IM, Cox DB,  et al. (August 2016). \"C2c2 is a single-component programmable RNA-guided RNA-targeting CRISPR effector\". Science. 353 (6299): aaf5573. doi:10.1126/science.aaf5573. PMC 5127784. PMID 27256883.\n","Baltimore D, Berg P, Botchan M, Carroll D, Charo RA, Church G, Corn JE, Daley GQ, Doudna JA, Fenner M, Greely HT, Jinek M, Martin GS, Penhoet E, Puck J, Sternberg SH, Weissman JS, Yamamoto KR (April 2015). \"Biotechnology. A prudent path forward for genomic engineering and germline gene modification\". Science. 348 (6230): 36–38. Bibcode:2015Sci...348...36B. doi:10.1126/science.aab1028. PMC 4394183. PMID 25791083.\n","Barrangou R (November 2015). \"Diversity of CRISPR-Cas immune systems and molecular machines\". Genome Biology. 16: 247. doi:10.1186/s13059-015-0816-9. PMC 4638107. PMID 26549499.\n","Barrangou R, Fremaux C, Deveau H, Richards M, Boyaval P, Moineau S,  et al. (March 2007). \"CRISPR provides acquired resistance against viruses in prokaryotes\". Science. 315 (5819): 1709–1712. Bibcode:2007Sci...315.1709B. doi:10.1126/science.1138140. hdl:20.500.11794/38902. PMID 17379808. S2CID 3888761.\n","Barrangou R, van der Oost J (2013). CRISPR-Cas Systems : RNA-mediated Adaptive Immunity in Bacteria and Archaea. Heidelberg: Springer. p. 6. ISBN 978-3-642-34656-9.\n","Bassett AR, Tibbit C, Ponting CP, Liu JL (July 2013). \"Highly efficient targeted mutagenesis of Drosophila with the CRISPR/Cas9 system\". Cell Reports. 4 (1): 220–228. doi:10.1016/j.celrep.2013.06.020. PMC 3714591. PMID 23827738.\n","Bolotin A, Quinquis B, Sorokin A, Ehrlich SD (August 2005). \"Clustered regularly interspaced short palindrome repeats (CRISPRs) have spacers of extrachromosomal origin\". Microbiology. 151 (Pt 8): 2551–2561. doi:10.1099/mic.0.28048-0. PMID 16079334.\n","Brouns SJ, Jore MM, Lundgren M, Westra ER, Slijkhuis RJ, Snijders AP, Dickman MJ, Makarova KS, Koonin EV, van der Oost J (August 2008). \"Small CRISPR RNAs guide antiviral defense in prokaryotes\". Science. 321 (5891): 960–964. Bibcode:2008Sci...321..960B. doi:10.1126/science.1159689. PMC 5898235. PMID 18703739.\n","Charpentier E, Richter H, van der Oost J, White MF (May 2015). \"Biogenesis pathways of RNA guides in archaeal and bacterial CRISPR-Cas adaptive immunity\". FEMS Microbiology Reviews. 39 (3): 428–441. doi:10.1093/femsre/fuv023. PMC 5965381. PMID 25994611.\n","Cong L, Ran FA, Cox D, Lin S, Barretto R, Habib N, Hsu PD, Wu X, Jiang W, Marraffini LA, Zhang F (February 2013). \"Multiplex genome engineering using CRISPR/Cas systems\". Science. 339 (6121): 819–823. Bibcode:2013Sci...339..819C. doi:10.1126/science.1231143. PMC 3795411. PMID 23287718.\n","Deltcheva E, Chylinski K, Sharma CM, Gonzales K, Chao Y, Pirzada ZA, Eckert MR, Vogel J, Charpentier E (March 2011). \"CRISPR RNA maturation by trans-encoded small RNA and host factor RNase III\". Nature. 471 (7340): 602–607. Bibcode:2011Natur.471..602D. doi:10.1038/nature09886. PMC 3070239. PMID 21455174.\n","DiCarlo JE, Norville JE, Mali P, Rios X, Aach J, Church GM (April 2013). \"Genome engineering in Saccharomyces cerevisiae using CRISPR-Cas systems\". Nucleic Acids Research. 41 (7): 4336–4343. doi:10.1093/nar/gkt135. PMC 3627607. PMID 23460208.\n","Fonfara I, Richter H, Bratovič M, Le Rhun A, Charpentier E (April 2016). \"The CRISPR-associated DNA-cleaving enzyme Cpf1 also processes precursor CRISPR RNA\". Nature. 532 (7600): 517–521. Bibcode:2016Natur.532..517F. doi:10.1038/nature17945. PMID 27096362. S2CID 2271552.\n","Friedland AE, Tzur YB, Esvelt KM, Colaiácovo MP, Church GM, Calarco JA (August 2013). \"Heritable genome editing in C. elegans via a CRISPR-Cas9 system\". Nature Methods. 10 (8): 741–743. doi:10.1038/nmeth.2532. PMC 3822328. PMID 23817069.\n","Garneau JE, Dupuis MÈ, Villion M, Romero DA, Barrangou R, Boyaval P,  et al. (November 2010). \"The CRISPR/Cas bacterial immune system cleaves bacteriophage and plasmid DNA\". Nature. 468 (7320): 67–71. Bibcode:2010Natur.468...67G. CiteSeerX 10.1.1.451.9645. doi:10.1038/nature09523. PMID 21048762. S2CID 205222849.\n","Gootenberg JS, Abudayyeh OO, Kellner MJ, Joung J, Collins JJ, Zhang F (April 2018). \"Multiplexed and portable nucleic acid detection platform with Cas13, Cas12a, and Csm6\". Science. 360 (6387): 439–444. Bibcode:2018Sci...360..439G. doi:10.1126/science.aaq0179. PMC 5961727. PMID 29449508.\n","Gootenberg JS, Abudayyeh OO, Lee JW, Essletzbichler P, Dy AJ, Joung J,  et al. (April 2017). \"Nucleic acid detection with CRISPR-Cas13a/C2c2\". Science. 356 (6336): 438–442. Bibcode:2017Sci...356..438G. doi:10.1126/science.aam9321. PMC 5526198. PMID 28408723.\n","Gratz SJ, Cummings AM, Nguyen JN, Hamm DC, Donohue LK, Harrison MM, Wildonger J, O'Connor-Giles KM (August 2013). \"Genome engineering of Drosophila with the CRISPR RNA-guided Cas9 nuclease\". Genetics. 194 (4): 1029–1035. doi:10.1534/genetics.113.152710. PMC 3730909. PMID 23709638.\n","Grissa I, Vergnaud G, Pourcel C (May 2007). \"The CRISPRdb database and tools to display CRISPRs and to generate dictionaries of spacers and repeats\". BMC Bioinformatics. 8: 172. doi:10.1186/1471-2105-8-172. PMC 1892036. PMID 17521438.\n","Groenen PM, Bunschoten AE, van Soolingen D, van Embden JD (December 1993). \"Nature of DNA polymorphism in the direct repeat cluster of Mycobacterium tuberculosis; application for strain differentiation by a novel typing method\". Molecular Microbiology. 10 (5): 1057–1065. doi:10.1111/j.1365-2958.1993.tb00976.x. PMID 7934856. S2CID 25304723.\n","Guo X, Li XJ (July 2015). \"Targeted genome editing in primate embryos\". Cell Research. 25 (7): 767–768. doi:10.1038/cr.2015.64. PMC 4493275. PMID 26032266.\n","Horvath P, Barrangou R (January 2010). \"CRISPR/Cas, the immune system of bacteria and archaea\". Science. 327 (5962): 167–170. Bibcode:2010Sci...327..167H. doi:10.1126/Science.1179555. PMID 20056882. S2CID 17960960.\n","Hsu PD, Lander ES, Zhang F (June 2014). \"Development and applications of CRISPR-Cas9 for genome engineering\". Cell. 157 (6): 1262–1278. doi:10.1016/j.cell.2014.05.010. PMC 4343198. PMID 24906146.\n","Hwang WY, Fu Y, Reyon D, Maeder ML, Tsai SQ, Sander JD, Peterson RT, Yeh JR, Joung JK (March 2013). \"Efficient genome editing in zebrafish using a CRISPR-Cas system\". Nature Biotechnology. 31 (3): 227–229. doi:10.1038/nbt.2501. PMC 3686313. PMID 23360964.\n","Ishino Y, Shinagawa H, Makino K, Amemura M, Nakata A (December 1987). \"Nucleotide sequence of the iap gene, responsible for alkaline phosphatase isozyme conversion in Escherichia coli, and identification of the gene product\". Journal of Bacteriology. 169 (12): 5429–5433. doi:10.1128/jb.169.12.5429-5433.1987. PMC 213968. PMID 3316184.\n","Iwasaki RS, Batey RT (2020). \"SPRINT: a Cas13a-based platform for detection of small molecules\". Nucleic Acids Research. 48 (17): e101. doi:10.1093/nar/gkaa673. PMC 7515716. PMID 32797156.\n","Jansen R, Embden JD, Gaastra W, Schouls LM (March 2002). \"Identification of genes that are associated with DNA repeats in prokaryotes\". Molecular Microbiology. 43 (6): 1565–1575. doi:10.1046/j.1365-2958.2002.02839.x. PMID 11952905. S2CID 23196085.\n","Jiang W, Zhou H, Bi H, Fromm M, Yang B, Weeks DP (November 2013). \"Demonstration of CRISPR/Cas9/sgRNA-mediated targeted gene modification in Arabidopsis, tobacco, sorghum and rice\". Nucleic Acids Research. 41 (20): e188. doi:10.1093/nar/gkt780. PMC 3814374. PMID 23999092.\n","Jinek M, Chylinski K, Fonfara I, Hauer M, Doudna JA, Charpentier E (August 2012). \"A programmable dual-RNA-guided DNA endonuclease in adaptive bacterial immunity\". Science. 337 (6096): 816–821. Bibcode:2012Sci...337..816J. doi:10.1126/science.1225829. PMC 6286148. PMID 22745249.\n","Kim H, Kim ST, Ryu J, Kang BC, Kim JS, and Kim SG (February 2017). \"CRISPR/Cpf1-mediated DNA-free plant genome editing\". Nature Communications. 8 (14406): 14406. Bibcode:2017NatCo...814406K. doi:10.1038/ncomms14406. PMC 5316869. PMID 28205546.\n","Kistler KE, Vosshall LB, Matthews BJ (April 2015). \"Genome engineering with CRISPR-Cas9 in the mosquito Aedes aegypti\". Cell Reports. 11 (1): 51–60. doi:10.1016/j.celrep.2015.03.009. PMC 4394034. PMID 25818303.\n","Lander ES (January 2016). \"The Heroes of CRISPR\". Cell. 164 (1–2): 18–28. doi:10.1016/j.cell.2015.12.041. PMID 26771483.\n","Larson MH, Gilbert LA, Wang X, Lim WA, Weissman JS, Qi LS (November 2013). \"CRISPR interference (CRISPRi) for sequence-specific control of gene expression\". Nature Protocols. 8 (11): 2180–2196. doi:10.1038/nprot.2013.132. PMC 3922765. PMID 24136345.\n","Liang P, Xu Y, Zhang X, Ding C, Huang R, Zhang Z,  et al. (May 2015). \"CRISPR/Cas9-mediated gene editing in human tripronuclear zygotes\". Protein & Cell. 6 (5): 363–372. doi:10.1007/s13238-015-0153-5. PMC 4417674. PMID 25894090.\n","Liu JJ, Kong II, Zhang GC, Jayakody LN, Kim H, Xia PF, Kwak S, Sung BH, Sohn JH, Walukiewicz HE, Rao CV, Jin YS (April 2016). \"Metabolic Engineering of Probiotic Saccharomyces boulardii\". Applied and Environmental Microbiology. 82 (8): 2280–2287. doi:10.1128/AEM.00057-16. PMC 4959471. PMID 26850302.\n","Makarova KS, Grishin NV, Shabalina SA, Wolf YI, Koonin EV (March 2006). \"A putative RNA-interference-based immune system in prokaryotes: computational analysis of the predicted enzymatic machinery, functional analogies with eukaryotic RNAi, and hypothetical mechanisms of action\". Biology Direct. 1: 7. doi:10.1186/1745-6150-1-7. PMC 1462988. PMID 16545108.\n","Mali P, Yang L, Esvelt KM, Aach J, Guell M, DiCarlo JE, Norville JE, Church GM (February 2013). \"RNA-guided human genome engineering via Cas9\". Science. 339 (6121): 823–826. Bibcode:2013Sci...339..823M. doi:10.1126/science.1232033. PMC 3712628. PMID 23287722.\n","Marraffini LA (October 2015). \"CRISPR-Cas immunity in prokaryotes\". Nature. 526 (7571): 55–61. Bibcode:2015Natur.526...55M. doi:10.1038/nature15386. PMID 26432244. S2CID 3718361.\n","Marraffini LA, Sontheimer EJ (March 2010). \"CRISPR interference: RNA-directed adaptive immunity in bacteria and archaea\". Nature Reviews Genetics. 11 (3): 181–190. doi:10.1038/nrg2749. PMC 2928866. PMID 20125085.\n","Mojica FJ, Díez-Villaseñor C, García-Martínez J, Soria E (February 2005). \"Intervening sequences of regularly spaced prokaryotic repeats derive from foreign genetic elements\". Journal of Molecular Evolution. 60 (2): 174–182. Bibcode:2005JMolE..60..174M. doi:10.1007/s00239-004-0046-3. PMID 15791728. S2CID 27481111.\n","Mojica FJ, Díez-Villaseñor C, Soria E, Juez G (April 2000). \"Biological significance of a family of regularly spaced repeats in the genomes of Archaea, Bacteria and mitochondria\". Molecular Microbiology. 36 (1): 244–246. doi:10.1046/j.1365-2958.2000.01838.x. PMID 10760181.\n","Mojica FJ, Montoliu L (2016). \"On the Origin of CRISPR-Cas Technology: From Prokaryotes to Mammals\". Trends in Microbiology. 24 (10): 811–820. doi:10.1016/j.tim.2016.06.005. PMID 27401123.\n","Mojica FJ, Rodriguez-Valera F (2016). \"The discovery of CRISPR in archaea and bacteria\" (PDF). The FEBS Journal. 283 (17): 3162–3169. doi:10.1111/febs.13766. hdl:10045/57676. PMID 27234458. S2CID 42827598.\n","Morange M (June 2015). \"What history tells us XXXVII. CRISPR-Cas: The discovery of an immune system in prokaryotes\". Journal of Biosciences. 40 (2): 221–223. doi:10.1007/s12038-015-9532-6. PMID 25963251.\n","Ng H, Dean N (2017). \"Candida albicans by Increased Single Guide RNA Expression\". mSphere. 2 (2): e00385–16. doi:10.1128/mSphere.00385-16. PMC 5397569. PMID 28435892.\n","Pennisi E (August 2013). \"The CRISPR craze\". News Focus. Science. 341 (6148): 833–836. Bibcode:2013Sci...341..833P. doi:10.1126/science.341.6148.833. PMID 23970676.\n","Pourcel C, Salvignol G, Vergnaud G (March 2005). \"CRISPR elements in Yersinia pestis acquire new repeats by preferential uptake of bacteriophage DNA, and provide additional tools for evolutionary studies\". Microbiology. 151 (Pt 3): 653–663. doi:10.1099/mic.0.27437-0. PMID 15758212.\n","Soni D, Wang DM, Regmi SC, Mittal M, Vogel SM, Schlüter D, Tiruppathi C (May 2018). \"Deubiquitinase function of A20 maintains and repairs endothelial barrier after lung vascular injury\". Cell Death Discovery. 4 (60): 60. doi:10.1038/s41420-018-0056-3. PMC 5955943. PMID 29796309.\n","Tang TH, Bachellerie JP, Rozhdestvensky T, Bortolin ML, Huber H, Drungowski M;  et al. (2002). \"Identification of 86 candidates for small non-messenger RNAs from the archaeon Archaeoglobus fulgidus\". Proc Natl Acad Sci U S A. 99 (11): 7536–41. Bibcode:2002PNAS...99.7536T. doi:10.1073/pnas.112047299. PMC 124276. PMID 12032318.\n","Trible W, Olivos-Cisneros L, McKenzie SK, Saragosti J, Chang NC, Matthews BJ, Oxley PR, Kronauer DJ (August 2017). \"orco Mutagenesis Causes Loss of Antennal Lobe Glomeruli and Impaired Social Behavior in Ants\". Cell. 170 (4): 727–735.e10. doi:10.1016/j.cell.2017.07.001. PMC 5556950. PMID 28802042.\n","Vyas VK, Barrasa MI, Fink GR (2015). \"Candida albicans CRISPR system permits genetic engineering of essential genes and gene families\". Science Advances. 1 (3): e1500248. Bibcode:2015SciA....1E0248V. doi:10.1126/sciadv.1500248. PMC 4428347. PMID 25977940.\n","Wang H, Yang H, Shivalila CS, Dawlaty MM, Cheng AW, Zhang F, Jaenisch R (May 2013). \"One-step generation of mice carrying mutations in multiple genes by CRISPR/Cas-mediated genome engineering\". Cell. 153 (4): 910–918. doi:10.1016/j.cell.2013.04.025. PMC 3969854. PMID 23643243.\n","Yan H, Opachaloemphan C, Mancini G, Yang H, Gallitto M, Mlejnek J, Leibholz A, Haight K, Ghaninia M, Huo L, Perry M, Slone J, Zhou X, Traficante M, Penick CA, Dolezal K, Gokhale K, Stevens K, Fetter-Pruneda I, Bonasio R, Zwiebel LJ, Berger SL, Liebig J, Reinberg D, Desplan C (August 2017). \"An Engineered orco Mutation Produces Aberrant Social Behavior and Defective Neural Development in Ants\". Cell. 170 (4): 736–747.e9. doi:10.1016/j.cell.2017.06.051. PMC 5587193. PMID 28802043.\n","Yan MY, Yan HQ, Ren GX, Zhao JP, Guo XP, Sun YC (September 2017). \"CRISPR-Cas12a-Assisted Recombineering in Bacteria\". Applied and Environmental Microbiology. 83 (17). doi:10.1128/AEM.00947-17. PMC 5561284. PMID 28646112.\n","Zetsche B, Gootenberg JS, Abudayyeh OO, Slaymaker IM, Makarova KS, Essletzbichler P, Volz SE, Joung J, van der Oost J, Regev A, Koonin EV, Zhang F (October 2015). \"Cpf1 is a single RNA-guided endonuclease of a class 2 CRISPR-Cas system\". Cell. 163 (3): 759–771. doi:10.1016/j.cell.2015.09.038. PMC 4638220. PMID 26422227.\n","Zhang GC, Kong II, Kim H, Liu JJ, Cate JH, Jin YS (December 2014). \"Construction of a quadruple auxotrophic mutant of an industrial polyploid saccharomyces cerevisiae strain by using RNA-guided Cas9 nuclease\". Applied and Environmental Microbiology. 80 (24): 7694–7701. doi:10.1128/AEM.02310-14. PMC 4249234. PMID 25281382.\n","van Soolingen D, de Haas PE, Hermans PW, Groenen PM, van Embden JD (August 1993). \"Comparison of various repetitive DNA elements as genetic markers for strain differentiation and epidemiology of Mycobacterium tuberculosis\". Journal of Clinical Microbiology. 31 (8): 1987–1995. doi:10.1128/JCM.31.8.1987-1995.1993. PMC 265684. PMID 7690367.\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"iUGFPUFokrQW","executionInfo":{"status":"ok","timestamp":1613610497149,"user_tz":-60,"elapsed":13444,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"6fb7f675-d35c-48cc-a3aa-e54fa16f3b02"},"source":["for section in revision.get_arno_sections():\n","  print(section.heading)\n","  if section.children:\n","    print('\\t', [c.heading for c in section.children])"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Contents\n","History\n","\t ['Repeated sequences', 'CRISPR-associated systems', 'Cas9', 'Cas12a (formerly Cpf1)', 'Cas13 (formerly C2c2)']\n","Repeated sequences\n","CRISPR-associated systems\n","Cas9\n","Cas12a (formerly Cpf1)\n","Cas13 (formerly C2c2)\n","Locus structure\n","\t ['Repeats and spacers', 'CRISPR RNA structures', 'Cas genes and CRISPR subtypes']\n","Repeats and spacers\n","CRISPR RNA structures\n","Cas genes and CRISPR subtypes\n","Mechanism\n","\t ['Spacer acquisition', 'Biogenesis', 'Interference']\n","Spacer acquisition\n","\t ['Protospacer adjacent motifs', 'Insertion variants']\n","Protospacer adjacent motifs\n","Insertion variants\n","Biogenesis\n","Interference\n","Evolution\n","\t ['Coevolution', 'Rates']\n","Coevolution\n","Rates\n","Identification\n","Use by phages\n","Applications\n","\t ['CRISPR gene editing', 'CRISPR as diagnostic tool']\n","CRISPR gene editing\n","CRISPR as diagnostic tool\n","See also\n","Notes\n","References\n","Further reading\n","External links\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"n-kefUUG2Lj3"},"source":["## Building the functions (prior to pushing them to the git)"]},{"cell_type":"code","metadata":{"id":"LpunGQ4F7m18"},"source":["class Section:\n","  \n","  def __init__(self, heading_html, text_html):\n","    self.heading_html = heading_html\n","    self.heading = ''.join(lxml.html.fromstring(heading_html).itertext()) if heading_html else ''\n","    self.text_html = text_html\n","    self.text = ''.join(lxml.html.fromstring(text_html).itertext()) if text_html else ''\n","    self.level = int(heading_html.strip()[2:3]) # so far, ony one-digit levels\n","\n","    self.children = []\n","    self.parent = None\n","    self.next = None\n","    self.previous = None\n","  \n","  def get_html(self):\n","    return self.heading_html + self.text_html\n","  \n","  def get_text(self):\n","    return self.text\n","\n","  def get_all_text(self):\n","    return self.text + '\\n\\n'.join(['\\n\\n'.join([child.heading, child.text]) for child in self.children])\n"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"LIZXCcsJvLno"},"source":["headline_range = range(1,7)\n","starts = [m.start() for m in re.finditer(r'|'.join([r'<h{0}.*?h{0}>'.format(i) for i in headline_range]), revision.html)]\n","ends = [m.end() for m in re.finditer(r'|'.join([r'<h{0}.*?h{0}>'.format(i) for i in headline_range]), revision.html)]\n","\n","headings_html = [revision.html[start:end] for start, end in zip(starts,ends)]\n","texts_html = [revision.html[end:start] for end, start in zip(ends, starts[1:])] + [revision.html[ends[-1]:]]\n","\n","sections = [Section(heading_html.strip(), text_html.strip()) for heading_html, text_html in zip(headings_html, texts_html)]\n","\n","last_parents = [None for i in range(0,10)]\n","last_section = None\n","for section in sections:\n","  if last_section:\n","    last_section.next = section\n","    if section.level > last_section.level:\n","      last_section.children.append(section)\n","      section.parent = last_section\n","      last_parents[section.level] = section\n","    elif section.level == last_section.level:\n","      if last_section.parent:\n","        section.parent = last_section.parent\n","        last_section.parent.children.append(section)\n","      last_parents[section.level] = section\n","    elif section.level < last_section.level:\n","      if last_parents[section.level]:\n","        if last_parents[section.level].parent:\n","          section.parent = last_parents[section.level].parent \n","          last_parents[section.level].parent .children.append(section)\n","    section.previous = last_section\n","  last_section = section\n","  last_parents[section.level] = section"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Te4e90DVgPJ9","executionInfo":{"status":"ok","timestamp":1613607777902,"user_tz":-60,"elapsed":562,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"65f0ffcf-283e-4093-94f5-180ee587fe1a"},"source":["pp.pprint(sections[17].get_all_text())"],"execution_count":null,"outputs":[{"output_type":"stream","text":["('CRISPR associated proteincrystal structure of a crispr-associated protein '\n"," 'from Thermus thermophilusIdentifiersSymbolCRISPR_assocPfamPF08798Pfam '\n"," 'clanCL0362InterProIPR010179CDDcd09727Available protein structures:Pfam\\n'\n"," ' \\xa0structures / ECOD\\n'\n"," ' \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\n'\n"," 'CRISPR associated protein Cas2crystal structure of a hypothetical protein '\n"," 'tt1823 from Thermus '\n"," 'thermophilusIdentifiersSymbolCRISPR_Cas2PfamPF09827InterProIPR019199CDDcd09638Available '\n"," 'protein structures:Pfam\\n'\n"," ' \\xa0structures / ECOD\\n'\n"," ' \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\n'\n"," 'CRISPR-associated protein '\n"," 'Cse1IdentifiersSymbolCRISPR_Cse1PfamPF09481InterProIPR013381CDDcd09729Available '\n"," 'protein structures:Pfam\\n'\n"," ' \\xa0structures / ECOD\\n'\n"," ' \\xa0PDBRCSB PDB; PDBe; PDBjPDBsumstructure summary\\n'\n"," 'The cas genes in the adaptor and effector modules of the CRISPR-Cas system '\n"," 'are believed to have evolved from two different ancestral modules. A '\n"," 'transposon-like element called casposon encoding the Cas1-like integrase and '\n"," 'potentially other components of the adaptation module was inserted next to '\n"," 'the ancestral effector module, which likely functioned as an independent '\n"," 'innate immune system.[141] The highly conserved cas1 and cas2 genes of the '\n"," 'adaptor module evolved from the ancestral module while a variety of class 1 '\n"," 'effector was genes evolved from the ancestral effector module.[142] The '\n"," 'evolution of these various class 1 effector module cas genes was guided by '\n"," 'various mechanisms, such as duplication events.[143] On the other hand, each '\n"," 'type of class 2 effector module arose from subsequent independent insertions '\n"," 'of mobile genetic elements.[144] These mobile genetic elements took the '\n"," 'place of the multiple gene effector modules to create single gene effector '\n"," 'modules that produce large proteins which perform all the necessary tasks of '\n"," 'the effector module.[144] The spacer regions of CRISPR-Cas systems are taken '\n"," 'directly from foreign mobile genetic elements and thus their long term '\n"," 'evolution is hard to trace.[145] The non-random evolution of these spacer '\n"," 'regions has been found to be highly dependent on the environment and the '\n"," 'particular foreign mobile genetic elements it contains.[146]\\n'\n"," 'CRISPR/Cas can immunize bacteria against certain phages and thus halt '\n"," 'transmission. For this reason, Koonin described CRISPR/Cas as a Lamarckian '\n"," 'inheritance mechanism.[147] However, this was disputed by a critic who '\n"," 'noted, \"We should remember [Lamarck] for the good he contributed to science, '\n"," 'not for things that resemble his theory only superficially. Indeed, thinking '\n"," 'of CRISPR and other phenomena as Lamarckian only obscures the simple and '\n"," 'elegant way evolution really works\".[148] But as more recent studies have '\n"," 'been conducted, it has become apparent that the acquired spacer regions of '\n"," 'CRISPR-Cas systems are indeed a form of Lamarckian evolution because they '\n"," 'are genetic mutations that are acquired and then passed on.[149] On the '\n"," 'other hand, the evolution of the Cas gene machinery that facilitates the '\n"," 'system evolves through classic Darwinian evolution.[149]\\n'\n"," 'Coevolution\\n'\n"," '\\n'\n"," 'Analysis of CRISPR sequences revealed coevolution of host and viral '\n"," 'genomes.[150] Cas9 proteins are highly enriched in pathogenic and commensal '\n"," 'bacteria. CRISPR/Cas-mediated gene regulation may contribute to the '\n"," 'regulation of endogenous bacterial genes, particularly during interaction '\n"," 'with eukaryotic hosts. For example, Francisella novicida uses a unique, '\n"," 'small, CRISPR/Cas-associated RNA (scaRNA) to repress an endogenous '\n"," 'transcript encoding a bacterial lipoprotein that is critical for F. novicida '\n"," 'to dampen host response and promote virulence.[151]\\n'\n"," 'The basic model of CRISPR evolution is newly incorporated spacers driving '\n"," 'phages to mutate their genomes to avoid the bacterial immune response, '\n"," 'creating diversity in both the phage and host populations. To resist a phage '\n"," 'infection, the sequence of the CRISPR spacer must correspond perfectly to '\n"," 'the sequence of the target phage gene. Phages can continue to infect their '\n"," 'hosts given point mutations in the spacer.[140] Similar stringency is '\n"," 'required in PAM or the bacterial strain remains phage sensitive.[113][140]\\n'\n"," '\\n'\n"," '\\n'\n"," 'Rates\\n'\n"," '\\n'\n"," 'A study of 124 S. thermophilus strains showed that 26% of all spacers were '\n"," 'unique and that different CRISPR loci showed different rates of spacer '\n"," 'acquisition.[112] Some CRISPR loci evolve more rapidly than others, which '\n"," \"allowed the strains' phylogenetic relationships to be determined. A \"\n"," 'comparative genomic analysis showed that E. coli and S. enterica evolve much '\n"," \"more slowly than S. thermophilus. The latter's strains that diverged 250 \"\n"," 'thousand years ago still contained the same spacer complement.[152]\\n'\n"," 'Metagenomic analysis of two acid-mine-drainage biofilms showed that one of '\n"," 'the analyzed CRISPRs contained extensive deletions and spacer additions '\n"," 'versus the other biofilm, suggesting a higher phage activity/prevalence in '\n"," 'one community than the other.[72] In the oral cavity, a temporal study '\n"," 'determined that 7–22% of spacers were shared over 17 months within an '\n"," 'individual while less than 2% were shared across individuals.[121]\\n'\n"," 'From the same environment a single strain was tracked using PCR primers '\n"," 'specific to its CRISPR system. Broad-level results of spacer '\n"," 'presence/absence showed significant diversity. However, this CRISPR added 3 '\n"," 'spacers over 17 months,[121] suggesting that even in an environment with '\n"," 'significant CRISPR diversity some loci evolve slowly.\\n'\n"," 'CRISPRs were analysed from the metagenomes produced for the human microbiome '\n"," 'project.[153] Although most were body-site specific, some within a body site '\n"," 'are widely shared among individuals. One of these loci originated from '\n"," 'streptococcal species and contained ≈15,000 spacers, 50% of which were '\n"," 'unique. Similar to the targeted studies of the oral cavity, some showed '\n"," 'little evolution over time.[153]\\n'\n"," 'CRISPR evolution was studied in chemostats using S. thermophilus to directly '\n"," 'examine spacer acquisition rates. In one week, S. thermophilus strains '\n"," 'acquired up to three spacers when challenged with a single phage.[154] '\n"," 'During the same interval the phage developed single nucleotide polymorphisms '\n"," 'that became fixed in the population, suggesting that targeting had prevented '\n"," 'phage replication absent these mutations.[154]\\n'\n"," 'Another S. thermophilus experiment showed that phages can infect and '\n"," 'replicate in hosts that have only one targeting spacer. Yet another showed '\n"," 'that sensitive hosts can exist in environments with high phage titres.[155] '\n"," 'The chemostat and observational studies suggest many nuances to CRISPR and '\n"," 'phage (co)evolution.\\n')\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"wsPyNY-tOR4U","executionInfo":{"status":"ok","timestamp":1613607503981,"user_tz":-60,"elapsed":550,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"431bcd40-dbcc-499b-ba8e-db9564024d14"},"source":["for s in sections:\n","  print(s.heading)\n","  if s.children:\n","    print('\\t', [c.heading for c in s.children])"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Contents\n","History\n","\t ['Repeated sequences', 'CRISPR-associated systems', 'Cas9', 'Cas12a (formerly Cpf1)', 'Cas13 (formerly C2c2)']\n","Repeated sequences\n","CRISPR-associated systems\n","Cas9\n","Cas12a (formerly Cpf1)\n","Cas13 (formerly C2c2)\n","Locus structure\n","\t ['Repeats and spacers', 'CRISPR RNA structures', 'Cas genes and CRISPR subtypes']\n","Repeats and spacers\n","CRISPR RNA structures\n","Cas genes and CRISPR subtypes\n","Mechanism\n","\t ['Spacer acquisition', 'Biogenesis', 'Interference']\n","Spacer acquisition\n","\t ['Protospacer adjacent motifs', 'Insertion variants']\n","Protospacer adjacent motifs\n","Insertion variants\n","Biogenesis\n","Interference\n","Evolution\n","\t ['Coevolution', 'Rates']\n","Coevolution\n","Rates\n","Identification\n","Use by phages\n","Applications\n","\t ['CRISPR gene editing', 'CRISPR as diagnostic tool']\n","CRISPR gene editing\n","CRISPR as diagnostic tool\n","See also\n","Notes\n","References\n","Further reading\n","External links\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"A4IQsgXc5x27"},"source":["# Evolution of headings"]},{"cell_type":"code","metadata":{"id":"cnMvIwDuNkAp"},"source":["articles = [    \n","  CRISPR_gene_editing_en,\n","  CRISPR_en,\n","]\n","\n","xpath_expression = \"|\".join([\".//\" + tag for tag in [\"h1\",\"h2\",\"h3\",\"h4\",\"h5\",\"h6\"]])\n","for article in articles:\n","  # print(article.name)\n","  unique_headings = set()\n","  evolving_headings = []\n","  for revision in article.yield_revisions():\n","    headings = tuple(\n","      (heading.tag, ''.join(heading.itertext()).split('[edit]')[0].strip()) \n","      for heading in revision.etree_from_html().xpath(xpath_expression)\n","    )\n","    if not headings in unique_headings:\n","      evolving_headings.append(\n","          (revision.index, revision.timestamp.string, revision.revid, revision.url, headings)\n","      )\n","    unique_headings.add(headings)\n","  df = pd.DataFrame(evolving_headings, columns=['revindx','timestamp','revid','url','new_headings',])\n","  # df.to_excel(path_extractions + '/wikipedia_entries/headings_history_{}.xlsx'.format(article.name))\n","\n","  # Output\n","  with open(path_extractions + '/wikipedia_entries/headings_history_{}.txt'.format(article.name), 'w') as writer:\n","    for indx, row in df.iterrows():\n","      writer.write(row.timestamp[:10] + ' ({}/{}) '.format(indx, row.revindx) + row.url + '\\n\\n')\n","      for i in row.new_headings:\n","        writer.write('\\t{}{}'.format('   '*(int(i[0][-1])-2) ,i[1]) + '\\n')\n","      writer.write('\\n')"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"IpZaf0Otpqth"},"source":["# Revision history into table"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":459},"id":"0UE9b4cap2gF","executionInfo":{"status":"ok","timestamp":1612892756295,"user_tz":-60,"elapsed":47502,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"176b2af7-c2e3-493c-f640-78b532d502f1"},"source":["articles = [    \n","  CRISPR_en,\n","  CRISPR_gene_editing_en\n","]\n","\n","for article in articles:\n","  print(article.name)\n","  differences = article.calculate_revision_size_difference()\n","  revisions = [\n","      (revision.index, revision.timestamp.string, revision.revid, revision.url, revision.user, revision.userid, revision.size, revision.comment)\n","      for revision in article.yield_revisions()\n","  ]\n","  df = pd.DataFrame(revisions, columns=['revindx','timestamp','revid','url','user','userid','size','comment',])\n","  df.insert(loc=7, column='diff', value=differences) # https://stackoverflow.com/questions/18674064/how-do-i-insert-a-column-at-a-specific-column-index-in-pandas\n","  df.to_excel(path_extractions + '/wikipedia_entries/revision_history_{}.xlsx'.format(article.name))\n","df"],"execution_count":null,"outputs":[{"output_type":"stream","text":["CRISPR_en\n","CRISPR_gene_editing_en\n"],"name":"stdout"},{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>revindx</th>\n","      <th>timestamp</th>\n","      <th>revid</th>\n","      <th>url</th>\n","      <th>user</th>\n","      <th>userid</th>\n","      <th>size</th>\n","      <th>diff</th>\n","      <th>comment</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>2019-02-17 06:32:48</td>\n","      <td>883727671</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Helito</td>\n","      <td>28440460</td>\n","      <td>77238</td>\n","      <td>77238</td>\n","      <td>Split from CRISPR page</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1</td>\n","      <td>2019-02-17 06:33:54</td>\n","      <td>883727741</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Helito</td>\n","      <td>28440460</td>\n","      <td>77551</td>\n","      <td>313</td>\n","      <td></td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>2</td>\n","      <td>2019-02-17 06:35:41</td>\n","      <td>883727860</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Helito</td>\n","      <td>28440460</td>\n","      <td>77771</td>\n","      <td>220</td>\n","      <td>rectify ref 8</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>3</td>\n","      <td>2019-02-17 06:37:14</td>\n","      <td>883727959</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Helito</td>\n","      <td>28440460</td>\n","      <td>78031</td>\n","      <td>260</td>\n","      <td>/* Editing */ rectify ref 25</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>4</td>\n","      <td>2019-02-17 06:39:37</td>\n","      <td>883728113</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Helito</td>\n","      <td>28440460</td>\n","      <td>78024</td>\n","      <td>-7</td>\n","      <td>/* Editing */</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>194</th>\n","      <td>194</td>\n","      <td>2020-12-10 20:28:06</td>\n","      <td>993476518</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Nisse Hulta</td>\n","      <td>11218895</td>\n","      <td>96234</td>\n","      <td>243</td>\n","      <td>Link to the Nobel Prize site</td>\n","    </tr>\n","    <tr>\n","      <th>195</th>\n","      <td>195</td>\n","      <td>2020-12-15 10:27:37</td>\n","      <td>994366355</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Smartse</td>\n","      <td>1640548</td>\n","      <td>95784</td>\n","      <td>-450</td>\n","      <td>rm [[WP:REFSPAM]] - see talk</td>\n","    </tr>\n","    <tr>\n","      <th>196</th>\n","      <td>196</td>\n","      <td>2020-12-19 07:34:37</td>\n","      <td>995107732</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>Bagumba</td>\n","      <td>8005368</td>\n","      <td>95699</td>\n","      <td>-85</td>\n","      <td>/* top */  [[WP:NAMB]]</td>\n","    </tr>\n","    <tr>\n","      <th>197</th>\n","      <td>197</td>\n","      <td>2020-12-22 17:57:54</td>\n","      <td>995747724</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>PaulTConley</td>\n","      <td>35393367</td>\n","      <td>95975</td>\n","      <td>276</td>\n","      <td>/* Predecessors */ Added detail on: zinc finge...</td>\n","    </tr>\n","    <tr>\n","      <th>198</th>\n","      <td>198</td>\n","      <td>2020-12-31 22:07:09</td>\n","      <td>997514523</td>\n","      <td>https://en.wikipedia.org/w/index.php?title=CRI...</td>\n","      <td>OAbot</td>\n","      <td>28481209</td>\n","      <td>96011</td>\n","      <td>36</td>\n","      <td>[[Wikipedia:OABOT|Open access bot]]: doi added...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>199 rows × 9 columns</p>\n","</div>"],"text/plain":["     revindx  ...                                            comment\n","0          0  ...                             Split from CRISPR page\n","1          1  ...                                                   \n","2          2  ...                                      rectify ref 8\n","3          3  ...                       /* Editing */ rectify ref 25\n","4          4  ...                                      /* Editing */\n","..       ...  ...                                                ...\n","194      194  ...                       Link to the Nobel Prize site\n","195      195  ...                       rm [[WP:REFSPAM]] - see talk\n","196      196  ...                             /* top */  [[WP:NAMB]]\n","197      197  ...  /* Predecessors */ Added detail on: zinc finge...\n","198      198  ...  [[Wikipedia:OABOT|Open access bot]]: doi added...\n","\n","[199 rows x 9 columns]"]},"metadata":{"tags":[]},"execution_count":4}]},{"cell_type":"markdown","metadata":{"id":"rRJN-eU96egO"},"source":["# Check which entry pointing to CRISPR really mentions CRISPR "]},{"cell_type":"markdown","metadata":{"id":"aXqC-apagkNN"},"source":["## Get incomming entries and users"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Pkxc15HZUGgZ","executionInfo":{"status":"ok","timestamp":1612911189567,"user_tz":-60,"elapsed":1881,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"afb1a6d2-8fd2-4760-992c-98b26b469a21"},"source":["target_entries =[\n","  'CRISPR',\n","  'CRISPR_gene_editing',\n","]\n","\n","incomming_entries_titles = []\n","incomming_entries_urls = []\n","\n","for target in target_entries:\n","  print(target)\n","  # url = 'https://en.wikipedia.org/w/index.php?title=Special:WhatLinksHere/{}&hideredirs=1&hidetrans=1&limit=500'.format(target) # note: only first 500...\n","  url = 'https://en.wikipedia.org/w/index.php?title=Special:WhatLinksHere/{}&namespace=0&limit=500&hideredirs=1&hidetrans=1'.format(target) # note: only first 500...\n","  target_soup = make_soup(url)\n","  for li in target_soup.find_all(attrs={\"id\": \"mw-whatlinkshere-list\"}):\n","    for i in li.find_all('a'):\n","      if i['href'].startswith('/wiki/'):\n","        if not ':' in i['href']:\n","          in_link_title = i['href'][6:]\n","          in_link_url = 'https://en.wikipedia.org' + i['href']\n","          incomming_entries_titles.append(in_link_title)\n","          incomming_entries_urls.append(in_link_url)\n","  # Save incoming entries\n","  with open(path_extractions + '/wikipedia_entries/{}_incomming_entries_titles.txt'.format(target), 'w') as f:\n","      f.writelines([i+'\\n' if not indx == len(incomming_entries_titles) - 1 else i for indx, i in enumerate(sorted(incomming_entries_titles))])\n","  with open(path_extractions + '/wikipedia_entries/{}_incomming_entries_urls.txt'.format(target), 'w') as f:\n","      f.writelines([i+'\\n' if not indx == len(incomming_entries_urls) - 1 else i for indx, i in enumerate(sorted(incomming_entries_urls))])"],"execution_count":null,"outputs":[{"output_type":"stream","text":["CRISPR\n","CRISPR_gene_editing\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"mhVGOeFIgnxt"},"source":["## Filter relevant entries"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Weh2T9C-h38L","executionInfo":{"status":"ok","timestamp":1612911478471,"user_tz":-60,"elapsed":137195,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"8e8dfcd4-fb62-43b6-b619-96507308c531"},"source":["with open(path_extractions + '/wikipedia_entries/CRISPR_incomming_entries_titles.txt', 'r') as f:\n","    CRISPR_incomming_entries_titles = {i.strip() for i in f.readlines()}\n","with open(path_extractions + '/wikipedia_entries/CRISPR_gene_editing_incomming_entries_titles.txt', 'r') as f:\n","    CRISPR_gene_editing_incomming_entries_titles = {i.strip() for i in f.readlines()}\n","incomming_entries_titles = sorted(CRISPR_incomming_entries_titles | CRISPR_gene_editing_incomming_entries_titles)\n","\n","relevant_entries_text = []\n","relevant_entries_else = []\n","irrelevant_entries = []\n","no_crispr = []\n","no_entries = []\n","\n","for title in incomming_entries_titles:\n","  print(title)\n","  url = 'https://en.wikipedia.org/wiki/{}'.format(title)\n","  soup = make_soup(url)\n","  if 'Wikipedia does not have an article with this exact name.' in soup.text: # just a quick hack...\n","    print('\\tNo Soup!')\n","    no_entries.append(title)\n","    continue\n","  if not 'CRISPR' in str(soup.html): # Sollte nicht vorkommen!\n","    print('\\tNo CRISPR!')\n","    no_crispr.append(title)\n","    continue\n","  # kill navboxes\n","  discard = soup.find_all(attrs={\"class\": \"navbox\"}) # https://stackoverflow.com/questions/39885359/beautifulsoup-decompose\n","  for match in discard:\n","      match.decompose()\n","  # kill 'series' boxes\n","  discard = soup.find_all(attrs={\"class\": \"sidebar vertical-navbox nomobile plainlist\"}) # https://stackoverflow.com/questions/39885359/beautifulsoup-decompose\n","  for match in discard:\n","      match.decompose()\n","  # Check for relevance\n","  if not 'CRISPR' in str(soup.html):\n","    irrelevant_entries.append(title)\n","    continue\n","  if 'CRISPR' in soup.text:\n","    relevant_entries_text.append(title)\n","  else:\n","    relevant_entries_else.append(title)\n","\n","print('relevant_entries')\n","pp.pprint(relevant_entries)\n","\n","# Save files\n","with open(path_extractions + '/wikipedia_entries/TEXT_incomming_entries_titles.txt', 'w') as f:\n","    f.writelines([i+'\\n' if not indx == len(relevant_entries_text) - 1 else i for indx, i in enumerate(relevant_entries_text)])\n","with open(path_extractions + '/wikipedia_entries/ELSE_incomming_entries_titles.txt', 'w') as f:\n","    f.writelines([i+'\\n' if not indx == len(relevant_entries_else) - 1 else i for indx, i in enumerate(relevant_entries_else)])\n","with open(path_extractions + '/wikipedia_entries/BOX_incomming_entries_titles.txt', 'w') as f:\n","    f.writelines([i+'\\n' if not indx == len(irrelevant_entries) - 1 else i for indx, i in enumerate(irrelevant_entries)])\n","with open(path_extractions + '/wikipedia_entries/NO_CRISPR_incomming_entries_titles.txt', 'w') as f:\n","    f.writelines([i+'\\n' if not indx == len(no_crispr) - 1 else i for indx, i in enumerate(no_crispr)])\n","with open(path_extractions + '/wikipedia_entries/NO_ENTRIES_incomming_entries_titles.txt', 'w') as f:\n","    f.writelines([i+'\\n' if not indx == len(no_entries) - 1 else i for indx, i in enumerate(no_entries)])"],"execution_count":null,"outputs":[{"output_type":"stream","text":["15-Cis-phytoene_desaturase\n","2010s\n","2012_in_science\n","2014_in_science\n","2015_in_science\n","2015_in_the_United_States\n","2016_in_science\n","2017_in_science\n","2017_in_the_United_States\n","2018_in_science\n","2019\n","2019_in_science\n","2020_in_science\n","ATUM\n","Aaron_Traywick\n","Accelerating_expansion_of_the_universe\n","Adam_Bogdanove\n","Addgene\n","Adenoviridae\n","Alex_Marson\n","Alu_element\n","Anatoly_B._Kolomeisky\n","Andrea_Crisanti_(scientist)\n","Anna_Dumitriu\n","Anti-CRISPR\n","Antibiotic\n","Antidote\n","Antisense_RNA\n","Antiviral_drug\n","April%E2%80%93June_2020_in_science\n","Arabidopsis_thaliana\n","Ardipithecus\n","Ardipithecus_ramidus\n","Ark_Invest\n","Artificial_cell\n","Asian_Scientist\n","Avian_malaria\n","BASF\n","BLESS\n","Bacteria\n","Bacterial_DNA_binding_protein\n","Bacterial_small_RNA\n","Bacteriophage\n","Behavioural_genetics\n","BioGRID\n","Biohub\n","Biotechnology_risk\n","Box_jellyfish\n","Breakthrough_of_the_Year\n","Brian_Hanley_(microbiologist)\n","Brief_Answers_to_the_Big_Questions\n","Bryan_R._Cullen\n","CAS\n","CAS2_(disambiguation)\n","CASPR\n","CITE-Seq\n","COVID-19_pandemic_in_Karnataka\n","COVID-19_testing\n","CRISPR\n","CRISPR-Display\n","CRISPR/Cas_Tools\n","CRISPR/Cpf1\n","CRISPR_Therapeutics\n","CRISPR_activation\n","CRISPR_gene_editing\n","CRISPR_interference\n","Cancer_immunotherapy\n","Cancer_pharmacogenomics\n","Candida_albicans\n","Carl_Zimmer\n","Carolyn_Lawrence-Dill\n","Cartagena_Protocol_on_Biosafety\n","Cas1\n","Cas2\n","Cas3\n","Cas4\n","Cas9\n","Cascade\n","Cell_culture\n","Cell_lineage\n","Cepheid_(company)\n","Chen_Hu_(physician)\n","Chick_culling\n","Choroideremia\n","Christoph_Bock\n","Christopher_Vakoc\n","Cis-natural_antisense_transcript\n","Clinical_metagenomic_sequencing\n","Cpf1\n","Crispa\n","Crisper\n","Cure_Rare_Disease\n","Cyanophage\n","Cyanophage_N-1\n","Cyanothece\n","D-Wave_Systems\n","DNA-binding_domain\n","DNA_digital_data_storage\n","DNA_repair\n","DNA_transposon\n","Daisy_chaining_DNA\n","Daniel_Suarez_(author)\n","Daniel_Voytas\n","Danisco\n","Dark_energy\n","David_M._Sabatini\n","David_and_Lucile_Packard_Foundation\n","Deng_Hongkui\n","Designer_baby\n","Dharmacon\n","Diatom\n","Direct_repeat\n","Directed_evolution_(transhumanism)\n","Do-it-yourself_biology\n","Dolly_(sheep)\n","Domestic_pig\n","Dorothy_Semenow\n","Editas_Medicine\n","Elche\n","Ellen_Jorgensen\n","Emmanuelle_Charpentier\n","Endocrine_disruptor\n","Entomological_warfare\n","Epigenome_editing\n","Eradication_of_suffering\n","Evolution\n","FGF5\n","Facioscapulohumeral_muscular_dystrophy\n","Feng_Zhang\n","First_observation_of_gravitational_waves\n","Flaminia_Catteruccia\n","Fleming_Prize_Lecture\n","Fondation_M%C3%A9rieux\n","Four_Thieves_Vinegar_Collective\n","Francisco_Mojica\n","Francisella_small_RNAs\n","GMO_conspiracy_theories\n","GW170817\n","Gene_amplification\n","Gene_cluster\n","Gene_conversion\n","Gene_delivery\n","Gene_drive\n","Gene_duplication\n","Gene_editing\n","Gene_family\n","Gene_knock-in\n","Gene_knockdown\n","Gene_knockout\n","Gene_silencing\n","Gene_therapy\n","Gene_therapy_for_epilepsy\n","Genetic_engineering\n","Genetic_engineering_techniques\n","Genetically_modified_animal\n","Genetically_modified_bacteria\n","Genetically_modified_crops\n","Genetically_modified_fish\n","Genetically_modified_food\n","Genetically_modified_food_controversies\n","Genetically_modified_insect\n","Genetically_modified_maize\n","Genetically_modified_mammal\n","Genetically_modified_organism\n","Genetically_modified_plant\n","Genetically_modified_potato\n","Genetically_modified_rice\n","Genetically_modified_soybean\n","Genetically_modified_virus\n","Genetics\n","Genodermatosis\n","Genome-wide_CRISPR-Cas9_knockout_screens\n","Genome_editing\n","Genomic_island\n","George_Church_(geneticist)\n","Germline_mutation\n","Gladstone_Institutes\n","Glossary_of_genetics\n","Gravitational_wave\n","Gregory_Hannon\n","Guide_RNA\n","HA-tag\n","HAP1_cells\n","HAT_transposon\n","HIV\n","HPTN_052\n","Hazards_of_synthetic_biology\n","He_Jiankui\n","He_Jiankui_affair\n","Hefei\n","Helitron_(biology)\n","Herpes_simplex_research\n","Higgs_boson\n","Hilo_High_School\n","History_of_genetic_engineering\n","History_of_genetics\n","History_of_research_on_Arabidopsis_thaliana\n","Horizon_Discovery\n","Horizontal_Environmental_Genetic_Alteration_Agents\n","How_to_Clone_a_Mammoth\n","Human_Nature_(2019_film)\n","Human_genetic_variation\n","Human_germline_engineering\n","Huntington%27s_disease\n","Hyperphosphatasia_with_mental_retardation_syndrome\n","IFT140\n","Immune_system\n","Induced_stem_cells\n","Institute_of_Genomics_and_Integrative_Biology\n","Intellia_Therapeutics\n","International_Society_for_Stem_Cell_Research\n","Interspersed_repeat\n","Inverted_repeat\n","Ishino\n","J%C3%B6rg_Vogel\n","J%C3%BCrgen_Brosius\n","Jacquin_Niles\n","Janice_Chen\n","January%E2%80%93March_2020_in_science\n","Jean-Fran%C3%A7ois_Gari%C3%A9py\n","Jennifer_Doudna\n","Jennifer_Kahn\n","Jim_Penman\n","Josiah_Zayner\n","Juvisy-sur-Orge\n","Katherine_Jones_(academic)\n","Kathy_Niakan\n","Kavli_Prize\n","Killifish\n","Kira_Makarova\n","Kiran_Musunuru\n","Knowing_Neurons\n","Kraig_Biocraft_Laboratories\n","Kyushu_University\n","LEAPER_gene_editing\n","LINE1\n","LMNA-related_congenital_muscular_dystrophy\n","LTR_retrotransposon\n","Lactobacillus_rhamnosus\n","Lamarckism\n","Lei_Stanley_Qi\n","List_of_French_inventions_and_discoveries\n","List_of_Iranian_Americans\n","List_of_Japanese_inventions_and_discoveries\n","List_of_Last_Week_Tonight_with_John_Oliver_episodes\n","List_of_Nobel_laureates_in_Chemistry\n","List_of_Pomona_College_people\n","List_of_RNAs\n","List_of_Spanish_inventions_and_discoveries\n","List_of_The_Good_Doctor_episodes\n","List_of_University_of_California,_Berkeley_faculty\n","List_of_awards_and_honors_received_by_Jennifer_Doudna\n","List_of_biochemists\n","List_of_covers_of_Time_magazine_(2010s)\n","List_of_drugs_by_year_of_discovery\n","List_of_inventions_and_discoveries_by_women\n","Lithuania\n","Live_single-cell_imaging\n","Locus_Biosciences\n","Long_interspersed_nuclear_element\n","Long_terminal_repeat\n","Low_copy_repeats\n","Luciano_Marraffini\n","MAGESTIC\n","MCR\n","MELK\n","MIR561\n","MISL_RNA_motif\n","MacVector\n","Mammalian-wide_interspersed_repeat\n","Margaret_Goodell\n","Maria_Vorontsova\n","Marine_viruses\n","Mary_Ann_Liebert\n","Mary_Dasso\n","Mechanism_of_action\n","Medical_genetics\n","Melanie_Ott\n","Merlin_Crossley\n","Methanocaldococcus_sp._FS406-22\n","Methylobacillus_flagellatus\n","MicroRNA\n","MicrobesOnline\n","Microsatellite\n","Middle_Eastern_Americans\n","Minisatellite\n","Mobile_genetic_elements\n","Molecular_engineering\n","Molecular_scale_electronics\n","Mutagenesis_(molecular_biology_technique)\n","Nanocircuitry\n","Nessa_Carey\n","Neutron_star_merger\n","NgAgo\n","Nitrososphaera_gargensis\n","No-SCAR_(Scarless_Cas9_Assisted_Recombineering)_Genome_Editing\n","Nobel_Prize_controversies\n","Non-coding_RNA\n","Nuclear_organization\n","Off-target_genome_editing\n","Open_synthetic_biology\n","Opisthorchiasis\n","Osaka_University\n","Outline_of_transhumanism\n","P_element\n","Pandemic_prevention\n","Paola_Arlotta\n","Pathogenicity_island\n","Paul_J._Tesar\n","Paul_Knoepfler\n","Pelizaeus%E2%80%93Merzbacher_disease\n","Perturb-seq\n","Phage_ecology\n","Phage_therapy\n","Pharmacoepigenetics\n","Phikzvirus\n","Philippe_Horvath\n","Physalis_pruinosa\n","PiggyBac_transposon_system\n","Piwi-interacting_RNA\n","Poincar%C3%A9_conjecture\n","Polinton\n","Polled_livestock\n","Pomona_College\n","Poribacteria\n","Protein_%26_Cell\n","Protospacer_adjacent_motif\n","Pseudogene\n","Pseudomonas_syringae\n","Public_participation_(decision_making)\n","Pusztai_affair\n","Quantum_machine\n","RNA\n","RNA-directed_DNA_methylation\n","RNA_editing\n","RNA_interference\n","Rampage_(2018_film)\n","Rampage_(2018_video_game)\n","Rat_Genome_Database\n","Recombinant_DNA\n","Red_flour_beetle\n","Regulation_of_genetic_engineering\n","Repeated_sequence_(DNA)\n","Repeatome\n","Research_Science_Institute\n","Restriction_enzyme\n","Retinitis_pigmentosa\n","Retroposon\n","Retrotransposon\n","Revival_of_the_woolly_mammoth\n","Ribonuclease\n","Ribosomal_DNA\n","Rodolphe_Barrangou\n","Rosetta_(spacecraft)\n","S%C3%A9ralini_affair\n","SKBR3\n","SaPI\n","Saccharomyces_boulardii\n","Samira_Kiani\n","Santa_Cruz_Biotechnology\n","Satellite_DNA\n","Sauerkraut\n","Schistosomiasis\n","Science_(journal)\n","Scott_W._Lowe\n","Selfish_genetic_element\n","Short_hairpin_RNA\n","Short_interspersed_nuclear_element\n","Shoukhrat_Mitalipov\n","Sickle_cell_disease\n","Single_cell_sequencing\n","Site-directed_mutagenesis\n","Small_interfering_RNA\n","Small_nucleolar_RNA\n","Small_temporal_RNA\n","Spirit_(rover)\n","StarLink_corn_recall\n","Stem_cell\n","Stephen_Tsang\n","Streptococcus_pyogenes\n","Surveyor_nuclease_assay\n","Synechocystis_sp._PCC_6803\n","Synthetic_biological_circuit\n","Synthetic_biology\n","Synthetic_genomics\n","Synthetic_immunology\n","T7_phage\n","Tandem_repeat\n","Tandemly_arrayed_genes\n","Taosheng_Huang\n","Tc1/mariner\n","Telomere\n","The_CRISPR_Journal\n","The_Hastings_Center\n","Theobroma_cacao\n","Theodore_Roosevelt_High_School_(Iowa)\n","Time_Person_of_the_Year\n","Timeline_of_biology_and_organic_chemistry\n","Timeline_of_biotechnology\n","Timeline_of_the_San_Francisco_Bay_Area\n","Timeline_of_women_in_science\n","Topologically_associating_domain\n","Trans-acting_siRNA\n","Trans-activating_crRNA\n","Transcription_activator-like_effector_nuclease\n","Transfer-messenger_RNA\n","Transhuman\n","Transib\n","Transposable_element\n","Trichome\n","Trinucleotide_repeat_disorder\n","Tubulin\n","Ume%C3%A5\n","University_of_California,_Berkeley\n","Unnatural_Selection\n","Unnatural_Selection_(TV_series)\n","VPS35\n","Variable_number_tandem_repeat\n","Vilnius\n","Viral_disease_testing\n","Virginijus_%C5%A0ik%C5%A1nys\n","Virome\n","Virotherapy\n","Virus\n","Whole_genome_sequencing\n","Woolly_mammoth\n","Xenobiology\n","Xenotransplantation\n","Xiaole_Shirley_Liu\n","Yoshizumi_Ishino\n","Zhong_Zhong_and_Hua_Hua\n","Zinc_finger_nuclease\n","relevant_entries\n","[   '15-Cis-phytoene_desaturase',\n","    '2010s',\n","    '2012_in_science',\n","    '2014_in_science',\n","    '2015_in_science',\n","    '2015_in_the_United_States',\n","    '2016_in_science',\n","    '2017_in_science',\n","    '2017_in_the_United_States',\n","    '2018_in_science',\n","    '2019',\n","    '2019_in_science',\n","    '2020_in_science',\n","    'ATUM',\n","    'Adam_Bogdanove',\n","    'Addgene',\n","    'Adenoviridae',\n","    'Alex_Marson',\n","    'Andrea_Crisanti_(scientist)',\n","    'Anna_Dumitriu',\n","    'Anti-CRISPR',\n","    'Antibiotic',\n","    'Antidote',\n","    'Antiviral_drug',\n","    'April%E2%80%93June_2020_in_science',\n","    'Arabidopsis_thaliana',\n","    'Asian_Scientist',\n","    'Avian_malaria',\n","    'BASF',\n","    'Bacteria',\n","    'Bacterial_DNA_binding_protein',\n","    'Bacteriophage',\n","    'Behavioural_genetics',\n","    'BioGRID',\n","    'Biohub',\n","    'Biotechnology_risk',\n","    'Box_jellyfish',\n","    'Breakthrough_of_the_Year',\n","    'Brian_Hanley_(microbiologist)',\n","    'CAS',\n","    'CAS2_(disambiguation)',\n","    'CASPR',\n","    'COVID-19_pandemic_in_Karnataka',\n","    'COVID-19_testing',\n","    'CRISPR',\n","    'CRISPR-Display',\n","    'CRISPR/Cas_Tools',\n","    'CRISPR/Cpf1',\n","    'CRISPR_Therapeutics',\n","    'CRISPR_activation',\n","    'CRISPR_interference',\n","    'Candida_albicans',\n","    'Carl_Zimmer',\n","    'Cas1',\n","    'Cas9',\n","    'Cascade',\n","    'Cell_culture',\n","    'Cell_lineage',\n","    'Cepheid_(company)',\n","    'Chen_Hu_(physician)',\n","    'Chick_culling',\n","    'Choroideremia',\n","    'Christoph_Bock',\n","    'Cpf1',\n","    'Crispa',\n","    'Crisper',\n","    'Cyanophage',\n","    'Cyanophage_N-1',\n","    'Cyanothece',\n","    'DNA-binding_domain',\n","    'DNA_digital_data_storage',\n","    'DNA_repair',\n","    'Daisy_chaining_DNA',\n","    'Daniel_Suarez_(author)',\n","    'Daniel_Voytas',\n","    'Danisco',\n","    'David_M._Sabatini',\n","    'David_and_Lucile_Packard_Foundation',\n","    'Deng_Hongkui',\n","    'Designer_baby',\n","    'Dharmacon',\n","    'Diatom',\n","    'Directed_evolution_(transhumanism)',\n","    'Do-it-yourself_biology',\n","    'Domestic_pig',\n","    'Editas_Medicine',\n","    'Elche',\n","    'Emmanuelle_Charpentier',\n","    'Endocrine_disruptor',\n","    'Entomological_warfare',\n","    'Epigenome_editing',\n","    'Eradication_of_suffering',\n","    'FGF5',\n","    'Facioscapulohumeral_muscular_dystrophy',\n","    'Feng_Zhang',\n","    'Flaminia_Catteruccia',\n","    'Francisco_Mojica',\n","    'Francisella_small_RNAs',\n","    'Gene_drive',\n","    'Gene_editing',\n","    'Gene_knock-in',\n","    'Gene_knockdown',\n","    'Gene_knockout',\n","    'Gene_silencing',\n","    'Gene_therapy',\n","    'Gene_therapy_for_epilepsy',\n","    'Genetic_engineering',\n","    'Genetic_engineering_techniques',\n","    'Genetically_modified_animal',\n","    'Genetically_modified_crops',\n","    'Genetically_modified_food',\n","    'Genetically_modified_insect',\n","    'Genetically_modified_organism',\n","    'Genetics',\n","    'Genodermatosis',\n","    'Genome-wide_CRISPR-Cas9_knockout_screens',\n","    'Genome_editing',\n","    'George_Church_(geneticist)',\n","    'Germline_mutation',\n","    'Gladstone_Institutes',\n","    'Glossary_of_genetics',\n","    'Gregory_Hannon',\n","    'Guide_RNA',\n","    'HA-tag',\n","    'HAP1_cells',\n","    'He_Jiankui_affair',\n","    'Hefei',\n","    'Herpes_simplex_research',\n","    'Hilo_High_School',\n","    'History_of_genetic_engineering',\n","    'History_of_genetics',\n","    'History_of_research_on_Arabidopsis_thaliana',\n","    'Horizon_Discovery',\n","    'Horizontal_Environmental_Genetic_Alteration_Agents',\n","    'How_to_Clone_a_Mammoth',\n","    'Human_germline_engineering',\n","    'Huntington%27s_disease',\n","    'Hyperphosphatasia_with_mental_retardation_syndrome',\n","    'IFT140',\n","    'Immune_system',\n","    'Induced_stem_cells',\n","    'Institute_of_Genomics_and_Integrative_Biology',\n","    'Intellia_Therapeutics',\n","    'International_Society_for_Stem_Cell_Research',\n","    'Ishino',\n","    'J%C3%B6rg_Vogel',\n","    'J%C3%BCrgen_Brosius',\n","    'Jacquin_Niles',\n","    'Janice_Chen',\n","    'January%E2%80%93March_2020_in_science',\n","    'Jennifer_Doudna',\n","    'Jim_Penman',\n","    'Josiah_Zayner',\n","    'Juvisy-sur-Orge',\n","    'Kathy_Niakan',\n","    'Kavli_Prize',\n","    'Killifish',\n","    'Kiran_Musunuru',\n","    'Knowing_Neurons',\n","    'Kraig_Biocraft_Laboratories',\n","    'Kyushu_University',\n","    'LEAPER_gene_editing',\n","    'LMNA-related_congenital_muscular_dystrophy',\n","    'Lactobacillus_rhamnosus',\n","    'Lamarckism',\n","    'List_of_French_inventions_and_discoveries',\n","    'List_of_Iranian_Americans',\n","    'List_of_Japanese_inventions_and_discoveries',\n","    'List_of_Last_Week_Tonight_with_John_Oliver_episodes',\n","    'List_of_Pomona_College_people',\n","    'List_of_RNAs',\n","    'List_of_Spanish_inventions_and_discoveries',\n","    'List_of_The_Good_Doctor_episodes',\n","    'List_of_University_of_California,_Berkeley_faculty',\n","    'List_of_awards_and_honors_received_by_Jennifer_Doudna',\n","    'List_of_biochemists',\n","    'List_of_covers_of_Time_magazine_(2010s)',\n","    'List_of_drugs_by_year_of_discovery',\n","    'List_of_inventions_and_discoveries_by_women',\n","    'Lithuania',\n","    'Live_single-cell_imaging',\n","    'Locus_Biosciences',\n","    'MCR',\n","    'MELK',\n","    'MacVector',\n","    'Margaret_Goodell',\n","    'Mary_Ann_Liebert',\n","    'Mechanism_of_action',\n","    'Medical_genetics',\n","    'Merlin_Crossley',\n","    'Methanocaldococcus_sp._FS406-22',\n","    'Methylobacillus_flagellatus',\n","    'MicrobesOnline',\n","    'Middle_Eastern_Americans',\n","    'Mobile_genetic_elements',\n","    'Molecular_engineering',\n","    'Mutagenesis_(molecular_biology_technique)',\n","    'Nessa_Carey',\n","    'NgAgo',\n","    'Nitrososphaera_gargensis',\n","    'No-SCAR_(Scarless_Cas9_Assisted_Recombineering)_Genome_Editing',\n","    'Nobel_Prize_controversies',\n","    'Non-coding_RNA',\n","    'Nuclear_organization',\n","    'Opisthorchiasis',\n","    'Osaka_University',\n","    'Outline_of_transhumanism',\n","    'Pandemic_prevention',\n","    'Paul_Knoepfler',\n","    'Pelizaeus%E2%80%93Merzbacher_disease',\n","    'Perturb-seq',\n","    'Phage_ecology',\n","    'Phage_therapy',\n","    'Phikzvirus',\n","    'Philippe_Horvath',\n","    'Polled_livestock',\n","    'Pomona_College',\n","    'Poribacteria',\n","    'Protein_%26_Cell',\n","    'Protospacer_adjacent_motif',\n","    'Pseudomonas_syringae',\n","    'Public_participation_(decision_making)',\n","    'RNA',\n","    'RNA_editing',\n","    'RNA_interference',\n","    'Rampage_(2018_film)',\n","    'Rat_Genome_Database',\n","    'Red_flour_beetle',\n","    'Repeated_sequence_(DNA)',\n","    'Research_Science_Institute',\n","    'Restriction_enzyme',\n","    'Retinitis_pigmentosa',\n","    'Revival_of_the_woolly_mammoth',\n","    'Ribonuclease',\n","    'Rodolphe_Barrangou',\n","    'SKBR3',\n","    'SaPI',\n","    'Saccharomyces_boulardii',\n","    'Samira_Kiani',\n","    'Santa_Cruz_Biotechnology',\n","    'Sauerkraut',\n","    'Schistosomiasis',\n","    'Scott_W._Lowe',\n","    'Selfish_genetic_element',\n","    'Shoukhrat_Mitalipov',\n","    'Sickle_cell_disease',\n","    'Single_cell_sequencing',\n","    'Site-directed_mutagenesis',\n","    'Small_interfering_RNA',\n","    'Stephen_Tsang',\n","    'Streptococcus_pyogenes',\n","    'Surveyor_nuclease_assay',\n","    'Synechocystis_sp._PCC_6803',\n","    'Synthetic_biology',\n","    'Synthetic_immunology',\n","    'T7_phage',\n","    'The_Hastings_Center',\n","    'Theobroma_cacao',\n","    'Theodore_Roosevelt_High_School_(Iowa)',\n","    'Time_Person_of_the_Year',\n","    'Timeline_of_biology_and_organic_chemistry',\n","    'Timeline_of_biotechnology',\n","    'Timeline_of_the_San_Francisco_Bay_Area',\n","    'Topologically_associating_domain',\n","    'Trans-activating_crRNA',\n","    'Transcription_activator-like_effector_nuclease',\n","    'Transhuman',\n","    'Trichome',\n","    'Tubulin',\n","    'Ume%C3%A5',\n","    'University_of_California,_Berkeley',\n","    'Unnatural_Selection',\n","    'Unnatural_Selection_(TV_series)',\n","    'VPS35',\n","    'Vilnius',\n","    'Virginijus_%C5%A0ik%C5%A1nys',\n","    'Virome',\n","    'Virotherapy',\n","    'Virus',\n","    'Woolly_mammoth',\n","    'Xenotransplantation',\n","    'Yoshizumi_Ishino',\n","    'Zhong_Zhong_and_Hua_Hua',\n","    'Zinc_finger_nuclease']\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"RYiX-CAS8F31"},"source":["# Extract keyphrase contexts"]},{"cell_type":"markdown","metadata":{"id":"A-dbVRM88bS_"},"source":["## Define keyphrases"]},{"cell_type":"code","metadata":{"id":"G3gVN7VqHAmw"},"source":["keyphrases = [                              \n","        # discoveries\n","        'first',\n","        'discover',\n","        'invent',\n","        'independent',\n","        'history',\n","        # naming things\n","        'name', # CRISPR acronym\n","        'acronym', # CRISPR acronym\n","        # applications/ patents\n","        'application',\n","        'editing',\n","        'efforts to edit genomes',\n","        'engineering',\n","        'patent',\n","        'property',\n","        'right',\n","        'trial',\n","        'clinical',\n","        'patient',\n","        'approve', # e.g. FDA approved...\n","        # awards\n","        'award',\n","        'prize',\n","        'nobel',\n","        # people:\n","        'doudna', \n","          'charpentier', # Doudna lab?\n","          'jinek', # Doudna lab/ first author of 2012 paper (Unsung Hero)\n","          'wiedenheft', # Doudna lab (Unsung Hero)\n","          'haurwitz', # Doudna lab (Unsung Hero)\n","\n","        'church',\n","          'mali', # Church lab/ first author of 2013 paper (Unsung Hero)\n","          'yang', # Church lab/ first author of 2015 paper (mentioned in Wikipedia?) (Unsung Hero)\n","\n","        'zhang', # Broad\n","          'cong', # Broad (Unsung Hero)\n","          ' ran ', # Broad (Unsung Hero)\n","                \n","        'šikšnys', # (Unsung Hero)\n","            'šiksnys', # only first accent\n","            'sikšnys', # only second accent\n","            'siksnys', # no accents\n","          'gašiūnas', # Siksnys lab/ first author of 2012 paper (Unsung Hero)\n","            'gasiūnas', # only first accent\n","            'gašiunas', # only second accent\n","            'gasiunas', # no accents\n","\n","        'horvath', # Danisco Boyz\n","          'barrangou', # Danisco Boyz\n","        \n","        'brouns',\n","        'mojica',\n","        'ishino',\n","        'jansen',\n","        'marraffini',\n","        'sontheimer',\n","        'koonin',\n","          'lamarck', # Koonin claimed that CRISPR was Lamarckian\n","        'moineau',\n","          'laval',\n","        'bolotin',\n","        'sorokin',\n","        'makarova',\n","        'pourcel',\n","          'adaptive immunity', # 2005 studies: 1) Pourcel, 2) Mojica, 3) Bolotin\n","          'plasmids', # 2005 studies: 1) Pourcel, 2) Mojica, 3) Bolotin\n","          'rejected', # 2005 studies: 1) Pourcel, 2) Mojica, 3) Bolotin\n","        'overbeek',\n","        'liang',\n","        ' xu', # added space to avoid false positives!\n","        'jiankui',\n","        'nishimasu',\n","        'shibata',\n","        'lander', # review\n","        # institutions:\n","        'osaka',\n","        'alicante',\n","        'danisco',\n","        'broad',\n","        'harvard',\n","        'university of california', \n","        'berkeley', \n","]"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"sHS-YLj18n7_"},"source":["## Extract and cluster keyphrase contexts WIKIPEDIA"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"gD5w946_-hmg","executionInfo":{"status":"ok","timestamp":1613778585606,"user_tz":-60,"elapsed":105510,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"0c5e8016-d2f8-4ac1-c47d-a842c650c9b8"},"source":["articles = [    \n","  CRISPR_en,\n","  # CRISPR_gene_editing_en\n","]\n","\n","for article in articles:\n","  print(article.name)\n","  for keyphrase in keyphrases:\n","    print('\\t', keyphrase)\n","    lr_contexts = [\n","      (revision.index, revision.timestamp.string, 0, left, keyphrase, right, 0, revision.revid, revision.url, revision.user, revision.userid, revision.comment,)\n","      for revision in article.yield_revisions()\n","      for left, right in revision.get_lr_contexts(keyphrase, width=100, lower=True)\n","      ]\n","    df = cluster_lr_contexts(lr_contexts, score_cutoff=75, scorer=fuzz.ratio, sort_by=['left_id','timestamp'])\n","    df.to_excel(path_extractions + '/keyphrase_contexts/{}/{}{}_{}.xlsx'.format(article.name, '' if len(df) > 0 else 'EMPTY_', keyphrase, article.name))"],"execution_count":null,"outputs":[{"output_type":"stream","text":["CRISPR_en\n","\t laval\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"AzERA0ZWRAjw"},"source":["## Extract keyphrase contexts EXTERNAL ACCOUNTS"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"oHZ2z_EaRGqn","executionInfo":{"status":"ok","timestamp":1611523897825,"user_tz":-60,"elapsed":480,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"5884cf5f-bfd3-467d-cd8b-cedb65cdf391"},"source":["accounts = {\n","  'Hsu_et_al_2014':'24906146',       \n","  'Doudna_Charpentier_2014':'25430774',       \n","  'Adli_2018':'29765029',       \n","  'Ledford_2015':'26040877',       \n","  'Pennisi_2013':'23970676',       \n","  'Lander_2016':'26771483',       \n","  'Mojica_Montoliu_2016':'27401123',       \n","  'Mojica_Rodriguez-Valera_2016':'27234458',       \n","  'Ishino_et_al_2018':'29358495',       \n","  'van_Erp_et_al_2015':'25914022',       \n","  'Morange_2015a':'25740136',       \n","  'Sherkov_2015':'25748913',       \n","  'Ledford_2016a':'27443723',       \n","  'Han_She_2017':'29150001',       \n","  'Morange_2015b':'26648028',       \n","  'Liang_et_al_2017':'28623876',       \n","  'Javed_et_al_2018':'30078067',       \n","  'Ledford_2016b':'27652544',       \n","  'Ledford_Callaway_2020':'33028993',       \n","}\n","\n","accounts = {k:'https://pubmed.ncbi.nlm.nih.gov/{}/'.format(v) for k,v in accounts.items()}\n","accounts"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["{'Adli_2018': 'https://pubmed.ncbi.nlm.nih.gov/29765029/',\n"," 'Doudna_Charpentier_2014': 'https://pubmed.ncbi.nlm.nih.gov/25430774/',\n"," 'Han_She_2017': 'https://pubmed.ncbi.nlm.nih.gov/29150001/',\n"," 'Hsu_et_al_2014': 'https://pubmed.ncbi.nlm.nih.gov/24906146/',\n"," 'Ishino_et_al_2018': 'https://pubmed.ncbi.nlm.nih.gov/29358495/',\n"," 'Javed_et_al_2018': 'https://pubmed.ncbi.nlm.nih.gov/30078067/',\n"," 'Lander_2016': 'https://pubmed.ncbi.nlm.nih.gov/26771483/',\n"," 'Ledford_2015': 'https://pubmed.ncbi.nlm.nih.gov/26040877/',\n"," 'Ledford_2016a': 'https://pubmed.ncbi.nlm.nih.gov/27443723/',\n"," 'Ledford_2016b': 'https://pubmed.ncbi.nlm.nih.gov/27652544/',\n"," 'Ledford_Callaway_2020': 'https://pubmed.ncbi.nlm.nih.gov/33028993/',\n"," 'Liang_et_al_2017': 'https://pubmed.ncbi.nlm.nih.gov/28623876/',\n"," 'Mojica_Montoliu_2016': 'https://pubmed.ncbi.nlm.nih.gov/27401123/',\n"," 'Mojica_Rodriguez-Valera_2016': 'https://pubmed.ncbi.nlm.nih.gov/27234458/',\n"," 'Morange_2015a': 'https://pubmed.ncbi.nlm.nih.gov/25740136/',\n"," 'Morange_2015b': 'https://pubmed.ncbi.nlm.nih.gov/26648028/',\n"," 'Pennisi_2013': 'https://pubmed.ncbi.nlm.nih.gov/23970676/',\n"," 'Sherkov_2015': 'https://pubmed.ncbi.nlm.nih.gov/25748913/',\n"," 'van_Erp_et_al_2015': 'https://pubmed.ncbi.nlm.nih.gov/25914022/'}"]},"metadata":{"tags":[]},"execution_count":4}]},{"cell_type":"markdown","metadata":{"id":"4rG5Bsw18IoO"},"source":["# Extract citation contexts"]},{"cell_type":"markdown","metadata":{"id":"oUTw67zW8-3y"},"source":["## Define citations"]},{"cell_type":"code","metadata":{"id":"-Tf7IXZRG5HT"},"source":["citations = [\n","        # {'DOI':'10.1016/j.cell.2014.05.010',    'PMID':'24906146',  'PMC':'',         'name':'Hsu_et_al_2014'},\n","        # {'DOI':'10.1126/science.1258096',       'PMID':'25430774',  'PMC':'',         'name':'Doudna_Charpentier_2014'},\n","        # {'DOI':'10.1038/s41467-018-04252-2',    'PMID':'29765029',  'PMC':'5953931',  'name':'Adli_2018'},\n","        {'DOI':'10.1038/522020a',               'PMID':'26040877',  'PMC':'',         'name':'Ledford_2015'},\n","        # {'DOI':'10.1126/science.341.6148.833',  'PMID':'23970676',  'PMC':'',         'name':'Pennisi_2013'},\n","        # {'DOI':'10.1016/j.cell.2015.12.041',    'PMID':'26771483',  'PMC':'',         'name':'Lander_2016'},\n","        # {'DOI':'10.1016/j.tim.2016.06.005',     'PMID':'27401123',  'PMC':'',         'name':'Mojica_Montoliu_2016'},\n","        # {'DOI':'10.1111/febs.13766',            'PMID':'27234458',  'PMC':'',         'name':'Mojica_Rodriguez-Valera_2016'},\n","        # {'DOI':'10.1128/JB.00580-17',           'PMID':'29358495',  'PMC':'5847661',  'name':'Ishino_et_al_2018'},\n","        {'DOI':'10.1016/j.coviro.2015.03.011',  'PMID':'25914022',  'PMC':'',         'name':'van_Erp_et_al_2015'},\n","        {'DOI':'10.1007/s12038-015-9532-6',     'PMID':'25740136',  'PMC':'',         'name':'Morange_2015a'},\n","        {'DOI':'10.1038/nbt.3160',              'PMID':'25748913',  'PMC':'',         'name':'Sherkov_2015'},\n","        {'DOI':'10.1007/978-3-642-34657-6_1',   'PMID':'',          'PMC':'',         'name':'Mojica_Garett_2013'},\n","        # {'DOI':'10.1038/535342a',               'PMID':'27443723',  'PMC':'',         'name':'Ledford_2016a'},\n","        # {'DOI':'10.1016/bs.pmbts.2017.10.001',  'PMID':'29150001',  'PMC':'',         'name':'Han_She_2017'},\n","        {'DOI':'10.1007/s12038-015-9575-8',     'PMID':'26648028',  'PMC':'',         'name':'Morange_2015b'},\n","        # {'DOI':'10.1002/jgm.2963',              'PMID':'28623876',  'PMC':'',         'name':'Liang_et_al_2017'},\n","        # {'DOI':'10.1007/s00284-018-1547-4',     'PMID':'30078067',  'PMC':'',         'name':'Javed_et_al_2018'},\n","        # {'DOI':'10.1038/537460a',               'PMID':'27652544',  'PMC':'',         'name':'Ledford_2016b'},\n","        {'DOI':'10.1007/978-3-642-34657-6_3',   'PMID':'',          'PMC':'',         'name':'Makarova_Koonin_2013'},\n","        {'DOI':'10.1007/978-3-642-34657-6_11',  'PMID':'',          'PMC':'',         'name':'Horvath_et_al_2013'},\n","        # {'DOI':'10.1038/d41586-020-02765-9',    'PMID':'33028993',  'PMC':'',         'name':'Ledford_Callaway_2020'}, \n","]"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"YFkaVr_I9AwQ"},"source":["## Extract and cluster citation contexts"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"jvpWEDWs9kmH","executionInfo":{"status":"ok","timestamp":1612875922633,"user_tz":-60,"elapsed":3274564,"user":{"displayName":"Arno Simons","photoUrl":"","userId":"02448447039943627700"}},"outputId":"8cb15790-d6da-4cb9-c265-b24d8d1ac602"},"source":["articles = [    \n","  CRISPR_en,\n","  # CRISPR_gene_editing_en,\n","]\n","\n","for article in articles:\n","  print(article.name)\n","  for citation in citations:\n","    print('\\t', citation['name'])\n","    lr_contexts = [\n","      (revision.index, revision.timestamp.string, 0, left, matched_citation.get_superscript(revision), right, 0, revision.revid, revision.url, revision.user, revision.userid, revision.comment,)\n","      for revision in article.yield_revisions()\n","      for id_type, id_num in {k:v for k,v in citation.items() if v and any(k==i for i in ['DOI', 'PMC', 'PMID'])}.items()\n","      for matched_citation in set(\n","          reference for reference in revision.get_references() if reference.get_identifiers()[id_type] == id_num\n","          )\n","      for left, right in revision.get_lr_contexts(\n","          r'\\[{}\\]'.format(matched_citation.get_superscript(revision).strip().strip('[]')), # I do this crap to make sure '[5]' becomes '\\\\[5\\\\]' or else finiter cuts the brakets\n","          width=100, \n","          lower=True, # search in lowercase text of revision\n","          )\n","      if matched_citation.get_superscript(revision)\n","      ]\n","    df = cluster_lr_contexts(lr_contexts, score_cutoff=75, scorer=fuzz.ratio, sort_by=['left_id','timestamp'])\n","    df.to_excel(path_extractions + '/citation_contexts/{}/{}{}_{}.xlsx'.format(article.name, '' if len(df) > 0 else 'EMPTY_', citation['name'], article.name))"],"execution_count":null,"outputs":[{"output_type":"stream","text":["CRISPR_en\n","\t Ledford_2015\n","\t van_Erp_et_al_2015\n","\t Morange_2015a\n","\t Sherkov_2015\n","\t Mojica_Garett_2013\n","\t Morange_2015b\n","\t Makarova_Koonin_2013\n","\t Horvath_et_al_2013\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"wSZjc1g51so2"},"source":["# Experiments and snippets"]},{"cell_type":"code","metadata":{"id":"Gi_C2GrJ1ufk"},"source":["# revision = CRISPR_en.get_revision(revid=725648335)\n","# print(revision.url, \"\\n\")\n","# pp.pprint(revision.get_lr_contexts('was'))\n","# references = revision.get_references()\n","# reference = references[1]\n","# print(\"REFERENCE\\n\", reference.get_text())\n","# print(\"REFERENCE ID\\n\", reference.get_id())\n","# print(\"REFERENCE Number\\n\", reference.get_number_via_id())\n","# print(\"REFERENCE Superscript\\n\", reference.get_superscript(revision))"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"3NCx3QPvG2hf"},"source":["# def occurance_spans(keyphrase, article):\n","#   ''' \n","#   returns a list of time spans during which a keyphrase 'keyphrase' continously appears in an article 'a'\n","#   '''\n","#   gen = article.yield_revisions()\n","#   last_rev_was_match = False\n","#   spans = []\n","#   for revision in gen:\n","#     if keyphrase in revision.get_text():\n","#     # if keyphrase in i.get_text().split('\\nLocus structure\\n')[0]: # Trick 17 to exclude everything after history section and practially only search in histroy section\n","#       last_rev_that_matched = revision\n","#       if not last_rev_was_match:\n","#         span_beg = revision\n","#         last_rev_was_match = True\n","#     else:\n","#       if last_rev_was_match:\n","#         spans.append((span_beg, last_rev_that_matched))\n","#         last_rev_was_match = False\n","#   if last_rev_was_match:\n","#     spans.append((span_beg, last_rev_that_matched))\n","#   print(\n","#       '\"{}\" was found in the following revisions of {}:\\n{}'.format(keyphrase, 'Test', '\\n'.join(['Span {}: {} - {}'.format(indx + 1, revision[0].timestamp, revision[1].timestamp) for indx,revision in enumerate(spans)])) \n","#       if spans else '\"{}\" was not found in any revision of \"{}\"'.format(keyphrase, 'Test')\n","#       )\n","#   return spans"],"execution_count":null,"outputs":[]}]}